<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">

<!--Converted with LaTeX2HTML 2002-2-1 (1.70)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Appendices</TITLE>
<META NAME="description" CONTENT="Appendices">
<META NAME="keywords" CONTENT="prog-manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-15">
<META NAME="Generator" CONTENT="LaTeX2HTML v2002-2-1">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="prog-manual.css">

<LINK REL="next" HREF="node6.html">
<LINK REL="previous" HREF="node4.html">
<LINK REL="up" HREF="prog-manual.html">
<LINK REL="next" HREF="node6.html">
</HEAD>

<BODY >
<!--Navigation Panel-->
<A NAME="tex2html1876"
  HREF="node6.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next" SRC="next.png"></A> 
<A NAME="tex2html1870"
  HREF="prog-manual.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up" SRC="up.png"></A> 
<A NAME="tex2html1864"
  HREF="node4.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous" SRC="prev.png"></A> 
<A NAME="tex2html1872"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents" SRC="contents.png"></A> 
<A NAME="tex2html1874"
  HREF="node6.html">
<IMG WIDTH="43" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A NAME="tex2html1877"
  HREF="node6.html">Index</A>
<B> Up:</B> <A NAME="tex2html1871"
  HREF="prog-manual.html">Plotting and Analysis for</A>
<B> Previous:</B> <A NAME="tex2html1865"
  HREF="node4.html">Bibliography</A>
 &nbsp; <B>  <A NAME="tex2html1873"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A NAME="tex2html1875"
  HREF="node6.html">Index</A></B> 
<BR>
<BR>
<!--End of Navigation Panel-->
<!--Table of Child-Links-->
<A NAME="CHILD_LINKS"><STRONG>Subsections</STRONG></A>

<UL>
<LI><UL>
<LI><A NAME="tex2html1878"
  HREF="node5.html#SECTION05010000000000000000">1 Function Reference</A>
<UL>
<LI><A NAME="tex2html1879"
  HREF="node5.html#SECTION05011000000000000000">1.1 Class <TT>chans_db</TT></A>
<UL>
<LI><A NAME="tex2html1880"
  HREF="node5.html#SECTION05011100000000000000">1.1.1 Constructor <TT>chans_db</TT></A>
<LI><A NAME="tex2html1881"
  HREF="node5.html#SECTION05011200000000000000">1.1.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1882"
  HREF="node5.html#SECTION05011300000000000000">1.1.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html1883"
  HREF="node5.html#SECTION05011400000000000000">1.1.4 Method <TT>plotInf</TT></A>
<LI><A NAME="tex2html1884"
  HREF="node5.html#SECTION05011500000000000000">1.1.5 Method <TT>display</TT></A>
<LI><A NAME="tex2html1885"
  HREF="node5.html#SECTION05011600000000000000">1.1.6 Method <TT>plotAllInf</TT></A>
<LI><A NAME="tex2html1886"
  HREF="node5.html#SECTION05011700000000000000">1.1.7 Method <TT>plotAllVars</TT></A>
<LI><A NAME="tex2html1887"
  HREF="node5.html#SECTION05011800000000000000">1.1.8 Method <TT>plotGateVars</TT></A>
</UL>
<LI><A NAME="tex2html1888"
  HREF="node5.html#SECTION05012000000000000000">1.2 Class <TT>cip_trace</TT></A>
<UL>
<LI><A NAME="tex2html1889"
  HREF="node5.html#SECTION05012100000000000000">1.2.1 Constructor <TT>cip_trace</TT></A>
<LI><A NAME="tex2html1890"
  HREF="node5.html#SECTION05012200000000000000">1.2.2 Method <TT>getPulseSpike</TT></A>
<LI><A NAME="tex2html1891"
  HREF="node5.html#SECTION05012300000000000000">1.2.3 Method <TT>plotData</TT></A>
<LI><A NAME="tex2html1892"
  HREF="node5.html#SECTION05012400000000000000">1.2.4 Method <TT>get</TT></A>
<LI><A NAME="tex2html1893"
  HREF="node5.html#SECTION05012500000000000000">1.2.5 Method <TT>periodRecSpont2</TT></A>
<LI><A NAME="tex2html1894"
  HREF="node5.html#SECTION05012600000000000000">1.2.6 Method <TT>periodRecSpontRestPeriod</TT></A>
<LI><A NAME="tex2html1895"
  HREF="node5.html#SECTION05012700000000000000">1.2.7 Method <TT>periodPulse</TT></A>
<LI><A NAME="tex2html1896"
  HREF="node5.html#SECTION05012800000000000000">1.2.8 Method <TT>getResults</TT></A>
<LI><A NAME="tex2html1897"
  HREF="node5.html#SECTION05012900000000000000">1.2.9 Method <TT>set</TT></A>
<LI><A NAME="tex2html1898"
  HREF="node5.html#SECTION050121000000000000000">1.2.10 Method <TT>periodPulseIni100msRest1</TT></A>
<LI><A NAME="tex2html1899"
  HREF="node5.html#SECTION050121100000000000000">1.2.11 Method <TT>periodIniSpont</TT></A>
<LI><A NAME="tex2html1900"
  HREF="node5.html#SECTION050121200000000000000">1.2.12 Method <TT>periodPulseIni100ms</TT></A>
<LI><A NAME="tex2html1901"
  HREF="node5.html#SECTION050121300000000000000">1.2.13 Method <TT>getCIPResults</TT></A>
<LI><A NAME="tex2html1902"
  HREF="node5.html#SECTION050121400000000000000">1.2.14 Method <TT>periodPulseHalf1</TT></A>
<LI><A NAME="tex2html1903"
  HREF="node5.html#SECTION050121500000000000000">1.2.15 Method <TT>periodRecSpont1</TT></A>
<LI><A NAME="tex2html1904"
  HREF="node5.html#SECTION050121600000000000000">1.2.16 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html1905"
  HREF="node5.html#SECTION050121700000000000000">1.2.17 Method <TT>getBurstResults</TT></A>
<LI><A NAME="tex2html1906"
  HREF="node5.html#SECTION050121800000000000000">1.2.18 Method <TT>periodPulseIni50msRest1</TT></A>
<LI><A NAME="tex2html1907"
  HREF="node5.html#SECTION050121900000000000000">1.2.19 Method <TT>measureNames</TT></A>
<LI><A NAME="tex2html1908"
  HREF="node5.html#SECTION050122000000000000000">1.2.20 Method <TT>display</TT></A>
<LI><A NAME="tex2html1909"
  HREF="node5.html#SECTION050122100000000000000">1.2.21 Method <TT>calcRecSpontPotAvg</TT></A>
<LI><A NAME="tex2html1910"
  HREF="node5.html#SECTION050122200000000000000">1.2.22 Method <TT>getRateResults</TT></A>
<LI><A NAME="tex2html1911"
  HREF="node5.html#SECTION050122300000000000000">1.2.23 Method <TT>calcPulsePotAvg</TT></A>
<LI><A NAME="tex2html1912"
  HREF="node5.html#SECTION050122400000000000000">1.2.24 Method <TT>periodRecSpontIniPeriod</TT></A>
<LI><A NAME="tex2html1913"
  HREF="node5.html#SECTION050122500000000000000">1.2.25 Method <TT>periodPulseIni100msRest2</TT></A>
<LI><A NAME="tex2html1914"
  HREF="node5.html#SECTION050122600000000000000">1.2.26 Method <TT>getRecSpontSpike</TT></A>
<LI><A NAME="tex2html1915"
  HREF="node5.html#SECTION050122700000000000000">1.2.27 Method <TT>spikes</TT></A>
<LI><A NAME="tex2html1916"
  HREF="node5.html#SECTION050122800000000000000">1.2.28 Method <TT>periodPulseIni50msRest2</TT></A>
<LI><A NAME="tex2html1917"
  HREF="node5.html#SECTION050122900000000000000">1.2.29 Method <TT>periodRecSpont</TT></A>
<LI><A NAME="tex2html1918"
  HREF="node5.html#SECTION050123000000000000000">1.2.30 Method <TT>calcPulsePotSag</TT></A>
<LI><A NAME="tex2html1919"
  HREF="node5.html#SECTION050123100000000000000">1.2.31 Method <TT>periodPulseIni50ms</TT></A>
<LI><A NAME="tex2html1920"
  HREF="node5.html#SECTION050123200000000000000">1.2.32 Method <TT>plot_abstract</TT></A>
<LI><A NAME="tex2html1921"
  HREF="node5.html#SECTION050123300000000000000">1.2.33 Method <TT>getProfileAllSpikes</TT></A>
</UL>
<LI><A NAME="tex2html1922"
  HREF="node5.html#SECTION05013000000000000000">1.3 Class <TT>cip_trace_allspikes_profile</TT></A>
<UL>
<LI><A NAME="tex2html1923"
  HREF="node5.html#SECTION05013100000000000000">1.3.1 Constructor <TT>cip_trace_allspikes_profile</TT></A>
<LI><A NAME="tex2html1924"
  HREF="node5.html#SECTION05013200000000000000">1.3.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1925"
  HREF="node5.html#SECTION05013300000000000000">1.3.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html1926"
  HREF="node5.html#SECTION05013400000000000000">1.3.4 Method <TT>plotRowSpontSpikeAnal</TT></A>
<LI><A NAME="tex2html1927"
  HREF="node5.html#SECTION05013500000000000000">1.3.5 Method <TT>display</TT></A>
</UL>
<LI><A NAME="tex2html1928"
  HREF="node5.html#SECTION05014000000000000000">1.4 Class <TT>cip_trace_profile</TT></A>
<UL>
<LI><A NAME="tex2html1929"
  HREF="node5.html#SECTION05014100000000000000">1.4.1 Constructor <TT>cip_trace_profile</TT></A>
<LI><A NAME="tex2html1930"
  HREF="node5.html#SECTION05014200000000000000">1.4.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1931"
  HREF="node5.html#SECTION05014300000000000000">1.4.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html1932"
  HREF="node5.html#SECTION05014400000000000000">1.4.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html1933"
  HREF="node5.html#SECTION05014500000000000000">1.4.5 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html1934"
  HREF="node5.html#SECTION05014600000000000000">1.4.6 Method <TT>display</TT></A>
</UL>
<LI><A NAME="tex2html1935"
  HREF="node5.html#SECTION05015000000000000000">1.5 Class <TT>cip_traces_dataset</TT></A>
<UL>
<LI><A NAME="tex2html1936"
  HREF="node5.html#SECTION05015100000000000000">1.5.1 Constructor <TT>cip_traces_dataset</TT></A>
<LI><A NAME="tex2html1937"
  HREF="node5.html#SECTION05015200000000000000">1.5.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1938"
  HREF="node5.html#SECTION05015300000000000000">1.5.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html1939"
  HREF="node5.html#SECTION05015400000000000000">1.5.4 Method <TT>getItemParams</TT></A>
<LI><A NAME="tex2html1940"
  HREF="node5.html#SECTION05015500000000000000">1.5.5 Method <TT>cip_trace_profile</TT></A>
<LI><A NAME="tex2html1941"
  HREF="node5.html#SECTION05015600000000000000">1.5.6 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html1942"
  HREF="node5.html#SECTION05015700000000000000">1.5.7 Method <TT>display</TT></A>
<LI><A NAME="tex2html1943"
  HREF="node5.html#SECTION05015800000000000000">1.5.8 Method <TT>loadItemProfile</TT></A>
<LI><A NAME="tex2html1944"
  HREF="node5.html#SECTION05015900000000000000">1.5.9 Method <TT>paramNames</TT></A>
</UL>
<LI><A NAME="tex2html1945"
  HREF="node5.html#SECTION05016000000000000000">1.6 Class <TT>cip_traceset</TT></A>
<UL>
<LI><A NAME="tex2html1946"
  HREF="node5.html#SECTION05016100000000000000">1.6.1 Constructor <TT>cip_traceset</TT></A>
<LI><A NAME="tex2html1947"
  HREF="node5.html#SECTION05016200000000000000">1.6.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1948"
  HREF="node5.html#SECTION05016300000000000000">1.6.3 Method <TT>getItemParams</TT></A>
<LI><A NAME="tex2html1949"
  HREF="node5.html#SECTION05016400000000000000">1.6.4 Method <TT>cip_trace_profile</TT></A>
<LI><A NAME="tex2html1950"
  HREF="node5.html#SECTION05016500000000000000">1.6.5 Method <TT>display</TT></A>
<LI><A NAME="tex2html1951"
  HREF="node5.html#SECTION05016600000000000000">1.6.6 Method <TT>loadItemProfile</TT></A>
<LI><A NAME="tex2html1952"
  HREF="node5.html#SECTION05016700000000000000">1.6.7 Method <TT>paramNames</TT></A>
</UL>
<LI><A NAME="tex2html1953"
  HREF="node5.html#SECTION05017000000000000000">1.7 Class <TT>cip_traceset_dataset</TT></A>
<UL>
<LI><A NAME="tex2html1954"
  HREF="node5.html#SECTION05017100000000000000">1.7.1 Constructor <TT>cip_traceset_dataset</TT></A>
<LI><A NAME="tex2html1955"
  HREF="node5.html#SECTION05017200000000000000">1.7.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1956"
  HREF="node5.html#SECTION05017300000000000000">1.7.3 Method <TT>readDBItems</TT></A>
<LI><A NAME="tex2html1957"
  HREF="node5.html#SECTION05017400000000000000">1.7.4 Method <TT>display</TT></A>
<LI><A NAME="tex2html1958"
  HREF="node5.html#SECTION05017500000000000000">1.7.5 Method <TT>loadItemProfile</TT></A>
</UL>
<LI><A NAME="tex2html1959"
  HREF="node5.html#SECTION05018000000000000000">1.8 Class <TT>cluster_db</TT></A>
<UL>
<LI><A NAME="tex2html1960"
  HREF="node5.html#SECTION05018100000000000000">1.8.1 Constructor <TT>cluster_db</TT></A>
<LI><A NAME="tex2html1961"
  HREF="node5.html#SECTION05018200000000000000">1.8.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1962"
  HREF="node5.html#SECTION05018300000000000000">1.8.3 Method <TT>plotHist</TT></A>
<LI><A NAME="tex2html1963"
  HREF="node5.html#SECTION05018400000000000000">1.8.4 Method <TT>plotQuality</TT></A>
<LI><A NAME="tex2html1964"
  HREF="node5.html#SECTION05018500000000000000">1.8.5 Method <TT>display</TT></A>
<LI><A NAME="tex2html1965"
  HREF="node5.html#SECTION05018600000000000000">1.8.6 Method <TT>plot_abstract</TT></A>
</UL>
<LI><A NAME="tex2html1966"
  HREF="node5.html#SECTION05019000000000000000">1.9 Class <TT>corrcoefs_db</TT></A>
<UL>
<LI><A NAME="tex2html1967"
  HREF="node5.html#SECTION05019100000000000000">1.9.1 Constructor <TT>corrcoefs_db</TT></A>
</UL>
<LI><A NAME="tex2html1968"
  HREF="node5.html#SECTION050110000000000000000">1.10 Class <TT>dataset_db_bundle</TT></A>
<UL>
<LI><A NAME="tex2html1969"
  HREF="node5.html#SECTION050110100000000000000">1.10.1 Constructor <TT>dataset_db_bundle</TT></A>
<LI><A NAME="tex2html1970"
  HREF="node5.html#SECTION050110200000000000000">1.10.2 Method <TT>matchingRow</TT></A>
<LI><A NAME="tex2html1971"
  HREF="node5.html#SECTION050110300000000000000">1.10.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html1972"
  HREF="node5.html#SECTION050110400000000000000">1.10.4 Method <TT>getNeuronRowIndex</TT></A>
<LI><A NAME="tex2html1973"
  HREF="node5.html#SECTION050110500000000000000">1.10.5 Method <TT>ctFromRows</TT></A>
<LI><A NAME="tex2html1974"
  HREF="node5.html#SECTION050110600000000000000">1.10.6 Method <TT>reportNeuron</TT></A>
<LI><A NAME="tex2html1975"
  HREF="node5.html#SECTION050110700000000000000">1.10.7 Method <TT>set</TT></A>
<LI><A NAME="tex2html1976"
  HREF="node5.html#SECTION050110800000000000000">1.10.8 Method <TT>constrainedMeasuresPreset</TT></A>
<LI><A NAME="tex2html1977"
  HREF="node5.html#SECTION050110900000000000000">1.10.9 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html1978"
  HREF="node5.html#SECTION0501101000000000000000">1.10.10 Method <TT>subsasgn</TT></A>
<LI><A NAME="tex2html1979"
  HREF="node5.html#SECTION0501101100000000000000">1.10.11 Method <TT>plotfICurve</TT></A>
<LI><A NAME="tex2html1980"
  HREF="node5.html#SECTION0501101200000000000000">1.10.12 Method <TT>display</TT></A>
<LI><A NAME="tex2html1981"
  HREF="node5.html#SECTION0501101300000000000000">1.10.13 Method <TT>rankingReportTeX</TT></A>
</UL>
<LI><A NAME="tex2html1982"
  HREF="node5.html#SECTION050111000000000000000">1.11 Class <TT>doc_generate</TT></A>
<UL>
<LI><A NAME="tex2html1983"
  HREF="node5.html#SECTION050111100000000000000">1.11.1 Constructor <TT>doc_generate</TT></A>
<LI><A NAME="tex2html1984"
  HREF="node5.html#SECTION050111200000000000000">1.11.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1985"
  HREF="node5.html#SECTION050111300000000000000">1.11.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html1986"
  HREF="node5.html#SECTION050111400000000000000">1.11.4 Method <TT>printTeXFile</TT></A>
<LI><A NAME="tex2html1987"
  HREF="node5.html#SECTION050111500000000000000">1.11.5 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html1988"
  HREF="node5.html#SECTION050111600000000000000">1.11.6 Method <TT>display</TT></A>
<LI><A NAME="tex2html1989"
  HREF="node5.html#SECTION050111700000000000000">1.11.7 Method <TT>getTeXString</TT></A>
</UL>
<LI><A NAME="tex2html1990"
  HREF="node5.html#SECTION050112000000000000000">1.12 Class <TT>doc_multi</TT></A>
<UL>
<LI><A NAME="tex2html1991"
  HREF="node5.html#SECTION050112100000000000000">1.12.1 Constructor <TT>doc_multi</TT></A>
<LI><A NAME="tex2html1992"
  HREF="node5.html#SECTION050112200000000000000">1.12.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1993"
  HREF="node5.html#SECTION050112300000000000000">1.12.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html1994"
  HREF="node5.html#SECTION050112400000000000000">1.12.4 Method <TT>getTeXString</TT></A>
</UL>
<LI><A NAME="tex2html1995"
  HREF="node5.html#SECTION050113000000000000000">1.13 Class <TT>doc_plot</TT></A>
<UL>
<LI><A NAME="tex2html1996"
  HREF="node5.html#SECTION050113100000000000000">1.13.1 Constructor <TT>doc_plot</TT></A>
<LI><A NAME="tex2html1997"
  HREF="node5.html#SECTION050113200000000000000">1.13.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html1998"
  HREF="node5.html#SECTION050113300000000000000">1.13.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html1999"
  HREF="node5.html#SECTION050113400000000000000">1.13.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html2000"
  HREF="node5.html#SECTION050113500000000000000">1.13.5 Method <TT>plot_abstract</TT></A>
<LI><A NAME="tex2html2001"
  HREF="node5.html#SECTION050113600000000000000">1.13.6 Method <TT>getTeXString</TT></A>
</UL>
<LI><A NAME="tex2html2002"
  HREF="node5.html#SECTION050114000000000000000">1.14 Class <TT>histogram_db</TT></A>
<UL>
<LI><A NAME="tex2html2003"
  HREF="node5.html#SECTION050114100000000000000">1.14.1 Constructor <TT>histogram_db</TT></A>
<LI><A NAME="tex2html2004"
  HREF="node5.html#SECTION050114200000000000000">1.14.2 Method <TT>plotEqSpaced</TT></A>
<LI><A NAME="tex2html2005"
  HREF="node5.html#SECTION050114300000000000000">1.14.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html2006"
  HREF="node5.html#SECTION050114400000000000000">1.14.4 Method <TT>calcMode</TT></A>
<LI><A NAME="tex2html2007"
  HREF="node5.html#SECTION050114500000000000000">1.14.5 Method <TT>plotRowMatrix</TT></A>
<LI><A NAME="tex2html2008"
  HREF="node5.html#SECTION050114600000000000000">1.14.6 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2009"
  HREF="node5.html#SECTION050114700000000000000">1.14.7 Method <TT>plotPages</TT></A>
<LI><A NAME="tex2html2010"
  HREF="node5.html#SECTION050114800000000000000">1.14.8 Method <TT>plot_abstract</TT></A>
</UL>
<LI><A NAME="tex2html2011"
  HREF="node5.html#SECTION050115000000000000000">1.15 Class <TT>model_ct_bundle</TT></A>
<UL>
<LI><A NAME="tex2html2012"
  HREF="node5.html#SECTION050115100000000000000">1.15.1 Constructor <TT>model_ct_bundle</TT></A>
<LI><A NAME="tex2html2013"
  HREF="node5.html#SECTION050115200000000000000">1.15.2 Method <TT>reportRankingToPhysiolNeuronsTeXFile</TT></A>
<LI><A NAME="tex2html2014"
  HREF="node5.html#SECTION050115300000000000000">1.15.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html2015"
  HREF="node5.html#SECTION050115400000000000000">1.15.4 Method <TT>plotComparefICurve</TT></A>
<LI><A NAME="tex2html2016"
  HREF="node5.html#SECTION050115500000000000000">1.15.5 Method <TT>getNeuronRowIndex</TT></A>
<LI><A NAME="tex2html2017"
  HREF="node5.html#SECTION050115600000000000000">1.15.6 Method <TT>ctFromRows</TT></A>
<LI><A NAME="tex2html2018"
  HREF="node5.html#SECTION050115700000000000000">1.15.7 Method <TT>getNeuronLabel</TT></A>
<LI><A NAME="tex2html2019"
  HREF="node5.html#SECTION050115800000000000000">1.15.8 Method <TT>addToDB</TT></A>
<LI><A NAME="tex2html2020"
  HREF="node5.html#SECTION050115900000000000000">1.15.9 Method <TT>set</TT></A>
<LI><A NAME="tex2html2021"
  HREF="node5.html#SECTION0501151000000000000000">1.15.10 Method <TT>reportCompareModelToPhysiolNeuron</TT></A>
<LI><A NAME="tex2html2022"
  HREF="node5.html#SECTION0501151100000000000000">1.15.11 Method <TT>rankMatching</TT></A>
<LI><A NAME="tex2html2023"
  HREF="node5.html#SECTION0501151200000000000000">1.15.12 Method <TT>collectPhysiolMatches</TT></A>
<LI><A NAME="tex2html2024"
  HREF="node5.html#SECTION0501151300000000000000">1.15.13 Method <TT>getTrialNum</TT></A>
<LI><A NAME="tex2html2025"
  HREF="node5.html#SECTION0501151400000000000000">1.15.14 Method <TT>plotCompareRanks</TT></A>
</UL>
<LI><A NAME="tex2html2026"
  HREF="node5.html#SECTION050116000000000000000">1.16 Class <TT>model_ranked_to_physiol_bundle</TT></A>
<UL>
<LI><A NAME="tex2html2027"
  HREF="node5.html#SECTION050116100000000000000">1.16.1 Constructor <TT>model_ranked_to_physiol_bundle</TT></A>
<LI><A NAME="tex2html2028"
  HREF="node5.html#SECTION050116200000000000000">1.16.2 Method <TT>plotfICurve</TT></A>
<LI><A NAME="tex2html2029"
  HREF="node5.html#SECTION050116300000000000000">1.16.3 Method <TT>plotCompareRanks</TT></A>
<LI><A NAME="tex2html2030"
  HREF="node5.html#SECTION050116400000000000000">1.16.4 Method <TT>comparisonReport</TT></A>
</UL>
<LI><A NAME="tex2html2031"
  HREF="node5.html#SECTION050117000000000000000">1.17 Class <TT>params_cip_trace_fileset</TT></A>
<UL>
<LI><A NAME="tex2html2032"
  HREF="node5.html#SECTION050117100000000000000">1.17.1 Constructor <TT>params_cip_trace_fileset</TT></A>
<LI><A NAME="tex2html2033"
  HREF="node5.html#SECTION050117200000000000000">1.17.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2034"
  HREF="node5.html#SECTION050117300000000000000">1.17.3 Method <TT>ctFromRows</TT></A>
<LI><A NAME="tex2html2035"
  HREF="node5.html#SECTION050117400000000000000">1.17.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html2036"
  HREF="node5.html#SECTION050117500000000000000">1.17.5 Method <TT>cip_trace_profile</TT></A>
<LI><A NAME="tex2html2037"
  HREF="node5.html#SECTION050117600000000000000">1.17.6 Method <TT>display</TT></A>
<LI><A NAME="tex2html2038"
  HREF="node5.html#SECTION050117700000000000000">1.17.7 Method <TT>cip_trace</TT></A>
<LI><A NAME="tex2html2039"
  HREF="node5.html#SECTION050117800000000000000">1.17.8 Method <TT>loadItemProfile</TT></A>
</UL>
<LI><A NAME="tex2html2040"
  HREF="node5.html#SECTION050118000000000000000">1.18 Class <TT>params_tests_dataset</TT></A>
<UL>
<LI><A NAME="tex2html2041"
  HREF="node5.html#SECTION050118100000000000000">1.18.1 Constructor <TT>params_tests_dataset</TT></A>
<LI><A NAME="tex2html2042"
  HREF="node5.html#SECTION050118200000000000000">1.18.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2043"
  HREF="node5.html#SECTION050118300000000000000">1.18.3 Method <TT>addItem</TT></A>
<LI><A NAME="tex2html2044"
  HREF="node5.html#SECTION050118400000000000000">1.18.4 Method <TT>readDBItems</TT></A>
<LI><A NAME="tex2html2045"
  HREF="node5.html#SECTION050118500000000000000">1.18.5 Method <TT>set</TT></A>
<LI><A NAME="tex2html2046"
  HREF="node5.html#SECTION050118600000000000000">1.18.6 Method <TT>getItemParams</TT></A>
<LI><A NAME="tex2html2047"
  HREF="node5.html#SECTION050118700000000000000">1.18.7 Method <TT>testNames</TT></A>
<LI><A NAME="tex2html2048"
  HREF="node5.html#SECTION050118800000000000000">1.18.8 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2049"
  HREF="node5.html#SECTION050118900000000000000">1.18.9 Method <TT>subsasgn</TT></A>
<LI><A NAME="tex2html2050"
  HREF="node5.html#SECTION0501181000000000000000">1.18.10 Method <TT>params_tests_db</TT></A>
<LI><A NAME="tex2html2051"
  HREF="node5.html#SECTION0501181100000000000000">1.18.11 Method <TT>display</TT></A>
<LI><A NAME="tex2html2052"
  HREF="node5.html#SECTION0501181200000000000000">1.18.12 Method <TT>itemResultsRow</TT></A>
<LI><A NAME="tex2html2053"
  HREF="node5.html#SECTION0501181300000000000000">1.18.13 Method <TT>getItem</TT></A>
</UL>
<LI><A NAME="tex2html2054"
  HREF="node5.html#SECTION050119000000000000000">1.19 Class <TT>params_tests_db</TT></A>
<UL>
<LI><A NAME="tex2html2055"
  HREF="node5.html#SECTION050119100000000000000">1.19.1 Constructor <TT>params_tests_db</TT></A>
<LI><A NAME="tex2html2056"
  HREF="node5.html#SECTION050119200000000000000">1.19.2 Method <TT>getParamNames</TT></A>
<LI><A NAME="tex2html2057"
  HREF="node5.html#SECTION050119300000000000000">1.19.3 Method <TT>matchingRow</TT></A>
<LI><A NAME="tex2html2058"
  HREF="node5.html#SECTION050119400000000000000">1.19.4 Method <TT>get</TT></A>
<LI><A NAME="tex2html2059"
  HREF="node5.html#SECTION050119500000000000000">1.19.5 Method <TT>joinRows</TT></A>
<LI><A NAME="tex2html2060"
  HREF="node5.html#SECTION050119600000000000000">1.19.6 Method <TT>rankVsDB</TT></A>
<LI><A NAME="tex2html2061"
  HREF="node5.html#SECTION050119700000000000000">1.19.7 Method <TT>plotParamsHists</TT></A>
<LI><A NAME="tex2html2062"
  HREF="node5.html#SECTION050119800000000000000">1.19.8 Method <TT>set</TT></A>
<LI><A NAME="tex2html2063"
  HREF="node5.html#SECTION050119900000000000000">1.19.9 Method <TT>scanParamAllRows</TT></A>
<LI><A NAME="tex2html2064"
  HREF="node5.html#SECTION0501191000000000000000">1.19.10 Method <TT>plotVarBoxMatrix</TT></A>
<LI><A NAME="tex2html2065"
  HREF="node5.html#SECTION0501191100000000000000">1.19.11 Method <TT>meanDuplicateParams</TT></A>
<LI><A NAME="tex2html2066"
  HREF="node5.html#SECTION0501191200000000000000">1.19.12 Method <TT>onlyRowsTests</TT></A>
<LI><A NAME="tex2html2067"
  HREF="node5.html#SECTION0501191300000000000000">1.19.13 Method <TT>paramsCoefs</TT></A>
<LI><A NAME="tex2html2068"
  HREF="node5.html#SECTION0501191400000000000000">1.19.14 Method <TT>unionCat</TT></A>
<LI><A NAME="tex2html2069"
  HREF="node5.html#SECTION0501191500000000000000">1.19.15 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2070"
  HREF="node5.html#SECTION0501191600000000000000">1.19.16 Method <TT>displayRankingsTeX</TT></A>
<LI><A NAME="tex2html2071"
  HREF="node5.html#SECTION0501191700000000000000">1.19.17 Method <TT>display</TT></A>
<LI><A NAME="tex2html2072"
  HREF="node5.html#SECTION0501191800000000000000">1.19.18 Method <TT>addParams</TT></A>
<LI><A NAME="tex2html2073"
  HREF="node5.html#SECTION0501191900000000000000">1.19.19 Method <TT>makeGenesisParFile</TT></A>
<LI><A NAME="tex2html2074"
  HREF="node5.html#SECTION0501192000000000000000">1.19.20 Method <TT>reIndexNeurons</TT></A>
<LI><A NAME="tex2html2075"
  HREF="node5.html#SECTION0501192100000000000000">1.19.21 Method <TT>rankVsAllDB</TT></A>
<LI><A NAME="tex2html2076"
  HREF="node5.html#SECTION0501192200000000000000">1.19.22 Method <TT>paramsParamsCoefs</TT></A>
<LI><A NAME="tex2html2077"
  HREF="node5.html#SECTION0501192300000000000000">1.19.23 Method <TT>mergeMultipleCIPsInOne</TT></A>
<LI><A NAME="tex2html2078"
  HREF="node5.html#SECTION0501192400000000000000">1.19.24 Method <TT>getParamRowIndices</TT></A>
<LI><A NAME="tex2html2079"
  HREF="node5.html#SECTION0501192500000000000000">1.19.25 Method <TT>testsHists</TT></A>
<LI><A NAME="tex2html2080"
  HREF="node5.html#SECTION0501192600000000000000">1.19.26 Method <TT>getDualCIPdb</TT></A>
<LI><A NAME="tex2html2081"
  HREF="node5.html#SECTION0501192700000000000000">1.19.27 Method <TT>invarParam</TT></A>
<LI><A NAME="tex2html2082"
  HREF="node5.html#SECTION0501192800000000000000">1.19.28 Method <TT>getProfile</TT></A>
<LI><A NAME="tex2html2083"
  HREF="node5.html#SECTION0501192900000000000000">1.19.29 Method <TT>delColumns</TT></A>
<LI><A NAME="tex2html2084"
  HREF="node5.html#SECTION0501193000000000000000">1.19.30 Method <TT>scaleParamsOneRow</TT></A>
<LI><A NAME="tex2html2085"
  HREF="node5.html#SECTION0501193100000000000000">1.19.31 Method <TT>paramsHists</TT></A>
<LI><A NAME="tex2html2086"
  HREF="node5.html#SECTION0501193200000000000000">1.19.32 Method <TT>paramsTestsCoefsHists</TT></A>
<LI><A NAME="tex2html2087"
  HREF="node5.html#SECTION0501193300000000000000">1.19.33 Method <TT>crossProd</TT></A>
<LI><A NAME="tex2html2088"
  HREF="node5.html#SECTION0501193400000000000000">1.19.34 Method <TT>invarParams</TT></A>
</UL>
<LI><A NAME="tex2html2089"
  HREF="node5.html#SECTION050120000000000000000">1.20 Class <TT>params_tests_fileset</TT></A>
<UL>
<LI><A NAME="tex2html2090"
  HREF="node5.html#SECTION050120100000000000000">1.20.1 Constructor <TT>params_tests_fileset</TT></A>
<LI><A NAME="tex2html2091"
  HREF="node5.html#SECTION050120200000000000000">1.20.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2092"
  HREF="node5.html#SECTION050120300000000000000">1.20.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html2093"
  HREF="node5.html#SECTION050120400000000000000">1.20.4 Method <TT>trace_profile</TT></A>
<LI><A NAME="tex2html2094"
  HREF="node5.html#SECTION050120500000000000000">1.20.5 Method <TT>getItemParams</TT></A>
<LI><A NAME="tex2html2095"
  HREF="node5.html#SECTION050120600000000000000">1.20.6 Method <TT>addFiles</TT></A>
<LI><A NAME="tex2html2096"
  HREF="node5.html#SECTION050120700000000000000">1.20.7 Method <TT>display</TT></A>
<LI><A NAME="tex2html2097"
  HREF="node5.html#SECTION050120800000000000000">1.20.8 Method <TT>loadItemProfile</TT></A>
<LI><A NAME="tex2html2098"
  HREF="node5.html#SECTION050120900000000000000">1.20.9 Method <TT>paramNames</TT></A>
<LI><A NAME="tex2html2099"
  HREF="node5.html#SECTION0501201000000000000000">1.20.10 Method <TT>trace</TT></A>
</UL>
<LI><A NAME="tex2html2100"
  HREF="node5.html#SECTION050121000000000000000">1.21 Class <TT>params_tests_profile</TT></A>
<UL>
<LI><A NAME="tex2html2101"
  HREF="node5.html#SECTION050121100000000000000">1.21.1 Constructor <TT>params_tests_profile</TT></A>
<LI><A NAME="tex2html2102"
  HREF="node5.html#SECTION050121200000000000000">1.21.2 Method <TT>get</TT></A>
</UL>
<LI><A NAME="tex2html2103"
  HREF="node5.html#SECTION050122000000000000000">1.22 Class <TT>period</TT></A>
<UL>
<LI><A NAME="tex2html2104"
  HREF="node5.html#SECTION050122100000000000000">1.22.1 Constructor <TT>period</TT></A>
<LI><A NAME="tex2html2105"
  HREF="node5.html#SECTION050122200000000000000">1.22.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2106"
  HREF="node5.html#SECTION050122300000000000000">1.22.3 Method <TT>SpikeTimesinPeriod</TT></A>
<LI><A NAME="tex2html2107"
  HREF="node5.html#SECTION050122400000000000000">1.22.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html2108"
  HREF="node5.html#SECTION050122500000000000000">1.22.5 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2109"
  HREF="node5.html#SECTION050122600000000000000">1.22.6 Method <TT>display</TT></A>
</UL>
<LI><A NAME="tex2html2110"
  HREF="node5.html#SECTION050123000000000000000">1.23 Class <TT>physiol_bundle</TT></A>
<UL>
<LI><A NAME="tex2html2111"
  HREF="node5.html#SECTION050123100000000000000">1.23.1 Constructor <TT>physiol_bundle</TT></A>
<LI><A NAME="tex2html2112"
  HREF="node5.html#SECTION050123200000000000000">1.23.2 Method <TT>matchingControlNeuron</TT></A>
<LI><A NAME="tex2html2113"
  HREF="node5.html#SECTION050123300000000000000">1.23.3 Method <TT>matchingRow</TT></A>
<LI><A NAME="tex2html2114"
  HREF="node5.html#SECTION050123400000000000000">1.23.4 Method <TT>get</TT></A>
<LI><A NAME="tex2html2115"
  HREF="node5.html#SECTION050123500000000000000">1.23.5 Method <TT>getNeuronRowIndex</TT></A>
<LI><A NAME="tex2html2116"
  HREF="node5.html#SECTION050123600000000000000">1.23.6 Method <TT>ctFromRows</TT></A>
<LI><A NAME="tex2html2117"
  HREF="node5.html#SECTION050123700000000000000">1.23.7 Method <TT>getNeuronLabel</TT></A>
<LI><A NAME="tex2html2118"
  HREF="node5.html#SECTION050123800000000000000">1.23.8 Method <TT>plotfICurveStats</TT></A>
<LI><A NAME="tex2html2119"
  HREF="node5.html#SECTION050123900000000000000">1.23.9 Method <TT>set</TT></A>
<LI><A NAME="tex2html2120"
  HREF="node5.html#SECTION0501231000000000000000">1.23.10 Method <TT>constrainedMeasuresPreset</TT></A>
<LI><A NAME="tex2html2121"
  HREF="node5.html#SECTION0501231100000000000000">1.23.11 Method <TT>bestMatchAllNeurons</TT></A>
</UL>
<LI><A NAME="tex2html2122"
  HREF="node5.html#SECTION050124000000000000000">1.24 Class <TT>physiol_cip_traceset</TT></A>
<UL>
<LI><A NAME="tex2html2123"
  HREF="node5.html#SECTION050124100000000000000">1.24.1 Constructor <TT>physiol_cip_traceset</TT></A>
<LI><A NAME="tex2html2124"
  HREF="node5.html#SECTION050124200000000000000">1.24.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2125"
  HREF="node5.html#SECTION050124300000000000000">1.24.3 Method <TT>setProp</TT></A>
<LI><A NAME="tex2html2126"
  HREF="node5.html#SECTION050124400000000000000">1.24.4 Method <TT>CIPform</TT></A>
<LI><A NAME="tex2html2127"
  HREF="node5.html#SECTION050124500000000000000">1.24.5 Method <TT>set</TT></A>
<LI><A NAME="tex2html2128"
  HREF="node5.html#SECTION050124600000000000000">1.24.6 Method <TT>getItemParams</TT></A>
<LI><A NAME="tex2html2129"
  HREF="node5.html#SECTION050124700000000000000">1.24.7 Method <TT>cip_trace_profile</TT></A>
<LI><A NAME="tex2html2130"
  HREF="node5.html#SECTION050124800000000000000">1.24.8 Method <TT>cip_trace</TT></A>
<LI><A NAME="tex2html2131"
  HREF="node5.html#SECTION050124900000000000000">1.24.9 Method <TT>loadItemProfile</TT></A>
<LI><A NAME="tex2html2132"
  HREF="node5.html#SECTION0501241000000000000000">1.24.10 Method <TT>itemResultsRow</TT></A>
<LI><A NAME="tex2html2133"
  HREF="node5.html#SECTION0501241100000000000000">1.24.11 Method <TT>paramNames</TT></A>
</UL>
<LI><A NAME="tex2html2134"
  HREF="node5.html#SECTION050125000000000000000">1.25 Class <TT>physiol_cip_traceset_fileset</TT></A>
<UL>
<LI><A NAME="tex2html2135"
  HREF="node5.html#SECTION050125100000000000000">1.25.1 Constructor <TT>physiol_cip_traceset_fileset</TT></A>
<LI><A NAME="tex2html2136"
  HREF="node5.html#SECTION050125200000000000000">1.25.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2137"
  HREF="node5.html#SECTION050125300000000000000">1.25.3 Method <TT>setProp</TT></A>
<LI><A NAME="tex2html2138"
  HREF="node5.html#SECTION050125400000000000000">1.25.4 Method <TT>physiol_bundle</TT></A>
<LI><A NAME="tex2html2139"
  HREF="node5.html#SECTION050125500000000000000">1.25.5 Method <TT>readDBItems</TT></A>
<LI><A NAME="tex2html2140"
  HREF="node5.html#SECTION050125600000000000000">1.25.6 Method <TT>set</TT></A>
<LI><A NAME="tex2html2141"
  HREF="node5.html#SECTION050125700000000000000">1.25.7 Method <TT>display</TT></A>
<LI><A NAME="tex2html2142"
  HREF="node5.html#SECTION050125800000000000000">1.25.8 Method <TT>vertcat</TT></A>
<LI><A NAME="tex2html2143"
  HREF="node5.html#SECTION050125900000000000000">1.25.9 Method <TT>cip_trace</TT></A>
<LI><A NAME="tex2html2144"
  HREF="node5.html#SECTION0501251000000000000000">1.25.10 Method <TT>loadItemProfile</TT></A>
<LI><A NAME="tex2html2145"
  HREF="node5.html#SECTION0501251100000000000000">1.25.11 Method <TT>neuronNameFromId</TT></A>
</UL>
<LI><A NAME="tex2html2146"
  HREF="node5.html#SECTION050126000000000000000">1.26 Class <TT>plot_abstract</TT></A>
<UL>
<LI><A NAME="tex2html2147"
  HREF="node5.html#SECTION050126100000000000000">1.26.1 Constructor <TT>plot_abstract</TT></A>
<LI><A NAME="tex2html2148"
  HREF="node5.html#SECTION050126200000000000000">1.26.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2149"
  HREF="node5.html#SECTION050126300000000000000">1.26.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2150"
  HREF="node5.html#SECTION050126400000000000000">1.26.4 Method <TT>setProp</TT></A>
<LI><A NAME="tex2html2151"
  HREF="node5.html#SECTION050126500000000000000">1.26.5 Method <TT>set</TT></A>
<LI><A NAME="tex2html2152"
  HREF="node5.html#SECTION050126600000000000000">1.26.6 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2153"
  HREF="node5.html#SECTION050126700000000000000">1.26.7 Method <TT>subsasgn</TT></A>
<LI><A NAME="tex2html2154"
  HREF="node5.html#SECTION050126800000000000000">1.26.8 Method <TT>axis</TT></A>
<LI><A NAME="tex2html2155"
  HREF="node5.html#SECTION050126900000000000000">1.26.9 Method <TT>matrixPlots</TT></A>
<LI><A NAME="tex2html2156"
  HREF="node5.html#SECTION0501261000000000000000">1.26.10 Method <TT>display</TT></A>
<LI><A NAME="tex2html2157"
  HREF="node5.html#SECTION0501261100000000000000">1.26.11 Method <TT>superposePlots</TT></A>
<LI><A NAME="tex2html2158"
  HREF="node5.html#SECTION0501261200000000000000">1.26.12 Method <TT>decorate</TT></A>
<LI><A NAME="tex2html2159"
  HREF="node5.html#SECTION0501261300000000000000">1.26.13 Method <TT>plotFigure</TT></A>
<LI><A NAME="tex2html2160"
  HREF="node5.html#SECTION0501261400000000000000">1.26.14 Method <TT>openAxis</TT></A>
</UL>
<LI><A NAME="tex2html2161"
  HREF="node5.html#SECTION050127000000000000000">1.27 Class <TT>plot_bars</TT></A>
<UL>
<LI><A NAME="tex2html2162"
  HREF="node5.html#SECTION050127100000000000000">1.27.1 Constructor <TT>plot_bars</TT></A>
<LI><A NAME="tex2html2163"
  HREF="node5.html#SECTION050127200000000000000">1.27.2 Method <TT>set</TT></A>
</UL>
<LI><A NAME="tex2html2164"
  HREF="node5.html#SECTION050128000000000000000">1.28 Class <TT>plot_errorbar</TT></A>
<UL>
<LI><A NAME="tex2html2165"
  HREF="node5.html#SECTION050128100000000000000">1.28.1 Constructor <TT>plot_errorbar</TT></A>
<LI><A NAME="tex2html2166"
  HREF="node5.html#SECTION050128200000000000000">1.28.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2167"
  HREF="node5.html#SECTION050128300000000000000">1.28.3 Method <TT>axis</TT></A>
</UL>
<LI><A NAME="tex2html2168"
  HREF="node5.html#SECTION050129000000000000000">1.29 Class <TT>plot_errorbars</TT></A>
<UL>
<LI><A NAME="tex2html2169"
  HREF="node5.html#SECTION050129100000000000000">1.29.1 Constructor <TT>plot_errorbars</TT></A>
</UL>
<LI><A NAME="tex2html2170"
  HREF="node5.html#SECTION050130000000000000000">1.30 Class <TT>plot_image</TT></A>
<UL>
<LI><A NAME="tex2html2171"
  HREF="node5.html#SECTION050130100000000000000">1.30.1 Constructor <TT>plot_image</TT></A>
<LI><A NAME="tex2html2172"
  HREF="node5.html#SECTION050130200000000000000">1.30.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2173"
  HREF="node5.html#SECTION050130300000000000000">1.30.3 Method <TT>axis</TT></A>
</UL>
<LI><A NAME="tex2html2174"
  HREF="node5.html#SECTION050131000000000000000">1.31 Class <TT>plot_inset</TT></A>
<UL>
<LI><A NAME="tex2html2175"
  HREF="node5.html#SECTION050131100000000000000">1.31.1 Constructor <TT>plot_inset</TT></A>
<LI><A NAME="tex2html2176"
  HREF="node5.html#SECTION050131200000000000000">1.31.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2177"
  HREF="node5.html#SECTION050131300000000000000">1.31.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2178"
  HREF="node5.html#SECTION050131400000000000000">1.31.4 Method <TT>set</TT></A>
</UL>
<LI><A NAME="tex2html2179"
  HREF="node5.html#SECTION050132000000000000000">1.32 Class <TT>plot_simple</TT></A>
<UL>
<LI><A NAME="tex2html2180"
  HREF="node5.html#SECTION050132100000000000000">1.32.1 Constructor <TT>plot_simple</TT></A>
<LI><A NAME="tex2html2181"
  HREF="node5.html#SECTION050132200000000000000">1.32.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2182"
  HREF="node5.html#SECTION050132300000000000000">1.32.3 Method <TT>set</TT></A>
</UL>
<LI><A NAME="tex2html2183"
  HREF="node5.html#SECTION050133000000000000000">1.33 Class <TT>plot_stack</TT></A>
<UL>
<LI><A NAME="tex2html2184"
  HREF="node5.html#SECTION050133100000000000000">1.33.1 Constructor <TT>plot_stack</TT></A>
<LI><A NAME="tex2html2185"
  HREF="node5.html#SECTION050133200000000000000">1.33.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2186"
  HREF="node5.html#SECTION050133300000000000000">1.33.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2187"
  HREF="node5.html#SECTION050133400000000000000">1.33.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html2188"
  HREF="node5.html#SECTION050133500000000000000">1.33.5 Method <TT>display</TT></A>
<LI><A NAME="tex2html2189"
  HREF="node5.html#SECTION050133600000000000000">1.33.6 Method <TT>superposePlots</TT></A>
<LI><A NAME="tex2html2190"
  HREF="node5.html#SECTION050133700000000000000">1.33.7 Method <TT>decorate</TT></A>
</UL>
<LI><A NAME="tex2html2191"
  HREF="node5.html#SECTION050134000000000000000">1.34 Class <TT>plot_superpose</TT></A>
<UL>
<LI><A NAME="tex2html2192"
  HREF="node5.html#SECTION050134100000000000000">1.34.1 Constructor <TT>plot_superpose</TT></A>
<LI><A NAME="tex2html2193"
  HREF="node5.html#SECTION050134200000000000000">1.34.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2194"
  HREF="node5.html#SECTION050134300000000000000">1.34.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2195"
  HREF="node5.html#SECTION050134400000000000000">1.34.4 Method <TT>set</TT></A>
<LI><A NAME="tex2html2196"
  HREF="node5.html#SECTION050134500000000000000">1.34.5 Method <TT>axis</TT></A>
<LI><A NAME="tex2html2197"
  HREF="node5.html#SECTION050134600000000000000">1.34.6 Method <TT>display</TT></A>
<LI><A NAME="tex2html2198"
  HREF="node5.html#SECTION050134700000000000000">1.34.7 Method <TT>superposePlots</TT></A>
<LI><A NAME="tex2html2199"
  HREF="node5.html#SECTION050134800000000000000">1.34.8 Method <TT>decorate</TT></A>
</UL>
<LI><A NAME="tex2html2200"
  HREF="node5.html#SECTION050135000000000000000">1.35 Class <TT>ranked_db</TT></A>
<UL>
<LI><A NAME="tex2html2201"
  HREF="node5.html#SECTION050135100000000000000">1.35.1 Constructor <TT>ranked_db</TT></A>
<LI><A NAME="tex2html2202"
  HREF="node5.html#SECTION050135200000000000000">1.35.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2203"
  HREF="node5.html#SECTION050135300000000000000">1.35.3 Method <TT>plotRowErrors</TT></A>
<LI><A NAME="tex2html2204"
  HREF="node5.html#SECTION050135400000000000000">1.35.4 Method <TT>joinOriginal</TT></A>
<LI><A NAME="tex2html2205"
  HREF="node5.html#SECTION050135500000000000000">1.35.5 Method <TT>plotDistMatrix</TT></A>
<LI><A NAME="tex2html2206"
  HREF="node5.html#SECTION050135600000000000000">1.35.6 Method <TT>set</TT></A>
<LI><A NAME="tex2html2207"
  HREF="node5.html#SECTION050135700000000000000">1.35.7 Method <TT>renameColumns</TT></A>
<LI><A NAME="tex2html2208"
  HREF="node5.html#SECTION050135800000000000000">1.35.8 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2209"
  HREF="node5.html#SECTION050135900000000000000">1.35.9 Method <TT>plotCompareDistMatx</TT></A>
<LI><A NAME="tex2html2210"
  HREF="node5.html#SECTION0501351000000000000000">1.35.10 Method <TT>displayRows</TT></A>
<LI><A NAME="tex2html2211"
  HREF="node5.html#SECTION0501351100000000000000">1.35.11 Method <TT>getDistMatrix</TT></A>
<LI><A NAME="tex2html2212"
  HREF="node5.html#SECTION0501351200000000000000">1.35.12 Method <TT>blockedDistances</TT></A>
</UL>
<LI><A NAME="tex2html2213"
  HREF="node5.html#SECTION050136000000000000000">1.36 Class <TT>results_profile</TT></A>
<UL>
<LI><A NAME="tex2html2214"
  HREF="node5.html#SECTION050136100000000000000">1.36.1 Constructor <TT>results_profile</TT></A>
<LI><A NAME="tex2html2215"
  HREF="node5.html#SECTION050136200000000000000">1.36.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2216"
  HREF="node5.html#SECTION050136300000000000000">1.36.3 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2217"
  HREF="node5.html#SECTION050136400000000000000">1.36.4 Method <TT>getResults</TT></A>
<LI><A NAME="tex2html2218"
  HREF="node5.html#SECTION050136500000000000000">1.36.5 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2219"
  HREF="node5.html#SECTION050136600000000000000">1.36.6 Method <TT>display</TT></A>
</UL>
<LI><A NAME="tex2html2220"
  HREF="node5.html#SECTION050137000000000000000">1.37 Class <TT>script_array</TT></A>
<UL>
<LI><A NAME="tex2html2221"
  HREF="node5.html#SECTION050137100000000000000">1.37.1 Constructor <TT>script_array</TT></A>
<LI><A NAME="tex2html2222"
  HREF="node5.html#SECTION050137200000000000000">1.37.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2223"
  HREF="node5.html#SECTION050137300000000000000">1.37.3 Method <TT>runJob</TT></A>
<LI><A NAME="tex2html2224"
  HREF="node5.html#SECTION050137400000000000000">1.37.4 Method <TT>runLast</TT></A>
<LI><A NAME="tex2html2225"
  HREF="node5.html#SECTION050137500000000000000">1.37.5 Method <TT>set</TT></A>
<LI><A NAME="tex2html2226"
  HREF="node5.html#SECTION050137600000000000000">1.37.6 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2227"
  HREF="node5.html#SECTION050137700000000000000">1.37.7 Method <TT>subsasgn</TT></A>
<LI><A NAME="tex2html2228"
  HREF="node5.html#SECTION050137800000000000000">1.37.8 Method <TT>runFirst</TT></A>
</UL>
<LI><A NAME="tex2html2229"
  HREF="node5.html#SECTION050138000000000000000">1.38 Class <TT>script_array_for_cluster</TT></A>
<UL>
<LI><A NAME="tex2html2230"
  HREF="node5.html#SECTION050138100000000000000">1.38.1 Constructor <TT>script_array_for_cluster</TT></A>
<LI><A NAME="tex2html2231"
  HREF="node5.html#SECTION050138200000000000000">1.38.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2232"
  HREF="node5.html#SECTION050138300000000000000">1.38.3 Method <TT>set</TT></A>
<LI><A NAME="tex2html2233"
  HREF="node5.html#SECTION050138400000000000000">1.38.4 Method <TT>runFirst</TT></A>
</UL>
<LI><A NAME="tex2html2234"
  HREF="node5.html#SECTION050139000000000000000">1.39 Class <TT>script_factory</TT></A>
<UL>
<LI><A NAME="tex2html2235"
  HREF="node5.html#SECTION050139100000000000000">1.39.1 Constructor <TT>script_factory</TT></A>
<LI><A NAME="tex2html2236"
  HREF="node5.html#SECTION050139200000000000000">1.39.2 Method <TT>get</TT></A>
</UL>
<LI><A NAME="tex2html2237"
  HREF="node5.html#SECTION050140000000000000000">1.40 Class <TT>spike_shape</TT></A>
<UL>
<LI><A NAME="tex2html2238"
  HREF="node5.html#SECTION050140100000000000000">1.40.1 Constructor <TT>spike_shape</TT></A>
<LI><A NAME="tex2html2239"
  HREF="node5.html#SECTION050140200000000000000">1.40.2 Method <TT>calcInitVmV2PPLocal</TT></A>
<LI><A NAME="tex2html2240"
  HREF="node5.html#SECTION050140300000000000000">1.40.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html2241"
  HREF="node5.html#SECTION050140400000000000000">1.40.4 Method <TT>calcMinVm</TT></A>
<LI><A NAME="tex2html2242"
  HREF="node5.html#SECTION050140500000000000000">1.40.5 Method <TT>calcInitVmSekerliV2</TT></A>
<LI><A NAME="tex2html2243"
  HREF="node5.html#SECTION050140600000000000000">1.40.6 Method <TT>calcInitVmMaxCurvature</TT></A>
<LI><A NAME="tex2html2244"
  HREF="node5.html#SECTION050140700000000000000">1.40.7 Method <TT>calcMaxVm</TT></A>
<LI><A NAME="tex2html2245"
  HREF="node5.html#SECTION050140800000000000000">1.40.8 Method <TT>getResults</TT></A>
<LI><A NAME="tex2html2246"
  HREF="node5.html#SECTION050140900000000000000">1.40.9 Method <TT>plotCompareMethodsSimple</TT></A>
<LI><A NAME="tex2html2247"
  HREF="node5.html#SECTION0501401000000000000000">1.40.10 Method <TT>set</TT></A>
<LI><A NAME="tex2html2248"
  HREF="node5.html#SECTION0501401100000000000000">1.40.11 Method <TT>plotPP</TT></A>
<LI><A NAME="tex2html2249"
  HREF="node5.html#SECTION0501401200000000000000">1.40.12 Method <TT>calcInitVmSlopeThresholdSupsample</TT></A>
<LI><A NAME="tex2html2250"
  HREF="node5.html#SECTION0501401300000000000000">1.40.13 Method <TT>calcInitVmV3hKpTinterp</TT></A>
<LI><A NAME="tex2html2251"
  HREF="node5.html#SECTION0501401400000000000000">1.40.14 Method <TT>plotCompareMethods</TT></A>
<LI><A NAME="tex2html2252"
  HREF="node5.html#SECTION0501401500000000000000">1.40.15 Method <TT>plotResults</TT></A>
<LI><A NAME="tex2html2253"
  HREF="node5.html#SECTION0501401600000000000000">1.40.16 Method <TT>display</TT></A>
<LI><A NAME="tex2html2254"
  HREF="node5.html#SECTION0501401700000000000000">1.40.17 Method <TT>calcInitVmLtdMaxCurv</TT></A>
<LI><A NAME="tex2html2255"
  HREF="node5.html#SECTION0501401800000000000000">1.40.18 Method <TT>calcInitVmMaxCurvPhasePlane</TT></A>
<LI><A NAME="tex2html2256"
  HREF="node5.html#SECTION0501401900000000000000">1.40.19 Method <TT>plotTPP</TT></A>
<LI><A NAME="tex2html2257"
  HREF="node5.html#SECTION0501402000000000000000">1.40.20 Method <TT>calcInitVm</TT></A>
<LI><A NAME="tex2html2258"
  HREF="node5.html#SECTION0501402100000000000000">1.40.21 Method <TT>calcWidthFall</TT></A>
<LI><A NAME="tex2html2259"
  HREF="node5.html#SECTION0501402200000000000000">1.40.22 Method <TT>calcInitVmSlopeThreshold</TT></A>
</UL>
<LI><A NAME="tex2html2260"
  HREF="node5.html#SECTION050141000000000000000">1.41 Class <TT>spike_shape_profile</TT></A>
<UL>
<LI><A NAME="tex2html2261"
  HREF="node5.html#SECTION050141100000000000000">1.41.1 Constructor <TT>spike_shape_profile</TT></A>
<LI><A NAME="tex2html2262"
  HREF="node5.html#SECTION050141200000000000000">1.41.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2263"
  HREF="node5.html#SECTION050141300000000000000">1.41.3 Method <TT>plot_abstract</TT></A>
</UL>
<LI><A NAME="tex2html2264"
  HREF="node5.html#SECTION050142000000000000000">1.42 Class <TT>spikes</TT></A>
<UL>
<LI><A NAME="tex2html2265"
  HREF="node5.html#SECTION050142100000000000000">1.42.1 Constructor <TT>spikes</TT></A>
<LI><A NAME="tex2html2266"
  HREF="node5.html#SECTION050142200000000000000">1.42.2 Method <TT>plotData</TT></A>
<LI><A NAME="tex2html2267"
  HREF="node5.html#SECTION050142300000000000000">1.42.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html2268"
  HREF="node5.html#SECTION050142400000000000000">1.42.4 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2269"
  HREF="node5.html#SECTION050142500000000000000">1.42.5 Method <TT>ISICV</TT></A>
<LI><A NAME="tex2html2270"
  HREF="node5.html#SECTION050142600000000000000">1.42.6 Method <TT>getResults</TT></A>
<LI><A NAME="tex2html2271"
  HREF="node5.html#SECTION050142700000000000000">1.42.7 Method <TT>plotISIs</TT></A>
<LI><A NAME="tex2html2272"
  HREF="node5.html#SECTION050142800000000000000">1.42.8 Method <TT>set</TT></A>
<LI><A NAME="tex2html2273"
  HREF="node5.html#SECTION050142900000000000000">1.42.9 Method <TT>periodWhole</TT></A>
<LI><A NAME="tex2html2274"
  HREF="node5.html#SECTION0501421000000000000000">1.42.10 Method <TT>getISIs</TT></A>
<LI><A NAME="tex2html2275"
  HREF="node5.html#SECTION0501421100000000000000">1.42.11 Method <TT>spikeRateISI</TT></A>
<LI><A NAME="tex2html2276"
  HREF="node5.html#SECTION0501421200000000000000">1.42.12 Method <TT>spikeRate</TT></A>
<LI><A NAME="tex2html2277"
  HREF="node5.html#SECTION0501421300000000000000">1.42.13 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2278"
  HREF="node5.html#SECTION0501421400000000000000">1.42.14 Method <TT>withinPeriodWOffset</TT></A>
<LI><A NAME="tex2html2279"
  HREF="node5.html#SECTION0501421500000000000000">1.42.15 Method <TT>spikeAmpSlope</TT></A>
<LI><A NAME="tex2html2280"
  HREF="node5.html#SECTION0501421600000000000000">1.42.16 Method <TT>display</TT></A>
<LI><A NAME="tex2html2281"
  HREF="node5.html#SECTION0501421700000000000000">1.42.17 Method <TT>vertcat</TT></A>
<LI><A NAME="tex2html2282"
  HREF="node5.html#SECTION0501421800000000000000">1.42.18 Method <TT>withinPeriod</TT></A>
<LI><A NAME="tex2html2283"
  HREF="node5.html#SECTION0501421900000000000000">1.42.19 Method <TT>intoPeriod</TT></A>
<LI><A NAME="tex2html2284"
  HREF="node5.html#SECTION0501422000000000000000">1.42.20 Method <TT>addSpikes</TT></A>
<LI><A NAME="tex2html2285"
  HREF="node5.html#SECTION0501422100000000000000">1.42.21 Method <TT>plotFreqVsTime</TT></A>
<LI><A NAME="tex2html2286"
  HREF="node5.html#SECTION0501422200000000000000">1.42.22 Method <TT>SFA</TT></A>
</UL>
<LI><A NAME="tex2html2287"
  HREF="node5.html#SECTION050143000000000000000">1.43 Class <TT>spikes_db</TT></A>
<UL>
<LI><A NAME="tex2html2288"
  HREF="node5.html#SECTION050143100000000000000">1.43.1 Constructor <TT>spikes_db</TT></A>
<LI><A NAME="tex2html2289"
  HREF="node5.html#SECTION050143200000000000000">1.43.2 Method <TT>plot_abstract</TT></A>
</UL>
<LI><A NAME="tex2html2290"
  HREF="node5.html#SECTION050144000000000000000">1.44 Class <TT>sql_portal</TT></A>
<UL>
<LI><A NAME="tex2html2291"
  HREF="node5.html#SECTION050144100000000000000">1.44.1 Constructor <TT>sql_portal</TT></A>
<LI><A NAME="tex2html2292"
  HREF="node5.html#SECTION050144200000000000000">1.44.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2293"
  HREF="node5.html#SECTION050144300000000000000">1.44.3 Method <TT>sql_table</TT></A>
<LI><A NAME="tex2html2294"
  HREF="node5.html#SECTION050144400000000000000">1.44.4 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2295"
  HREF="node5.html#SECTION050144500000000000000">1.44.5 Method <TT>sql_statement</TT></A>
<LI><A NAME="tex2html2296"
  HREF="node5.html#SECTION050144600000000000000">1.44.6 Method <TT>tests_db</TT></A>
</UL>
<LI><A NAME="tex2html2297"
  HREF="node5.html#SECTION050145000000000000000">1.45 Class <TT>stats_db</TT></A>
<UL>
<LI><A NAME="tex2html2298"
  HREF="node5.html#SECTION050145100000000000000">1.45.1 Constructor <TT>stats_db</TT></A>
<LI><A NAME="tex2html2299"
  HREF="node5.html#SECTION050145200000000000000">1.45.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2300"
  HREF="node5.html#SECTION050145300000000000000">1.45.3 Method <TT>compareStats</TT></A>
<LI><A NAME="tex2html2301"
  HREF="node5.html#SECTION050145400000000000000">1.45.4 Method <TT>plotVarMatrix</TT></A>
<LI><A NAME="tex2html2302"
  HREF="node5.html#SECTION050145500000000000000">1.45.5 Method <TT>plotColorVar</TT></A>
<LI><A NAME="tex2html2303"
  HREF="node5.html#SECTION050145600000000000000">1.45.6 Method <TT>set</TT></A>
<LI><A NAME="tex2html2304"
  HREF="node5.html#SECTION050145700000000000000">1.45.7 Method <TT>onlyRowsTests</TT></A>
<LI><A NAME="tex2html2305"
  HREF="node5.html#SECTION050145800000000000000">1.45.8 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2306"
  HREF="node5.html#SECTION050145900000000000000">1.45.9 Method <TT>plotVar</TT></A>
<LI><A NAME="tex2html2307"
  HREF="node5.html#SECTION0501451000000000000000">1.45.10 Method <TT>plot_bars_ax</TT></A>
<LI><A NAME="tex2html2308"
  HREF="node5.html#SECTION0501451100000000000000">1.45.11 Method <TT>plotYTests</TT></A>
<LI><A NAME="tex2html2309"
  HREF="node5.html#SECTION0501451200000000000000">1.45.12 Method <TT>plot_bars</TT></A>
<LI><A NAME="tex2html2310"
  HREF="node5.html#SECTION0501451300000000000000">1.45.13 Method <TT>plot_abstract</TT></A>
</UL>
<LI><A NAME="tex2html2311"
  HREF="node5.html#SECTION050146000000000000000">1.46 Class <TT>tests_3D_db</TT></A>
<UL>
<LI><A NAME="tex2html2312"
  HREF="node5.html#SECTION050146100000000000000">1.46.1 Constructor <TT>tests_3D_db</TT></A>
<LI><A NAME="tex2html2313"
  HREF="node5.html#SECTION050146200000000000000">1.46.2 Method <TT>get</TT></A>
<LI><A NAME="tex2html2314"
  HREF="node5.html#SECTION050146300000000000000">1.46.3 Method <TT>plotParamPairImage</TT></A>
<LI><A NAME="tex2html2315"
  HREF="node5.html#SECTION050146400000000000000">1.46.4 Method <TT>mergePages</TT></A>
<LI><A NAME="tex2html2316"
  HREF="node5.html#SECTION050146500000000000000">1.46.5 Method <TT>diff2D</TT></A>
<LI><A NAME="tex2html2317"
  HREF="node5.html#SECTION050146600000000000000">1.46.6 Method <TT>set</TT></A>
<LI><A NAME="tex2html2318"
  HREF="node5.html#SECTION050146700000000000000">1.46.7 Method <TT>histograms</TT></A>
<LI><A NAME="tex2html2319"
  HREF="node5.html#SECTION050146800000000000000">1.46.8 Method <TT>plotVarBox</TT></A>
<LI><A NAME="tex2html2320"
  HREF="node5.html#SECTION050146900000000000000">1.46.9 Method <TT>display</TT></A>
<LI><A NAME="tex2html2321"
  HREF="node5.html#SECTION0501461000000000000000">1.46.10 Method <TT>joinPages</TT></A>
<LI><A NAME="tex2html2322"
  HREF="node5.html#SECTION0501461100000000000000">1.46.11 Method <TT>corrCoefs</TT></A>
<LI><A NAME="tex2html2323"
  HREF="node5.html#SECTION0501461200000000000000">1.46.12 Method <TT>paramsTestsHistsStats</TT></A>
<LI><A NAME="tex2html2324"
  HREF="node5.html#SECTION0501461300000000000000">1.46.13 Method <TT>swapRowsPages</TT></A>
</UL>
<LI><A NAME="tex2html2325"
  HREF="node5.html#SECTION050147000000000000000">1.47 Class <TT>tests_db</TT></A>
<UL>
<LI><A NAME="tex2html2326"
  HREF="node5.html#SECTION050147100000000000000">1.47.1 Constructor <TT>tests_db</TT></A>
<LI><A NAME="tex2html2327"
  HREF="node5.html#SECTION050147200000000000000">1.47.2 Method <TT>setRows</TT></A>
<LI><A NAME="tex2html2328"
  HREF="node5.html#SECTION050147300000000000000">1.47.3 Method <TT>corrcoef</TT></A>
<LI><A NAME="tex2html2329"
  HREF="node5.html#SECTION050147400000000000000">1.47.4 Method <TT>allocateRows</TT></A>
<LI><A NAME="tex2html2330"
  HREF="node5.html#SECTION050147500000000000000">1.47.5 Method <TT>matchingRow</TT></A>
<LI><A NAME="tex2html2331"
  HREF="node5.html#SECTION050147600000000000000">1.47.6 Method <TT>get</TT></A>
<LI><A NAME="tex2html2332"
  HREF="node5.html#SECTION050147700000000000000">1.47.7 Method <TT>tests2idx</TT></A>
<LI><A NAME="tex2html2333"
  HREF="node5.html#SECTION050147800000000000000">1.47.8 Method <TT>invarValues</TT></A>
<LI><A NAME="tex2html2334"
  HREF="node5.html#SECTION050147900000000000000">1.47.9 Method <TT>shufflerows</TT></A>
<LI><A NAME="tex2html2335"
  HREF="node5.html#SECTION0501471000000000000000">1.47.10 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2336"
  HREF="node5.html#SECTION0501471100000000000000">1.47.11 Method <TT>setProp</TT></A>
<LI><A NAME="tex2html2337"
  HREF="node5.html#SECTION0501471200000000000000">1.47.12 Method <TT>physiol_bundle</TT></A>
<LI><A NAME="tex2html2338"
  HREF="node5.html#SECTION0501471300000000000000">1.47.13 Method <TT>mtimes</TT></A>
<LI><A NAME="tex2html2339"
  HREF="node5.html#SECTION0501471400000000000000">1.47.14 Method <TT>addColumns</TT></A>
<LI><A NAME="tex2html2340"
  HREF="node5.html#SECTION0501471500000000000000">1.47.15 Method <TT>rdivide</TT></A>
<LI><A NAME="tex2html2341"
  HREF="node5.html#SECTION0501471600000000000000">1.47.16 Method <TT>tests2log</TT></A>
<LI><A NAME="tex2html2342"
  HREF="node5.html#SECTION0501471700000000000000">1.47.17 Method <TT>transpose</TT></A>
<LI><A NAME="tex2html2343"
  HREF="node5.html#SECTION0501471800000000000000">1.47.18 Method <TT>joinRows</TT></A>
<LI><A NAME="tex2html2344"
  HREF="node5.html#SECTION0501471900000000000000">1.47.19 Method <TT>histogram</TT></A>
<LI><A NAME="tex2html2345"
  HREF="node5.html#SECTION0501472000000000000000">1.47.20 Method <TT>enumerateColumns</TT></A>
<LI><A NAME="tex2html2346"
  HREF="node5.html#SECTION0501472100000000000000">1.47.21 Method <TT>uop</TT></A>
<LI><A NAME="tex2html2347"
  HREF="node5.html#SECTION0501472200000000000000">1.47.22 Method <TT>approxMappingNNet</TT></A>
<LI><A NAME="tex2html2348"
  HREF="node5.html#SECTION0501472300000000000000">1.47.23 Method <TT>isnan</TT></A>
<LI><A NAME="tex2html2349"
  HREF="node5.html#SECTION0501472400000000000000">1.47.24 Method <TT>minus</TT></A>
<LI><A NAME="tex2html2350"
  HREF="node5.html#SECTION0501472500000000000000">1.47.25 Method <TT>displayRowsTeX</TT></A>
<LI><A NAME="tex2html2351"
  HREF="node5.html#SECTION0501472600000000000000">1.47.26 Method <TT>plotXRows</TT></A>
<LI><A NAME="tex2html2352"
  HREF="node5.html#SECTION0501472700000000000000">1.47.27 Method <TT>plotCovar</TT></A>
<LI><A NAME="tex2html2353"
  HREF="node5.html#SECTION0501472800000000000000">1.47.28 Method <TT>noNaNRows</TT></A>
<LI><A NAME="tex2html2354"
  HREF="node5.html#SECTION0501472900000000000000">1.47.29 Method <TT>princomp</TT></A>
<LI><A NAME="tex2html2355"
  HREF="node5.html#SECTION0501473000000000000000">1.47.30 Method <TT>cov</TT></A>
<LI><A NAME="tex2html2356"
  HREF="node5.html#SECTION0501473100000000000000">1.47.31 Method <TT>set</TT></A>
<LI><A NAME="tex2html2357"
  HREF="node5.html#SECTION0501473200000000000000">1.47.32 Method <TT>plotrow</TT></A>
<LI><A NAME="tex2html2358"
  HREF="node5.html#SECTION0501473300000000000000">1.47.33 Method <TT>checkConsistentCols</TT></A>
<LI><A NAME="tex2html2359"
  HREF="node5.html#SECTION0501473400000000000000">1.47.34 Method <TT>ne</TT></A>
<LI><A NAME="tex2html2360"
  HREF="node5.html#SECTION0501473500000000000000">1.47.35 Method <TT>std</TT></A>
<LI><A NAME="tex2html2361"
  HREF="node5.html#SECTION0501473600000000000000">1.47.36 Method <TT>sortrows</TT></A>
<LI><A NAME="tex2html2362"
  HREF="node5.html#SECTION0501473700000000000000">1.47.37 Method <TT>processDimNonNaNInf</TT></A>
<LI><A NAME="tex2html2363"
  HREF="node5.html#SECTION0501473800000000000000">1.47.38 Method <TT>factoran</TT></A>
<LI><A NAME="tex2html2364"
  HREF="node5.html#SECTION0501473900000000000000">1.47.39 Method <TT>plotrows</TT></A>
<LI><A NAME="tex2html2365"
  HREF="node5.html#SECTION0501474000000000000000">1.47.40 Method <TT>mean</TT></A>
<LI><A NAME="tex2html2366"
  HREF="node5.html#SECTION0501474100000000000000">1.47.41 Method <TT>rankMatching</TT></A>
<LI><A NAME="tex2html2367"
  HREF="node5.html#SECTION0501474200000000000000">1.47.42 Method <TT>onlyRowsTests</TT></A>
<LI><A NAME="tex2html2368"
  HREF="node5.html#SECTION0501474300000000000000">1.47.43 Method <TT>times</TT></A>
<LI><A NAME="tex2html2369"
  HREF="node5.html#SECTION0501474400000000000000">1.47.44 Method <TT>renameColumns</TT></A>
<LI><A NAME="tex2html2370"
  HREF="node5.html#SECTION0501474500000000000000">1.47.45 Method <TT>statsAll</TT></A>
<LI><A NAME="tex2html2371"
  HREF="node5.html#SECTION0501474600000000000000">1.47.46 Method <TT>getColNames</TT></A>
<LI><A NAME="tex2html2372"
  HREF="node5.html#SECTION0501474700000000000000">1.47.47 Method <TT>addLastRow</TT></A>
<LI><A NAME="tex2html2373"
  HREF="node5.html#SECTION0501474800000000000000">1.47.48 Method <TT>isinf</TT></A>
<LI><A NAME="tex2html2374"
  HREF="node5.html#SECTION0501474900000000000000">1.47.49 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2375"
  HREF="node5.html#SECTION0501475000000000000000">1.47.50 Method <TT>eq</TT></A>
<LI><A NAME="tex2html2376"
  HREF="node5.html#SECTION0501475100000000000000">1.47.51 Method <TT>plotBox</TT></A>
<LI><A NAME="tex2html2377"
  HREF="node5.html#SECTION0501475200000000000000">1.47.52 Method <TT>uminus</TT></A>
<LI><A NAME="tex2html2378"
  HREF="node5.html#SECTION0501475300000000000000">1.47.53 Method <TT>subsasgn</TT></A>
<LI><A NAME="tex2html2379"
  HREF="node5.html#SECTION0501475400000000000000">1.47.54 Method <TT>dbsize</TT></A>
<LI><A NAME="tex2html2380"
  HREF="node5.html#SECTION0501475500000000000000">1.47.55 Method <TT>sqrt</TT></A>
<LI><A NAME="tex2html2381"
  HREF="node5.html#SECTION0501475600000000000000">1.47.56 Method <TT>plotTestsHistsMatrix</TT></A>
<LI><A NAME="tex2html2382"
  HREF="node5.html#SECTION0501475700000000000000">1.47.57 Method <TT>anyRows</TT></A>
<LI><A NAME="tex2html2383"
  HREF="node5.html#SECTION0501475800000000000000">1.47.58 Method <TT>gt</TT></A>
<LI><A NAME="tex2html2384"
  HREF="node5.html#SECTION0501475900000000000000">1.47.59 Method <TT>hist3</TT></A>
<LI><A NAME="tex2html2385"
  HREF="node5.html#SECTION0501476000000000000000">1.47.60 Method <TT>kmeansCluster</TT></A>
<LI><A NAME="tex2html2386"
  HREF="node5.html#SECTION0501476100000000000000">1.47.61 Method <TT>compareRows</TT></A>
<LI><A NAME="tex2html2387"
  HREF="node5.html#SECTION0501476200000000000000">1.47.62 Method <TT>statsMeanStd</TT></A>
<LI><A NAME="tex2html2388"
  HREF="node5.html#SECTION0501476300000000000000">1.47.63 Method <TT>display</TT></A>
<LI><A NAME="tex2html2389"
  HREF="node5.html#SECTION0501476400000000000000">1.47.64 Method <TT>groupBy</TT></A>
<LI><A NAME="tex2html2390"
  HREF="node5.html#SECTION0501476500000000000000">1.47.65 Method <TT>statsBounds</TT></A>
<LI><A NAME="tex2html2391"
  HREF="node5.html#SECTION0501476600000000000000">1.47.66 Method <TT>end</TT></A>
<LI><A NAME="tex2html2392"
  HREF="node5.html#SECTION0501476700000000000000">1.47.67 Method <TT>isnanrows</TT></A>
<LI><A NAME="tex2html2393"
  HREF="node5.html#SECTION0501476800000000000000">1.47.68 Method <TT>le</TT></A>
<LI><A NAME="tex2html2394"
  HREF="node5.html#SECTION0501476900000000000000">1.47.69 Method <TT>plotUniquesStatsStacked3D</TT></A>
<LI><A NAME="tex2html2395"
  HREF="node5.html#SECTION0501477000000000000000">1.47.70 Method <TT>plotScatter3D</TT></A>
<LI><A NAME="tex2html2396"
  HREF="node5.html#SECTION0501477100000000000000">1.47.71 Method <TT>rop</TT></A>
<LI><A NAME="tex2html2397"
  HREF="node5.html#SECTION0501477200000000000000">1.47.72 Method <TT>vertcat</TT></A>
<LI><A NAME="tex2html2398"
  HREF="node5.html#SECTION0501477300000000000000">1.47.73 Method <TT>lt</TT></A>
<LI><A NAME="tex2html2399"
  HREF="node5.html#SECTION0501477400000000000000">1.47.74 Method <TT>unique</TT></A>
<LI><A NAME="tex2html2400"
  HREF="node5.html#SECTION0501477500000000000000">1.47.75 Method <TT>addRow</TT></A>
<LI><A NAME="tex2html2401"
  HREF="node5.html#SECTION0501477600000000000000">1.47.76 Method <TT>plotYTests</TT></A>
<LI><A NAME="tex2html2402"
  HREF="node5.html#SECTION0501477700000000000000">1.47.77 Method <TT>plotUniquesStatsBars</TT></A>
<LI><A NAME="tex2html2403"
  HREF="node5.html#SECTION0501477800000000000000">1.47.78 Method <TT>plotScatter</TT></A>
<LI><A NAME="tex2html2404"
  HREF="node5.html#SECTION0501477900000000000000">1.47.79 Method <TT>assignRowsTests</TT></A>
<LI><A NAME="tex2html2405"
  HREF="node5.html#SECTION0501478000000000000000">1.47.80 Method <TT>plus</TT></A>
<LI><A NAME="tex2html2406"
  HREF="node5.html#SECTION0501478100000000000000">1.47.81 Method <TT>displayRows</TT></A>
<LI><A NAME="tex2html2407"
  HREF="node5.html#SECTION0501478200000000000000">1.47.82 Method <TT>testsHists</TT></A>
<LI><A NAME="tex2html2408"
  HREF="node5.html#SECTION0501478300000000000000">1.47.83 Method <TT>ge</TT></A>
<LI><A NAME="tex2html2409"
  HREF="node5.html#SECTION0501478400000000000000">1.47.84 Method <TT>swapRowsPages</TT></A>
<LI><A NAME="tex2html2410"
  HREF="node5.html#SECTION0501478500000000000000">1.47.85 Method <TT>plotUniquesStats2D</TT></A>
<LI><A NAME="tex2html2411"
  HREF="node5.html#SECTION0501478600000000000000">1.47.86 Method <TT>plot_bars</TT></A>
<LI><A NAME="tex2html2412"
  HREF="node5.html#SECTION0501478700000000000000">1.47.87 Method <TT>diff</TT></A>
<LI><A NAME="tex2html2413"
  HREF="node5.html#SECTION0501478800000000000000">1.47.88 Method <TT>delColumns</TT></A>
<LI><A NAME="tex2html2414"
  HREF="node5.html#SECTION0501478900000000000000">1.47.89 Method <TT>meanDuplicateRows</TT></A>
<LI><A NAME="tex2html2415"
  HREF="node5.html#SECTION0501479000000000000000">1.47.90 Method <TT>statsMeanSE</TT></A>
<LI><A NAME="tex2html2416"
  HREF="node5.html#SECTION0501479100000000000000">1.47.91 Method <TT>sum</TT></A>
<LI><A NAME="tex2html2417"
  HREF="node5.html#SECTION0501479200000000000000">1.47.92 Method <TT>plot_abstract</TT></A>
<LI><A NAME="tex2html2418"
  HREF="node5.html#SECTION0501479300000000000000">1.47.93 Method <TT>crossProd</TT></A>
<LI><A NAME="tex2html2419"
  HREF="node5.html#SECTION0501479400000000000000">1.47.94 Method <TT>tests2cols</TT></A>
<LI><A NAME="tex2html2420"
  HREF="node5.html#SECTION0501479500000000000000">1.47.95 Method <TT>rows2Struct</TT></A>
<LI><A NAME="tex2html2421"
  HREF="node5.html#SECTION0501479600000000000000">1.47.96 Method <TT>max</TT></A>
</UL>
<LI><A NAME="tex2html2422"
  HREF="node5.html#SECTION050148000000000000000">1.48 Class <TT>trace</TT></A>
<UL>
<LI><A NAME="tex2html2423"
  HREF="node5.html#SECTION050148100000000000000">1.48.1 Constructor <TT>trace</TT></A>
<LI><A NAME="tex2html2424"
  HREF="node5.html#SECTION050148200000000000000">1.48.2 Method <TT>plotData</TT></A>
<LI><A NAME="tex2html2425"
  HREF="node5.html#SECTION050148300000000000000">1.48.3 Method <TT>get</TT></A>
<LI><A NAME="tex2html2426"
  HREF="node5.html#SECTION050148400000000000000">1.48.4 Method <TT>plot</TT></A>
<LI><A NAME="tex2html2427"
  HREF="node5.html#SECTION050148500000000000000">1.48.5 Method <TT>setProp</TT></A>
<LI><A NAME="tex2html2428"
  HREF="node5.html#SECTION050148600000000000000">1.48.6 Method <TT>calcMax</TT></A>
<LI><A NAME="tex2html2429"
  HREF="node5.html#SECTION050148700000000000000">1.48.7 Method <TT>getResults</TT></A>
<LI><A NAME="tex2html2430"
  HREF="node5.html#SECTION050148800000000000000">1.48.8 Method <TT>calcAvg</TT></A>
<LI><A NAME="tex2html2431"
  HREF="node5.html#SECTION050148900000000000000">1.48.9 Method <TT>analyzeSpikesInPeriod</TT></A>
<LI><A NAME="tex2html2432"
  HREF="node5.html#SECTION0501481000000000000000">1.48.10 Method <TT>set</TT></A>
<LI><A NAME="tex2html2433"
  HREF="node5.html#SECTION0501481100000000000000">1.48.11 Method <TT>periodWhole</TT></A>
<LI><A NAME="tex2html2434"
  HREF="node5.html#SECTION0501481200000000000000">1.48.12 Method <TT>subsref</TT></A>
<LI><A NAME="tex2html2435"
  HREF="node5.html#SECTION0501481300000000000000">1.48.13 Method <TT>calcMin</TT></A>
<LI><A NAME="tex2html2436"
  HREF="node5.html#SECTION0501481400000000000000">1.48.14 Method <TT>runAvg</TT></A>
<LI><A NAME="tex2html2437"
  HREF="node5.html#SECTION0501481500000000000000">1.48.15 Method <TT>display</TT></A>
<LI><A NAME="tex2html2438"
  HREF="node5.html#SECTION0501481600000000000000">1.48.16 Method <TT>getDy</TT></A>
<LI><A NAME="tex2html2439"
  HREF="node5.html#SECTION0501481700000000000000">1.48.17 Method <TT>findFilteredSpikes</TT></A>
<LI><A NAME="tex2html2440"
  HREF="node5.html#SECTION0501481800000000000000">1.48.18 Method <TT>withinPeriod</TT></A>
<LI><A NAME="tex2html2441"
  HREF="node5.html#SECTION0501481900000000000000">1.48.19 Method <TT>spikes</TT></A>
<LI><A NAME="tex2html2442"
  HREF="node5.html#SECTION0501482000000000000000">1.48.20 Method <TT>getSpike</TT></A>
<LI><A NAME="tex2html2443"
  HREF="node5.html#SECTION0501482100000000000000">1.48.21 Method <TT>plot_abstract</TT></A>
<LI><A NAME="tex2html2444"
  HREF="node5.html#SECTION0501482200000000000000">1.48.22 Method <TT>spike_shape</TT></A>
</UL>
<LI><A NAME="tex2html2445"
  HREF="node5.html#SECTION050149000000000000000">1.49 Class <TT>trace_profile</TT></A>
<UL>
<LI><A NAME="tex2html2446"
  HREF="node5.html#SECTION050149100000000000000">1.49.1 Constructor <TT>trace_profile</TT></A>
<LI><A NAME="tex2html2447"
  HREF="node5.html#SECTION050149200000000000000">1.49.2 Method <TT>get</TT></A>
</UL>
<LI><A NAME="tex2html2448"
  HREF="node5.html#SECTION050150000000000000000">1.50 Utility functions</A>
<UL>
<LI><A NAME="tex2html2449"
  HREF="node5.html#SECTION050150100000000000000">1.50.1 Function <TT>TeXtable</TT></A>
<LI><A NAME="tex2html2450"
  HREF="node5.html#SECTION050150200000000000000">1.50.2 Function <TT>logLevels</TT></A>
<LI><A NAME="tex2html2451"
  HREF="node5.html#SECTION050150300000000000000">1.50.3 Function <TT>parseGenesisFilename</TT></A>
<LI><A NAME="tex2html2452"
  HREF="node5.html#SECTION050150400000000000000">1.50.4 Function <TT>gettracelist2</TT></A>
<LI><A NAME="tex2html2453"
  HREF="node5.html#SECTION050150500000000000000">1.50.5 Function <TT>loadtraces</TT></A>
<LI><A NAME="tex2html2454"
  HREF="node5.html#SECTION050150600000000000000">1.50.6 Function <TT>properTeXLabel</TT></A>
<LI><A NAME="tex2html2455"
  HREF="node5.html#SECTION050150700000000000000">1.50.7 Function <TT>defaultValue</TT></A>
<LI><A NAME="tex2html2456"
  HREF="node5.html#SECTION050150800000000000000">1.50.8 Function <TT>ns_CIPlist</TT></A>
<LI><A NAME="tex2html2457"
  HREF="node5.html#SECTION050150900000000000000">1.50.9 Function <TT>diff3T</TT></A>
<LI><A NAME="tex2html2458"
  HREF="node5.html#SECTION0501501000000000000000">1.50.10 Function <TT>findspikes_old</TT></A>
<LI><A NAME="tex2html2459"
  HREF="node5.html#SECTION0501501100000000000000">1.50.11 Function <TT>properTeXFilename</TT></A>
<LI><A NAME="tex2html2460"
  HREF="node5.html#SECTION0501501200000000000000">1.50.12 Function <TT>mergeStructs</TT></A>
<LI><A NAME="tex2html2461"
  HREF="node5.html#SECTION0501501300000000000000">1.50.13 Function <TT>fillederrorbar</TT></A>
<LI><A NAME="tex2html2462"
  HREF="node5.html#SECTION0501501400000000000000">1.50.14 Function <TT>struct2DB</TT></A>
<LI><A NAME="tex2html2463"
  HREF="node5.html#SECTION0501501500000000000000">1.50.15 Function <TT>getfuzzyfield</TT></A>
<LI><A NAME="tex2html2464"
  HREF="node5.html#SECTION0501501600000000000000">1.50.16 Function <TT>maxima</TT></A>
<LI><A NAME="tex2html2465"
  HREF="node5.html#SECTION0501501700000000000000">1.50.17 Function <TT>chanTables2DB</TT></A>
<LI><A NAME="tex2html2466"
  HREF="node5.html#SECTION0501501800000000000000">1.50.18 Function <TT>prefixStruct</TT></A>
<LI><A NAME="tex2html2467"
  HREF="node5.html#SECTION0501501900000000000000">1.50.19 Function <TT>array2str</TT></A>
<LI><A NAME="tex2html2468"
  HREF="node5.html#SECTION0501502000000000000000">1.50.20 Function <TT>cell2TeX</TT></A>
<LI><A NAME="tex2html2469"
  HREF="node5.html#SECTION0501502100000000000000">1.50.21 Function <TT>setAxisNonNaN</TT></A>
<LI><A NAME="tex2html2470"
  HREF="node5.html#SECTION0501502200000000000000">1.50.22 Function <TT>diff2T_h4</TT></A>
<LI><A NAME="tex2html2471"
  HREF="node5.html#SECTION0501502300000000000000">1.50.23 Function <TT>plotImage</TT></A>
<LI><A NAME="tex2html2472"
  HREF="node5.html#SECTION0501502400000000000000">1.50.24 Function <TT>diff2T</TT></A>
<LI><A NAME="tex2html2473"
  HREF="node5.html#SECTION0501502500000000000000">1.50.25 Function <TT>boxutilp</TT></A>
<LI><A NAME="tex2html2474"
  HREF="node5.html#SECTION0501502600000000000000">1.50.26 Function <TT>ns_load_tracesets</TT></A>
<LI><A NAME="tex2html2475"
  HREF="node5.html#SECTION0501502700000000000000">1.50.27 Function <TT>string2File</TT></A>
<LI><A NAME="tex2html2476"
  HREF="node5.html#SECTION0501502800000000000000">1.50.28 Function <TT>sortedUniqueValues</TT></A>
<LI><A NAME="tex2html2477"
  HREF="node5.html#SECTION0501502900000000000000">1.50.29 Function <TT>meanSpikeFreq</TT></A>
<LI><A NAME="tex2html2478"
  HREF="node5.html#SECTION0501503000000000000000">1.50.30 Function <TT>diff3T_h4</TT></A>
<LI><A NAME="tex2html2479"
  HREF="node5.html#SECTION0501503100000000000000">1.50.31 Function <TT>growRange</TT></A>
<LI><A NAME="tex2html2480"
  HREF="node5.html#SECTION0501503200000000000000">1.50.32 Function <TT>readNeuronVecBin</TT></A>
<LI><A NAME="tex2html2481"
  HREF="node5.html#SECTION0501503300000000000000">1.50.33 Function <TT>colormapBlueCrossRed</TT></A>
<LI><A NAME="tex2html2482"
  HREF="node5.html#SECTION0501503400000000000000">1.50.34 Function <TT>diffT</TT></A>
<LI><A NAME="tex2html2483"
  HREF="node5.html#SECTION0501503500000000000000">1.50.35 Function <TT>interpValByIndex</TT></A>
<LI><A NAME="tex2html2484"
  HREF="node5.html#SECTION0501503600000000000000">1.50.36 Function <TT>subTextLabel</TT></A>
<LI><A NAME="tex2html2485"
  HREF="node5.html#SECTION0501503700000000000000">1.50.37 Function <TT>calcGraphNormPtsRatio</TT></A>
<LI><A NAME="tex2html2486"
  HREF="node5.html#SECTION0501503800000000000000">1.50.38 Function <TT>balanceInputProbs</TT></A>
<LI><A NAME="tex2html2487"
  HREF="node5.html#SECTION0501503900000000000000">1.50.39 Function <TT>findspikes</TT></A>
<LI><A NAME="tex2html2488"
  HREF="node5.html#SECTION0501504000000000000000">1.50.40 Function <TT>boxplotp</TT></A>
<LI><A NAME="tex2html2489"
  HREF="node5.html#SECTION0501504100000000000000">1.50.41 Function <TT>collectspikes</TT></A>
<LI><A NAME="tex2html2490"
  HREF="node5.html#SECTION0501504200000000000000">1.50.42 Function <TT>findVectorInMatrix</TT></A>
<LI><A NAME="tex2html2491"
  HREF="node5.html#SECTION0501504300000000000000">1.50.43 Function <TT>uniqueValues</TT></A>
<LI><A NAME="tex2html2492"
  HREF="node5.html#SECTION0501504400000000000000">1.50.44 Function <TT>makeIdx</TT></A>
</UL></UL></UL></UL>
<!--End of Table of Child-Links-->
<HR>

<H1><A NAME="SECTION05000000000000000000">
Appendices</A>
</H1>

<P>

<H1><A NAME="SECTION05010000000000000000"></A><A NAME="sec:Function-Reference"></A>
<BR>
1 Function Reference
</H1>

<P>
See Section <A HREF="node1.html#sub:Finding-your-way-around">1.6</A> on how to get online
help about the software within M<SMALL>ATLAB</SMALL>.

<P>


<P>

<H2><A NAME="SECTION05011000000000000000">
1.1 Class <TT>chans_db</TT></A>
</H2><A NAME="15527"></A><A NAME="ref_chans_db"></A>
<H3><A NAME="SECTION05011100000000000000">
1.1.1 Constructor <TT>chans_db/chans_db</TT></A>
</H3><A NAME="15529"></A><A NAME="ref_chans_db__chans_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of channel activation and kinetics.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_chans_db = chans_db(data, col_names, channel_info, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Channel tables can be imported from
 Genesis using the utils/chanTables2DB script.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>Database contents.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>The channel variable names.
</DD>
<DT><STRONG><TT>channel_info</TT>:</STRONG></DT>
<DD>Structure that holds scalar data elements such as Gbar.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, channel_info, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15531"></A>, <TT>chanTables2DB</TT> (p.&nbsp;<A HREF="#ref_chanTables2DB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15533"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/06/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011200000000000000">
1.1.2 Method <TT>chans_db/get</TT></A>
</H3><A NAME="15535"></A><A NAME="ref_chans_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011300000000000000">
1.1.3 Method <TT>chans_db/set</TT></A>
</H3><A NAME="15537"></A><A NAME="ref_chans_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011400000000000000">
1.1.4 Method <TT>chans_db/plotInf</TT></A>
</H3><A NAME="15539"></A><A NAME="ref_chans_db__plotInf"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plot the product of minf variables and the gmax of the given channel.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotInf(a_chans_db, chan_name, gate_subnames, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_chans_db</TT>:</STRONG></DT>
<DD>A chans_db describing channel variables.
</DD>
<DT><STRONG><TT>chan_name</TT>:</STRONG></DT>
<DD>Name of channel that make up the stem of variable
<P>
names.
</DD>
<DT><STRONG><TT>gate_subnames</TT>:</STRONG></DT>
<DD>Gate names of the channel.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.

<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15541"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15543"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15545"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/07/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011500000000000000">
1.1.5 Method <TT>chans_db/display</TT></A>
</H3><A NAME="15547"></A><A NAME="ref_chans_db__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011600000000000000">
1.1.6 Method <TT>chans_db/plotAllInf</TT></A>
</H3><A NAME="15549"></A><A NAME="ref_chans_db__plotAllInf"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots the steady-state (infinity) response of all channels.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotAllInf(a_chans_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_chans_db</TT>:</STRONG></DT>
<DD>a chans_db
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(rest passed to matrixPlots.)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15551"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15553"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15555"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/03/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011700000000000000">
1.1.7 Method <TT>chans_db/plotAllVars</TT></A>
</H3><A NAME="15557"></A><A NAME="ref_chans_db__plotAllVars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plot all channel variables by grouping activation and time constant curves per channel.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotAllVars(a_chans_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_chans_db</TT>:</STRONG></DT>
<DD>A chans_db describing channel variables.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>String that identify the source of the tables structure.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15559"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15561"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15563"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/03/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05011800000000000000">
1.1.8 Method <TT>chans_db/plotGateVars</TT></A>
</H3><A NAME="15565"></A><A NAME="ref_chans_db__plotGateVars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plot given channel gate variables of the same channel superposed.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotGateVars(a_chans_db, chan_name, gate_subnames, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_chans_db</TT>:</STRONG></DT>
<DD>A chans_db describing channel variables.
</DD>
<DT><STRONG><TT>chan_name</TT>:</STRONG></DT>
<DD>Name of channel that make up the stem of variable
<P>
names.
</DD>
<DT><STRONG><TT>gate_subnames</TT>:</STRONG></DT>
<DD>Gate names of the channel.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>usePowers</TT>:</STRONG></DT>
<DD>Use the gate powers, Luke.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15567"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15569"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15571"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/07/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05012000000000000000">
1.2 Class <TT>cip_trace</TT></A>
</H2><A NAME="15573"></A><A NAME="ref_cip_trace"></A>
<H3><A NAME="SECTION05012100000000000000">
1.2.1 Constructor <TT>cip_trace/cip_trace</TT></A>
</H3><A NAME="15575"></A><A NAME="ref_cip_trace__cip_trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A trace with a current injection pulse (CIP).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = cip_trace(datasrc, dt, dy,
		  pulse_time_start, pulse_time_width, id, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>datasrc</TT>:</STRONG></DT>
<DD>A vector of data points containing the spike shape.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s].
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>pulse_time_start, pulse_time_width</TT>:</STRONG></DT>
<DD><P>
Start and width of the pulse [dt]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, such as:
<DL>
<DD></DD>
<DT><STRONG><TT>trace_time_start</TT>:</STRONG></DT>
<DD>Samples in the beginning to discard [dt]
<P>
(see trace for more)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
trace, pulse_time_start, pulse_time_width, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15577"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15579"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15581"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15583"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012200000000000000">
1.2.2 Method <TT>cip_trace/getPulseSpike</TT></A>
</H3><A NAME="15585"></A><A NAME="ref_cip_trace__getPulseSpike"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert a spike in the CIP period to a spike_shape object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = getPulseSpike(trace, spikes, spike_num, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spike_shape object from desired spike. Calls trace/getSpike method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>spikes</TT>:</STRONG></DT>
<DD>(Optional) A spikes object obtained from trace, 
<P>
calculated automatically if given as [].
</DD>
<DT><STRONG><TT>spike_num</TT>:</STRONG></DT>
<DD>The index of spike to extract.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to getSpike.
</DD>
</DL></DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Get 2nd pulse spike and plot it:
<BR>&#187; plotFigure(plotResults(getPulseSpike(t, [], 2)))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15587"></A>, <TT>trace/getSpike</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__getSpike"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15589"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012300000000000000">
1.2.3 Method <TT>cip_trace/plotData</TT></A>
</H3><A NAME="15591"></A><A NAME="ref_cip_trace__plotData"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a trace by calling trace/plotData but also adds optionaldecorations.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotData(t, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If t is a vector of traces, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>stimBar</TT>:</STRONG></DT>
<DD>If true, put a bar indicating the CIP duration.
<P>
(rest passed to trace/plotData)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15593"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15595"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15597"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012400000000000000">
1.2.4 Method <TT>cip_trace/get</TT></A>
</H3><A NAME="15599"></A><A NAME="ref_cip_trace__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012500000000000000">
1.2.5 Method <TT>cip_trace/periodRecSpont2</TT></A>
</H3><A NAME="15601"></A><A NAME="ref_cip_trace__periodRecSpont2"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the second half of the recovery spontaneous
		 activity period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodRecSpont2(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15603"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15605"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15607"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012600000000000000">
1.2.6 Method <TT>cip_trace/periodRecSpontRestPeriod</TT></A>
</H3><A NAME="15609"></A><A NAME="ref_cip_trace__periodRecSpontRestPeriod"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodRecSpont(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>iniPeriod</TT>:</STRONG></DT>
<DD>the time following pulse offset that is ignored. The rest of
<P>
the time is kept
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15611"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15613"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15615"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;,Tom Sangrey 2006/01/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012700000000000000">
1.2.7 Method <TT>cip_trace/periodPulse</TT></A>
</H3><A NAME="15617"></A><A NAME="ref_cip_trace__periodPulse"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the CIP period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulse(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15619"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15621"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15623"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012800000000000000">
1.2.8 Method <TT>cip_trace/getResults</TT></A>
</H3><A NAME="15625"></A><A NAME="ref_cip_trace__getResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculate test results given a_spikes object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getResults(a_cip_trace, a_spikes)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names with result values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15627"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15629"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15631"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05012900000000000000">
1.2.9 Method <TT>cip_trace/set</TT></A>
</H3><A NAME="15633"></A><A NAME="ref_cip_trace__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121000000000000000">
1.2.10 Method <TT>cip_trace/periodPulseIni100msRest1</TT></A>
</H3><A NAME="15635"></A><A NAME="ref_cip_trace__periodPulseIni100msRest1"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni50msRest1(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15637"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15639"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15641"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121100000000000000">
1.2.11 Method <TT>cip_trace/periodIniSpont</TT></A>
</H3><A NAME="15643"></A><A NAME="ref_cip_trace__periodIniSpont"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the initial spontaneous activity period of 
		cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodIniSpont(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15645"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15647"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15649"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121200000000000000">
1.2.12 Method <TT>cip_trace/periodPulseIni100ms</TT></A>
</H3><A NAME="15651"></A><A NAME="ref_cip_trace__periodPulseIni100ms"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the first 100ms of the CIP period of 
			cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni100ms(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15653"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15655"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15657"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121300000000000000">
1.2.13 Method <TT>cip_trace/getCIPResults</TT></A>
</H3><A NAME="15659"></A><A NAME="ref_cip_trace__getCIPResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculate test results about CIP protocol.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getCIPResults(a_cip_trace, a_spikes)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names with result values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15661"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15663"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15665"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121400000000000000">
1.2.14 Method <TT>cip_trace/periodPulseHalf1</TT></A>
</H3><A NAME="15667"></A><A NAME="ref_cip_trace__periodPulseHalf1"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the first half of the CIP period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseHalf1(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15669"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15671"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15673"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121500000000000000">
1.2.15 Method <TT>cip_trace/periodRecSpont1</TT></A>
</H3><A NAME="15675"></A><A NAME="ref_cip_trace__periodRecSpont1"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the first half of the recovery spontaneous
		 activity period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodRecSpont1(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15677"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15679"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15681"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121600000000000000">
1.2.16 Method <TT>cip_trace/subsref</TT></A>
</H3><A NAME="15683"></A><A NAME="ref_cip_trace__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121700000000000000">
1.2.17 Method <TT>cip_trace/getBurstResults</TT></A>
</H3><A NAME="15685"></A><A NAME="ref_cip_trace__getBurstResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculate test results related to Burst behavior.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getRateResults(a_cip_trace, a_spikes)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names with result values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15687"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15689"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15691"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/30, Tom Sangrey
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121800000000000000">
1.2.18 Method <TT>cip_trace/periodPulseIni50msRest1</TT></A>
</H3><A NAME="15693"></A><A NAME="ref_cip_trace__periodPulseIni50msRest1"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the first half of the rest after the 
			first 50ms of the CIP period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni50msRest1(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15695"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15697"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15699"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121900000000000000">
1.2.19 Method <TT>cip_trace/measureNames</TT></A>
</H3><A NAME="15701"></A><A NAME="ref_cip_trace__measureNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns taxonomy of measurements collected by cip_trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">measures = measureNames(a_cip_trace)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a static method, in the sense that it does not need the object passed as argument.
 Therefore it can be called directly by using the default constructor; e.g., measureNames(cip_trace).
 The measure names are required for merging columns of a database generated by profiling these objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object. It can be created by the the default constructor 'cip_trace'.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
measures: A structure with cell arrays of types of measures, and measure names inside.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>getResults</TT> (p.&nbsp;<A HREF="#ref_getResults"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15703"></A>, <TT>getProfileAllSpikes</TT> (p.&nbsp;<A HREF="#ref_getProfileAllSpikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15705"></A>, <TT>mergeMultipleCIPsInOne</TT> (p.&nbsp;<A HREF="#ref_mergeMultipleCIPsInOne"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15707"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122000000000000000">
1.2.20 Method <TT>cip_trace/display</TT></A>
</H3><A NAME="15709"></A><A NAME="ref_cip_trace__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122100000000000000">
1.2.21 Method <TT>cip_trace/calcRecSpontPotAvg</TT></A>
</H3><A NAME="15711"></A><A NAME="ref_cip_trace__calcRecSpontPotAvg"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the average potential value of the 
			recovery period of the cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">avg_val = calcRecSpontPotAvg(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
avg_val: The avg value [dy].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15713"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15715"></A>, <TT>trace/calcAvg</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__calcAvg"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15717"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122200000000000000">
1.2.22 Method <TT>cip_trace/getRateResults</TT></A>
</H3><A NAME="15719"></A><A NAME="ref_cip_trace__getRateResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculate test results related to spike rate.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getRateResults(a_cip_trace, a_spikes)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names with result values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15721"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15723"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15725"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122300000000000000">
1.2.23 Method <TT>cip_trace/calcPulsePotAvg</TT></A>
</H3><A NAME="15727"></A><A NAME="ref_cip_trace__calcPulsePotAvg"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the average potential value of the 
		CIP period of the cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">avg_val = calcPulsePotAvg(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
avg_val: The avg value [dy].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15729"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15731"></A>, <TT>trace/calcAvg</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__calcAvg"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15733"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122400000000000000">
1.2.24 Method <TT>cip_trace/periodRecSpontIniPeriod</TT></A>
</H3><A NAME="15735"></A><A NAME="ref_cip_trace__periodRecSpontIniPeriod"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodRecSpont(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>iniPeriod</TT>:</STRONG></DT>
<DD>the time following pulse offset that is kept, the rest of
<P>
the time is ignored.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15737"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15739"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15741"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;,Tom Sangrey 2006/01/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122500000000000000">
1.2.25 Method <TT>cip_trace/periodPulseIni100msRest2</TT></A>
</H3><A NAME="15743"></A><A NAME="ref_cip_trace__periodPulseIni100msRest2"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni50msRest2(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15745"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15747"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15749"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122600000000000000">
1.2.26 Method <TT>cip_trace/getRecSpontSpike</TT></A>
</H3><A NAME="15751"></A><A NAME="ref_cip_trace__getRecSpontSpike"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert a spike in the CIP period to a spike_shape object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = getRecSpontSpike(trace, spikes, spike_num, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spike_shape object from desired spike.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>spikes</TT>:</STRONG></DT>
<DD>A spikes object on trace.
</DD>
<DT><STRONG><TT>spike_num</TT>:</STRONG></DT>
<DD>The index of spike to extract.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15753"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122700000000000000">
1.2.27 Method <TT>cip_trace/spikes</TT></A>
</H3><A NAME="15755"></A><A NAME="ref_cip_trace__spikes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert cip_trace to spikes object for spike timing calculations.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = spikes(trace, plotit)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spikes object by finding the spikes in the three 
 separate periods, initial spontaneous activity period, CIP period, and
 final recovery period.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, a plot is generated for showing spikes found
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15757"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15759"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122800000000000000">
1.2.28 Method <TT>cip_trace/periodPulseIni50msRest2</TT></A>
</H3><A NAME="15761"></A><A NAME="ref_cip_trace__periodPulseIni50msRest2"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the second half of the rest after the 
			first 50ms of the CIP period of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni50msRest2(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15763"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15765"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15767"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122900000000000000">
1.2.29 Method <TT>cip_trace/periodRecSpont</TT></A>
</H3><A NAME="15769"></A><A NAME="ref_cip_trace__periodRecSpont"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the recovery spontaneous activity period 
		of cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodRecSpont(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15771"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15773"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15775"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123000000000000000">
1.2.30 Method <TT>cip_trace/calcPulsePotSag</TT></A>
</H3><A NAME="15777"></A><A NAME="ref_cip_trace__calcPulsePotSag"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the minimal sag and sag amount of the CIP period of the cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[min_val, min_idx, sag_val] = calcPulsePotSag(t)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The minimal sag is the minimal potential value of the 
 first half of the CIP period. The sag amount is calculated by 
 comparing this to the steady-state value at the end of the CIP period.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
min_val: The min value [dy].
	min_idx: The index of the min value [dt].
	sag_val: The sag amount [dy].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15779"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15781"></A>, <TT>trace/calcMin</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__calcMin"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15783"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123100000000000000">
1.2.31 Method <TT>cip_trace/periodPulseIni50ms</TT></A>
</H3><A NAME="15785"></A><A NAME="ref_cip_trace__periodPulseIni50ms"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the first 50ms of the CIP period of 
			cip_trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">the_period = periodPulseIni50ms(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: A period object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15787"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15789"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15791"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123200000000000000">
1.2.32 Method <TT>cip_trace/plot_abstract</TT></A>
</H3><A NAME="15793"></A><A NAME="ref_cip_trace__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a trace by calling plotData.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(t, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If t is a vector of traces, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>timeScale</TT>:</STRONG></DT>
<DD>'s' for seconds, or 'ms' for milliseconds.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15795"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15797"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15799"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123300000000000000">
1.2.33 Method <TT>cip_trace/getProfileAllSpikes</TT></A>
</H3><A NAME="15801"></A><A NAME="ref_cip_trace__getProfileAllSpikes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a cip_trace_allspikes_profile object by collecting test results of a cip_trace, analyzing each individual spike.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">profile_obj = getProfileAllSpikes(a_cip_trace)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Analyzes the spontaneous (periodIniSpont), pulse (periodPulse) and the
 recovery (periodRecSpont) periods separately and produces spike shape
 distribution results. Rate and CIP measurements are added to these.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
profile_obj: A cip_trace_allspikes_profile object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15803"></A>, <TT>cip_trace_allspikes_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace_allspikes_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15805"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05013000000000000000">
1.3 Class <TT>cip_trace_allspikes_profile</TT></A>
</H2><A NAME="15807"></A><A NAME="ref_cip_trace_allspikes_profile"></A>
<H3><A NAME="SECTION05013100000000000000">
1.3.1 Constructor <TT>cip_trace_allspikes_profile/cip_trace_allspikes_profile</TT></A>
</H3><A NAME="15809"></A><A NAME="ref_cip_trace_allspikes_profile__cip_trace_allspikes_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates and collects test results of a cip_trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = 
   cip_trace_allspikes_profile(a_cip_trace, a_spikes, a_spont_spike_shape, 
				results, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of results_profile. It is made to be used from 
 subclass constructors.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>spont_spikes_db, pulse_spikes_db, recov_spikes_db</TT>:</STRONG></DT>
<DD><P>
tests_dbs with spontaneous, pulse and recovery period spike info.
</DD>
<DT><STRONG><TT>results_obj</TT>:</STRONG></DT>
<DD>A results_profile object with test results.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
trace, spikes, spont_spikes_db, 
	pulse_spikes_db, recov_spikes_db, props
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15811"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15813"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15815"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05013200000000000000">
1.3.2 Method <TT>cip_trace_allspikes_profile/get</TT></A>
</H3><A NAME="15817"></A><A NAME="ref_cip_trace_allspikes_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05013300000000000000">
1.3.3 Method <TT>cip_trace_allspikes_profile/set</TT></A>
</H3><A NAME="15819"></A><A NAME="ref_cip_trace_allspikes_profile__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05013400000000000000">
1.3.4 Method <TT>cip_trace_allspikes_profile/plotRowSpontSpikeAnal</TT></A>
</H3><A NAME="15821"></A><A NAME="ref_cip_trace_allspikes_profile__plotRowSpontSpikeAnal"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a row of plots that show spontaneous spikes, starting from the whole trace, zooming into the individual spike.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotRowSpontSpikeAnal(prof, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>prof</TT>:</STRONG></DT>
<DD>A cip_trace_allspikes_profile object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15823"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15825"></A>, <TT>spike_shape/plotCompareMethodsSimple</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape__plotCompareMethodsSimple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15827"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15829"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/23
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05013500000000000000">
1.3.5 Method <TT>cip_trace_allspikes_profile/display</TT></A>
</H3><A NAME="15831"></A><A NAME="ref_cip_trace_allspikes_profile__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05014000000000000000">
1.4 Class <TT>cip_trace_profile</TT></A>
</H2><A NAME="15833"></A><A NAME="ref_cip_trace_profile"></A>
<H3><A NAME="SECTION05014100000000000000">
1.4.1 Constructor <TT>cip_trace_profile/cip_trace_profile</TT></A>
</H3><A NAME="15835"></A><A NAME="ref_cip_trace_profile__cip_trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates and collects test results of a cip_trace.
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The first usage is fully customizable to be used from subclass constructors.
 The second usage generates the spikes and spont_spike_shape objects, and
 collects some generic test results from them. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data_src</TT>:</STRONG></DT>
<DD>The trace column OR the filename.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>pulse_time_start, pulse_time_width</TT>:</STRONG></DT>
<DD><P>
Start and width of the pulse [dt]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>See trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
trace, spikes, spont_spike_shape, results, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15837"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15839"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15841"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05014200000000000000">
1.4.2 Method <TT>cip_trace_profile/get</TT></A>
</H3><A NAME="15843"></A><A NAME="ref_cip_trace_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05014300000000000000">
1.4.3 Method <TT>cip_trace_profile/plot</TT></A>
</H3><A NAME="15845"></A><A NAME="ref_cip_trace_profile__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a cip_trace_profile object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plot(t)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Plots contents of this object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A cip_trace_profile object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
h: Plot handle(s).
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05014400000000000000">
1.4.4 Method <TT>cip_trace_profile/set</TT></A>
</H3><A NAME="15847"></A><A NAME="ref_cip_trace_profile__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05014500000000000000">
1.4.5 Method <TT>cip_trace_profile/subsref</TT></A>
</H3><A NAME="15849"></A><A NAME="ref_cip_trace_profile__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05014600000000000000">
1.4.6 Method <TT>cip_trace_profile/display</TT></A>
</H3><A NAME="15851"></A><A NAME="ref_cip_trace_profile__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05015000000000000000">
1.5 Class <TT>cip_traces_dataset</TT></A>
</H2><A NAME="15853"></A><A NAME="ref_cip_traces_dataset"></A>
<H3><A NAME="SECTION05015100000000000000">
1.5.1 Constructor <TT>cip_traces_dataset/cip_traces_dataset</TT></A>
</H3><A NAME="15855"></A><A NAME="ref_cip_traces_dataset__cip_traces_dataset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Dataset of cip_traces objects, each with varying cip magnitudes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = cip_traces_dataset(ts, cipmag, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_fileset.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>ts</TT>:</STRONG></DT>
<DD>A cell array of cip_traces objects.
</DD>
<DT><STRONG><TT>cipmag</TT>:</STRONG></DT>
<DD>A single cip magnitude to trace take from objects.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identification string for the whole dataset.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to cip_trace_profile.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_dataset,
	cipmag, props (see above).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_traces</TT> (p.&nbsp;<A HREF="#ref_cip_traces"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15857"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15859"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15861"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015200000000000000">
1.5.2 Method <TT>cip_traces_dataset/get</TT></A>
</H3><A NAME="15863"></A><A NAME="ref_cip_traces_dataset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015300000000000000">
1.5.3 Method <TT>cip_traces_dataset/set</TT></A>
</H3><A NAME="15865"></A><A NAME="ref_cip_traces_dataset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015400000000000000">
1.5.4 Method <TT>cip_traces_dataset/getItemParams</TT></A>
</H3><A NAME="15867"></A><A NAME="ref_cip_traces_dataset__getItemParams"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">params_row = getParams(dataset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15869"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15871"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15873"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15875"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015500000000000000">
1.5.5 Method <TT>cip_traces_dataset/cip_trace_profile</TT></A>
</H3><A NAME="15877"></A><A NAME="ref_cip_traces_dataset__cip_trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a raw cip_trace_profile given a index 
		      to this dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace_profile = cip_trace_profile(dataset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of file in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace_profile: A cip_trace_profile object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15879"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15881"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015600000000000000">
1.5.6 Method <TT>cip_traces_dataset/subsref</TT></A>
</H3><A NAME="15883"></A><A NAME="ref_cip_traces_dataset__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015700000000000000">
1.5.7 Method <TT>cip_traces_dataset/display</TT></A>
</H3><A NAME="15885"></A><A NAME="ref_cip_traces_dataset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015800000000000000">
1.5.8 Method <TT>cip_traces_dataset/loadItemProfile</TT></A>
</H3><A NAME="15887"></A><A NAME="ref_cip_traces_dataset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a profile object from a raw data item in the dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(dataset, index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclasses should overload this function to load the specific profile
 object they desire. The profile class should define a getResults method
 which is used in the itemResultsRow method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15889"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15891"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15893"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15895"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05015900000000000000">
1.5.9 Method <TT>cip_traces_dataset/paramNames</TT></A>
</H3><A NAME="15897"></A><A NAME="ref_cip_traces_dataset__paramNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the only parameter, 'pAcip,' for this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">param_names = paramNames(fileset)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks at the filename of the first file to find the parameter names.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_names: Cell array with ordered parameter names.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15899"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15901"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15903"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05016000000000000000">
1.6 Class <TT>cip_traceset</TT></A>
</H2><A NAME="15905"></A><A NAME="ref_cip_traceset"></A>
<H3><A NAME="SECTION05016100000000000000">
1.6.1 Constructor <TT>cip_traceset/cip_traceset</TT></A>
</H3><A NAME="15907"></A><A NAME="ref_cip_traceset__cip_traceset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A traceset with varying cip magnitudes from a single cip_traces object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = cip_traceset(ct, cip_mags, dy, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_fileset. This traceset can create a 
 mini-database form a single cip_traces object. The list contains cip_mags.
 cip_traceset_dataset should be used to load multiple cip_traceset objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>ct</TT>:</STRONG></DT>
<DD>A cip_traces object.
</DD>
<DT><STRONG><TT>cip_mags</TT>:</STRONG></DT>
<DD>An array of cip magnitudes to select from the object.
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution, [V] or [A] (default=1e-3).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>offsetPotential</TT>:</STRONG></DT>
<DD>Add this to physiology trace as compensation.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_dataset,
	ct, props (see above).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_traces</TT> (p.&nbsp;<A HREF="#ref_cip_traces"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15909"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15911"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15913"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016200000000000000">
1.6.2 Method <TT>cip_traceset/get</TT></A>
</H3><A NAME="15915"></A><A NAME="ref_cip_traceset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016300000000000000">
1.6.3 Method <TT>cip_traceset/getItemParams</TT></A>
</H3><A NAME="15917"></A><A NAME="ref_cip_traceset__getItemParams"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">params_row = getParams(dataset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
<DT><STRONG><TT>a_profile</TT>:</STRONG></DT>
<DD>A profile object for the item (optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15919"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15921"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15923"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15925"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016400000000000000">
1.6.4 Method <TT>cip_traceset/cip_trace_profile</TT></A>
</H3><A NAME="15927"></A><A NAME="ref_cip_traceset__cip_trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a raw cip_trace_profile given an index in this traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace_profile = cip_trace_profile(traceset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A cip_traceset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace_profile: A cip_trace_profile object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15929"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15931"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016500000000000000">
1.6.5 Method <TT>cip_traceset/display</TT></A>
</H3><A NAME="15933"></A><A NAME="ref_cip_traceset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016600000000000000">
1.6.6 Method <TT>cip_traceset/loadItemProfile</TT></A>
</H3><A NAME="15935"></A><A NAME="ref_cip_traceset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a profile object from a raw data item in the dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(dataset, index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclasses should overload this function to load the specific profile
 object they desire. The profile class should define a getResults method
 which is used in the itemResultsRow method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15937"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15939"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15941"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15943"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05016700000000000000">
1.6.7 Method <TT>cip_traceset/paramNames</TT></A>
</H3><A NAME="15945"></A><A NAME="ref_cip_traceset__paramNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the only parameter, 'pAcip,' for this traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">param_names = paramNames(traceset)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks at the filename of the first file to find the parameter names.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A cip_traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_names: Cell array with ordered parameter names.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15947"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15949"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15951"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05017000000000000000">
1.7 Class <TT>cip_traceset_dataset</TT></A>
</H2><A NAME="15953"></A><A NAME="ref_cip_traceset_dataset"></A>
<H3><A NAME="SECTION05017100000000000000">
1.7.1 Constructor <TT>cip_traceset_dataset/cip_traceset_dataset</TT></A>
</H3><A NAME="15955"></A><A NAME="ref_cip_traceset_dataset__cip_traceset_dataset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Dataset of multiple cip magnitudes from cip_traces objects .
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = cip_traceset_dataset(cts, cip_mags, dy, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_dataset. Designed to extract a trace
 for each cip magnitude from the cip_traceset objects contained. Uses cip_traceset
 objects to extract multiple traces from each cip_traces object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>cts</TT>:</STRONG></DT>
<DD>Array or cell array of cip_traces objects.
</DD>
<DT><STRONG><TT>cip_mags</TT>:</STRONG></DT>
<DD>An array of cip magnitudes to select from each cip_traces object.
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution, [V] or [A] (default = 1e-3).
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to cip_traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_dataset, cip_mags
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_cip_traceset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15957"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15959"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15961"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/28
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05017200000000000000">
1.7.2 Method <TT>cip_traceset_dataset/get</TT></A>
</H3><A NAME="15963"></A><A NAME="ref_cip_traceset_dataset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05017300000000000000">
1.7.3 Method <TT>cip_traceset_dataset/readDBItems</TT></A>
</H3><A NAME="15965"></A><A NAME="ref_cip_traceset_dataset__readDBItems"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Reads all items to generate a params_tests_db object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params, param_names, tests, test_names] = readDBItems(obj)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a specific method to convert from cip_traceset_dataset to
 a params_tests_db, or a subclass. Output of this function can be 
 directly fed to the constructor of a params_tests_db or a subclass.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params, param_names, tests, test_names: See params_tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15967"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15969"></A>, <TT>itemResultsRow
	    testNames</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow______testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15971"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15973"></A>, <TT>physiol_cip_traceset_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15975"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/28
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05017400000000000000">
1.7.4 Method <TT>cip_traceset_dataset/display</TT></A>
</H3><A NAME="15977"></A><A NAME="ref_cip_traceset_dataset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05017500000000000000">
1.7.5 Method <TT>cip_traceset_dataset/loadItemProfile</TT></A>
</H3><A NAME="15979"></A><A NAME="ref_cip_traceset_dataset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace_profile object from a raw data file in the fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(fileset, neuron_id, trace_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>neuron_id </TT>:</STRONG></DT>
<DD>tells which item in fileset (corresponds to cells_filename) to use grab the cell information 
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of file in traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15981"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15983"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15985"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15987"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14 and Tom Sangrey
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05018000000000000000">
1.8 Class <TT>cluster_db</TT></A>
</H2><A NAME="15989"></A><A NAME="ref_cluster_db"></A>
<H3><A NAME="SECTION05018100000000000000">
1.8.1 Constructor <TT>cluster_db/cluster_db</TT></A>
</H3><A NAME="15991"></A><A NAME="ref_cluster_db__cluster_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of cluster centroids generated by a clustering algorithm from a rows of orig_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cluster_db = cluster_db(data, col_names, orig_db, cluster_idx, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Use one of the clustering methods in 
 tests_db, such as kmeansCluster, to get an instance of this class.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>Database contents.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>The column names.
</DD>
<DT><STRONG><TT>orig_db</TT>:</STRONG></DT>
<DD>DB whose rows are clustered.
</DD>
<DT><STRONG><TT>cluster_idx</TT>:</STRONG></DT>
<DD>Array of cluster numbers that correspond to each row in orig_db.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>sumDistances</TT>:</STRONG></DT>
<DD>Total distance of elements within each cluster.
</DD>
<DT><STRONG><TT>distanceMeasure</TT>:</STRONG></DT>
<DD>Measure used to find clusters (Default='correlation')
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, 
	orig_db: original DB from which clusters were obtained, 
	cluster_idx: Array associating rows of orig_db to each cluster here.
	props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15993"></A>, <TT>tests_db/kmeansCluster</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__kmeansCluster"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="15995"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05018200000000000000">
1.8.2 Method <TT>cluster_db/get</TT></A>
</H3><A NAME="15997"></A><A NAME="ref_cluster_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05018300000000000000">
1.8.3 Method <TT>cluster_db/plotHist</TT></A>
</H3><A NAME="15999"></A><A NAME="ref_cluster_db__plotHist"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a histogram plot showing the clustering memberships.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotHist(a_cluster_db, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cluster_db</TT>:</STRONG></DT>
<DD>A cluster_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16001"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16003"></A>, <TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16005"></A>, <TT>histogram_db/plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db__plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16007"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05018400000000000000">
1.8.4 Method <TT>cluster_db/plotQuality</TT></A>
</H3><A NAME="16009"></A><A NAME="ref_cluster_db__plotQuality"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a plot_abstract of the silhouette plot showing the clustering quality.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotQuality(a_cluster_db, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cluster_db</TT>:</STRONG></DT>
<DD>A cluster_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16011"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16013"></A>, <TT>silhouette</TT> (p.&nbsp;<A HREF="#ref_silhouette"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16015"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05018500000000000000">
1.8.5 Method <TT>cluster_db/display</TT></A>
</H3><A NAME="16017"></A><A NAME="ref_cluster_db__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION05018600000000000000">
1.8.6 Method <TT>cluster_db/plot_abstract</TT></A>
</H3><A NAME="16019"></A><A NAME="ref_cluster_db__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a vertical plot_stack of silhouette and membership histograms for the clusters.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(a_cluster_db, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cluster_db</TT>:</STRONG></DT>
<DD>A cluster_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cluster_db/plotQuality</TT> (p.&nbsp;<A HREF="node5.html#ref_cluster_db__plotQuality"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16021"></A>, <TT>cluster_db/plotHist</TT> (p.&nbsp;<A HREF="node5.html#ref_cluster_db__plotHist"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16023"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION05019000000000000000">
1.9 Class <TT>corrcoefs_db</TT></A>
</H2><A NAME="16025"></A><A NAME="ref_corrcoefs_db"></A>
<H3><A NAME="SECTION05019100000000000000">
1.9.1 Constructor <TT>corrcoefs_db/corrcoefs_db</TT></A>
</H3><A NAME="16027"></A><A NAME="ref_corrcoefs_db__corrcoefs_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of correlation coefficients generated from 
		a column of another database.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_coef_db = corrcoefs_db(col_name, coefs, coef_names, pages, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_3d_db. Allows generating a plot, etc.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>col_name</TT>:</STRONG></DT>
<DD>The column with which the others are correlated.
</DD>
<DT><STRONG><TT>coefs</TT>:</STRONG></DT>
<DD>Matrix where each column has another coefficient.
</DD>
<DT><STRONG><TT>coef_names</TT>:</STRONG></DT>
<DD>Cell array of column names corresponding to coefficients.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>Column vector of page indices pointing to the tests_3d_db.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16029"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16031"></A>, <TT>tests_db/histogram</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__histogram"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16033"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050110000000000000000">
1.10 Class <TT>dataset_db_bundle</TT></A>
</H2><A NAME="16035"></A><A NAME="ref_dataset_db_bundle"></A>
<H3><A NAME="SECTION050110100000000000000">
1.10.1 Constructor <TT>dataset_db_bundle/dataset_db_bundle</TT></A>
</H3><A NAME="16037"></A><A NAME="ref_dataset_db_bundle__dataset_db_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>The dataset and the DB created from it bundled together.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_bundle = dataset_db_bundle(a_dataset, a_db, a_joined_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This class is made to enable operations that require seamless connection between 
 the high-level DB and the raw data. The raw DB is only required to bridge the gap 
 between the high-level DB and the dataset. Therefore it only needs to contain 
 columns necessary to make this connection. It is not required to include all 
 raw DB columns, which is inefficient.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset object or a subclass.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>The raw tests_db object (or a subclass) created from the dataset.
</DD>
<DT><STRONG><TT>a_joined_db</TT>:</STRONG></DT>
<DD>The processed DB created from the raw DB.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
dataset, db, joined_db, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16039"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16041"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110200000000000000">
1.10.2 Method <TT>dataset_db_bundle/matchingRow</TT></A>
</H3><A NAME="16043"></A><A NAME="ref_dataset_db_bundle__matchingRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a criterion database for matching the tests of a row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">crit_db = matchingRow(a_bundle, row, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Copies selected test values from row as the first row into the 
 criterion db. Adds a second row for the STD of each column in the db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row index to match.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>distDB</TT>:</STRONG></DT>
<DD>Take the standard deviation from this db instead.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
crit_db: A tests_db with two rows for values and STDs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">physiol_bundle has an overloaded matchingRow method that
<BR>
takes the TracesetIndex as argument:
<BR>&#187; crit_db = matchingRow(pbundle, 61)
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16045"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16047"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16049"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110300000000000000">
1.10.3 Method <TT>dataset_db_bundle/get</TT></A>
</H3><A NAME="16051"></A><A NAME="ref_dataset_db_bundle__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110400000000000000">
1.10.4 Method <TT>dataset_db_bundle/getNeuronRowIndex</TT></A>
</H3><A NAME="16053"></A><A NAME="ref_dataset_db_bundle__getNeuronRowIndex"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the neuron index from bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_row_index = getNeuronRowIndex(a_bundle, an_index, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a polymorphic method. Therefor it is not defined for this class, 
 but see subclasses of dataset_db_bundle for its more meaningful implementations.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle subclass object.
</DD>
<DT><STRONG><TT>an_index</TT>:</STRONG></DT>
<DD>An index number of neuron, or a DB row containing this.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_row_index: A row index of neuron in a_bundle.joined_db.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; displayRows(mbundle.joined_db(getNeuronRowIndex(mbundle, 98364), :))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16055"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110500000000000000">
1.10.5 Method <TT>dataset_db_bundle/ctFromRows</TT></A>
</H3><A NAME="16057"></A><A NAME="ref_dataset_db_bundle__ctFromRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from a raw data file in the a_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = ctFromRows(a_bundle, a_db, cip_levels, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is not implemented for the generic dataset_db_bundle class. See 
 subclass implementations.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB created by the dataset in the a_bundle to read the neuron index numbers from.
</DD>
<DT><STRONG><TT>cip_levels</TT>:</STRONG></DT>
<DD>A column vector of CIP-levels to be loaded.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(passed to a_bundle.dataset/cip_trace)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: One or more cip_trace objects that hold the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>model_ct_bundle/ctFromRows</TT> (p.&nbsp;<A HREF="node5.html#ref_model_ct_bundle__ctFromRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16059"></A>, <TT>physiol_bundle/ctFromRows</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_bundle__ctFromRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16061"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110600000000000000">
1.10.6 Method <TT>dataset_db_bundle/reportNeuron</TT></A>
</H3><A NAME="16063"></A><A NAME="ref_dataset_db_bundle__reportNeuron"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a report of neuron at given an_index of a_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc = reportNeuron(a_bundle, an_index, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a report document with preset layouts of annotated plots of
 the selected neuron. See reportLayout below for presets.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>a dataset_db_bundle object which contains the neuron
</DD>
<DT><STRONG><TT>an_index</TT>:</STRONG></DT>
<DD>The index to pass to ctFromRows method of a_bundle.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>reportLayout</TT>:</STRONG></DT>
<DD>Allows choosing one of predefined report types (strings):
<DL>
<DD></DD>
<DT><STRONG><TT>1</TT>:</STRONG></DT>
<DD>Only +/- 100 pA traces in one plot (default).
<P>
1a/b: Either one of the +/- 100 pA traces in one plot.
</DD>
<DT><STRONG><TT>2</TT>:</STRONG></DT>
<DD>Only +/- 100 pA traces and spike shapes in one horiz. plot.
</DD>
<DT><STRONG><TT>2a</TT>:</STRONG></DT>
<DD>+/- 100 pA traces, f-I curve and spike shapes in one horiz. plot.
</DD>
<DT><STRONG><TT>3</TT>:</STRONG></DT>
<DD>+100 pA raw trace and rate profile stacked vertically.
</DD>
<DT><STRONG><TT>3b</TT>:</STRONG></DT>
<DD>-100 pA raw trace and rate profile stacked vertically.
</DD>
<DT><STRONG><TT>4</TT>:</STRONG></DT>
<DD>Horiz stack of +/- 100 pA raw trace with rate profiles underneath.
</DD>
<DT><STRONG><TT>5</TT>:</STRONG></DT>
<DD>5-piece trace, spike shape, f-I curve, f-t curve quad-plot.
</DD>
</DL></DD>
<DT><STRONG><TT>numTraces</TT>:</STRONG></DT>
<DD>Limit number of traces to show in plot (&gt;=1).
</DD>
<DT><STRONG><TT>traces</TT>:</STRONG></DT>
<DD>List of acceptable traces to load.
</DD>
<DT><STRONG><TT>traceAxisLimits</TT>:</STRONG></DT>
<DD>If given, use these limits for trace plots.
</DD>
<DT><STRONG><TT>rateAxisLimits</TT>:</STRONG></DT>
<DD>If given, use these limits for rate plots.
</DD>
<DT><STRONG><TT>fIAxisLimits</TT>:</STRONG></DT>
<DD>If given, use these limits for fIcurve plots.
</DD>
<DT><STRONG><TT>fIstats</TT>:</STRONG></DT>
<DD>Add a fI-stats plot in addition to the curve.
</DD>
<DT><STRONG><TT>sshapeAxisLimits</TT>:</STRONG></DT>
<DD>If given, use these limits for spike shape plots.
</DD>
<DT><STRONG><TT>sshapeResults</TT>:</STRONG></DT>
<DD>If 1, plot measures on the spike shape (default=1).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_doc: A doc_generate object, or a subclass, that can be plotted, or
		printed as a PS or PDF file.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; printTeXFile(reportNeuron(mbundle, 2222), 'a.tex')
<BR>
or:
<BR>&#187; plotFigure(get(reportNeuron(mbundle, 2222), 'plot'))
<BR>
or if the result is one or many doc_plot objects:
<BR>&#187; plot(reportNeuron(mbundle, 2222:2224, struct('reportLayout', '5')))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_multi</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_multi"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16065"></A>, <TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16067"></A>, <TT>doc_generate/printTeXFile</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate__printTeXFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16069"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110700000000000000">
1.10.7 Method <TT>dataset_db_bundle/set</TT></A>
</H3><A NAME="16071"></A><A NAME="ref_dataset_db_bundle__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110800000000000000">
1.10.8 Method <TT>dataset_db_bundle/constrainedMeasuresPreset</TT></A>
</H3><A NAME="16073"></A><A NAME="ref_dataset_db_bundle__constrainedMeasuresPreset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a dataset_db_bundle with constrained measures according to chosen preset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_bundle test_names] = constrainedMeasuresPreset(a_bundle, preset, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object.
</DD>
<DT><STRONG><TT>preset</TT>:</STRONG></DT>
<DD>Choose preset measure list (default=1).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_bundle: Modified bundle.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_bundle/constrainedMeasuresPreset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_bundle__constrainedMeasuresPreset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16075"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050110900000000000000">
1.10.9 Method <TT>dataset_db_bundle/subsref</TT></A>
</H3><A NAME="16077"></A><A NAME="ref_dataset_db_bundle__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines indexing for tests_db objects for () and . operations. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = obj(rows, tests)
 obj = obj.attribute</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Returns attributes or selects the given test columns and rows
 and returns in a new tests_db object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>A logical or index vector of rows. If ':', all rows.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Cell array of test names or column indices. If ':', all tests.
</DD>
<DT><STRONG><TT>attribute</TT>:</STRONG></DT>
<DD>A tests_db class attribute.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16079"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16081"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501101000000000000000">
1.10.10 Method <TT>dataset_db_bundle/subsasgn</TT></A>
</H3><A NAME="16083"></A><A NAME="ref_dataset_db_bundle__subsasgn"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic index-based assignment for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501101100000000000000">
1.10.11 Method <TT>dataset_db_bundle/plotfICurve</TT></A>
</H3><A NAME="16085"></A><A NAME="ref_dataset_db_bundle__plotfICurve"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a f-I curve doc_plot for neuron at given an_index in a_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotfICurve(a_bundle, trial_num, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object.
</DD>
<DT><STRONG><TT>an_index</TT>:</STRONG></DT>
<DD>An index with which to address the a_bundle.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>shortCaption</TT>:</STRONG></DT>
<DD>This appears in the figure caption.
</DD>
<DT><STRONG><TT>plotMStats</TT>:</STRONG></DT>
<DD>If set, add the a_bundle stats plot.
</DD>
<DT><STRONG><TT>captionToStats</TT>:</STRONG></DT>
<DD>Use this as its legend label. 
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>if given, no title is produced
<P>
(passed to plot_superpose)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_superpose that contains a f-I curve plot.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_p = plotfICurve(r, 1);
<BR>&#187; plotFigure(a_p, 'The f-I curve of best matching model');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16087"></A>, <TT>plot_superpose</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_superpose"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16089"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501101200000000000000">
1.10.12 Method <TT>dataset_db_bundle/display</TT></A>
</H3><A NAME="16091"></A><A NAME="ref_dataset_db_bundle__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501101300000000000000">
1.10.13 Method <TT>dataset_db_bundle/rankingReportTeX</TT></A>
</H3><A NAME="16093"></A><A NAME="ref_dataset_db_bundle__rankingReportTeX"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a report by comparing a_bundle with the given match criteria, crit_db from crit_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = rankingReportTeX(a_bundle, crit_bundle, crit_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a LaTeX document with:
	- (optional) Raw traces compared with some best matches at different distances
	- Values of some top matching a_db rows and match errors in a floating table.
	- colored-plot of measure errors for some top matches.
	- Parameter distributions of 50 best matches as a bar graph.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object that contains the DB to compare rows from.
</DD>
<DT><STRONG><TT>crit_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object that contains the criterion dataset.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>A tests_db object holding the match criterion tests and STDs
<P>
which can be created with matchingRow.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Identification of the criterion db (not needed/used?).
</DD>
<DT><STRONG><TT>num_matches</TT>:</STRONG></DT>
<DD>Number of best matches to display (default=10).
</DD>
<DT><STRONG><TT>rotate</TT>:</STRONG></DT>
<DD>Rotation angle for best matches table (default=90).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX document string.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>displayRowsTeX</TT> (p.&nbsp;<A HREF="#ref_displayRowsTeX"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16095"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050111000000000000000">
1.11 Class <TT>doc_generate</TT></A>
</H2><A NAME="16097"></A><A NAME="ref_doc_generate"></A>
<H3><A NAME="SECTION050111100000000000000">
1.11.1 Constructor <TT>doc_generate/doc_generate</TT></A>
</H3><A NAME="16099"></A><A NAME="ref_doc_generate__doc_generate"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic class to help generate printed or annotated documents with results.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc = doc_generate(text_string, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This constitutes the base class for other doc_ classes. For convenience,
 this class holds a text_string to be printed when the document is generated
 with the printTeXFile option.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>text_string</TT>:</STRONG></DT>
<DD>Contents of this document.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
text, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16101"></A>, <TT>doc_multi</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_multi"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16103"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111200000000000000">
1.11.2 Method <TT>doc_generate/get</TT></A>
</H3><A NAME="16105"></A><A NAME="ref_doc_generate__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111300000000000000">
1.11.3 Method <TT>doc_generate/set</TT></A>
</H3><A NAME="16107"></A><A NAME="ref_doc_generate__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111400000000000000">
1.11.4 Method <TT>doc_generate/printTeXFile</TT></A>
</H3><A NAME="16109"></A><A NAME="ref_doc_generate__printTeXFile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a TeX file with the contents of this document.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">printTeXFile(a_doc, filename, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calls getTeXString to generate the contents. The filename is adjusted with 
 a call to properFilename to generate an acceptable TeX filename. TeX-specific
 should only be added at this point or at getTeXString, because before we want
 the object to be a generic document container.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>filename</TT>:</STRONG></DT>
<DD>To write the TeX string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: A string that contains TeX commands, which upon writing to a file,
	  can be interpreted by the TeX engine to produce a document.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_doc = doc_plot(a_plot, 'Results from cell.', 'Results.', struct, ''); 
<BR>&#187; printTeXFile(a_doc, 'my_doc.tex')
<BR>
then my_doc.tex can be used by including from a valid LaTeX document.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16111"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16113"></A>, <TT>string2File</TT> (p.&nbsp;<A HREF="#ref_string2File"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16115"></A>, <TT>properFilename</TT> (p.&nbsp;<A HREF="#ref_properFilename"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16117"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111500000000000000">
1.11.5 Method <TT>doc_generate/subsref</TT></A>
</H3><A NAME="16119"></A><A NAME="ref_doc_generate__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111600000000000000">
1.11.6 Method <TT>doc_generate/display</TT></A>
</H3><A NAME="16121"></A><A NAME="ref_doc_generate__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050111700000000000000">
1.11.7 Method <TT>doc_generate/getTeXString</TT></A>
</H3><A NAME="16123"></A><A NAME="ref_doc_generate__getTeXString"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the TeX representation for the document.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = getTeXString(a_doc, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is an abstract placeholder for this method. It specifies what this 
 method should do in the subclasses that implement it. This method should
 create all the auxiliary files needed by the document. The generated tex_string
 should be ready to be visualized.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: A string that contains TeX commands, which upon writing to a file,
	  can be interpreted by the TeX engine to produce a document.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">doc_plot has an overloaded getTeXString method:
<BR>&#187; tex_string = getTeXString(a_doc_plot)
<BR>&#187; string2File(tex_string, 'my_doc.tex')
<BR>
then my_doc.tex can be used by including from a valid LaTeX document.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16125"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16127"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050112000000000000000">
1.12 Class <TT>doc_multi</TT></A>
</H2><A NAME="16129"></A><A NAME="ref_doc_multi"></A>
<H3><A NAME="SECTION050112100000000000000">
1.12.1 Constructor <TT>doc_multi/doc_multi</TT></A>
</H3><A NAME="16131"></A><A NAME="ref_doc_multi__doc_multi"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A document that is composed of multiple other doc_generate objects.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc = doc_multi(docs, id, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>docs</TT>:</STRONG></DT>
<DD>A vector of doc_generate objects.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
docs, doc_generate.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; mydoc = doc_multi([doc_plot(a_plot1), doc_plot(a_plot2)], 'Two plots')
<BR>&#187; printTeXFile(mydoc, 'two_plots.tex')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16133"></A>, <TT>getTeXString</TT> (p.&nbsp;<A HREF="#ref_getTeXString"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16135"></A>, <TT>doc_generate/printTeXFile</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate__printTeXFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16137"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050112200000000000000">
1.12.2 Method <TT>doc_multi/get</TT></A>
</H3><A NAME="16139"></A><A NAME="ref_doc_multi__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050112300000000000000">
1.12.3 Method <TT>doc_multi/set</TT></A>
</H3><A NAME="16141"></A><A NAME="ref_doc_multi__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050112400000000000000">
1.12.4 Method <TT>doc_multi/getTeXString</TT></A>
</H3><A NAME="16143"></A><A NAME="ref_doc_multi__getTeXString"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the TeX representation for the document.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = getTeXString(a_doc, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Concatenates TeX representations of doc_generate, or subclass, objects it contains.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: A string that contains TeX commands, which upon writing to a file,
	  can be interpreted by the TeX engine to produce a document.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">doc_plot has an overloaded getTeXString method:
<BR>&#187; tex_string = getTeXString(a_doc_plot)
<BR>&#187; string2File(tex_string, 'my_doc.tex')
<BR>
then my_doc.tex can be used by including from a valid LaTeX document.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16145"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16147"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050113000000000000000">
1.13 Class <TT>doc_plot</TT></A>
</H2><A NAME="16149"></A><A NAME="ref_doc_plot"></A>
<H3><A NAME="SECTION050113100000000000000">
1.13.1 Constructor <TT>doc_plot/doc_plot</TT></A>
</H3><A NAME="16151"></A><A NAME="ref_doc_plot__doc_plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a formatted plot for printing, annotated with captions.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc = doc_plot(a_plot, caption, plot_filename, float_props, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The generated file may take an extension according to chosen format.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract ready to be visualized.
</DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Long caption to appear under the figure.
</DD>
<DT><STRONG><TT>plot_filename</TT>:</STRONG></DT>
<DD>Filename to be generated from the plot.
</DD>
<DT><STRONG><TT>float_props</TT>:</STRONG></DT>
<DD>Formatting instructions passed to TeXtable. 
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>orient</TT>:</STRONG></DT>
<DD>Passed to the orient command before printing to figure file.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot, caption, plot_filename, float_props, doc_generate.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_doc = doc_plot(plotData(my_cip_trace), 'My CIP trace. Very interesting.', ...
<BR>'trace1', struct, 'first doc');
<BR>&#187; printTeXFile(a_doc, 'my_doc.tex'); </div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16153"></A>, <TT>TeXtable</TT> (p.&nbsp;<A HREF="#ref_TeXtable"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16155"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050113200000000000000">
1.13.2 Method <TT>doc_plot/get</TT></A>
</H3><A NAME="16157"></A><A NAME="ref_doc_plot__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050113300000000000000">
1.13.3 Method <TT>doc_plot/plot</TT></A>
</H3><A NAME="16159"></A><A NAME="ref_doc_plot__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Default plot method to preview the contained plot in a new figure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">figure_handle = plot(a_doc, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Only generate the contained plot for previewing. Opens a new figure.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A doc_plot object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
figure_handle: Handle of newly opened figure.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; figure_handle = plot(a_doc_plot)
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16161"></A>, <TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16163"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16165"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050113400000000000000">
1.13.4 Method <TT>doc_plot/set</TT></A>
</H3><A NAME="16167"></A><A NAME="ref_doc_plot__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050113500000000000000">
1.13.5 Method <TT>doc_plot/plot_abstract</TT></A>
</H3><A NAME="16169"></A><A NAME="ref_doc_plot__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the plot_abstract object within the doc_plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(a_doc, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If a_doc is a vector, returns a vector of plot_abstract objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A doc_plot object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to replace plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object or vector that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_plot/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16171"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16173"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050113600000000000000">
1.13.6 Method <TT>doc_plot/getTeXString</TT></A>
</H3><A NAME="16175"></A><A NAME="ref_doc_plot__getTeXString"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the TeX representation for the plot document.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = getTeXString(a_doc, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Plots, prints EPS files and generates the necessary LaTeX code.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_doc</TT>:</STRONG></DT>
<DD>A doc_plot object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: A string that contains TeX commands, which upon writing to a file,
	  can be interpreted by the TeX engine to produce a document.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">doc_plot has an overloaded getTeXString method:
<BR>&#187; tex_string = getTeXString(a_doc_plot)
<BR>&#187; string2File(tex_string, 'my_doc.tex')
<BR>
then my_doc.tex can be used by including from a valid LaTeX document.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16177"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16179"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050114000000000000000">
1.14 Class <TT>histogram_db</TT></A>
</H2><A NAME="16181"></A><A NAME="ref_histogram_db"></A>
<H3><A NAME="SECTION050114100000000000000">
1.14.1 Constructor <TT>histogram_db/histogram_db</TT></A>
</H3><A NAME="16183"></A><A NAME="ref_histogram_db__histogram_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of histogram values generated for a column of another database.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_hist_db = histogram_db(col_name, bins, hist_results, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Allows generating a histogram plot,
 etc. The histogram count is entered as a column named histVal.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>col_name</TT>:</STRONG></DT>
<DD>The column name of the histogrammed value.
</DD>
<DT><STRONG><TT>bins</TT>:</STRONG></DT>
<DD>The values for which the histogram values are calculated.
</DD>
<DT><STRONG><TT>hist_results</TT>:</STRONG></DT>
<DD>A column vector of histogram values.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, props.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; [hist_results, bins] = hist(my_data);
<BR>&#187; a_hist_db = histogram_db('firing_rate', bins, hist_results, 'rate histogram db');
<BR>&#187; plot(a_hist_db);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16185"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16187"></A>, <TT>tests_db/histogram</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__histogram"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16189"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114200000000000000">
1.14.2 Method <TT>histogram_db/plotEqSpaced</TT></A>
</H3><A NAME="16191"></A><A NAME="ref_histogram_db__plotEqSpaced"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a histogram plot where the values are equally spaced on the x-axis. For use with non-linear parameter values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotEqSpaced(a_hist_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a plot_simple object from this histogram.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_hist_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Optional title string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties passed to plot_abstract.
<DL>
<DD></DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_abstract or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16193"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16195"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114300000000000000">
1.14.3 Method <TT>histogram_db/get</TT></A>
</H3><A NAME="16197"></A><A NAME="ref_histogram_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114400000000000000">
1.14.4 Method <TT>histogram_db/calcMode</TT></A>
</H3><A NAME="16199"></A><A NAME="ref_histogram_db__calcMode"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds the mode of the distribution, that is, the bin with the highest value.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[mode_val, mode_mag] = calcMode(a_hist_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_hist_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
mode_val: The center of the bin that has most members.
	mode_mag: The value of the histogram bin.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16201"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114500000000000000">
1.14.5 Method <TT>histogram_db/plotRowMatrix</TT></A>
</H3><A NAME="16203"></A><A NAME="ref_histogram_db__plotRowMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a subplot matrix of measure columns versus rows of databases. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotRowMatrix(hist_dbs, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Each row in the hist_dbs is assumed to come from a different DB. Columns represent histograms 
 of different measurements. The plot is made such that histograms in each row have the same
 maximal count, and histograms in each column have the same axis limits.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>hist_dbs</TT>:</STRONG></DT>
<DD>A matrix of histogram_db objects.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Title to go at the top.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>rowLabels</TT>:</STRONG></DT>
<DD>Cell array of y-axis labels for each row.
<P>
(rest passed to histogram_db/plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_abstract or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16205"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114600000000000000">
1.14.6 Method <TT>histogram_db/subsref</TT></A>
</H3><A NAME="16207"></A><A NAME="ref_histogram_db__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114700000000000000">
1.14.7 Method <TT>histogram_db/plotPages</TT></A>
</H3><A NAME="16209"></A><A NAME="ref_histogram_db__plotPages"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a plot containing subplots from each page of histograms.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotPages(a_hist_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>For each page of the histogram, a histogram is placed in a subplot.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_hist_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_stack.
<DL>
<DD></DD>
<DT><STRONG><TT>an_orient</TT>:</STRONG></DT>
<DD>Stack orientation. One of 'x', or 'y' (Default='y'; vertical).
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, only display given title_str.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_abstract or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plotPages</TT> (p.&nbsp;<A HREF="#ref_plotPages"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16211"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16213"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050114800000000000000">
1.14.8 Method <TT>histogram_db/plot_abstract</TT></A>
</H3><A NAME="16215"></A><A NAME="ref_histogram_db__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a bar plot from the histogram.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(a_hist_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a plot_simple object from this histogram.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_hist_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties passed to plot_abstract.
<DL>
<DD></DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plot command (Optional, default='bar')
</DD>
<DT><STRONG><TT>lineSpec</TT>:</STRONG></DT>
<DD>Line specification passed to bar command.
</DD>
<DT><STRONG><TT>logScale</TT>:</STRONG></DT>
<DD>If 1, use logarithmic y-scale.
</DD>
<DT><STRONG><TT>shading</TT>:</STRONG></DT>
<DD>'faceted' (default) or 'flat'.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_abstract or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16217"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16219"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050115000000000000000">
1.15 Class <TT>model_ct_bundle</TT></A>
</H2><A NAME="16221"></A><A NAME="ref_model_ct_bundle"></A>
<H3><A NAME="SECTION050115100000000000000">
1.15.1 Constructor <TT>model_ct_bundle/model_ct_bundle</TT></A>
</H3><A NAME="16223"></A><A NAME="ref_model_ct_bundle__model_ct_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>The model cip_trace dataset and the DB created from it bundled together.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_bundle = model_ct_bundle(a_dataset, a_db, a_joined_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of dataset_db_bundle, specialized for model datasets. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>A params_cip_trace_fileset object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>The raw params_tests_db object created from the dataset. It only needs
<P>
to have the pAcip, trial, and ItemIndex columns.
</DD>
<DT><STRONG><TT>a_joined_db</TT>:</STRONG></DT>
<DD>The one-model-per-line DB created from the raw DB.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
dataset_db_bundle.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_joined_db = mergeMultipleCIPsInOne(a_db, ...)
<BR>&#187; a_bundle = model_ct_bundle(a_params_cip_trace_fileset, a_db, a_joined_db)
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16225"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16227"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16229"></A>, <TT>params_tests_db/mergeMultipleCIPsInOne</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db__mergeMultipleCIPsInOne"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16231"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115200000000000000">
1.15.2 Method <TT>model_ct_bundle/reportRankingToPhysiolNeuronsTeXFile</TT></A>
</H3><A NAME="16233"></A><A NAME="ref_model_ct_bundle__reportRankingToPhysiolNeuronsTeXFile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Compare model DB to given physiol criterion and create a report.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_filename = reportRankingToPhysiolNeuronsTeXFile(m_bundle, p_bundle, a_crit_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>A LaTeX report is generated 
 following the example in physiol_bundle/matchingRow. The filename contains the neuron
 name, followed by the traceset index as an identifier of pharmacological applications,
 as in gpd0421c_s34. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>m_bundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>p_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>a_crit_db</TT>:</STRONG></DT>
<DD>The criterion neuron chosen with a matchingRow method.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>filenameSuffix</TT>:</STRONG></DT>
<DD>Append this identifier to the TeX filename.
<P>
(others passed to rankMatching)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns: </STRONG></DT>
<DD>&nbsp;

<P>
tex_filename: Name of LaTeX file generated.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/rankMatching</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16235"></A>, <TT>physiol_cip_traceset/cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset__cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16237"></A>, <TT>physiol_bundle/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_bundle__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16239"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115300000000000000">
1.15.3 Method <TT>model_ct_bundle/get</TT></A>
</H3><A NAME="16241"></A><A NAME="ref_model_ct_bundle__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115400000000000000">
1.15.4 Method <TT>model_ct_bundle/plotComparefICurve</TT></A>
</H3><A NAME="16243"></A><A NAME="ref_model_ct_bundle__plotComparefICurve"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a f-I curve doc_plot comparing m_trial and to_index.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotComparefICurve(m_bundle, m_trial, to_bundle, to_index, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Note that this is not a general method. to_bundle should have been able to accept
 any type of bundle. Most probably this method is redundant and deprecated.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>m_bundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>m_trial</TT>:</STRONG></DT>
<DD>Trial number of model.
</DD>
<DT><STRONG><TT>to_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>to_index</TT>:</STRONG></DT>
<DD>TracesetIndex of neuron.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>shortCaption</TT>:</STRONG></DT>
<DD>This appears in the figure caption.
</DD>
<DT><STRONG><TT>plotMStats</TT>:</STRONG></DT>
<DD>If set, add the m_bundle stats plot.
</DD>
<DT><STRONG><TT>plotToStats</TT>:</STRONG></DT>
<DD>If set, add the to_bundle stats plot.
</DD>
<DT><STRONG><TT>captionToStats</TT>:</STRONG></DT>
<DD>Use this as its legend label. 
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>if given, no title is produced
<P>
(passed to plot_superpose)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_superpose that contains a f-I curve plot.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_p = plotComparefICurve(r, 1);
<BR>&#187; plotFigure(a_p, 'The f-I curve of best matching model');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16245"></A>, <TT>plot_superpose</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_superpose"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16247"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115500000000000000">
1.15.5 Method <TT>model_ct_bundle/getNeuronRowIndex</TT></A>
</H3><A NAME="16249"></A><A NAME="ref_model_ct_bundle__getNeuronRowIndex"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the neuron index from bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_row_index = getNeuronRowIndex(a_bundle, trial_num, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>trial_num</TT>:</STRONG></DT>
<DD>The trial number of model neuron, or a DB row containing this.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_row_index: A row index of neuron in a_bundle.joined_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16251"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115600000000000000">
1.15.6 Method <TT>model_ct_bundle/ctFromRows</TT></A>
</H3><A NAME="16253"></A><A NAME="ref_model_ct_bundle__ctFromRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from a raw data file in the a_mbundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = ctFromRows(a_mbundle, a_db|trials, cip_levels, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is an overloaded method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_mbundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB created by the dataset in the a_mbundle to read the trial numbers from.
</DD>
<DT><STRONG><TT>trials</TT>:</STRONG></DT>
<DD>A column vector with trial numbers.
</DD>
<DT><STRONG><TT>cip_levels</TT>:</STRONG></DT>
<DD>A column vector of CIP-levels to be loaded.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(passed to a_mbundle.dataset/cip_trace)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: One or more cip_trace objects that hold the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle/ctFromRows</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle__ctFromRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16255"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/07/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115700000000000000">
1.15.7 Method <TT>model_ct_bundle/getNeuronLabel</TT></A>
</H3><A NAME="16257"></A><A NAME="ref_model_ct_bundle__getNeuronLabel"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Constructs the neuron label from bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_label = getNeuronLabel(a_bundle, trial_num, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>trial_num</TT>:</STRONG></DT>
<DD>The trial number of model neuron.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_label: A string label identifying selected neuron in bundle.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16259"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115800000000000000">
1.15.8 Method <TT>model_ct_bundle/addToDB</TT></A>
</H3><A NAME="16261"></A><A NAME="ref_model_ct_bundle__addToDB"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Concatenate to existing DB in the bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_mbundle = addToDB(a_mbundle, a_raw_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If joinedDb is not given in props, calls treatSimDB to get the joined_db from this raw DB. 
 Then concats to both db and joined_db in bundle.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_mbundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>a_crit_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle having a crit_db as its joined_db.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>joinedDb</TT>:</STRONG></DT>
<DD>The joined version of a_raw_db.
</DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>If given, this one is used to replace the fileset in the bundle.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_mbundle: a model_ct_bundle object containing the added DB.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; mbundle = addToDB(mbundle, params_tests_db(mfileset, [19684:59956]))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_fileset/addFiles</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset__addFiles"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16263"></A>, <TT>multi_fileset_gpsim_cns2005/addFileDir</TT> (p.&nbsp;<A HREF="#ref_multi_fileset_gpsim_cns2005__addFileDir"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16265"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050115900000000000000">
1.15.9 Method <TT>model_ct_bundle/set</TT></A>
</H3><A NAME="16267"></A><A NAME="ref_model_ct_bundle__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501151000000000000000">
1.15.10 Method <TT>model_ct_bundle/reportCompareModelToPhysiolNeuron</TT></A>
</H3><A NAME="16269"></A><A NAME="ref_model_ct_bundle__reportCompareModelToPhysiolNeuron"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a report by comparing given model neuron to given physiol neuron.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc_multi = reportCompareModelToPhysiolNeuron(m_bundle, trial_num, p_bundle, 
						  traceset_index, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a report document with:
	- Figure displaying raw traces of the physiol neuron compared with the model neuron
	- Figure comparing f-I curves of the two neurons.
	- Figure comparing spont and pulse spike shapes of the two neurons.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>m_bundle, p_bundle</TT>:</STRONG></DT>
<DD>dataset_db_bundle objects of the model and physiology neurons.
</DD>
<DT><STRONG><TT>trial_num</TT>:</STRONG></DT>
<DD>Trial number of desired model neuron in m_bundle.
</DD>
<DT><STRONG><TT>traceset_index</TT>:</STRONG></DT>
<DD>TracesetIndex of desired neuron in p_bundle.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>horizRow</TT>:</STRONG></DT>
<DD>If defined, create a row-figure with all plots.
</DD>
<DT><STRONG><TT>numPhysTraces</TT>:</STRONG></DT>
<DD>Number of physiology traces to show in plot (&gt;=1).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_doc_multi: A doc_multi object that can be printed as a PS or PDF file.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; printTeXFile(reportCompareModelToPhysiolNeuron(mbundle, 2222, pbundle, 34), 'a.tex')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_multi</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_multi"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16271"></A>, <TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16273"></A>, <TT>doc_generate/printTeXFile</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate__printTeXFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16275"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501151100000000000000">
1.15.11 Method <TT>model_ct_bundle/rankMatching</TT></A>
</H3><A NAME="16277"></A><A NAME="ref_model_ct_bundle__rankMatching"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a ranked_db from given criterion db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ranked_db = rankMatching(a_mbundle, a_crit_db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_mbundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>a_crit_db</TT>:</STRONG></DT>
<DD>A crit_db created by a matchingRow method.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(passed to tests_db/rankMatching)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_ranked_db: a ranked_db object containing the rankings.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/rankMatching</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16279"></A>, <TT>ranked_db</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16281"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501151200000000000000">
1.15.12 Method <TT>model_ct_bundle/collectPhysiolMatches</TT></A>
</H3><A NAME="16283"></A><A NAME="ref_model_ct_bundle__collectPhysiolMatches"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Compare model DB to given physiol criteria and return some top matches.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">row_index = collectPhysiolMatches(a_mbundle, a_crit_bundle, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_mbundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>a_crit_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object that holds the criterion neuron.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>showTopmost</TT>:</STRONG></DT>
<DD>Number of top matching models to return (default=50)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns: </STRONG></DT>
<DD>&nbsp;

<P>
row_index: Row indices of best matching models.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501151300000000000000">
1.15.13 Method <TT>model_ct_bundle/getTrialNum</TT></A>
</H3><A NAME="16285"></A><A NAME="ref_model_ct_bundle__getTrialNum"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Extracts identifying neuron trial number from DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">trial_num = getTrialNum(a_bundle, a_db|trial_num, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>DB rows representing deisred model neuron(s).
</DD>
<DT><STRONG><TT>trial_num</TT>:</STRONG></DT>
<DD>Trial numbers. If specified, this function does nothing but return them.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
trial_num: The trial number(s) identifying selected neuron(s) in bundle.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16287"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501151400000000000000">
1.15.14 Method <TT>model_ct_bundle/plotCompareRanks</TT></A>
</H3><A NAME="16289"></A><A NAME="ref_model_ct_bundle__plotCompareRanks"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a plots of given ranks from the ranked_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">plots = plotCompareRanks(m_bundle, p_bundle, a_ranked_db, ranks, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>m_bundle</TT>:</STRONG></DT>
<DD>A model_ct_bundle object.
</DD>
<DT><STRONG><TT>p_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object that originated the criterion.
</DD>
<DT><STRONG><TT>a_ranked_db</TT>:</STRONG></DT>
<DD>A ranked_db generated from ranking m_bundle.
</DD>
<DT><STRONG><TT>ranks</TT>:</STRONG></DT>
<DD>Vector of rank indices for which to generate the plots.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
plots: A structure that contains the joined_db, and the plot vectors 
	  trace_d100_plots and trace_h100_plots.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plots = plotCompareRanks(r, 1:10);
<BR>&#187; plotFigure(plots.trace_d100_plots(1), 'The best matching +100 pA CIP trace');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050116000000000000000">
1.16 Class <TT>model_ranked_to_physiol_bundle</TT></A>
</H2><A NAME="16291"></A><A NAME="ref_model_ranked_to_physiol_bundle"></A>
<H3><A NAME="SECTION050116100000000000000">
1.16.1 Constructor <TT>model_ranked_to_physiol_bundle/model_ranked_to_physiol_bundle</TT></A>
</H3><A NAME="16293"></A><A NAME="ref_model_ranked_to_physiol_bundle__model_ranked_to_physiol_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A DB bundled with its dataset, ranked to a physiology DB bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">r_bundle = model_ranked_to_physiol_bundle(a_dataset, a_db, a_ranked_db, a_crit_bundle, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of model_ct_bundle, specialized for model datasets. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>A params_cip_trace_fileset object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>The raw params_tests_db object created from the dataset. It only needs
<P>
to have the pAcip, trial, and ItemIndex columns.
</DD>
<DT><STRONG><TT>a_ranked_db</TT>:</STRONG></DT>
<DD>The one-model-per-line DB created from the raw DB.
</DD>
<DT><STRONG><TT>a_crit_bundle</TT>:</STRONG></DT>
<DD>The bundle object associated with crit_db that caused the ranking in a_ranked_db.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
crit_bundle, model_ct_bundle.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>model_ct_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_model_ct_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16295"></A>, <TT>ranked_db</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16297"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16299"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050116200000000000000">
1.16.2 Method <TT>model_ranked_to_physiol_bundle/plotfICurve</TT></A>
</H3><A NAME="16301"></A><A NAME="ref_model_ranked_to_physiol_bundle__plotfICurve"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc = docfICurve(r_bundle, crit_bundle, crit_db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>r_bundle</TT>:</STRONG></DT>
<DD>A ranked_bundle object.
</DD>
<DT><STRONG><TT>rank_num</TT>:</STRONG></DT>
<DD>Rank index for which to generate the a_doc.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_doc: A doc_plot that contains a f-I curve plot and associated captions.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_d = docfICurve(r, 1);
<BR>&#187; plot(a_d, 'The f-I curve of best matching model');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>doc_generate</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_generate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16303"></A>, <TT>doc_plot</TT> (p.&nbsp;<A HREF="node5.html#ref_doc_plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16305"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050116300000000000000">
1.16.3 Method <TT>model_ranked_to_physiol_bundle/plotCompareRanks</TT></A>
</H3><A NAME="16307"></A><A NAME="ref_model_ranked_to_physiol_bundle__plotCompareRanks"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>OBSOLETE - Generates a plots of given ranks from the ranked_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">plots = plotCompareRanks(r_bundle, crit_bundle, crit_db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>r_bundle</TT>:</STRONG></DT>
<DD>A ranked_bundle object.
</DD>
<DT><STRONG><TT>ranks</TT>:</STRONG></DT>
<DD>Vector of rank indices for which to generate the plots.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
plots: A structure that contains the joined_db, and the plot vectors 
	  trace_d100_plots and trace_h100_plots.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plots = plotCompareRanks(r, 1:10);
<BR>&#187; plotFigure(plots.trace_d100_plots(1), 'The best matching +100 pA CIP trace');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050116400000000000000">
1.16.4 Method <TT>model_ranked_to_physiol_bundle/comparisonReport</TT></A>
</H3><A NAME="16309"></A><A NAME="ref_model_ranked_to_physiol_bundle__comparisonReport"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>OBSOLETE - Generates a report by comparing r_bundle with the given match criteria, crit_db from crit_bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_doc_multi = comparisonReport(r_bundle, crit_bundle, crit_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a LaTeX document with:
	- (optional) Raw traces compared with some best matches at different distances
	- Values of some top matching a_db rows and match errors in a floating table.
	- colored-plot of measure errors for some top matches.
	- Parameter distributions of 50 best matches as a bar graph.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>r_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object that contains the DB to compare rows from.
</DD>
<DT><STRONG><TT>crit_bundle</TT>:</STRONG></DT>
<DD>A dataset_db_bundle object that contains the criterion dataset.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>A tests_db object holding the match criterion tests and STDs
<P>
which can be created with matchingRow.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Identification of the criterion db (not needed/used?).
</DD>
<DT><STRONG><TT>num_matches</TT>:</STRONG></DT>
<DD>Number of best matches to display (default=10).
</DD>
<DT><STRONG><TT>rotate</TT>:</STRONG></DT>
<DD>Rotation angle for best matches table (default=90).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX document string.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>displayRowsTeX</TT> (p.&nbsp;<A HREF="#ref_displayRowsTeX"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16311"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050117000000000000000">
1.17 Class <TT>params_cip_trace_fileset</TT></A>
</H2><A NAME="16313"></A><A NAME="ref_params_cip_trace_fileset"></A>
<H3><A NAME="SECTION050117100000000000000">
1.17.1 Constructor <TT>params_cip_trace_fileset/params_cip_trace_fileset</TT></A>
</H3><A NAME="16315"></A><A NAME="ref_params_cip_trace_fileset__params_cip_trace_fileset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Description of a raw dataset consisting of cip_trace files varying 
	with parameter values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = params_cip_trace_fileset(file_pattern, dt, dy, 
				 pulse_time_start, pulse_time_width, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_fileset.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>file_pattern</TT>:</STRONG></DT>
<DD>File pattern mathing all files to be loaded.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>pulse_time_start, pulse_time_width</TT>:</STRONG></DT>
<DD><P>
Start and width of the pulse [dt]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identification string
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>profile_class_name</TT>:</STRONG></DT>
<DD>Use this profile class (Default: 'cip_trace_profile').
<P>
(All other props are passed to cip_trace objects)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_fileset,
	pulse_time_start, pulse_time_width.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; fileset = params_cip_trace_fileset('/home/abc/data/*.bin', 1e-4, 1e-3, 20001, 10000, 'sim dataset gpsc0501', struct('trace_time_start', 10001, 'type', 'sim', 'scale_y', 1e3))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16317"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16319"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117200000000000000">
1.17.2 Method <TT>params_cip_trace_fileset/get</TT></A>
</H3><A NAME="16321"></A><A NAME="ref_params_cip_trace_fileset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117300000000000000">
1.17.3 Method <TT>params_cip_trace_fileset/ctFromRows</TT></A>
</H3><A NAME="16323"></A><A NAME="ref_params_cip_trace_fileset__ctFromRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from raw data files in the fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = ctFromRows(m_fileset, m_dball, a_db|itemIndices, cip_levels, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>m_fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>m_dball</TT>:</STRONG></DT>
<DD>A DB created by this fileset that contains the trial, pAcip, and ItemIndex cols.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB that has one trial for each cip_trace to be loaded.
</DD>
<DT><STRONG><TT>itemIndices</TT>:</STRONG></DT>
<DD>A column vector with ItemIndex numbers.
</DD>
<DT><STRONG><TT>cip_levels</TT>:</STRONG></DT>
<DD>A column vector of CIP-levels to be loaded.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>neuronLabel</TT>:</STRONG></DT>
<DD>appropriate unique neuron label generated by the bundle.
<P>
(passed to params_cip_trace_fileset/cip_trace)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: One or more cip_trace objects that hold the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16325"></A>, <TT>physiol_cip_traceset/cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset__cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16327"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/07/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117400000000000000">
1.17.4 Method <TT>params_cip_trace_fileset/set</TT></A>
</H3><A NAME="16329"></A><A NAME="ref_params_cip_trace_fileset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117500000000000000">
1.17.5 Method <TT>params_cip_trace_fileset/cip_trace_profile</TT></A>
</H3><A NAME="16331"></A><A NAME="ref_params_cip_trace_fileset__cip_trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a raw cip_trace_profile given a file_index 
		      to this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace_profile = cip_trace_profile(fileset, file_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>Index of file in fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace_profile: A cip_trace_profile object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16333"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16335"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117600000000000000">
1.17.6 Method <TT>params_cip_trace_fileset/display</TT></A>
</H3><A NAME="16337"></A><A NAME="ref_params_cip_trace_fileset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117700000000000000">
1.17.7 Method <TT>params_cip_trace_fileset/cip_trace</TT></A>
</H3><A NAME="16339"></A><A NAME="ref_params_cip_trace_fileset__cip_trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads raw cip_traces for each given file_index in this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = cip_trace(fileset, file_index|a_db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>A single or array of indices of files in fileset.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB created by this fileset to read the item indices from.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>neuronLabel</TT>:</STRONG></DT>
<DD>Used for annotation purposes.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: A cip_trace object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16341"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16343"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050117800000000000000">
1.17.8 Method <TT>params_cip_trace_fileset/loadItemProfile</TT></A>
</H3><A NAME="16345"></A><A NAME="ref_params_cip_trace_fileset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace_profile object from a raw data file in the fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params_row, tests_row] = loadItemProfile(fileset, file_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>Index of file in fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16347"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16349"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16351"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16353"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050118000000000000000">
1.18 Class <TT>params_tests_dataset</TT></A>
</H2><A NAME="16355"></A><A NAME="ref_params_tests_dataset"></A>
<H3><A NAME="SECTION050118100000000000000">
1.18.1 Constructor <TT>params_tests_dataset/params_tests_dataset</TT></A>
</H3><A NAME="16357"></A><A NAME="ref_params_tests_dataset__params_tests_dataset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Contains a set of data objects or files of raw data varying with parameter values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = params_tests_dataset(list, dt, dy, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is an abstract base class for keeping dataset information separate
 from the parameters-results database (params_tests_db). The list contents
 can be filenames or objects (such as cip_traces) from which to get the raw data.
 The dataset should have all the necessary information to create a db when
 needed. This is an abstract class, thet it it cannot act on its own. Only 
 fully implemented subclasses can actually hold datasets. See methods below.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>list</TT>:</STRONG></DT>
<DD>Array of dataset items (filenames, objects, etc.).
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [integral V, A, etc.]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>type</TT>:</STRONG></DT>
<DD>type of file (default = '')
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
list, dt, dy, id, props (see above).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16359"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16361"></A>, <TT>cip_traces_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_traces_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16363"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118200000000000000">
1.18.2 Method <TT>params_tests_dataset/get</TT></A>
</H3><A NAME="16365"></A><A NAME="ref_params_tests_dataset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118300000000000000">
1.18.3 Method <TT>params_tests_dataset/addItem</TT></A>
</H3><A NAME="16367"></A><A NAME="ref_params_tests_dataset__addItem"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the new dataset with the added item.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">dataset = addItem(dataset, item)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Note that, this is NOT the way to create a dataset. It is only intended for 
 small additions to an existing dataset. This method is too slow
 for creating large datasets. The normal method for creating datasets is
 providing the full list of items to the class constructor.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>item</TT>:</STRONG></DT>
<DD>New item to add in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
dataset: With the added item.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16369"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16371"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16373"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16375"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118400000000000000">
1.18.4 Method <TT>params_tests_dataset/readDBItems</TT></A>
</H3><A NAME="16377"></A><A NAME="ref_params_tests_dataset__readDBItems"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Reads all items to generate a params_tests_db object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params, param_names, tests, test_names] = readDBItems(obj, items)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a generic method to convert from params_tests_fileset to
 a params_tests_db, or a subclass. This method depends on the  
 paramNames, testNames, and itemResultsRow functions. 
 Outputs of this function can be directly fed to the constructor of
 a params_tests_db or a subclass.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A params_tests_fileset object.
</DD>
<DT><STRONG><TT>items</TT>:</STRONG></DT>
<DD>(Optional) List of item indices to use to create the db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params, param_names, tests, test_names: See params_tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16379"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16381"></A>, <TT>itemResultsRow
	    testNames</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow______testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16383"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16385"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118500000000000000">
1.18.5 Method <TT>params_tests_dataset/set</TT></A>
</H3><A NAME="16387"></A><A NAME="ref_params_tests_dataset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118600000000000000">
1.18.6 Method <TT>params_tests_dataset/getItemParams</TT></A>
</H3><A NAME="16389"></A><A NAME="ref_params_tests_dataset__getItemParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Get the parameter values of a dataset item.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">params_row = getItemParams(dataset, index, a_profile)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method can retrieve the item parameters by using either the 
 dataset and the index to find the item or simply by using
 the item profile, a_profile.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
<DT><STRONG><TT>a_profile</TT>:</STRONG></DT>
<DD>An item profile.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16391"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16393"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16395"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16397"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118700000000000000">
1.18.7 Method <TT>params_tests_dataset/testNames</TT></A>
</H3><A NAME="16399"></A><A NAME="ref_params_tests_dataset__testNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the ordered names of tests for this dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">test_names = testNames(dataset, item)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks at the results of the first file to find the test names.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_names: Cell array with ordered parameter names.
	item: (Optional) If given, read names by loading item at this index.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16401"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16403"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16405"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118800000000000000">
1.18.8 Method <TT>params_tests_dataset/subsref</TT></A>
</H3><A NAME="16407"></A><A NAME="ref_params_tests_dataset__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050118900000000000000">
1.18.9 Method <TT>params_tests_dataset/subsasgn</TT></A>
</H3><A NAME="16409"></A><A NAME="ref_params_tests_dataset__subsasgn"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic index-based assignment for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501181000000000000000">
1.18.10 Method <TT>params_tests_dataset/params_tests_db</TT></A>
</H3><A NAME="16411"></A><A NAME="ref_params_tests_dataset__params_tests_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a params_tests_db object from the dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">db_obj = params_tests_db(obj, items, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a converter method to convert from params_tests_dataset to
 params_tests_db. Uses readDBItems to read the files.
 A customized subclass should provide the correct 
 paramNames, testNames, and itemResultsRow functions. Adds a ItemIndex
 column to the DB to keep track of raw data files after shuffling.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A params_tests_dataset object.
</DD>
<DT><STRONG><TT>items</TT>:</STRONG></DT>
<DD>(Optional) List of item indices to use to create the db.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Any optional params to pass to params_tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
db_obj: A params_tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>readDBItems</TT> (p.&nbsp;<A HREF="#ref_readDBItems"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16413"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16415"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16417"></A>, <TT>itemResultsRow
	    testNames</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow______testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16419"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16421"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501181100000000000000">
1.18.11 Method <TT>params_tests_dataset/display</TT></A>
</H3><A NAME="16423"></A><A NAME="ref_params_tests_dataset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501181200000000000000">
1.18.12 Method <TT>params_tests_dataset/itemResultsRow</TT></A>
</H3><A NAME="16425"></A><A NAME="ref_params_tests_dataset__itemResultsRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Processes a raw data file from the dataset and return
		its parameter and test values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params_row, tests_row] = itemResultsRow(dataset, index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is designed to be reused from subclasses as long as the
 loadItemProfile method is properly overloaded. Adds an Index
 column to the DB to keep track of raw data items after shuffling.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of file in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
	tests_row: Test values in the same order with testNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16427"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16429"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16431"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16433"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501181300000000000000">
1.18.13 Method <TT>params_tests_dataset/getItem</TT></A>
</H3><A NAME="16435"></A><A NAME="ref_params_tests_dataset__getItem"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the dataset item at given index.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">item = getItem(dataset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
item: Object, filename, etc.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16437"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16439"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16441"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16443"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/03
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050119000000000000000">
1.19 Class <TT>params_tests_db</TT></A>
</H2><A NAME="16445"></A><A NAME="ref_params_tests_db"></A>
<H3><A NAME="SECTION050119100000000000000">
1.19.1 Constructor <TT>params_tests_db/params_tests_db</TT></A>
</H3><A NAME="16447"></A><A NAME="ref_params_tests_db__params_tests_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A generic database of test results varying with parameter values, organized in a matrix format.
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Defines all operations on this
 structure so that subclasses can use them.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>num_params</TT>:</STRONG></DT>
<DD>Number of parameters.
</DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db upon which to build the params_tests_db.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db
	num_params: Number of variable parameters in simulations.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16449"></A>, <TT>test_variable_db (N/I)</TT> (p.&nbsp;<A HREF="#ref_test_variable_db__N__I_"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16451"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119200000000000000">
1.19.2 Method <TT>params_tests_db/getParamNames</TT></A>
</H3><A NAME="16453"></A><A NAME="ref_params_tests_db__getParamNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Gets parameter column names.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">col_names = getParamNames(db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Convenience function that delegates to getColNames.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
col_names: A cell array of strings.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/getColNames</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__getColNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16455"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16457"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/03
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119300000000000000">
1.19.3 Method <TT>params_tests_db/matchingRow</TT></A>
</H3><A NAME="16459"></A><A NAME="ref_params_tests_db__matchingRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a criterion database for matching the tests of a row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">crit_db = matchingRow(a_db, row, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Overloaded method for skipping parameter values. STD for param values will be NaNs.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row index to match.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>distDB</TT>:</STRONG></DT>
<DD>Take the standard deviation from this db instead.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
crit_db: A tests_db with two rows for values and STDs.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16461"></A>, <TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16463"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16465"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16467"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119400000000000000">
1.19.4 Method <TT>params_tests_db/get</TT></A>
</H3><A NAME="16469"></A><A NAME="ref_params_tests_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119500000000000000">
1.19.5 Method <TT>params_tests_db/joinRows</TT></A>
</H3><A NAME="16471"></A><A NAME="ref_params_tests_db__joinRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Joins the rows of the given db with rows of with_db with matching
  	RowIndex values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = joinRows(db, with_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Takes the desired columns in with_db with rows having a 
 row index and joins them next to dedired columns from the current db. 
 Assumes each row index only appears once in with_db. The created
 db preserves the ordering of with_db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A param_tests_db object.
</DD>
<DT><STRONG><TT>with_db</TT>:</STRONG></DT>
<DD>A tests_db object with a RowIndex column.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>indexColName</TT>:</STRONG></DT>
<DD>(Optional) Name of row index column (default='RowIndex').
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A params_tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16473"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119600000000000000">
1.19.6 Method <TT>params_tests_db/rankVsDB</TT></A>
</H3><A NAME="16475"></A><A NAME="ref_params_tests_db__rankVsDB"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a ranking DB by comparing rows of this db with the given test criteria.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ranked_db = rankVsDB(a_db, crit_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Distance is each measure difference divided by the STD in to_db, squared and
 summed. Returned DB contains only the selected tests from crit_db and the parameters
 from initial a_db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object to compare rows from.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>A tests_db object holding the match criterion tests and STDs
<P>
which can be created with matchingRow.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_ranked_db: The created DB with original rows and a distance measure, 
		   in ascending order. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>matchingRow</TT> (p.&nbsp;<A HREF="#ref_matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16477"></A>, <TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16479"></A>, <TT>joinRows</TT> (p.&nbsp;<A HREF="#ref_joinRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16481"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119700000000000000">
1.19.7 Method <TT>params_tests_db/plotParamsHists</TT></A>
</H3><A NAME="16483"></A><A NAME="ref_params_tests_db__plotParamsHists"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a horizontal plot_stack of parameter histograms.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ps = plotParamsHists(a_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>Do not display the DB id on the plot title.
</DD>
<DT><STRONG><TT>barAxisProps</TT>:</STRONG></DT>
<DD>passed to plotEqSpaced for each bar axis.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_ps: A horizontal plot_stack of plots
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_stack</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_stack"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16485"></A>, <TT>paramsHists</TT> (p.&nbsp;<A HREF="#ref_paramsHists"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16487"></A>, <TT>plotEqSpaced</TT> (p.&nbsp;<A HREF="#ref_plotEqSpaced"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16489"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119800000000000000">
1.19.8 Method <TT>params_tests_db/set</TT></A>
</H3><A NAME="16491"></A><A NAME="ref_params_tests_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050119900000000000000">
1.19.9 Method <TT>params_tests_db/scanParamAllRows</TT></A>
</H3><A NAME="16493"></A><A NAME="ref_params_tests_db__scanParamAllRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Scans given parameter range for each row in DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_params_db = scanParamAllRows(a_db, param, min_val, max_val, num_levels, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Produces rows by replacing the desired parameter value, in all rows of DB, 
 with num_levels values between the given boundaries, min_val and max_val. 
 This results in a DB with num_levels times more rows than the original DB. 
 Then, makeGenesisParFile can be used to generate a parameter file from 
 this DB to drive new simulations.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object whose first row is subject to modifications.
</DD>
<DT><STRONG><TT>param</TT>:</STRONG></DT>
<DD>The parameter to be varied (see tests2cols for param description).
</DD>
<DT><STRONG><TT>min_val, max_val</TT>:</STRONG></DT>
<DD>The low and high boundaries for the parameter value.
</DD>
<DT><STRONG><TT>num_levels</TT>:</STRONG></DT>
<DD>Number of levels to produce, including the boundaries.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>renameTrial</TT>:</STRONG></DT>
<DD>If given, the 'trial' column is renamed to this name.
</DD>
<DT><STRONG><TT>levelFunc</TT>:</STRONG></DT>
<DD>Use this function to get the parameter range with 
<P>
feval(levelFunc, min_val, max_val, num_levels). Example: 'logLevels'
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_params_db: A db only with params.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Sets NaF to given range with 100 levels:
<BR>&#187; naf_rows_db = scanParamAllRows(a_db(desired_rows, :), 'NaF', 0, 1000, 100);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>makeGenesisParFile</TT> (p.&nbsp;<A HREF="#ref_makeGenesisParFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16495"></A>, <TT>scaleParamsOneRow</TT> (p.&nbsp;<A HREF="#ref_scaleParamsOneRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16497"></A>, <TT>ranked_db/blockedDistances</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db__blockedDistances"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16499"></A>, <TT>getParamRowIndices</TT> (p.&nbsp;<A HREF="#ref_getParamRowIndices"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16501"></A>, <TT>logLevels</TT> (p.&nbsp;<A HREF="#ref_logLevels"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16503"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191000000000000000">
1.19.10 Method <TT>params_tests_db/plotVarBoxMatrix</TT></A>
</H3><A NAME="16505"></A><A NAME="ref_params_tests_db__plotVarBoxMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a stack of parameter-test variation plots 
		organized in a matrix.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot_stack = plotVarBoxMatrix(a_db, p_t3ds)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>p_t3ds</TT>:</STRONG></DT>
<DD>Cell array of invariant parameter databases.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot_stack: A plot_stack with the plots organized in matrix form
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16507"></A>, <TT>plotVar</TT> (p.&nbsp;<A HREF="#ref_plotVar"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16509"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191100000000000000">
1.19.11 Method <TT>params_tests_db/meanDuplicateParams</TT></A>
</H3><A NAME="16511"></A><A NAME="ref_params_tests_db__meanDuplicateParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Takes the mean of all measures for rows that have the same parameter columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_params_tests_db = meanDuplicateParams(db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calls tests_db/meanDuplicateRows with params as main_cols and tests
 as rest_cols.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Structure with optional parameters.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_params_tests_db: The db object of with the means on page 1 
		    and standard deviations on page 2.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/meanDuplicateRows</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__meanDuplicateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16513"></A>, <TT>tests_db/mean</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__mean"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16515"></A>, <TT>tests_db/std</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__std"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16517"></A>, <TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16519"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191200000000000000">
1.19.12 Method <TT>params_tests_db/onlyRowsTests</TT></A>
</H3><A NAME="16521"></A><A NAME="ref_params_tests_db__onlyRowsTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a tests_db that only contains the desired 
		tests and rows (and pages).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = onlyRowsTests(obj, rows, tests, pages)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Selects the given dimensions and returns in a new tests_db
 object. Makes sure num_params remains correct.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows, tests</TT>:</STRONG></DT>
<DD>A logical or index vector of rows, or cell array of
<P>
names of rows. If ':', all rows. For names, regular expressions are
supported if quoted with slashes (e.g., '/a.*/'). See tests2idx.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>(Optional) A logical or index vector of pages. ':' for all pages.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16523"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16525"></A>, <TT>test2idx</TT> (p.&nbsp;<A HREF="#ref_test2idx"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16527"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191300000000000000">
1.19.13 Method <TT>params_tests_db/paramsCoefs</TT></A>
</H3><A NAME="16529"></A><A NAME="ref_params_tests_db__paramsCoefs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates a corrcoefs_db for each param and collects them in a cell array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">p_coefs = paramsCoefs(a_db, p_t3ds)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>p_t3ds</TT>:</STRONG></DT>
<DD>Cell array of invariant parameter databases.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
p_coefs: A cell array of corrcoefs_dbs for each param in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16531"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191400000000000000">
1.19.14 Method <TT>params_tests_db/unionCat</TT></A>
</H3><A NAME="16533"></A><A NAME="ref_params_tests_db__unionCat"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Vertically concatenate two databases with different parameters or tests.
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The parameters and tests in the result are a union of both. Adds 0 for
 parameter and NaN for tests in the rows which didn't have the additional
 columns before.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li Su
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191500000000000000">
1.19.15 Method <TT>params_tests_db/subsref</TT></A>
</H3><A NAME="16535"></A><A NAME="ref_params_tests_db__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191600000000000000">
1.19.16 Method <TT>params_tests_db/displayRankingsTeX</TT></A>
</H3><A NAME="16537"></A><A NAME="ref_params_tests_db__displayRankingsTeX"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates and displays a ranking DB by comparing rows of a_db with the given match criteria.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = displayRankingsTeX(a_db, crit_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a LaTeX document with:
	- Values of 10 best matching a_db rows in a floating table.
	- (optional) Raw traces compared with some best matches at different distances
	- Parameter distributions of 50 best matches as a bar graph.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object to compare rows from.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>A tests_db object holding the match criterion tests and STDs
<P>
which can be created with matchingRow.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Identification of the criterion db (not needed/used?).
</DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>Dataset for a_db.
</DD>
<DT><STRONG><TT>a_dball</TT>:</STRONG></DT>
<DD>The non-joined DB for for a_db.
</DD>
<DT><STRONG><TT>crit_dataset</TT>:</STRONG></DT>
<DD>Dataset for crit_db.
</DD>
<DT><STRONG><TT>crit_dball</TT>:</STRONG></DT>
<DD>Dataset for crit_db.
</DD>
<DT><STRONG><TT>num_matches</TT>:</STRONG></DT>
<DD>Number of best matches to display (default=10).
</DD>
<DT><STRONG><TT>rotate</TT>:</STRONG></DT>
<DD>Rotation angle for best matches table (default=90).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX document string.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankVsDB</TT> (p.&nbsp;<A HREF="#ref_rankVsDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16539"></A>, <TT>displayRowsTeX</TT> (p.&nbsp;<A HREF="#ref_displayRowsTeX"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16541"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191700000000000000">
1.19.17 Method <TT>params_tests_db/display</TT></A>
</H3><A NAME="16543"></A><A NAME="ref_params_tests_db__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191800000000000000">
1.19.18 Method <TT>params_tests_db/addParams</TT></A>
</H3><A NAME="16545"></A><A NAME="ref_params_tests_db__addParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Inserts new parameter columns to tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = addParams(obj, param_names, param_columns)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Adds new columns to the database and returns the new DB.
   This operation is expensive in the sense that the whole database matrix
   needs to be enlarged just to add a 
   single new column. The method of allocating a matrix, filling it up, and
   then providing it to the tests_db constructor is the preferred method 
   of creating tests_db objects. This method may be used for 
   measures obtained by operating on raw measures.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>param_names</TT>:</STRONG></DT>
<DD>A cell array of param names to be added.
</DD>
<DT><STRONG><TT>param_columns</TT>:</STRONG></DT>
<DD>Data matrix of columns to be added.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that includes the new columns.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16547"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16549"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501191900000000000000">
1.19.19 Method <TT>params_tests_db/makeGenesisParFile</TT></A>
</H3><A NAME="16551"></A><A NAME="ref_params_tests_db__makeGenesisParFile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a Genesis parameter file with all the parameter values in a_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">makeGenesisParFile(a_db, filename, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>For each a_db row, print the parameter names in a
 file formatted for Genesis.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>filename</TT>:</STRONG></DT>
<DD>Genesis parameter file to be created.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>trialStart</TT>:</STRONG></DT>
<DD>If given, adds/replaces the trial parameter and counts forward.
</DD>
<DT><STRONG><TT>makeParamDesc</TT>:</STRONG></DT>
<DD>If 1, put the parameter names in a parameter description file with
<P>
with a .txt extension.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
nothing.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; blocked_rows_db = makeModifiedParamDB(ranked_for_gps0501a_db, 1, [1, 2], 10, [-100 100]);
<BR>&#187; makeGenesisParFile(blocked_rows_db, 'blocked_gps0501-03.par')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>makeModifiedParamDB</TT> (p.&nbsp;<A HREF="#ref_makeModifiedParamDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16553"></A>, <TT>scanParamAllRows</TT> (p.&nbsp;<A HREF="#ref_scanParamAllRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16555"></A>, <TT>scaleParamsOneRow</TT> (p.&nbsp;<A HREF="#ref_scaleParamsOneRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16557"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/03/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192000000000000000">
1.19.20 Method <TT>params_tests_db/reIndexNeurons</TT></A>
</H3><A NAME="16559"></A><A NAME="ref_params_tests_db__reIndexNeurons"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Re-index neurons with accending numbers. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">new_db = reIndexNeurons(a_db, startNum, colName)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is useful for avoiding NeuronId conflicts when concatenating two
 databases. It can also remove the unused number 'hole' (e.g. after
 deleting rows) and make the indices continuous.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>a tests_db object
</DD>
<DT><STRONG><TT>startNum</TT>:</STRONG></DT>
<DD>the starting index number (default = 1)
</DD>
<DT><STRONG><TT>colName</TT>:</STRONG></DT>
<DD>the column name or number of neuron index. (default = 'NeuronId')
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
new_db: a new database with new neuron index.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16561"></A>, <TT>tests_db/physiol_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__physiol_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16563"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li Su. 03/21/2008
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192100000000000000">
1.19.21 Method <TT>params_tests_db/rankVsAllDB</TT></A>
</H3><A NAME="16565"></A><A NAME="ref_params_tests_db__rankVsAllDB"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates ranking DBs by comparing rows of a_db with each row of to_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = rankVsAllDB(a_db, to_db, a_dataset, to_dataset)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Distance is each measure difference divided by the STD in to_db, squared and
 summed. Returned DB contains only the selected to_tests and the parameters
 from initial DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object to compare rows from.
</DD>
<DT><STRONG><TT>to_db</TT>:</STRONG></DT>
<DD>A tests_db object to compare it with.
</DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>Dataset for a_db.
</DD>
<DT><STRONG><TT>to_dataset</TT>:</STRONG></DT>
<DD>Dataset for crit_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ranked_dbs: Array of created DBs with original rows and a distance 
		   measure, in ascending order. 
	tex_string: A LaTeX string for all tables created.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankVsDB</TT> (p.&nbsp;<A HREF="#ref_rankVsDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16567"></A>, <TT>matchingRow</TT> (p.&nbsp;<A HREF="#ref_matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16569"></A>, <TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16571"></A>, <TT>joinRows</TT> (p.&nbsp;<A HREF="#ref_joinRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16573"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192200000000000000">
1.19.22 Method <TT>params_tests_db/paramsParamsCoefs</TT></A>
</H3><A NAME="16575"></A><A NAME="ref_params_tests_db__paramsParamsCoefs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates a corrcoefs_db for each param from correlations of variant params and invariant param coefs and collects them in a cell array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">pp_coefs = paramsParamsCoefs(a_db, p_t3ds, p_coefs)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>p_t3ds</TT>:</STRONG></DT>
<DD>Cell array of invariant parameter databases.
</DD>
<DT><STRONG><TT>p_coefs</TT>:</STRONG></DT>
<DD>Cell array of tests coefficients for each parameter.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
pp_coefs: A cell array of corrcoefs_dbs for each param 
		  combination in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16577"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192300000000000000">
1.19.23 Method <TT>params_tests_db/mergeMultipleCIPsInOne</TT></A>
</H3><A NAME="16579"></A><A NAME="ref_params_tests_db__mergeMultipleCIPsInOne"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Merges multiple rows with different CIP data into one, generating a database of one row per neuron.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = mergeMultipleCIPsInOne(db, names_tests_cell, index_col_name, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It calls invarParam to separate DB into pages with different CIP level
 data.  Then uses the names_tests_cell to choose tests from each page to be
 merged into the final database row. The tests will be suffixed with the
 field name so that they can be distinguished. RowIndex columns will be
 automatically included, and one of them can be chosen with index_col_name
 that has values for all cells. The suffixed for needs to be used to choose
 index_col_name, such as 'RowIndex_H100pA', assuming 'H100pA' was the field
 name in names_tests_cell that corresponds to page -100 pA.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>names_tests_cell</TT>:</STRONG></DT>
<DD>A cell array alternating suffix names and test column vectors.
<P>
The order of names correspond to each unique CIP level in db, 
with increasing order.
</DD>
<DT><STRONG><TT>index_col_name</TT>:</STRONG></DT>
<DD>(Optional) Name of row index column 

<P>
(default is 'RowIndex' suffixed with the first field name).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>cipLevels</TT>:</STRONG></DT>
<DD>In case db is missing some levels, provides a list of 
<P>
cip levels that correspond to names_tests_cell db. Missing 
levels are replaced with NaN values. DB is filtered to
remove other CIP levels.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A params_tests_db object of organized values.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; control_phys_sdb = 
<BR>
mergeMultipleCIPsInOne(control_phys_db, 
<BR>
struct('_H100pA', [1:10], '_D100pA', [1:10 16:18]), 
<BR>'RowIndex_H100pA')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>invarValues</TT> (p.&nbsp;<A HREF="#ref_invarValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16581"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16583"></A>, <TT>corrCoefs</TT> (p.&nbsp;<A HREF="#ref_corrCoefs"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16585"></A>, <TT>tests_3D_db/plotVarBox</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__plotVarBox"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16587"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192400000000000000">
1.19.24 Method <TT>params_tests_db/getParamRowIndices</TT></A>
</H3><A NAME="16589"></A><A NAME="ref_params_tests_db__getParamRowIndices"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns indices of rows with matching parameter values from rows of this db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">row_indices = getParamRowIndices(a_db, rows, to_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>rows to find indices for.
</DD>
<DT><STRONG><TT>to_db</TT>:</STRONG></DT>
<DD>Where to find the matching rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
row_indices: Array of row indices.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>makeModifiedParamDB</TT> (p.&nbsp;<A HREF="#ref_makeModifiedParamDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16591"></A>, <TT>scanParamAllRows</TT> (p.&nbsp;<A HREF="#ref_scanParamAllRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16593"></A>, <TT>scaleParamsOneRow</TT> (p.&nbsp;<A HREF="#ref_scaleParamsOneRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16595"></A>, <TT>makeGenesisParFile</TT> (p.&nbsp;<A HREF="#ref_makeGenesisParFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16597"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192500000000000000">
1.19.25 Method <TT>params_tests_db/testsHists</TT></A>
</H3><A NAME="16599"></A><A NAME="ref_params_tests_db__testsHists"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates histograms for all tests and returns them in a cell array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">t_hists = testsHists(a_db, num_bins)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>One or more tests_db objects in an array.
</DD>
<DT><STRONG><TT>num_bins</TT>:</STRONG></DT>
<DD>Number of histogram bins (Optional, default=100), or
<P>
vector of histogram bin centers.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
t_hists: An array of histograms for each test in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16601"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192600000000000000">
1.19.26 Method <TT>params_tests_db/getDualCIPdb</TT></A>
</H3><A NAME="16603"></A><A NAME="ref_params_tests_db__getDualCIPdb"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a database by merging selected tests of depolarizing and hyperpolarizing cip results.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = getDualCIPdb(db, depol_tests, hyper_tests, depol_suffix, hyper_suffix)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>depol_tests need to have the RowIndex column in it.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A params_tests_db object of organized values.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; control_phys_sdb = getDualCIPdb(control_phys_db, depol_tests, hyper_tests, '', 'Hyp100pA')
<BR>
where depol_tests and hyper_tests are cell arrays of selected tests.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>invarValues</TT> (p.&nbsp;<A HREF="#ref_invarValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16605"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16607"></A>, <TT>corrCoefs</TT> (p.&nbsp;<A HREF="#ref_corrCoefs"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16609"></A>, <TT>tests_3D_db/plotPair</TT> (p.&nbsp;<A HREF="#ref_tests_3D_db__plotPair"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16611"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192700000000000000">
1.19.27 Method <TT>params_tests_db/invarParam</TT></A>
</H3><A NAME="16613"></A><A NAME="ref_params_tests_db__invarParam"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a 3D database of invariant values of a parameter and all test columns. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_3D_db = invarParam(db, param)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Finds all combinations when the rest of the parameters are fixed,
 and saves the variation of the selected parameter and all tests in
 a new database.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>param</TT>:</STRONG></DT>
<DD>A parameter name/column number
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_3D_db: A tests_3D_db object of organized values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>invarValues</TT> (p.&nbsp;<A HREF="#ref_invarValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16615"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16617"></A>, <TT>corrCoefs</TT> (p.&nbsp;<A HREF="#ref_corrCoefs"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16619"></A>, <TT>tests_3D_db/plotPair</TT> (p.&nbsp;<A HREF="#ref_tests_3D_db__plotPair"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16621"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192800000000000000">
1.19.28 Method <TT>params_tests_db/getProfile</TT></A>
</H3><A NAME="16623"></A><A NAME="ref_params_tests_db__getProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a profile object from a params_tests_db by collecting
			 statistics.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pt_profile = getProfile(a_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calculates the following results items:
	idx: Name-index pairs for accessing results arrays.
	t_hists: Cell array of histograms of each test.
	p_t3ds: Cell array of invariant relations of each parameter with all tests.
	pt_hists: Cell array of separate test value histograms 
		for uniques value of each parameter.
	p_stats: Cell array of test stats for each param.
	p_coefs: Cell array of correlation coefficients 
		for each parameter with all tests.
	pt_coefs_hists: Cell matrix of histograms of coefficients from 
		correlations of each parameter with each test.
	pp_coefs: Cell 3D matrix of mean coefficients from 
		correlations of each parameter with correlation 
		coefficients of each parameter with each test.		
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a params_tests_profile object.</STRONG></DT>
<DD>&nbsp;

<P>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16625"></A>, <TT>results_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_results_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16627"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16629"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16631"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16633"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16635"></A>, <TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16637"></A>, <TT>stats_db</TT> (p.&nbsp;<A HREF="node5.html#ref_stats_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16639"></A>, <TT>corrcoefs_db</TT> (p.&nbsp;<A HREF="node5.html#ref_corrcoefs_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16641"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501192900000000000000">
1.19.29 Method <TT>params_tests_db/delColumns</TT></A>
</H3><A NAME="16643"></A><A NAME="ref_params_tests_db__delColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Deletes columns from tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">index = delColumns(obj, tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Overloaded function that maintains correct number of parameters. See
 original tests_db/delColumns.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Numbers or names of tests (see tests2cols)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that is missing the columns.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/delColumns</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__delColumns"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16645"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501193000000000000000">
1.19.30 Method <TT>params_tests_db/scaleParamsOneRow</TT></A>
</H3><A NAME="16647"></A><A NAME="ref_params_tests_db__scaleParamsOneRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Scales chosen parameters in a row by multiplying with levels to create a new parameter db with as many rows as values in levels.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_params_db = scaleParamsOneRow(a_db, params, levels)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Produces rows by multiplying desired params, in the first row of DB, 
 with each value in levels. Then, makeGenesisParFile can be used to generate
 a parameter file from this DB to drive new simulations.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object whose first row is subject to modifications.
</DD>
<DT><STRONG><TT>params</TT>:</STRONG></DT>
<DD>Parameters to be varied (see tests2cols for param description).
</DD>
<DT><STRONG><TT>levels</TT>:</STRONG></DT>
<DD>Column vector of parameter value multipliers (1=unity).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_params_db: A db only with params.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Blocks NaF from 0&#187; naf_rows_db = scanOneParam(a_db(desired_row, :), 'NaF', 0:0.1:1);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>ranked_db/blockedDistances</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db__blockedDistances"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16649"></A>, <TT>getParamRowIndices</TT> (p.&nbsp;<A HREF="#ref_getParamRowIndices"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16651"></A>, <TT>makeGenesisParFile</TT> (p.&nbsp;<A HREF="#ref_makeGenesisParFile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16653"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501193100000000000000">
1.19.31 Method <TT>params_tests_db/paramsHists</TT></A>
</H3><A NAME="16655"></A><A NAME="ref_params_tests_db__paramsHists"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates histograms for all parameters and returns in a 
		cell array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">p_hists = paramsHists(a_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test. Useful for looking at subset databases and
 find out what parameter values are used most.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
p_hists: An array of histograms for each parameter in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16657"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501193200000000000000">
1.19.32 Method <TT>params_tests_db/paramsTestsCoefsHists</TT></A>
</H3><A NAME="16659"></A><A NAME="ref_params_tests_db__paramsTestsCoefsHists"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates histograms for all pairs of params 
		  and tests coefficients and returns in a cell array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">pt_coefs_hists = paramsTestsCoefsHists(a_db, p_coefs)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>p_coefs</TT>:</STRONG></DT>
<DD>Cell array of tests coefficients for each parameter.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
pt_coefs_hists: A cell array of corrcoefs_dbs for each param in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16661"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501193300000000000000">
1.19.33 Method <TT>params_tests_db/crossProd</TT></A>
</H3><A NAME="16663"></A><A NAME="ref_params_tests_db__crossProd"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a DB by taking the cross product of two database row sets.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">cross_db = crossProd(a_db, b_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Overloaded function to maintain correct number of parameters after
 cross product operation. See original in tests_db/crossProd.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db, b_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
cross_db: The tests_db object with all combinations of rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/crossProd</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__crossProd"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16665"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501193400000000000000">
1.19.34 Method <TT>params_tests_db/invarParams</TT></A>
</H3><A NAME="16667"></A><A NAME="ref_params_tests_db__invarParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates invariant param dbs for all parameters and returns in an array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">p_t3ds = invarParams(a_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
p_t3ds: An array of tests_3D_dbs for each param in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16669"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050120000000000000000">
1.20 Class <TT>params_tests_fileset</TT></A>
</H2><A NAME="16671"></A><A NAME="ref_params_tests_fileset"></A>
<H3><A NAME="SECTION050120100000000000000">
1.20.1 Constructor <TT>params_tests_fileset/params_tests_fileset</TT></A>
</H3><A NAME="16673"></A><A NAME="ref_params_tests_fileset__params_tests_fileset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Description of a set of data files of raw data varying with parameter values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = params_tests_fileset(file_pattern, dt, dy, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_dataset. This class is used to generate 
 params_tests_db objects and keep 
 a connection to the raw data files. This class only keeps names of
 files and loads raw data files whenever it's requested. A database
 object can easily be generated using the convertion methods.
 Most methods defined here can 
 be used as-is, however some should be overloaded in subclasses. 
 The specific methods are loadItemProfile.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>file_pattern</TT>:</STRONG></DT>
<DD>File pattern, or cell array of patterns, matching all 
<P>
files to be loaded.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identification string
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>num_params</TT>:</STRONG></DT>
<DD>Number of parameters that appear in filenames.
</DD>
<DT><STRONG><TT>param_trial_name</TT>:</STRONG></DT>
<DD>Use this name on the filename as the 'trial' parameter.
</DD>
<DT><STRONG><TT>param_row_filename</TT>:</STRONG></DT>
<DD>If given, the 'trial' parameter will be used
<P>
to address rows from this file and acquire parameters.
</DD>
<DT><STRONG><TT>param_desc_filename</TT>:</STRONG></DT>
<DD>Contains the parameter range descriptions one per 

<P>
each row. The parameter names are acquired from this file.
</DD>
<DT><STRONG><TT>param_names</TT>:</STRONG></DT>
<DD>Cell array of parameter names corresponding to the 

<P>
param_row_filename columns can be specified as an alternative to
specifying param_desc_filename. These names are not for the 
parameters present in the data filename.
</DD>
<DT><STRONG><TT>profile_method_name</TT>:</STRONG></DT>
<DD>It can be one of the profile-creating methods in this

<P>
class. E.g., 'trace_profile', 'srp_trace_profile', etc.
(See parent classes and cip_trace object for more props)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_dataset,
	path: The pathname to files.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16675"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16677"></A>, <TT>test_variable_db (N/I)</TT> (p.&nbsp;<A HREF="#ref_test_variable_db__N__I_"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16679"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120200000000000000">
1.20.2 Method <TT>params_tests_fileset/get</TT></A>
</H3><A NAME="16681"></A><A NAME="ref_params_tests_fileset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120300000000000000">
1.20.3 Method <TT>params_tests_fileset/set</TT></A>
</H3><A NAME="16683"></A><A NAME="ref_params_tests_fileset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120400000000000000">
1.20.4 Method <TT>params_tests_fileset/trace_profile</TT></A>
</H3><A NAME="16685"></A><A NAME="ref_params_tests_fileset__trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a raw trace_profile given a file_index to this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_trace_profile = trace_profile(fileset, file_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>Index of file in fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_trace_profile: A trace_profile object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16687"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16689"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120500000000000000">
1.20.5 Method <TT>params_tests_fileset/getItemParams</TT></A>
</H3><A NAME="16691"></A><A NAME="ref_params_tests_fileset__getItemParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Get the parameter values of a dataset item.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">params_row = getItemParams(dataset, index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16693"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16695"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16697"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16699"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/03
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120600000000000000">
1.20.6 Method <TT>params_tests_fileset/addFiles</TT></A>
</H3><A NAME="16701"></A><A NAME="ref_params_tests_fileset__addFiles"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Adds to existing list of files in set.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_fileset, index_list] = addFiles(a_fileset, file_pattern, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset object.
</DD>
<DT><STRONG><TT>file_pattern</TT>:</STRONG></DT>
<DD>File pattern, or cell array of patterns, matching additional files.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>param_row_filename</TT>:</STRONG></DT>
<DD>Update parameters from here. The 'trial' parameter is used
<P>
to address rows from this file and acquire parameters.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_fileset: The augmented fileset object.
	index_list: The vector of index numbers of the new files added. Can be used
		to selectively load the new files into a DB using params_test_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16703"></A>, <TT>params_tests_dataset/params_test_db.</TT> (p.&nbsp;<A HREF="#ref_params_tests_dataset__params_test_db."><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16705"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120700000000000000">
1.20.7 Method <TT>params_tests_fileset/display</TT></A>
</H3><A NAME="16707"></A><A NAME="ref_params_tests_fileset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120800000000000000">
1.20.8 Method <TT>params_tests_fileset/loadItemProfile</TT></A>
</H3><A NAME="16709"></A><A NAME="ref_params_tests_fileset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a profile object from a raw data file in the fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(fileset, file_index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclasses should overload this function to load the specific profile
 object they desire. The profile class should define a getResults method
 which is used in the itemResultsRow method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>Index of file in fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16711"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16713"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16715"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16717"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050120900000000000000">
1.20.9 Method <TT>params_tests_fileset/paramNames</TT></A>
</H3><A NAME="16719"></A><A NAME="ref_params_tests_fileset__paramNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the ordered names of parameters for this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">param_names = paramNames(fileset, item)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks at the filename of the first file to find the parameter names.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>item</TT>:</STRONG></DT>
<DD>(Optional) If given, read param names by loading item at this index.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_names: Cell array with ordered parameter names.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16721"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16723"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16725"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501201000000000000000">
1.20.10 Method <TT>params_tests_fileset/trace</TT></A>
</H3><A NAME="16727"></A><A NAME="ref_params_tests_fileset__trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a raw trace given a file_index to this fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_trace = trace(fileset, file_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A params_tests_fileset.
</DD>
<DT><STRONG><TT>file_index</TT>:</STRONG></DT>
<DD>Index of file in fileset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_trace: A trace object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16729"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16731"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050121000000000000000">
1.21 Class <TT>params_tests_profile</TT></A>
</H2><A NAME="16733"></A><A NAME="ref_params_tests_profile"></A>
<H3><A NAME="SECTION050121100000000000000">
1.21.1 Constructor <TT>params_tests_profile/params_tests_profile</TT></A>
</H3><A NAME="16735"></A><A NAME="ref_params_tests_profile__params_tests_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Holds the results profile from a params_tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pt_profile = params_tests_profile(results, a_db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>results</TT>:</STRONG></DT>
<DD>A structure containing test results.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
results_profile: Contains results of tests.
	db: The params_tests_db.
	props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>results_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_results_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16737"></A>, <TT>params_tests_db/params_tests_profile</TT> (p.&nbsp;<A HREF="#ref_params_tests_db__params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16739"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050121200000000000000">
1.21.2 Method <TT>params_tests_profile/get</TT></A>
</H3><A NAME="16741"></A><A NAME="ref_params_tests_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050122000000000000000">
1.22 Class <TT>period</TT></A>
</H2><A NAME="16743"></A><A NAME="ref_period"></A>
<H3><A NAME="SECTION050122100000000000000">
1.22.1 Constructor <TT>period/period</TT></A>
</H3><A NAME="16745"></A><A NAME="ref_period__period"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Start and end times of a period in terms of the dt of the trace
	to which belongs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = period(start_time, end_time)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;

<P>
(see below for the rest)</DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
start_time, end_time: Inclusive period [dt].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16747"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16749"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16751"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122200000000000000">
1.22.2 Method <TT>period/get</TT></A>
</H3><A NAME="16753"></A><A NAME="ref_period__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122300000000000000">
1.22.3 Method <TT>period/SpikeTimesinPeriod</TT></A>
</H3><A NAME="16755"></A><A NAME="ref_period__SpikeTimesinPeriod"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">SpkTimes=Interval(times, period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>times</TT>:</STRONG></DT>
<DD>an array of spike times.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>A period object
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
the_period: The cropped set of spike times that fall within a period.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16757"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16759"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16761"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16763"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Tom Sangrey, 2006/01/26
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122400000000000000">
1.22.4 Method <TT>period/set</TT></A>
</H3><A NAME="16765"></A><A NAME="ref_period__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122500000000000000">
1.22.5 Method <TT>period/subsref</TT></A>
</H3><A NAME="16767"></A><A NAME="ref_period__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050122600000000000000">
1.22.6 Method <TT>period/display</TT></A>
</H3><A NAME="16769"></A><A NAME="ref_period__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050123000000000000000">
1.23 Class <TT>physiol_bundle</TT></A>
</H2><A NAME="16771"></A><A NAME="ref_physiol_bundle"></A>
<H3><A NAME="SECTION050123100000000000000">
1.23.1 Constructor <TT>physiol_bundle/physiol_bundle</TT></A>
</H3><A NAME="16773"></A><A NAME="ref_physiol_bundle__physiol_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>The physiology dataset and the DB created from it bundled together.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_bundle = physiol_bundle(a_cell, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of dataset_db_bundle, specialized for physiology datasets. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cell</TT>:</STRONG></DT>
<DD>A cell array that contains the following elements:
</DD>
<DT><STRONG><TT>a_dataset</TT>:</STRONG></DT>
<DD>A cell-enclosed physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>The raw params_tests_db object created from the dataset. 
<P>
It only needs to have the pAcip, pAbias, TracesetIndex, and ItemIndex columns.
</DD>
<DT><STRONG><TT>a_joined_db</TT>:</STRONG></DT>
<DD>The one-treatment-per-line DB created from the raw DB.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>controlDB</TT>:</STRONG></DT>
<DD>Use this as the ontrol DB rather than computing.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
dataset_db_bundle, 
	joined_control_db: DB of control neurons (no pharmacological applications).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16775"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16777"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16779"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123200000000000000">
1.23.2 Method <TT>physiol_bundle/matchingControlNeuron</TT></A>
</H3><A NAME="16781"></A><A NAME="ref_physiol_bundle__matchingControlNeuron"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a criterion database for matching the neuron at traceset_index.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_crit_bundle = matchingControlNeuron(a_bundle, neuron_id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Copies selected test values from row as the first row into the 
 criterion db. Adds a second row for the STD of each column in the db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>neuron_id</TT>:</STRONG></DT>
<DD>A NeuronId of the neuron to match.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_crit_bundle: A tests_db with two rows for values and STDs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Matches gpd0421c from cip_traces_all_axoclamp.txt:
<BR>&#187; a_crit_bundle = matchingControlNeuron(pbundle, 33)
<BR>(see example in matchingRow)
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16783"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16785"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16787"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123300000000000000">
1.23.3 Method <TT>physiol_bundle/matchingRow</TT></A>
</H3><A NAME="16789"></A><A NAME="ref_physiol_bundle__matchingRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a criterion database for matching the neuron at traceset_index.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_crit_db = matchingRow(p_bundle, traceset_index, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Copies selected test values from row as the first row into the 
 criterion db. Adds a second row for the STD of each column in the db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>traceset_index</TT>:</STRONG></DT>
<DD>A TracesetIndex of the neuron and treatments to match.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_crit_db: A tests_db with two rows for values and STDs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">physiol_bundle has an overloaded matchingRow method that
<BR>
takes the TracesetIndex as argument:
<BR>&#187; a_crit_bundle = matchingRow(pbundle, 61)
<BR>&#187; a_ranked_bundle = rankMatching(mbundle, a_crit_bundle);
<BR>&#187; printTeXFile(comparisonReport(a_ranked_bundle), 'my_report.tex')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16791"></A>, <TT>tests_db/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16793"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123400000000000000">
1.23.4 Method <TT>physiol_bundle/get</TT></A>
</H3><A NAME="16795"></A><A NAME="ref_physiol_bundle__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123500000000000000">
1.23.5 Method <TT>physiol_bundle/getNeuronRowIndex</TT></A>
</H3><A NAME="16797"></A><A NAME="ref_physiol_bundle__getNeuronRowIndex"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the neuron index from bundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_row_index = getNeuronRowIndex(a_bundle, traceset_index, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>traceset_index</TT>:</STRONG></DT>
<DD>The TracesetIndex number of neuron, or a DB row containing this.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_row_index: A row index of neuron in a_bundle.joined_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle/getNeuronRowIndex</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle__getNeuronRowIndex"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16799"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123600000000000000">
1.23.6 Method <TT>physiol_bundle/ctFromRows</TT></A>
</H3><A NAME="16801"></A><A NAME="ref_physiol_bundle__ctFromRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from a raw data file in the a_pbundle.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = ctFromRows(a_pbundle, a_db|traceset_idx, cip_levels, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_pbundle</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB created by this fileset to read the traceset indices from.
</DD>
<DT><STRONG><TT>traceset_idx</TT>:</STRONG></DT>
<DD>A column vector with traceset indices.
</DD>
<DT><STRONG><TT>cip_levels</TT>:</STRONG></DT>
<DD>A column vector of CIP-levels to be loaded.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>traces</TT>:</STRONG></DT>
<DD>column vector of trace indices to load.
</DD>
<DT><STRONG><TT>showParamsList</TT>:</STRONG></DT>
<DD>Cell array of params or treatments to include in the id field.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: One or more cip_trace object that holds the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16803"></A>, <TT>physiol_cip_traceset/cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset__cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16805"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/07/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123700000000000000">
1.23.7 Method <TT>physiol_bundle/getNeuronLabel</TT></A>
</H3><A NAME="16807"></A><A NAME="ref_physiol_bundle__getNeuronLabel"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Constructs the neuron label from dataset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_label = getNeuronLabel(a_bundle, traceset_index, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_bundle</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>traceset_index</TT>:</STRONG></DT>
<DD>The traceset index of neuron.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_label: A string label identifying selected neuron in bundle.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16809"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123800000000000000">
1.23.8 Method <TT>physiol_bundle/plotfICurveStats</TT></A>
</H3><A NAME="16811"></A><A NAME="ref_physiol_bundle__plotfICurveStats"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a f-I curve mean-std plot of physiology DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotfICurveStats(p_bundle, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>if given, no title is produced
<P>
(passed to plot_superpose)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: An f-I curve plot.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plotfICurveStats(pbundle));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>dataset_db_bundle/plotfICurve</TT> (p.&nbsp;<A HREF="node5.html#ref_dataset_db_bundle__plotfICurve"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16813"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16815"></A>, <TT>plot_superpose</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_superpose"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16817"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050123900000000000000">
1.23.9 Method <TT>physiol_bundle/set</TT></A>
</H3><A NAME="16819"></A><A NAME="ref_physiol_bundle__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501231000000000000000">
1.23.10 Method <TT>physiol_bundle/constrainedMeasuresPreset</TT></A>
</H3><A NAME="16821"></A><A NAME="ref_physiol_bundle__constrainedMeasuresPreset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a physiol_bundle with constrained measures according to chosen preset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_pbundle test_names] = constrainedMeasuresPreset(a_pbundle, preset, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_pbundle</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>preset</TT>:</STRONG></DT>
<DD>Choose preset measure list (default=1).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_pbundle: One or more cip_trace object that holds the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16823"></A>, <TT>physiol_cip_traceset/cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset__cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16825"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501231100000000000000">
1.23.11 Method <TT>physiol_bundle/bestMatchAllNeurons</TT></A>
</H3><A NAME="16827"></A><A NAME="ref_physiol_bundle__bestMatchAllNeurons"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds the best match among given database for each physiology neuron.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">all_ranks_db = bestMatchAllNeurons(p_bundle, joined_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Returns a database of best matching entries from joined_db for each
 entry in p_bundle.joined_control_db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_bundle</TT>:</STRONG></DT>
<DD>A physiol_bundle object.
</DD>
<DT><STRONG><TT>joined_db</TT>:</STRONG></DT>
<DD>A database with neuron representations to rank against
<P>
neurons.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.

<P>
(passed to rankMatching)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
all_ranks_db: DB of best matching from joined_db. Each row
		corresponds to p_bundle.joined_control_db rows.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; all_ranks_db = ...
<BR>
bestMatchAllNeurons(constrainedMeasuresPreset(pbundle2, 6), mbundle_maxcond.joined_db)
<BR>&#187; plotXRows(all_ranks_db, 'Distance', 'maxcond DB distance per neuron', 'maxcond', ...
<BR>
struct('LineStyle', '-', 'quiet', 1, 'PaperPosition', [0 0 4 3]))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/rankMatching</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16829"></A>, <TT>tests_db/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16831"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050124000000000000000">
1.24 Class <TT>physiol_cip_traceset</TT></A>
</H2><A NAME="16833"></A><A NAME="ref_physiol_cip_traceset"></A>
<H3><A NAME="SECTION050124100000000000000">
1.24.1 Constructor <TT>physiol_cip_traceset/physiol_cip_traceset</TT></A>
</H3><A NAME="16835"></A><A NAME="ref_physiol_cip_traceset__physiol_cip_traceset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Dataset of cip traces from same PCDX file.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = physiol_cip_traceset(trace_str, data_src, chaninfo, dt, dy, treatments, neuron_id, props);</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_dataset. Each trace varies in bias, 
 pulse times and cip magnitude.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace_str</TT>:</STRONG></DT>
<DD>Trace list in the format for loadtraces or just a Matlab vector.
</DD>
<DT><STRONG><TT>data_src</TT>:</STRONG></DT>
<DD>Absolute path of PCDX data source.
</DD>
<DT><STRONG><TT>chaninfo</TT>:</STRONG></DT>
<DD>4-element array containing vchan, ichan, vgain, igain
<DL>
<DD></DD>
<DT><STRONG><TT>vchan, ichan</TT>:</STRONG></DT>
<DD>Current and voltage channels.
</DD>
<DT><STRONG><TT>vgain, igain</TT>:</STRONG></DT>
<DD>External gain factors for voltage channel and current 
<P>
channel
(vgain does NOT include the 10X amplification from the Axoclamp,
so vgain = 1 would mean no additional amplification beyond the 10X.)
</DD>
</DL></DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s].
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>Y-axis resolution [V] or [A].
</DD>
<DT><STRONG><TT>treatments</TT>:</STRONG></DT>
<DD>Structure containing the names and concentrations
<P>
of compounds.
</DD>
<DT><STRONG><TT>neuron_id</TT>:</STRONG></DT>
<DD>Neuron name.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>nsHDF5</TT>:</STRONG></DT>
<DD>For NeuroSAGE HDF5 files, processing is faster if the output
<P>
of ns_open_file is given here. Must be defined to allow
special NeuroSAGE processing.
</DD>
<DT><STRONG><TT>profile_class_name</TT>:</STRONG></DT>
<DD>Use this cip_trace function to return a

<P>
profile (Default: 'getProfileAllSpikes').
</DD>
<DT><STRONG><TT>cip_list</TT>:</STRONG></DT>
<DD>Vector of cip levels to which the current trace will be matched.

<P>
(All other props are passed to cip_trace objects)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
params_tests_dataset,
	data_src, ichan, vchan, vgain, igain, treatments.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_traces</TT> (p.&nbsp;<A HREF="#ref_cip_traces"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16837"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16839"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16841"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt; and Thomas Sangrey, 2005/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124200000000000000">
1.24.2 Method <TT>physiol_cip_traceset/get</TT></A>
</H3><A NAME="16843"></A><A NAME="ref_physiol_cip_traceset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124300000000000000">
1.24.3 Method <TT>physiol_cip_traceset/setProp</TT></A>
</H3><A NAME="16845"></A><A NAME="ref_physiol_cip_traceset__setProp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting optional object properties.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = setProp(obj, prop1, val1, prop2, val2, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Modifies or adds property values. As many property name-value 
 pairs can be specified.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>Any object that has a props field.
</DD>
<DT><STRONG><TT>attr</TT>:</STRONG></DT>
<DD>Property name
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>Property value.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with the updated properties.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124400000000000000">
1.24.4 Method <TT>physiol_cip_traceset/CIPform</TT></A>
</H3><A NAME="16847"></A><A NAME="ref_physiol_cip_traceset__CIPform"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Extracts current bias and pulse information from the current channel.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[ciptype, on, off, finish, bias, pulse] = ns_CIPform(traceset,trace_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of item in traceset
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_traces</TT> (p.&nbsp;<A HREF="#ref_cip_traces"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16849"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16851"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16853"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Thomas Sangrey, 2005
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124500000000000000">
1.24.5 Method <TT>physiol_cip_traceset/set</TT></A>
</H3><A NAME="16855"></A><A NAME="ref_physiol_cip_traceset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124600000000000000">
1.24.6 Method <TT>physiol_cip_traceset/getItemParams</TT></A>
</H3><A NAME="16857"></A><A NAME="ref_physiol_cip_traceset__getItemParams"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Get the parameter values of a dataset item.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">params_row = getItemParams(dataset, index, a_profile)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of item in dataset.
</DD>
<DT><STRONG><TT>a_profile</TT>:</STRONG></DT>
<DD>cip_trace_profile object
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16859"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16861"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16863"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16865"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124700000000000000">
1.24.7 Method <TT>physiol_cip_traceset/cip_trace_profile</TT></A>
</H3><A NAME="16867"></A><A NAME="ref_physiol_cip_traceset__cip_trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace_profile object from a raw data file in the traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = cip_trace_profile(traceset, trace_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of file in traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16869"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16871"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16873"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16875"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt; and Thomas Sangrey, 2005/01/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124800000000000000">
1.24.8 Method <TT>physiol_cip_traceset/cip_trace</TT></A>
</H3><A NAME="16877"></A><A NAME="ref_physiol_cip_traceset__cip_trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from a raw data file in the traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cip_trace = cip_trace(traceset, trace_index, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of file in traceset.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>showParamsList</TT>:</STRONG></DT>
<DD>Cell array of params to add to id field.
</DD>
<DT><STRONG><TT>showName</TT>:</STRONG></DT>
<DD>Show the name of the cell in the id field (default=1).
</DD>
<DT><STRONG><TT>TracesetIndex</TT>:</STRONG></DT>
<DD>Indicates in the id field.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: A cip_trace object that holds the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16879"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16881"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16883"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16885"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/07/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050124900000000000000">
1.24.9 Method <TT>physiol_cip_traceset/loadItemProfile</TT></A>
</H3><A NAME="16887"></A><A NAME="ref_physiol_cip_traceset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace_profile object from a raw data file in the traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(traceset, trace_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of file in traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16889"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16891"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16893"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16895"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501241000000000000000">
1.24.10 Method <TT>physiol_cip_traceset/itemResultsRow</TT></A>
</H3><A NAME="16897"></A><A NAME="ref_physiol_cip_traceset__itemResultsRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Processes a raw data file from the dataset and return
		its parameter and test values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params_row, tests_row] = itemResultsRow(dataset, index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is designed to be reused from subclasses as long as the
 loadItemProfile method is properly overloaded. Adds an Index
 column to the DB to keep track of raw data items after shuffling.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>dataset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>Index of file in dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params_row: Parameter values in the same order of paramNames
	tests_row: Test values in the same order with testNames
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16899"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16901"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16903"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16905"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501241100000000000000">
1.24.11 Method <TT>physiol_cip_traceset/paramNames</TT></A>
</H3><A NAME="16907"></A><A NAME="ref_physiol_cip_traceset__paramNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the parameter names for this traceset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">param_names = paramNames(traceset)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks at the filename of the first file to find the parameter names.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset</TT>:</STRONG></DT>
<DD>A params_tests_dataset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
param_names: Cell array with ordered parameter names.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16909"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16911"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16913"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050125000000000000000">
1.25 Class <TT>physiol_cip_traceset_fileset</TT></A>
</H2><A NAME="16915"></A><A NAME="ref_physiol_cip_traceset_fileset"></A>
<H3><A NAME="SECTION050125100000000000000">
1.25.1 Constructor <TT>physiol_cip_traceset_fileset/physiol_cip_traceset_fileset</TT></A>
</H3><A NAME="16917"></A><A NAME="ref_physiol_cip_traceset_fileset__physiol_cip_traceset_fileset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Physiological fileset of traceset objects (concatenated).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = physiol_cip_traceset_fileset(traceset_items, dt, dy, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of params_tests_dataset. It contains a set of
 physiol_cip_traceset items that are tied to physical data sources. Each
 traceset can load a set of traces for an experimental recording. Most
 flexible usage is obtained when the input traceset_items is given as a
 cell array of physiol_cip_traceset objects. These objects can each link to
 PCDX or NeuroSAGE HDF5 files independent of each other. A regular Matlab
 script can be used to create such a cell array. If a function is defined
 to return such an array, it can be passed as
 traceset_items. Alternatively, the cell array can be constructed from an
 ASCII file as described below, such as for deprecated PCDX data files.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>traceset_items</TT>:</STRONG></DT>
<DD>It can be a function handle, cell array or filename
<P>
string. Function should return a cell array of physiol_cip_traceset
items. Alternatively a preconstructed cell array can be provided directly.
If it is an ASCII filename, then it should contain the following tab-delimited items:
1. Neuron ID (name to associate with the neuron). If left blank, use
the filename with the '.all' extension removed.
2. The absolute path of the data file
3. The trace numbers to load, space-delimited (e.g. 1-21 24 26 27)
4. Vchan: voltage channel number
5. Ichan: current channel number
6. Vgain: external gain on voltage channel IN ADDITION to the 10X that
automatically comes from the Axoclamp 2B.
7. Igain: external gain on current channel.
8. Pairs of condition names and molar concentrations in any order
e.g.: TTX       1e-8    apamin  2e-7    picrotoxin      1e-4
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
neuron_idx: A structure that points from neuron names to NeuronId numbers.
	params_tests_dataset
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_cip_traceset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16919"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16921"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16923"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt; and Thomas Sangrey, 2005/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125200000000000000">
1.25.2 Method <TT>physiol_cip_traceset_fileset/get</TT></A>
</H3><A NAME="16925"></A><A NAME="ref_physiol_cip_traceset_fileset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125300000000000000">
1.25.3 Method <TT>physiol_cip_traceset_fileset/setProp</TT></A>
</H3><A NAME="16927"></A><A NAME="ref_physiol_cip_traceset_fileset__setProp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting optional object properties.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = setProp(obj, prop1, val1, prop2, val2, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Modifies or adds property values. As many property name-value 
 pairs can be specified.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>Any object that has a props field.
</DD>
<DT><STRONG><TT>attr</TT>:</STRONG></DT>
<DD>Property name
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>Property value.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with the updated properties.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125400000000000000">
1.25.4 Method <TT>physiol_cip_traceset_fileset/physiol_bundle</TT></A>
</H3><A NAME="16929"></A><A NAME="ref_physiol_cip_traceset_fileset__physiol_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads the database and then creates the physiol_bundle object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pbundle = physiol_bundle(fileset, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calls params_tests_db to get the db, and then calls
 tests_db/physiol_bundle to do transformations.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(Passed to tests_db/physiol_bundle)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_physiol_bundle: One or more physiol_bundle object that holds the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/physiol_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__physiol_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16931"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125500000000000000">
1.25.5 Method <TT>physiol_cip_traceset_fileset/readDBItems</TT></A>
</H3><A NAME="16933"></A><A NAME="ref_physiol_cip_traceset_fileset__readDBItems"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Reads all items to generate a params_tests_db object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[params, param_names, tests, test_names] = readDBItems(obj, items)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a specific method to convert from physiol_cip_traceset_fileset to
 a params_tests_db, or a subclass. 
 Outputs of this function can be directly fed to the constructor of
 a params_tests_db or a subclass.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset 
</DD>
<DT><STRONG><TT>items</TT>:</STRONG></DT>
<DD>(Optional) List of item indices to use to create the db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
params, param_names, tests, test_names: See params_tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16935"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16937"></A>, <TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16939"></A></DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125600000000000000">
1.25.6 Method <TT>physiol_cip_traceset_fileset/set</TT></A>
</H3><A NAME="16941"></A><A NAME="ref_physiol_cip_traceset_fileset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125700000000000000">
1.25.7 Method <TT>physiol_cip_traceset_fileset/display</TT></A>
</H3><A NAME="16943"></A><A NAME="ref_physiol_cip_traceset_fileset__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125800000000000000">
1.25.8 Method <TT>physiol_cip_traceset_fileset/vertcat</TT></A>
</H3><A NAME="16945"></A><A NAME="ref_physiol_cip_traceset_fileset__vertcat"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Concatenates multiple physiol_cip_traceset_fileset objects.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = vertcat(obj, obj2)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Concatenates the list contents, and combines the neuron_idx
 structures. The properties such as dt, dy and props are retained from
 first object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj, obj2</TT>:</STRONG></DT>
<DD>Two physiol_cip_traceset_fileset objects without
<P>
overlapping neuron_id items.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with combined contents.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_cip_traceset_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16947"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050125900000000000000">
1.25.9 Method <TT>physiol_cip_traceset_fileset/cip_trace</TT></A>
</H3><A NAME="16949"></A><A NAME="ref_physiol_cip_traceset_fileset__cip_trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace object from a raw data file in the fileset.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>traceset_index</TT>:</STRONG></DT>
<DD>Index of traceset item in this fileset (corresponds 
<P>
to row in cells_filename) to find the cell information.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of item in the traceset.
</DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A DB created by this fileset to read the traceset and item indices from.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to physiol_cip_traceset/cip_trace.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cip_trace: One or more cip_trace object that holds the raw data.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>loadItemProfile</TT> (p.&nbsp;<A HREF="#ref_loadItemProfile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16951"></A>, <TT>physiol_cip_traceset/cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset__cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16953"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/07/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501251000000000000000">
1.25.10 Method <TT>physiol_cip_traceset_fileset/loadItemProfile</TT></A>
</H3><A NAME="16955"></A><A NAME="ref_physiol_cip_traceset_fileset__loadItemProfile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Loads a cip_trace_profile object from a raw data file in the fileset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_profile = loadItemProfile(fileset, traceset_index, trace_index)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset object.
</DD>
<DT><STRONG><TT>traceset_index </TT>:</STRONG></DT>
<DD>Index of traceset item in this fileset (corresponds 
<P>
to row in cells_filename) to use grab the cell information.
</DD>
<DT><STRONG><TT>trace_index</TT>:</STRONG></DT>
<DD>Index of item in the traceset.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_profile: A profile object that implements the getResults method.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>itemResultsRow</TT> (p.&nbsp;<A HREF="#ref_itemResultsRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16957"></A>, <TT>params_tests_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16959"></A>, <TT>paramNames</TT> (p.&nbsp;<A HREF="#ref_paramNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16961"></A>, <TT>testNames</TT> (p.&nbsp;<A HREF="#ref_testNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16963"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14 and Tom Sangrey
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501251100000000000000">
1.25.11 Method <TT>physiol_cip_traceset_fileset/neuronNameFromId</TT></A>
</H3><A NAME="16965"></A><A NAME="ref_physiol_cip_traceset_fileset__neuronNameFromId"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns string neuron names from a list of neuron ids.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">name_strs = neuronNameFromId(fileset, neuron_ids)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>fileset</TT>:</STRONG></DT>
<DD>A physiol_cip_traceset_fileset object.
</DD>
<DT><STRONG><TT>neuron_ids</TT>:</STRONG></DT>
<DD>One or more neuron ids in a vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
name_strs: Cell array of neuron names corresponding to the ids given.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_cip_traceset_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16967"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/11/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050126000000000000000">
1.26 Class <TT>plot_abstract</TT></A>
</H2><A NAME="16969"></A><A NAME="ref_plot_abstract"></A>
<H3><A NAME="SECTION050126100000000000000">
1.26.1 Constructor <TT>plot_abstract/plot_abstract</TT></A>
</H3><A NAME="16971"></A><A NAME="ref_plot_abstract__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A plot that can be directly visualized or included in subplots.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = plot_abstract(data, axis_labels, title, legend, command, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Base class that holds the necessary data to draw a plot. This data
 can then be used to generate different plots. Subclasses define specific
 plots with additional data. Subclasses should conform to the standard 
 that the series of commands found in plotFigure should produce a valid
 figure.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>A cell array of data arrays (x, y, z, etc.) that can be 
<P>
fed to plot commands.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings.
</DD>
<DT><STRONG><TT>title</TT>:</STRONG></DT>
<DD>Plot description string.
</DD>
<DT><STRONG><TT>legend</TT>:</STRONG></DT>
<DD>Cell array of descriptions for each item plotted.
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plotting command to use (Optional, default='plot')
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>axisLimits</TT>:</STRONG></DT>
<DD>Sets axis limits of non-NaN values in vector.
</DD>
<DT><STRONG><TT>tightLimits</TT>:</STRONG></DT>
<DD>If 1, issues an "axis tight" command (default=0)
</DD>
<DT><STRONG><TT>border</TT>:</STRONG></DT>
<DD>Relative size of border spacing around axis, between 0 - 1. (default=0)
<P>
If a scalar, equal border on all sides, give a four-element vector 
[left bottom right top] to define borders for each side.
</DD>
<DT><STRONG><TT>colormap</TT>:</STRONG></DT>
<DD>Figure colormap passed to the colormap function. If 

<P>
function handle, its output is passed instead.
</DD>
<DT><STRONG><TT>grid</TT>:</STRONG></DT>
<DD>Display dashed grid in background.
</DD>
<DT><STRONG><TT>noXLabel</TT>:</STRONG></DT>
<DD>No X-axis label.
</DD>
<DT><STRONG><TT>noYLabel</TT>:</STRONG></DT>
<DD>No Y-axis label.
</DD>
<DT><STRONG><TT>noTitle</TT>:</STRONG></DT>
<DD>No title.
</DD>
<DT><STRONG><TT>rotateXLabel</TT>:</STRONG></DT>
<DD>Rotates the X-axis label for smaller width.
</DD>
<DT><STRONG><TT>rotateYLabel</TT>:</STRONG></DT>
<DD>Rotates the Y-axis label for smaller width.
</DD>
<DT><STRONG><TT>numXTicks</TT>:</STRONG></DT>
<DD>Number of ticks on X-axis.
</DD>
<DT><STRONG><TT>formatXTickLabels</TT>:</STRONG></DT>
<DD>The sprintf format string for tick labels.
</DD>
<DT><STRONG><TT>XTick, YTick</TT>:</STRONG></DT>
<DD>Point locations for axis ticks.
</DD>
<DT><STRONG><TT>XTickLabel, YTickLabel</TT>:</STRONG></DT>
<DD>Axis tick labels.
</DD>
<DT><STRONG><TT>ColorOrder</TT>:</STRONG></DT>
<DD>Set the ColorOrder of the axis.
</DD>
<DT><STRONG><TT>LineStyleOrder</TT>:</STRONG></DT>
<DD>Set the LineStyleOrder of the axis.
</DD>
<DT><STRONG><TT>legendLocation</TT>:</STRONG></DT>
<DD>Passed to legend(..., 'location', legendLocation).
</DD>
<DT><STRONG><TT>legendOrientation</TT>:</STRONG></DT>
<DD>Passed to legend(..., 'orientation', legendLocation).
</DD>
<DT><STRONG><TT>noLegends</TT>:</STRONG></DT>
<DD>If exists, no legends are displayed.
</DD>
<DT><STRONG><TT>axisProps</TT>:</STRONG></DT>
<DD>Passed to set properties of the axis drawn.
</DD>
<DT><STRONG><TT>plotProps</TT>:</STRONG></DT>
<DD>Passed to set properties of the plot drawn.
</DD>
<DT><STRONG><TT>figureProps</TT>:</STRONG></DT>
<DD>Passed to set properties of the figure drawn.
</DD>
<DT><STRONG><TT>PaperPosition</TT>:</STRONG></DT>
<DD>Sets the figure property for printing at this size.
</DD>
<DT><STRONG><TT>resizeControl</TT>:</STRONG></DT>
<DD>If 0, drawing after resize is disabled and prints at screen 

<P>
size, if 1 (default), redraws figure after each resize event and 
prints at PaperPosition size.
</DD>
<DT><STRONG><TT>fixedSize</TT>:</STRONG></DT>
<DD>Vector of width and height in inches passed to

<P>
PaperPosition property. Implies resizeControl=0.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
data, axis_labels, title, legend, command, props
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16973"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16975"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126200000000000000">
1.26.2 Method <TT>plot_abstract/get</TT></A>
</H3><A NAME="16977"></A><A NAME="ref_plot_abstract__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126300000000000000">
1.26.3 Method <TT>plot_abstract/plot</TT></A>
</H3><A NAME="16979"></A><A NAME="ref_plot_abstract__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Draws this plot in the current axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = plot(a_plot, layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot (Optional). 
<P>
If NaN, doesn't open a new axis.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16981"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126400000000000000">
1.26.4 Method <TT>plot_abstract/setProp</TT></A>
</H3><A NAME="16983"></A><A NAME="ref_plot_abstract__setProp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting optional object properties.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = setProp(obj, prop1, val1, prop2, val2, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Modifies or adds property values. As many property name-value 
 pairs can be specified.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>Any object that has a props field.
</DD>
<DT><STRONG><TT>attr</TT>:</STRONG></DT>
<DD>Property name
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>Property value.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with the updated properties.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126500000000000000">
1.26.5 Method <TT>plot_abstract/set</TT></A>
</H3><A NAME="16985"></A><A NAME="ref_plot_abstract__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126600000000000000">
1.26.6 Method <TT>plot_abstract/subsref</TT></A>
</H3><A NAME="16987"></A><A NAME="ref_plot_abstract__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126700000000000000">
1.26.7 Method <TT>plot_abstract/subsasgn</TT></A>
</H3><A NAME="16989"></A><A NAME="ref_plot_abstract__subsasgn"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic index-based assignment for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126800000000000000">
1.26.8 Method <TT>plot_abstract/axis</TT></A>
</H3><A NAME="16991"></A><A NAME="ref_plot_abstract__axis"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the estimated axis ranges of this plot according to its data.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">ranges = axis(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ranges: The ranges as a vector in the same way 'axis' would return.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16993"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16995"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050126900000000000000">
1.26.9 Method <TT>plot_abstract/matrixPlots</TT></A>
</H3><A NAME="16997"></A><A NAME="ref_plot_abstract__matrixPlots"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superpose multiple plots with common command onto a single axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = matrixPlots(plots, axis_labels, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Array of plot_abstract or subclass objects.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings (optional, taken from plots).
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description string (optional, taken from plots).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to the Y stack_plot.
<DL>
<DD></DD>
<DT><STRONG><TT>titlesPos, yLabelsPos, yTicksPos</TT>:</STRONG></DT>
<DD>if specified, passed to the X stack_plots.
</DD>
<DT><STRONG><TT>rotateYLabel</TT>:</STRONG></DT>
<DD>if specified, passed to the X stack_plots.
</DD>
<DT><STRONG><TT>axisLimits</TT>:</STRONG></DT>
<DD>if specified, passed to the X stack_plots.
</DD>
<DT><STRONG><TT>goldratio</TT>:</STRONG></DT>
<DD>try to make the figure in this aspect ratio.
</DD>
<DT><STRONG><TT>width, height</TT>:</STRONG></DT>
<DD>if specified, make the figure have this many plots in 
<P>
corresponding dimension.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="16999"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17001"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17003"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501261000000000000000">
1.26.10 Method <TT>plot_abstract/display</TT></A>
</H3><A NAME="17005"></A><A NAME="ref_plot_abstract__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501261100000000000000">
1.26.11 Method <TT>plot_abstract/superposePlots</TT></A>
</H3><A NAME="17007"></A><A NAME="ref_plot_abstract__superposePlots"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superpose multiple plots with common command onto a single axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = superposePlots(plots, axis_labels, title_str, command, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The plot decoration will be taken from the last plot in the list, 
 with the exception of legend labels.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Array of plot_abstract or subclass objects.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings (optional, taken from plots).
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description string (optional, taken from plots).
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plotting command to use (optional, taken from plots)
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>noLegends</TT>:</STRONG></DT>
<DD>If exists, no legends are created.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17009"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17011"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17013"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/23
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501261200000000000000">
1.26.12 Method <TT>plot_abstract/decorate</TT></A>
</H3><A NAME="17015"></A><A NAME="ref_plot_abstract__decorate"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Places decorations (titles, labels, ticks, etc.) on the plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = decorate(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17017"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17019"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501261300000000000000">
1.26.13 Method <TT>plot_abstract/plotFigure</TT></A>
</H3><A NAME="17021"></A><A NAME="ref_plot_abstract__plotFigure"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Draws this plot alone in a new figure window.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handle = plotFigure(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handle: Handle of new figure.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17023"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17025"></A>, <TT>plot_abstract/decorate</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__decorate"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17027"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501261400000000000000">
1.26.14 Method <TT>plot_abstract/openAxis</TT></A>
</H3><A NAME="17029"></A><A NAME="ref_plot_abstract__openAxis"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the extents for the axis of this plot and opens it.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[axis_handle, layout_axis] = openAxis(a_plot, layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot (Optional). 
<P>
If NaN, doesn't open a new axis.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17031"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050127000000000000000">
1.27 Class <TT>plot_bars</TT></A>
</H2><A NAME="17033"></A><A NAME="ref_plot_bars"></A>
<H3><A NAME="SECTION050127100000000000000">
1.27.1 Constructor <TT>plot_bars/plot_bars</TT></A>
</H3><A NAME="17035"></A><A NAME="ref_plot_bars__plot_bars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Bar plot with error lines in individual axes for each variable.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_bars(mid_vals, lo_vals, hi_vals, n_vals, x_labels, y_labels, ...
		     title, axis_limits, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_stack. The plot_abstract/plot command can be used to
 plot this data. Rows of *_vals will create grouped bars, columns will
 create new axes.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>mid_vals</TT>:</STRONG></DT>
<DD>Middle points of error bars.
</DD>
<DT><STRONG><TT>lo_vals</TT>:</STRONG></DT>
<DD>Low points of error bars.
</DD>
<DT><STRONG><TT>hi_vals</TT>:</STRONG></DT>
<DD>High points of error bars.
</DD>
<DT><STRONG><TT>n_vals</TT>:</STRONG></DT>
<DD>Number of samples used for the statistic (Optional).
</DD>
<DT><STRONG><TT>x_labels, y_labels</TT>:</STRONG></DT>
<DD>Axis labels for each bar group. Must match with data columns.
</DD>
<DT><STRONG><TT>title</TT>:</STRONG></DT>
<DD>Plot description.
</DD>
<DT><STRONG><TT>axis_limits</TT>:</STRONG></DT>
<DD>If given, all plots contained will have these axis limits.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>dispBarsLines</TT>:</STRONG></DT>
<DD>Choose between using 'bars' or 'lines' to connect the errorbars.
</DD>
<DT><STRONG><TT>dispErrorbars</TT>:</STRONG></DT>
<DD>If 1, display errorbars for lo_vals and hi_vals deviation from mid_vals 
<P>
(default=1).
</DD>
<DT><STRONG><TT>dispNvals</TT>:</STRONG></DT>
<DD>If 1, display n_vals on top of each bar (default=1).
</DD>
<DT><STRONG><TT>groupValues</TT>:</STRONG></DT>
<DD>List of within-group labels passed to XTickLabels,
<P>
instead of just a sequence of numbers.
</DD>
<DT><STRONG><TT>truncateDecDigits</TT>:</STRONG></DT>
<DD>Truncate labels to this many decimal digits.
</DD>
<DT><STRONG><TT>barAxisProps</TT>:</STRONG></DT>
<DD>props passed to plot_abstract objects with bar commands
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17037"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17039"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050127200000000000000">
1.27.2 Method <TT>plot_bars/set</TT></A>
</H3><A NAME="17041"></A><A NAME="ref_plot_bars__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050128000000000000000">
1.28 Class <TT>plot_errorbar</TT></A>
</H2><A NAME="17043"></A><A NAME="ref_plot_errorbar"></A>
<H3><A NAME="SECTION050128100000000000000">
1.28.1 Constructor <TT>plot_errorbar/plot_errorbar</TT></A>
</H3><A NAME="17045"></A><A NAME="ref_plot_errorbar__plot_errorbar"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic errorbar plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_errorbar(x_vals, mid_vals, lo_vals, hi_vals, line_spec, 
			 axis_labels, title, legend, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. The plot_abstract/plot command can be used to
 plot this data. Needed to create this as a separate class to have the
 axis ranges method to measure the errorbars.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x_vals</TT>:</STRONG></DT>
<DD>X coordinates of errorbars.
</DD>
<DT><STRONG><TT>mid_vals</TT>:</STRONG></DT>
<DD>Middle points of error bars.
</DD>
<DT><STRONG><TT>lo_vals</TT>:</STRONG></DT>
<DD>Low points of error bars.
</DD>
<DT><STRONG><TT>hi_vals</TT>:</STRONG></DT>
<DD>High points of error bars.
</DD>
<DT><STRONG><TT>line_spec</TT>:</STRONG></DT>
<DD>Plot line spec to be passed to errorbar
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array for X, Y axis labels.
</DD>
<DT><STRONG><TT>title</TT>:</STRONG></DT>
<DD>Plot description.
</DD>
<DT><STRONG><TT>legend</TT>:</STRONG></DT>
<DD>For multiple errorbar plots (matrix form), description of each plot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties to be passed to plot_abstract.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17047"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17049"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050128200000000000000">
1.28.2 Method <TT>plot_errorbar/get</TT></A>
</H3><A NAME="17051"></A><A NAME="ref_plot_errorbar__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050128300000000000000">
1.28.3 Method <TT>plot_errorbar/axis</TT></A>
</H3><A NAME="17053"></A><A NAME="ref_plot_errorbar__axis"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the estimated axis ranges of this plot according to its data.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">ranges = axis(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ranges: The ranges as a vector in the same way 'axis' would return.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17055"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17057"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050129000000000000000">
1.29 Class <TT>plot_errorbars</TT></A>
</H2><A NAME="17059"></A><A NAME="ref_plot_errorbars"></A>
<H3><A NAME="SECTION050129100000000000000">
1.29.1 Constructor <TT>plot_errorbars/plot_errorbars</TT></A>
</H3><A NAME="17061"></A><A NAME="ref_plot_errorbars__plot_errorbars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots distributions of variables with errorbars in separate axes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_errorbars(labels, mid_vals, lo_vals, hi_vals, labels, 
			 title, axis_limits, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_stack. The plot_abstract/plot command can be used to
 plot this data. Each of mid_vals, lo_vals, and hi_vals plot its rows in
 the same axis and columns in different axes.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>labels</TT>:</STRONG></DT>
<DD>Labels of parameters to appear at bottom of each errorbar.
</DD>
<DT><STRONG><TT>mid_vals</TT>:</STRONG></DT>
<DD>Middle points of error bars.
</DD>
<DT><STRONG><TT>lo_vals</TT>:</STRONG></DT>
<DD>Low points of error bars.
</DD>
<DT><STRONG><TT>hi_vals</TT>:</STRONG></DT>
<DD>High points of error bars.
</DD>
<DT><STRONG><TT>title</TT>:</STRONG></DT>
<DD>Plot description.
</DD>
<DT><STRONG><TT>axis_limits</TT>:</STRONG></DT>
<DD>If given, all plots contained will have these axis limits.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract, labels.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17063"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17065"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050130000000000000000">
1.30 Class <TT>plot_image</TT></A>
</H2><A NAME="17067"></A><A NAME="ref_plot_image"></A>
<H3><A NAME="SECTION050130100000000000000">
1.30.1 Constructor <TT>plot_image/plot_image</TT></A>
</H3><A NAME="17069"></A><A NAME="ref_plot_image__plot_image"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic image plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_image(image_data, axis_labels, colorbar_label, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. The plot_abstract/plot command can be used to
 plot this data. Needed to create this as a separate class to have the
 axis ranges method implemented and take advantage of plot_abstract props.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>image_data</TT>:</STRONG></DT>
<DD>2D matrix with image data.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array for X, Y axis labels.
</DD>
<DT><STRONG><TT>colorbar_label</TT>:</STRONG></DT>
<DD>String to appear next to colorbar.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>colorbar</TT>:</STRONG></DT>
<DD>If defined, show colorbar next to plot.
</DD>
<DT><STRONG><TT>numGrads</TT>:</STRONG></DT>
<DD>Number of poles in the colormap gradient. If 1 (default),
<P>
it will be a monocolor gradient (e.g., gray-level); if 2, it will
be a dual-color gradient (e.g., blue to red) with a black
crossing point. This point is determined by the minValue
(below). Default numGrads=2, if negative values exist in
image_data after scaling.
</DD>
<DT><STRONG><TT>minValue,maxValue</TT>:</STRONG></DT>
<DD>Use these value to scale the data by 
<P>
(image_data - minValue) / (maxValue - minValue). 
Otherwise, its min and max is used.
</DD>
<DT><STRONG><TT>colormap</TT>:</STRONG></DT>
<DD>Colormap vector, function name or handle to colormap (e.g., 'jet').
</DD>
<DT><STRONG><TT>numColors</TT>:</STRONG></DT>
<DD>Number of colors in colormap.

<P>
(Rest passed to plotImage.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plot_image(rand(5), 'r1', 'r2', 'rand', 'random matrix'))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17071"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17073"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050130200000000000000">
1.30.2 Method <TT>plot_image/get</TT></A>
</H3><A NAME="17075"></A><A NAME="ref_plot_image__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050130300000000000000">
1.30.3 Method <TT>plot_image/axis</TT></A>
</H3><A NAME="17077"></A><A NAME="ref_plot_image__axis"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the estimated axis ranges of this plot according to its data.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">ranges = axis(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract, or subclass, object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ranges: The ranges as a vector in the same way 'axis' would return.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17079"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17081"></A>, <TT>axis</TT> (p.&nbsp;<A HREF="#ref_axis"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17083"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050131000000000000000">
1.31 Class <TT>plot_inset</TT></A>
</H2><A NAME="17085"></A><A NAME="ref_plot_inset"></A>
<H3><A NAME="SECTION050131100000000000000">
1.31.1 Constructor <TT>plot_inset/plot_inset</TT></A>
</H3><A NAME="17087"></A><A NAME="ref_plot_inset__plot_inset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superpose multiple plots with individual axis at arbitrary locations.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_inset(plots, axis_locations, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. Contains other plot_abstract objects or
 subclasses thereof to be layout in arbitaray format. Allows overlapping
 and therefore good for insets and special plots.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Cell array of plot_abstract or subclass objects.
</DD>
<DT><STRONG><TT>axis_locations</TT>:</STRONG></DT>
<DD>Matrix of four-element vectors for each given plot.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Title to go on top of the stack
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>positioning</TT>:</STRONG></DT>
<DD>axis_locations interpreted as 'absolute' values or
<P>
'relative' to the 1st plot (default='absolute'). 
Relative positioning doesn't work well.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract, plots, axis_locations.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17089"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17091"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/06/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050131200000000000000">
1.31.2 Method <TT>plot_inset/get</TT></A>
</H3><A NAME="17093"></A><A NAME="ref_plot_inset__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050131300000000000000">
1.31.3 Method <TT>plot_inset/plot</TT></A>
</H3><A NAME="17095"></A><A NAME="ref_plot_inset__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superposes contained plots in their own axes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = plot(a_plot, layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_superpose object.
</DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot (Optional). 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17097"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/06/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050131400000000000000">
1.31.4 Method <TT>plot_inset/set</TT></A>
</H3><A NAME="17099"></A><A NAME="ref_plot_inset__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050132000000000000000">
1.32 Class <TT>plot_simple</TT></A>
</H2><A NAME="17101"></A><A NAME="ref_plot_simple"></A>
<H3><A NAME="SECTION050132100000000000000">
1.32.1 Constructor <TT>plot_simple/plot_simple</TT></A>
</H3><A NAME="17103"></A><A NAME="ref_plot_simple__plot_simple"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Abstract description of a single plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_simple(data_x, data_y, title, 
		       label_x, label_y, legend, command, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. The plot_abstract/plot command can be used to
 plot this data.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data_x</TT>:</STRONG></DT>
<DD>X-axis values for the plot.
</DD>
<DT><STRONG><TT>data_y</TT>:</STRONG></DT>
<DD>Y-axis values for the plot.
</DD>
<DT><STRONG><TT>title</TT>:</STRONG></DT>
<DD>Plot description.
</DD>
<DT><STRONG><TT>label_x</TT>:</STRONG></DT>
<DD>X-axis label string.
</DD>
<DT><STRONG><TT>label_y</TT>:</STRONG></DT>
<DD>Y-axis label string.
</DD>
<DT><STRONG><TT>legend</TT>:</STRONG></DT>
<DD>Short description of data points.
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plotting command to use (Optional, default='plot')
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17105"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17107"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050132200000000000000">
1.32.2 Method <TT>plot_simple/get</TT></A>
</H3><A NAME="17109"></A><A NAME="ref_plot_simple__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050132300000000000000">
1.32.3 Method <TT>plot_simple/set</TT></A>
</H3><A NAME="17111"></A><A NAME="ref_plot_simple__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050133000000000000000">
1.33 Class <TT>plot_stack</TT></A>
</H2><A NAME="17113"></A><A NAME="ref_plot_stack"></A>
<H3><A NAME="SECTION050133100000000000000">
1.33.1 Constructor <TT>plot_stack/plot_stack</TT></A>
</H3><A NAME="17115"></A><A NAME="ref_plot_stack__plot_stack"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A horizontal or vertical stack of plots.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_stack(plots, axis_limits, orientation, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. Contains other plot_abstract objects or
 subclasses thereof to be layout in stack format. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Cell array of plot_abstract or subclass objects.
</DD>
<DT><STRONG><TT>axis_limits</TT>:</STRONG></DT>
<DD>If given, all plots contained will have these axis
<P>
limits. In this vector, NaNs are untouched, Infs are
replaced by minimal and maximal ranges of the
stacked plots.
</DD>
<DT><STRONG><TT>orientation</TT>:</STRONG></DT>
<DD>Stack orientation 'x' for horizontal, 'y' for vertical, etc.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Title to go on top of the stack
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>yLabelsPos</TT>:</STRONG></DT>
<DD>'left' means only put y-axis label to leftmost plot.
</DD>
<DT><STRONG><TT>yTicksPos</TT>:</STRONG></DT>
<DD>'left' means only put y-axis ticks to leftmost plot.
</DD>
<DT><STRONG><TT>xLabelsPos</TT>:</STRONG></DT>
<DD>'bottom' means only put x-axis label to lowest plot.
</DD>
<DT><STRONG><TT>xTicksPos</TT>:</STRONG></DT>
<DD>'bottom' means only put x-axis ticks to lowest plot.
</DD>
<DT><STRONG><TT>titlesPos</TT>:</STRONG></DT>
<DD>'top' means only put title to top plot.
</DD>
<DT><STRONG><TT>relaxedLimits</TT>:</STRONG></DT>
<DD>Add 10
</DD>
<DT><STRONG><TT>relativeSizes</TT>:</STRONG></DT>
<DD>An array specifying relative size of each plot with one value.
<P>
(Example: relativeSizes=[1 2] makes second plot twice wider than first.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract, plots, axis_limits, orient.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17117"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17119"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133200000000000000">
1.33.2 Method <TT>plot_stack/get</TT></A>
</H3><A NAME="17121"></A><A NAME="ref_plot_stack__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133300000000000000">
1.33.3 Method <TT>plot_stack/plot</TT></A>
</H3><A NAME="17123"></A><A NAME="ref_plot_stack__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Draws this plot in the current axis or at the position in
	layout_axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = plot(a_plot, layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot (Optional). 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_stack</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_stack"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17125"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17127"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133400000000000000">
1.33.4 Method <TT>plot_stack/set</TT></A>
</H3><A NAME="17129"></A><A NAME="ref_plot_stack__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133500000000000000">
1.33.5 Method <TT>plot_stack/display</TT></A>
</H3><A NAME="17131"></A><A NAME="ref_plot_stack__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133600000000000000">
1.33.6 Method <TT>plot_stack/superposePlots</TT></A>
</H3><A NAME="17133"></A><A NAME="ref_plot_stack__superposePlots"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superpose multiple plot_stack objects that contain exact same contents.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = superposePlots(plots, axis_labels, title_str, command, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The plot decoration will be taken from the last plot in the list, 
 with the exception of legend labels.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Array of plot_stack objects.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings (optional, taken from plots).
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description string (optional, taken from plots).
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plotting command to use (optional, taken from plots)
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>noLegends</TT>:</STRONG></DT>
<DD>If exists, no legends are created.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_stack object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17135"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17137"></A>, <TT>plot_abstract/plotFigure</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17139"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050133700000000000000">
1.33.7 Method <TT>plot_stack/decorate</TT></A>
</H3><A NAME="17141"></A><A NAME="ref_plot_stack__decorate"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>No additional decorations for stacked plots.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_histogram_db = decorate(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17143"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17145"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050134000000000000000">
1.34 Class <TT>plot_superpose</TT></A>
</H2><A NAME="17147"></A><A NAME="ref_plot_superpose"></A>
<H3><A NAME="SECTION050134100000000000000">
1.34.1 Constructor <TT>plot_superpose/plot_superpose</TT></A>
</H3><A NAME="17149"></A><A NAME="ref_plot_superpose__plot_superpose"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Multiple plot_abstract objects superposed on the same axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = plot_superpose(plots, axis_labels, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Subclass of plot_abstract. Contains multiple plot_abstract objects to be
 plotted on the same axis. This is different than the
 plot_abstract/superpose, where only using the same plot command is
 allowed.  Here, each plot_abstract can have its own special plotting
 command. Subclasses of plot_abstract is also allowed here. The decorations
 comes from this object and not children plots. This behavior is different
 than plot_stack, where each plot has its own decorations. If you want each
 plot to have its own axis (e.g. an inset, or plot with multiple axis
 labels) then you should use plot_inset.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Cell array of plot_abstract or subclass objects.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties (passed to plot_abstract).
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
plot_abstract, plots
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/superpose</TT> (p.&nbsp;<A HREF="#ref_plot_abstract__superpose"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17151"></A>, <TT>plot_superpose/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_superpose__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17153"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134200000000000000">
1.34.2 Method <TT>plot_superpose/get</TT></A>
</H3><A NAME="17155"></A><A NAME="ref_plot_superpose__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134300000000000000">
1.34.3 Method <TT>plot_superpose/plot</TT></A>
</H3><A NAME="17157"></A><A NAME="ref_plot_superpose__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Draws this plot in the current axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = plot(a_plot, layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_superpose object.
</DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot (Optional). 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17159"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134400000000000000">
1.34.4 Method <TT>plot_superpose/set</TT></A>
</H3><A NAME="17161"></A><A NAME="ref_plot_superpose__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134500000000000000">
1.34.5 Method <TT>plot_superpose/axis</TT></A>
</H3><A NAME="17163"></A><A NAME="ref_plot_superpose__axis"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the maximal axis ranges according to superposed subplots.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">ranges = axis(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ranges: The ranges as a vector in the same way 'axis' would return.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17165"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17167"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134600000000000000">
1.34.6 Method <TT>plot_superpose/display</TT></A>
</H3><A NAME="17169"></A><A NAME="ref_plot_superpose__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134700000000000000">
1.34.7 Method <TT>plot_superpose/superposePlots</TT></A>
</H3><A NAME="17171"></A><A NAME="ref_plot_superpose__superposePlots"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Superpose multiple plot_superpose objects by merging them into one.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = superposePlots(plots, axis_labels, title_str, command, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>plots</TT>:</STRONG></DT>
<DD>Array of plot_superpose objects.
</DD>
<DT><STRONG><TT>axis_labels</TT>:</STRONG></DT>
<DD>Cell array of axis label strings (optional, taken from plots).
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Plot description string (optional, taken from plots).
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>Plotting command to use (optional, taken from plots)
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>noLegends</TT>:</STRONG></DT>
<DD>If exists, no legends are created.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_superpose object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/superposePlots</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__superposePlots"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17173"></A>, <TT>plot_stack/superposePlots</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_stack__superposePlots"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17175"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050134800000000000000">
1.34.8 Method <TT>plot_superpose/decorate</TT></A>
</H3><A NAME="17177"></A><A NAME="ref_plot_superpose__decorate"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Places decorations using the first plot of the superposed plots.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = decorate(a_plot)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_plot</TT>:</STRONG></DT>
<DD>A plot_abstract object, or a subclass object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles of graphical objects drawn.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17179"></A>, <TT>plot_abstract/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17181"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050135000000000000000">
1.35 Class <TT>ranked_db</TT></A>
</H2><A NAME="17183"></A><A NAME="ref_ranked_db"></A>
<H3><A NAME="SECTION050135100000000000000">
1.35.1 Constructor <TT>ranked_db/ranked_db</TT></A>
</H3><A NAME="17185"></A><A NAME="ref_ranked_db__ranked_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of distance values generated by ranking rows of orig_db with the criterion in crit_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ranked_db = ranked_db(data, col_names, orig_db, crit_db, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. It should contain a Distance column. A
 more general ranked db class may be needed later. Use the rankMatching method
 to get an instance of this class.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>Database contents.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>The column names.
</DD>
<DT><STRONG><TT>orig_db</TT>:</STRONG></DT>
<DD>DB whose rows are ranked.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>The criterion DB used for generating the ranking scores.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>tolerateNaNs</TT>:</STRONG></DT>
<DD>If 0, rows with any NaN values are skipped (default=1).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, orig_db, crit_db, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17187"></A>, <TT>tests_db/rankMatching</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17189"></A>, <TT>tests_db/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17191"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135200000000000000">
1.35.2 Method <TT>ranked_db/get</TT></A>
</H3><A NAME="17193"></A><A NAME="ref_ranked_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135300000000000000">
1.35.3 Method <TT>ranked_db/plotRowErrors</TT></A>
</H3><A NAME="17195"></A><A NAME="ref_ranked_db__plotRowErrors"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create plot of rankings with errors associated with each measure color-coded.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotRowErrors(a_ranked_db, rows, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_ranked_db</TT>:</STRONG></DT>
<DD>A ranked_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in a_ranked_db.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>sortMeasures</TT>:</STRONG></DT>
<DD>If specified, measure order is determined with increasing 
<P>
overall distance.
</DD>
<DT><STRONG><TT>RowName</TT>:</STRONG></DT>
<DD>Label to show on X-axis (default='Ranks')
</DD>
<DT><STRONG><TT>rowSteps</TT>:</STRONG></DT>
<DD>Steps to jump in labeling rows on the x-axis.
</DD>
<DT><STRONG><TT>superposeDistances</TT>:</STRONG></DT>
<DD>Superpose a white-colored distance line plot.
<P>
(rest passed to plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>ranked_db</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17197"></A>, <TT>tests_db/rankMatching</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17199"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17201"></A>, <TT>plotImage</TT> (p.&nbsp;<A HREF="#ref_plotImage"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17203"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135400000000000000">
1.35.4 Method <TT>ranked_db/joinOriginal</TT></A>
</H3><A NAME="17205"></A><A NAME="ref_ranked_db__joinOriginal"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Joins the distance values to the original db rows with matching row indices.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = joinOriginal(a_ranked_db, rows)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Takes the parameter columns from orig_db and all tests from crit_db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_ranked_db</TT>:</STRONG></DT>
<DD>A ranked_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Join only the given rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A params_tests_db object (same type as a_ranked_db.orig_db) containing 
		the desired rows in ascending order of distance.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17207"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135500000000000000">
1.35.5 Method <TT>ranked_db/plotDistMatrix</TT></A>
</H3><A NAME="17209"></A><A NAME="ref_ranked_db__plotDistMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a color-coded matrix plot of with total errors from the ranked DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotDistMatrix(db, rows, col_size, col_name, num_col_labels, 
			  row_name, num_row_labels, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The col_size parameter is used to find the number of rows that make up the 
 x-dimension of the color matrix plot.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A ranked_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db after joining (and sorting).
</DD>
<DT><STRONG><TT>col_size</TT>:</STRONG></DT>
<DD>Number of rows to take from DB to form the columns of matrix plot.
</DD>
<DT><STRONG><TT>col_name, row_name</TT>:</STRONG></DT>
<DD>DB column to use for the figure column and row, respectively.
</DD>
<DT><STRONG><TT>num_col_labels, num_row_labels</TT>:</STRONG></DT>
<DD>Number of labels to put on each axis.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>If non-empty, replaces generic title with db name. 
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>sortBy</TT>:</STRONG></DT>
<DD>If specified, db is sorted after being joined with original using this column.
</DD>
<DT><STRONG><TT>colorbar</TT>:</STRONG></DT>
<DD>Put a colorbar on the figure.
<P>
(also passed to plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plotDistMatrix(scored_blocked_sk_gps0503b_control_db, ':', 10, 'SK', 10, 'trial', 10, 'gps0503b (control), preset 6 - top 50 matches', struct('sortBy', 'trial', 'colorbar', 1, 'PaperPosition', [0 0 5 3])));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>ranked_db</TT> (p.&nbsp;<A HREF="node5.html#ref_ranked_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17211"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17213"></A>, <TT>getDistMatrix</TT> (p.&nbsp;<A HREF="#ref_getDistMatrix"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17215"></A>, <TT>plotCompareDistMatx</TT> (p.&nbsp;<A HREF="#ref_plotCompareDistMatx"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17217"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135600000000000000">
1.35.6 Method <TT>ranked_db/set</TT></A>
</H3><A NAME="17219"></A><A NAME="ref_ranked_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135700000000000000">
1.35.7 Method <TT>ranked_db/renameColumns</TT></A>
</H3><A NAME="17221"></A><A NAME="ref_ranked_db__renameColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Rename an existing column or columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = renameColumns(a_db, test_names, new_names)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is an overloaded method for ranked_db that keeps consistent
 the column names of the ranked, criterion and original DBs. The other
 DBs are not renamed for the Distance and RowIndex columns.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A ranked_db object.
</DD>
<DT><STRONG><TT>test_names</TT>:</STRONG></DT>
<DD>A cell array of existing test names.
</DD>
<DT><STRONG><TT>new_names</TT>:</STRONG></DT>
<DD>New names to replace existing ones.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The ranked_db object that includes the new columns.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/renameColumns</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__renameColumns"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17223"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135800000000000000">
1.35.8 Method <TT>ranked_db/subsref</TT></A>
</H3><A NAME="17225"></A><A NAME="ref_ranked_db__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050135900000000000000">
1.35.9 Method <TT>ranked_db/plotCompareDistMatx</TT></A>
</H3><A NAME="17227"></A><A NAME="ref_ranked_db__plotCompareDistMatx"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Compare differences and correlations of distance matrices from two ranked DBs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotCompareDistMatx(db, rows, col_size, col_name, num_col_labels, 
			  row_name, num_row_labels, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Produces three plots: (1) distance difference matrix, (2) 2D cross-correlogram, 
 and (3) repeated 1D cross-correlogram for each row.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db, w_db</TT>:</STRONG></DT>
<DD>The ranked_db objects to be compared.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db after joining (and sorting) for both DBs.
</DD>
<DT><STRONG><TT>col_size</TT>:</STRONG></DT>
<DD>Number of rows to take from DB to form the columns of matrix plot.
</DD>
<DT><STRONG><TT>col_name, row_name</TT>:</STRONG></DT>
<DD>DB column to use fot the figure column and row, respectively.
</DD>
<DT><STRONG><TT>num_col_labels, num_row_labels</TT>:</STRONG></DT>
<DD>Number of labels to put on each axis.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>If non-empty, replaces generic title with db name. 
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>sortBy</TT>:</STRONG></DT>
<DD>If specified, db is sorted after being joined with original using this column.
</DD>
<DT><STRONG><TT>colorbar</TT>:</STRONG></DT>
<DD>Put a colorbar on the figure.
<P>
(also passed to plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17229"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17231"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501351000000000000000">
1.35.10 Method <TT>ranked_db/displayRows</TT></A>
</H3><A NAME="17233"></A><A NAME="ref_ranked_db__displayRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Displays rows of rankings together with errors associated with each measure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = displayRows(db, rows)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: A structure of column name and value pairs.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17235"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501351100000000000000">
1.35.11 Method <TT>ranked_db/getDistMatrix</TT></A>
</H3><A NAME="17237"></A><A NAME="ref_ranked_db__getDistMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a matrix of total errors from the ranked DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">distmatx = getDistMatrix(db, rows, col_size, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The col_size parameter is used to find the number of rows that make up the 
 x-dimension of the matrix.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db after joining (and sorting).
</DD>
<DT><STRONG><TT>col_size</TT>:</STRONG></DT>
<DD>Number of rows to take from DB to form the columns of matrix plot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>sortBy</TT>:</STRONG></DT>
<DD>If specified, db is sorted after being joined with original using this column.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17239"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17241"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501351200000000000000">
1.35.12 Method <TT>ranked_db/blockedDistances</TT></A>
</H3><A NAME="17243"></A><A NAME="ref_ranked_db__blockedDistances"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a db of distances to blocked versions of top ranks.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_db, ranked_dbs] = 
   blockedDistances(a_ranked_db, rows, blocked_db, blocked_param_indices, 
	  	     block_levels, crit_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_ranked_db</TT>:</STRONG></DT>
<DD>A ranked_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Use the given row rankings.
</DD>
<DT><STRONG><TT>blocked_db</TT>:</STRONG></DT>
<DD>db with blocked versions of original ranks.
</DD>
<DT><STRONG><TT>blocked_param_indices</TT>:</STRONG></DT>
<DD>Indices of parameters to be blocked.
</DD>
<DT><STRONG><TT>block_levels</TT>:</STRONG></DT>
<DD>Number of parameter levels for blocking.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>Calculate distance from this criterion.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object with the matrix of distances.
	ranked_dbs: A cell array of ranked_dbs for each row.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; dist_matx_db = blockedDistances(rankMatching(super_db, matchingRow(rsuper_phys_db, 20)), 1:5, super_blocker_db, [1 2], 10, matchingRow(rsuper_phys_db, 21))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>makeModifiedParamDB</TT> (p.&nbsp;<A HREF="#ref_makeModifiedParamDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17245"></A>, <TT>getParamRowIndices</TT> (p.&nbsp;<A HREF="#ref_getParamRowIndices"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17247"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050136000000000000000">
1.36 Class <TT>results_profile</TT></A>
</H2><A NAME="17249"></A><A NAME="ref_results_profile"></A>
<H3><A NAME="SECTION050136100000000000000">
1.36.1 Constructor <TT>results_profile/results_profile</TT></A>
</H3><A NAME="17251"></A><A NAME="ref_results_profile__results_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates and collects result profiles for data objects.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = results_profile(results, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is the base class for all profile classes.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>results</TT>:</STRONG></DT>
<DD>A structure containing test results.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
results, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17253"></A>, <TT>cip_trace_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17255"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050136200000000000000">
1.36.2 Method <TT>results_profile/get</TT></A>
</H3><A NAME="17257"></A><A NAME="ref_results_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050136300000000000000">
1.36.3 Method <TT>results_profile/plot</TT></A>
</H3><A NAME="17259"></A><A NAME="ref_results_profile__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method to plot a tests_db or a subclass. Requires a 
	plot_abstract method to be defined for this object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plot(a_tests_db, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
h: The figure handle created.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17261"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17263"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050136400000000000000">
1.36.4 Method <TT>results_profile/getResults</TT></A>
</H3><A NAME="17265"></A><A NAME="ref_results_profile__getResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Return the results profile structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getResults(p)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p</TT>:</STRONG></DT>
<DD>A result_profile object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names to values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>results_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_results_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17267"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050136500000000000000">
1.36.5 Method <TT>results_profile/subsref</TT></A>
</H3><A NAME="17269"></A><A NAME="ref_results_profile__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050136600000000000000">
1.36.6 Method <TT>results_profile/display</TT></A>
</H3><A NAME="17271"></A><A NAME="ref_results_profile__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050137000000000000000">
1.37 Class <TT>script_array</TT></A>
</H2><A NAME="17273"></A><A NAME="ref_script_array"></A>
<H3><A NAME="SECTION050137100000000000000">
1.37.1 Constructor <TT>script_array/script_array</TT></A>
</H3><A NAME="17275"></A><A NAME="ref_script_array__script_array"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic class that provides the scripts for a repetitive array job.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = script_array(num_runs, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is the base class for all script_array classes. Runs the runJob method as 
 num_runs many times.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>num_runs</TT>:</STRONG></DT>
<DD>The number of times the runJob script should be evoked.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>runJobFunc</TT>:</STRONG></DT>
<DD>A function name or handle to be used instead of default runJob.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
num_runs, id, props.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; func1 = inline('x&#94;2')
<BR>&#187; runFirst(script_array(10, 'squares numbers up to 10'), struct('runJobFunc', func1))
<BR>
ans = [  1]    [  4]    [  9]    [ 16]    [ 25]    [ 36]    [ 49]    [ 64]    [ 81]    [100]
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>runFirst</TT> (p.&nbsp;<A HREF="#ref_runFirst"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17277"></A>, <TT>runLast</TT> (p.&nbsp;<A HREF="#ref_runLast"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17279"></A>, <TT>runJob</TT> (p.&nbsp;<A HREF="#ref_runJob"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17281"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137200000000000000">
1.37.2 Method <TT>script_array/get</TT></A>
</H3><A NAME="17283"></A><A NAME="ref_script_array__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137300000000000000">
1.37.3 Method <TT>script_array/runJob</TT></A>
</H3><A NAME="17285"></A><A NAME="ref_script_array__runJob"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Method to be called for each of the script_array jobs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">job_result = runJob(a_script_array, vector_index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is provided as a placeholder and does nothing. If the run_job_func
 property is defined, it will call that function.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_script_array</TT>:</STRONG></DT>
<DD>A script_array object.
</DD>
<DT><STRONG><TT>vector_index</TT>:</STRONG></DT>
<DD>The index within the vector job.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
job_result: Any output produced by the job.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">See real example in script_array. Call the 5th job:
<BR>&#187; runJob(script_array(10, 'this one does nothing for 10 times'), 5);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>runLast</TT> (p.&nbsp;<A HREF="#ref_runLast"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17287"></A>, <TT>runFirst</TT> (p.&nbsp;<A HREF="#ref_runFirst"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17289"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137400000000000000">
1.37.4 Method <TT>script_array/runLast</TT></A>
</H3><A NAME="17291"></A><A NAME="ref_script_array__runLast"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Method to be called last after the script_array jobs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">job_results = runLast(a_script_array, job_results)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method is provided as a placeholder and does nothing. It can filter-out the
 results returned from the jobs run. Normally it is invoked internally by the runFirst
 method, after running and collecting results from the vector jobs with the runJob method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_script_array</TT>:</STRONG></DT>
<DD>A script_array object.
</DD>
<DT><STRONG><TT>job_results</TT>:</STRONG></DT>
<DD>The index within the vector job.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
job_results: Any output produced by the job.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Call it directly:
<BR>&#187; runLast(script_array(10, 'this one does nothing for 10 times'), );
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>runJob</TT> (p.&nbsp;<A HREF="#ref_runJob"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17293"></A>, <TT>runFirst</TT> (p.&nbsp;<A HREF="#ref_runFirst"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17295"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137500000000000000">
1.37.5 Method <TT>script_array/set</TT></A>
</H3><A NAME="17297"></A><A NAME="ref_script_array__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137600000000000000">
1.37.6 Method <TT>script_array/subsref</TT></A>
</H3><A NAME="17299"></A><A NAME="ref_script_array__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137700000000000000">
1.37.7 Method <TT>script_array/subsasgn</TT></A>
</H3><A NAME="17301"></A><A NAME="ref_script_array__subsasgn"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic index-based assignment for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050137800000000000000">
1.37.8 Method <TT>script_array/runFirst</TT></A>
</H3><A NAME="17303"></A><A NAME="ref_script_array__runFirst"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Method to be called at beginning of script_array jobs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">job_results = runFirst(a_script_array)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method initiates the script_array jobs. It loops and calls runJob and 
 finally calls runLast.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_script_array</TT>:</STRONG></DT>
<DD>A script_array object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
job_results: A cell array of results collected from each item of the vector jobs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; runFirst(script_array(10, 'this one does nothing for 10 times'));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>runLast</TT> (p.&nbsp;<A HREF="#ref_runLast"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17305"></A>, <TT>runJob</TT> (p.&nbsp;<A HREF="#ref_runJob"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17307"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050138000000000000000">
1.38 Class <TT>script_array_for_cluster</TT></A>
</H2><A NAME="17309"></A><A NAME="ref_script_array_for_cluster"></A>
<H3><A NAME="SECTION050138100000000000000">
1.38.1 Constructor <TT>script_array_for_cluster/script_array_for_cluster</TT></A>
</H3><A NAME="17311"></A><A NAME="ref_script_array_for_cluster__script_array_for_cluster"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic class defining a repetitive vector job to be run on a Sun Grid Engine (SGE) computing cluster.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_script_cluster = script_array_for_cluster(num_runs, sge_wrapper_script, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of the script_array class. The runFirst method spawns num_runs
 copies of the runJob method in parallel on the cluster, followed by the invocation 
 of the runLast method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>num_runs</TT>:</STRONG></DT>
<DD>The number of times the runJob script should be evoked.
</DD>
<DT><STRONG><TT>sge_wrapper_script</TT>:</STRONG></DT>
<DD>A script that can be submitted with qsub and can execute arbitrary
<P>
Matlab commands on the cluster nodes. It can have qsub options prepended to it
such as '-p -100 -q all.q &lt;abs_path_to&gt;/sge_matlab.sh'.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>notifyByMail</TT>:</STRONG></DT>
<DD>An SGE notification email is sent to this address after lastJob.
<P>
(others passed to script_array)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
num_runs, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>runFirst</TT> (p.&nbsp;<A HREF="#ref_runFirst"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17313"></A>, <TT>runLast</TT> (p.&nbsp;<A HREF="#ref_runLast"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17315"></A>, <TT>runJob</TT> (p.&nbsp;<A HREF="#ref_runJob"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17317"></A>, <TT>script_array</TT> (p.&nbsp;<A HREF="node5.html#ref_script_array"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17319"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050138200000000000000">
1.38.2 Method <TT>script_array_for_cluster/get</TT></A>
</H3><A NAME="17321"></A><A NAME="ref_script_array_for_cluster__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050138300000000000000">
1.38.3 Method <TT>script_array_for_cluster/set</TT></A>
</H3><A NAME="17323"></A><A NAME="ref_script_array_for_cluster__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050138400000000000000">
1.38.4 Method <TT>script_array_for_cluster/runFirst</TT></A>
</H3><A NAME="17325"></A><A NAME="ref_script_array_for_cluster__runFirst"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Method to be called at beginning of script_array_for_cluster jobs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">job_results = runFirst(a_script_cluster)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This method initiates the script_array_for_cluster jobs. It submits an SGE
 vector job for running each runJob and finally runLast. There is no way of
 collecting outputs from individual runJob calls.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_script_cluster</TT>:</STRONG></DT>
<DD>A script_array_for_cluster object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
job_results: A cell array of results collected from each item of the vector jobs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; runFirst(script_array_for_cluster(10, 'this one does nothing for 10 times'));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>script_array_for_cluster</TT> (p.&nbsp;<A HREF="node5.html#ref_script_array_for_cluster"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17327"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050139000000000000000">
1.39 Class <TT>script_factory</TT></A>
</H2><A NAME="17329"></A><A NAME="ref_script_factory"></A>
<H3><A NAME="SECTION050139100000000000000">
1.39.1 Constructor <TT>script_factory/script_factory</TT></A>
</H3><A NAME="17331"></A><A NAME="ref_script_factory__script_factory"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic class to automatically create a set of scripts.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = script_factory(num_scripts, out_name, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is the base class for all script_factory classes.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>num_scripts</TT>:</STRONG></DT>
<DD>Number of scripts to create.
</DD>
<DT><STRONG><TT>out_name</TT>:</STRONG></DT>
<DD>The file name for the output scripts. A '<P>
filename corresponds to the script number.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
num_scripts, out_name, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>script_factory/writeScripts</TT> (p.&nbsp;<A HREF="#ref_script_factory__writeScripts"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17333"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/11/28
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050139200000000000000">
1.39.2 Method <TT>script_factory/get</TT></A>
</H3><A NAME="17335"></A><A NAME="ref_script_factory__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050140000000000000000">
1.40 Class <TT>spike_shape</TT></A>
</H2><A NAME="17337"></A><A NAME="ref_spike_shape"></A>
<H3><A NAME="SECTION050140100000000000000">
1.40.1 Constructor <TT>spike_shape/spike_shape</TT></A>
</H3><A NAME="17339"></A><A NAME="ref_spike_shape__spike_shape"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>An action potential shape trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = spike_shape(data, dt, dy, id)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>A vector of data points containing the spike shape.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s].
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>baseline</TT>:</STRONG></DT>
<DD>Resting potential.
</DD>
<DT><STRONG><TT>threshold</TT>:</STRONG></DT>
<DD>Spike threshold.
</DD>
<DT><STRONG><TT>init_Vm_method</TT>:</STRONG></DT>
<DD>Method to obtain spike initiation voltage.
<P>
1- maximum acceleration point
2- threshold crossing of acceleration (needs threshold)
3- threshold crossing of slope (needs threshold)
4- maximum acceleration in phase space
(optionally specify maximal threshold as init_threshold)
5- point of maximum curvature, when slope is between 
init_lo_thr and init_hi_thr
6- local maximum of second derivative in the phase space
nearest slope crossing init_threshold
7- threshold crossing of interpolated slope (needs threshold)
8- maximum curvature in phase-plane
9- Combined curvature and inflection method in time-domain.
</DD>
<DT><STRONG><TT>init_threshold</TT>:</STRONG></DT>
<DD>Spike initiation threshold (deriv or accel).
<P>
(see above methods and implementation in calcInitVm)
</DD>
<DT><STRONG><TT>init_lo_thr, init_hi_thr</TT>:</STRONG></DT>
<DD>Low and high thresholds for slope.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
trace, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace/spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17341"></A>, <TT>trace/analyzeSpikesInPeriod</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__analyzeSpikesInPeriod"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17343"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17345"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17347"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17349"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140200000000000000">
1.40.2 Method <TT>spike_shape/calcInitVmV2PPLocal</TT></A>
</H3><A NAME="17351"></A><A NAME="ref_spike_shape__calcInitVmV2PPLocal"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the action potential threshold by finding the local second derivative maximum in voltage-time slope versus voltage phase plane, nearest a slope threshold crossing.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmV2PPLocal(s, max_idx, min_idx, lo_thr, plotit)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>lo_thr</TT>:</STRONG></DT>
<DD>Lower threshold for time voltage slope.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: Its index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17353"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140300000000000000">
1.40.3 Method <TT>spike_shape/get</TT></A>
</H3><A NAME="17355"></A><A NAME="ref_spike_shape__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140400000000000000">
1.40.4 Method <TT>spike_shape/calcMinVm</TT></A>
</H3><A NAME="17357"></A><A NAME="ref_spike_shape__calcMinVm"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the minimal value of the spike_shape, s. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[min_val, min_idx, max_min_time] = calcMinVm(s, max_idx)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
min_val: The min value [dy].
	min_idx: Its index in the spike_shape [dt].
	max_min_time: Time from max to min [dt].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17359"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17361"></A>, <TT>trace/calcMin</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__calcMin"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17363"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140500000000000000">
1.40.5 Method <TT>spike_shape/calcInitVmSekerliV2</TT></A>
</H3><A NAME="17365"></A><A NAME="ref_spike_shape__calcInitVmSekerliV2"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the action potential threshold using the maximum second derivative of the phase space of voltage-time slope versus voltage.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmSekerliV2(s, max_idx, min_idx, plotit)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: Its index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17367"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140600000000000000">
1.40.6 Method <TT>spike_shape/calcInitVmMaxCurvature</TT></A>
</H3><A NAME="17369"></A><A NAME="ref_spike_shape__calcInitVmMaxCurvature"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the action potential threshold using the
			maximum of the curvature equation.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmMaxCurvature(s, max_idx, min_idx, plotit)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Point of maximum curvature: Kp = V''[1 + (V')&#94;2]&#94;(-3/2)
 Taken from Sekerli, Del Negro, Lee and Butera. 
 IEEE Trans. Biomed. Eng., 51(9): 1665-71, 2004.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: AP threshold index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17371"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140700000000000000">
1.40.7 Method <TT>spike_shape/calcMaxVm</TT></A>
</H3><A NAME="17373"></A><A NAME="ref_spike_shape__calcMaxVm"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the maximal value of the spike_shape, s. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[max_val, max_idx] = calcMaxVm(s)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
max_val: The max value.
	max_idx: Its index in the spike_shape [dt].
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17375"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17377"></A>, <TT>trace/calcMax</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__calcMax"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17379"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140800000000000000">
1.40.8 Method <TT>spike_shape/getResults</TT></A>
</H3><A NAME="17381"></A><A NAME="ref_spike_shape__getResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Runs all tests defined by this class and return them in a 
		structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[results, a_plot] = getResults(s, plotit)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names to values in ms and mV.
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17383"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050140900000000000000">
1.40.9 Method <TT>spike_shape/plotCompareMethodsSimple</TT></A>
</H3><A NAME="17385"></A><A NAME="ref_spike_shape__plotCompareMethodsSimple"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a multi-plot comparing different action potential
			threshold finding methods.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotCompareMethodsSimple(s, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Title suffix (optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17387"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17389"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401000000000000000">
1.40.10 Method <TT>spike_shape/set</TT></A>
</H3><A NAME="17391"></A><A NAME="ref_spike_shape__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401100000000000000">
1.40.11 Method <TT>spike_shape/plotPP</TT></A>
</H3><A NAME="17393"></A><A NAME="ref_spike_shape__plotPP"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots the dV/dt vs. V phase-plane representation of the spike shape.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotPP(s)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17395"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17397"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401200000000000000">
1.40.12 Method <TT>spike_shape/calcInitVmSlopeThresholdSupsample</TT></A>
</H3><A NAME="17399"></A><A NAME="ref_spike_shape__calcInitVmSlopeThresholdSupsample"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimates the AP threshold as the first slope threshold crossing by first supersampling the data using cubic spline interpolation.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmSlopeThresholdSupsample(s, max_idx, min_idx, thr, plotit)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>thr</TT>:</STRONG></DT>
<DD>Threshold for time derivative of voltage.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: AP threshold index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17401"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/03/23
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401300000000000000">
1.40.13 Method <TT>spike_shape/calcInitVmV3hKpTinterp</TT></A>
</H3><A NAME="17403"></A><A NAME="ref_spike_shape__calcInitVmV3hKpTinterp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates candidates for action potential threshold using the first three time-domain derivatives.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = 
   calcInitVmV3hKpTinterp(s, max_idx, min_idx, lo_thr, hi_thr, plotit)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>First uses interpolation to increase time points. Calculates h,
 the second derivative of phase-plane (d&#94;2 v'/dv&#94;2), in terms of 
 time-domain derivatives. Also calculates Kp = V''[1 + (V')&#94;2]&#94;(-3/2), 
 the curvature. The maxima of these functions are used as candidates 
 for AP thresholds.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>lo_thr, hi_thr</TT>:</STRONG></DT>
<DD>Lower and higher thresholds for time derivative of voltage.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: Indices of threshold candidates in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17405"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401400000000000000">
1.40.14 Method <TT>spike_shape/plotCompareMethods</TT></A>
</H3><A NAME="17407"></A><A NAME="ref_spike_shape__plotCompareMethods"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a multi-plot comparing different action potential
			threshold finding methods.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotCompareMethods(s, title_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Title suffix (optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17409"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17411"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401500000000000000">
1.40.15 Method <TT>spike_shape/plotResults</TT></A>
</H3><A NAME="17413"></A><A NAME="ref_spike_shape__plotResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots the spike shape annotated with result characteristics.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotResults(s, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
	title_str: (Optional) String to append to plot title.
	props: A structure with any optional properties, passed to trace/plotData.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17415"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17417"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401600000000000000">
1.40.16 Method <TT>spike_shape/display</TT></A>
</H3><A NAME="17419"></A><A NAME="ref_spike_shape__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401700000000000000">
1.40.17 Method <TT>spike_shape/calcInitVmLtdMaxCurv</TT></A>
</H3><A NAME="17421"></A><A NAME="ref_spike_shape__calcInitVmLtdMaxCurv"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the action potential threshold using the maximum of the curvature equation only in the limited range given with two voltage slope thresholds.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmLtdMaxCurv(s, max_idx, min_idx, lo_thr, hi_thr, plotit)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Point of maximum curvature: Kp = V''[1 + (V')&#94;2]&#94;(-3/2)
 Taken from Sekerli, Del Negro, Lee and Butera. 
 IEEE Trans. Biomed. Eng., 51(9): 1665-71, 2004.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>lo_thr, hi_thr</TT>:</STRONG></DT>
<DD>Lower and higher thresholds for time derivative of voltage.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: AP threshold index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17423"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401800000000000000">
1.40.18 Method <TT>spike_shape/calcInitVmMaxCurvPhasePlane</TT></A>
</H3><A NAME="17425"></A><A NAME="ref_spike_shape__calcInitVmMaxCurvPhasePlane"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the voltage at the maximum curvature in the phase plane as action potential threshold.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, max_d1o, a_plot, fail_cond] = 
	calcInitVmMaxCurvPhasePlane(s, max_idx, min_idx, plotit)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>First take the phase-plane v'-v from the beginning to max(v'). Then regulate 
 intervals by interpolation. Point of maximum curvature: Kp = V''[1 + (V')&#94;2]&#94;(-3/2)
 Taken from Sekerli, Del Negro, Lee and Butera. 
 IEEE Trans. Biomed. Eng., 51(9): 1665-71, 2004.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: AP threshold index in the spike_shape [dt].
	max_d1o: Maximal value of first voltage derivative [dy].
	a_plot: plot_abstract, if requested.
	fail_cond: True if this algorithm fails to be trustable.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17427"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501401900000000000000">
1.40.19 Method <TT>spike_shape/plotTPP</TT></A>
</H3><A NAME="17429"></A><A NAME="ref_spike_shape__plotTPP"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots the dV/dt vs. V phase-plane representation of the spike shape.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotTPP(s)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses the Taylor series estimation for finding the derivative dV/dt.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17431"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17433"></A>, <TT>diffT</TT> (p.&nbsp;<A HREF="#ref_diffT"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17435"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501402000000000000000">
1.40.20 Method <TT>spike_shape/calcInitVm</TT></A>
</H3><A NAME="17437"></A><A NAME="ref_spike_shape__calcInitVm"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates spike threshold related measures of the spike_shape, s. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_val, init_idx, rise_time, amplitude,
  peak_mag, peak_idx, max_d1o, a_plot] = 
	calcInitVm(s, max_idx, min_idx)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_val: The potential value [dy].
	init_idx: Its index in the spike_shape [dt].
	rise_time: Time from initiation to maximum [dt].
	amplitude: Magnitude from initiation to max [dy].
	peak_mag: Peak value [dy].
	peak_idx: Extrapolated spike peak index [dt].
	max_d1o: Maximal value of first voltage derivative [dy].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17439"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501402100000000000000">
1.40.21 Method <TT>spike_shape/calcWidthFall</TT></A>
</H3><A NAME="17441"></A><A NAME="ref_spike_shape__calcWidthFall"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the spike width and fall information of the spike_shape, s. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[base_width, half_width, half_Vm, fall_time, min_idx, min_val, 
  max_ahp, ahp_decay_constant, dahp_mag, dahp_idx] = ...
      calcWidthFall(s, max_idx, max_val, init_idx, init_val)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>max_* can be the peak_* from calcInitVm.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point [dt].
</DD>
<DT><STRONG><TT>max_val</TT>:</STRONG></DT>
<DD>The value of the maximal point [dy].
</DD>
<DT><STRONG><TT>init_idx</TT>:</STRONG></DT>
<DD>The index of spike initiation point [dt].
</DD>
<DT><STRONG><TT>init_val</TT>:</STRONG></DT>
<DD>The value of spike initiation point [dy].
</DD>
<DT><STRONG><TT>fixed_Vm</TT>:</STRONG></DT>
<DD>The desired height for width calculation [V].
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
base_width: Width of spike at base [dt]
	half_width: Width of spike at half_Vm [dt]
	half_Vm: Half height of spike [dy]
	fall_time: Time from peak to initialization level [dt].
	min_idx: The index of the minimal point of the spike_shape [dt].
	max_ahp: Magnitude from initiation to minimum [dy].
	ahp_decay_constant: Approximation to refractory decay after maxAHP [dt].
	dahp_mag: Magnitude of the double AHP peak
	dahp_mag: Index of the double AHP peak
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17443"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501402200000000000000">
1.40.22 Method <TT>spike_shape/calcInitVmSlopeThreshold</TT></A>
</H3><A NAME="17445"></A><A NAME="ref_spike_shape__calcInitVmSlopeThreshold"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the AP threshold using the slope threhold crossing.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[init_idx, a_plot] = calcInitVmSlopeThreshold(s, max_idx, min_idx, thr, plotit)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>max_idx</TT>:</STRONG></DT>
<DD>The index of the maximal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>min_idx</TT>:</STRONG></DT>
<DD>The index of the minimal point of the spike_shape [dt].
</DD>
<DT><STRONG><TT>thr</TT>:</STRONG></DT>
<DD>Threshold for time derivative of voltage.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, plot a graph annotating the test results 
<P>
(optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
init_idx: AP threshold index in the spike_shape [dt].
	a_plot: plot_abstract, if requested.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>calcInitVm</TT> (p.&nbsp;<A HREF="#ref_calcInitVm"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17447"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050141000000000000000">
1.41 Class <TT>spike_shape_profile</TT></A>
</H2><A NAME="17449"></A><A NAME="ref_spike_shape_profile"></A>
<H3><A NAME="SECTION050141100000000000000">
1.41.1 Constructor <TT>spike_shape_profile/spike_shape_profile</TT></A>
</H3><A NAME="17451"></A><A NAME="ref_spike_shape_profile__spike_shape_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Holds the results profile from a spike_shape object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ss_profile = spike_shape_profile(results, a_spike_shape, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>results</TT>:</STRONG></DT>
<DD>A structure containing test results.
</DD>
<DT><STRONG><TT>a_spike_shape</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
results_profile: Contains results of tests.
	spike_shape: The spike_shape object from which results were obtained.
	props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>results_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_results_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17453"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050141200000000000000">
1.41.2 Method <TT>spike_shape_profile/get</TT></A>
</H3><A NAME="17455"></A><A NAME="ref_spike_shape_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050141300000000000000">
1.41.3 Method <TT>spike_shape_profile/plot_abstract</TT></A>
</H3><A NAME="17457"></A><A NAME="ref_spike_shape_profile__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots the spike shape with measurements marked in red.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(s, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spike_shape object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>absolute_peak_time</TT>:</STRONG></DT>
<DD>Shift the peak to this point on the plot.
</DD>
<DT><STRONG><TT>no_plot_spike</TT>:</STRONG></DT>
<DD>Do not plot the spike shape first.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17459"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17461"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050142000000000000000">
1.42 Class <TT>spikes</TT></A>
</H2><A NAME="17463"></A><A NAME="ref_spikes"></A>
<H3><A NAME="SECTION050142100000000000000">
1.42.1 Constructor <TT>spikes/spikes</TT></A>
</H3><A NAME="17465"></A><A NAME="ref_spikes__spikes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Spike times from a trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = spikes(times, num_samples, dt, id)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>times</TT>:</STRONG></DT>
<DD>The spike times [dt].
</DD>
<DT><STRONG><TT>num_samples</TT>:</STRONG></DT>
<DD>Number of samples in the original trace.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s].
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
times, num_samples, dt, id.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace/spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17467"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17469"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17471"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17473"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142200000000000000">
1.42.2 Method <TT>spikes/plotData</TT></A>
</H3><A NAME="17475"></A><A NAME="ref_spikes__plotData"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a spikes object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotData(s, title_str)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If s is a vector of spikes objects, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
	title_str: (Optional) String to append to plot title.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17477"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17479"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142300000000000000">
1.42.3 Method <TT>spikes/get</TT></A>
</H3><A NAME="17481"></A><A NAME="ref_spikes__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142400000000000000">
1.42.4 Method <TT>spikes/plot</TT></A>
</H3><A NAME="17483"></A><A NAME="ref_spikes__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots spikes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plot(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
h: Handle to figure object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17485"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17487"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142500000000000000">
1.42.5 Method <TT>spikes/ISICV</TT></A>
</H3><A NAME="17489"></A><A NAME="ref_spikes__ISICV"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the coefficient of variation (CV) of the 
	inter-spike-intervals (ISI).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">cv = ISICV(s, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The period where spikes were found (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
cv: Coefficient of variation.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17491"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17493"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142600000000000000">
1.42.6 Method <TT>spikes/getResults</TT></A>
</H3><A NAME="17495"></A><A NAME="ref_spikes__getResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Runs all tests defined by this class and return them in a 
		structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getResults(s)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names to values 
		in ms and mV (or mA).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17497"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142700000000000000">
1.42.7 Method <TT>spikes/plotISIs</TT></A>
</H3><A NAME="17499"></A><A NAME="ref_spikes__plotISIs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a spikes object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotISIs(s, title_str)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If s is a vector of spikes objects, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
	title_str: (Optional) String to append to plot title.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17501"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17503"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142800000000000000">
1.42.8 Method <TT>spikes/set</TT></A>
</H3><A NAME="17505"></A><A NAME="ref_spikes__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050142900000000000000">
1.42.9 Method <TT>spikes/periodWhole</TT></A>
</H3><A NAME="17507"></A><A NAME="ref_spikes__periodWhole"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the boundaries of the whole period of spikes, s. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">whole_period = periodWhole(s)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17509"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17511"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421000000000000000">
1.42.10 Method <TT>spikes/getISIs</TT></A>
</H3><A NAME="17513"></A><A NAME="ref_spikes__getISIs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the firing rate of the spikes found in the given 
		period with an averaged inter-spike-interval approach.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">isi = getISIs(s, period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>The period where spikes were found (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
isi: Inter-spike-interval vector [dt]
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17515"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17517"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17519"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421100000000000000">
1.42.11 Method <TT>spikes/spikeRateISI</TT></A>
</H3><A NAME="17521"></A><A NAME="ref_spikes__spikeRateISI"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the firing rate of the spikes found in the given 
		period with an averaged inter-spike-interval approach.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">freq = spikeRateISI(s, trace_index, times, period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>The period where spikes were found (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
freq: Firing rate [Hz]
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17523"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17525"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17527"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421200000000000000">
1.42.12 Method <TT>spikes/spikeRate</TT></A>
</H3><A NAME="17529"></A><A NAME="ref_spikes__spikeRate"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the average firing rate [Hz] of the given spike train.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">freq = spikeRate(s, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The period where spikes were found (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
freq: Firing rate [Hz]
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17531"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17533"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421300000000000000">
1.42.13 Method <TT>spikes/subsref</TT></A>
</H3><A NAME="17535"></A><A NAME="ref_spikes__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421400000000000000">
1.42.14 Method <TT>spikes/withinPeriodWOffset</TT></A>
</H3><A NAME="17537"></A><A NAME="ref_spikes__withinPeriodWOffset"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a spikes object valid only within the given period, keeps the offset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = withinPeriodWOffset(s, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The desired period 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: A spikes object
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17539"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17541"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421500000000000000">
1.42.15 Method <TT>spikes/spikeAmpSlope</TT></A>
</H3><A NAME="17543"></A><A NAME="ref_spikes__spikeAmpSlope"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the time constant and steady-state value
		      of the spike amplitude for slow inactivating decays.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_tau, da_inf] = spikeAmpSlope(a_spikes, a_trace, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The desired period (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tau: Approximate amplitude decay constant.
	da_inf: Delta change in final spike peak value from initial.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17545"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17547"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17549"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421600000000000000">
1.42.16 Method <TT>spikes/display</TT></A>
</H3><A NAME="17551"></A><A NAME="ref_spikes__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421700000000000000">
1.42.17 Method <TT>spikes/vertcat</TT></A>
</H3><A NAME="17553"></A><A NAME="ref_spikes__vertcat"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Vertical concatanation [a_spikes;with_spikes;...] operator.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_spikes = vertcat(a_spikes, with_spikes, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Concatanates spike times of with_spikes with that of a_spikes. Overrides the built-in
 vertcat function that is called when [a_spikes;with_spikes] is executed.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;

<P>
a_spikes, with_spikes, ...: Spikes objects.</DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_spikes: A tests_spikes that contains times of all given spikes objects.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>vertcat</TT> (p.&nbsp;<A HREF="#ref_vertcat"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17555"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17557"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421800000000000000">
1.42.18 Method <TT>spikes/withinPeriod</TT></A>
</H3><A NAME="17559"></A><A NAME="ref_spikes__withinPeriod"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a spikes object valid only within the given period, subtracts the offset.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = withinPeriod(s, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The desired period 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: A spikes object
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17561"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17563"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/31
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501421900000000000000">
1.42.19 Method <TT>spikes/intoPeriod</TT></A>
</H3><A NAME="17565"></A><A NAME="ref_spikes__intoPeriod"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Shifts the spikes times to be within the given period.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = intoPeriod(s, a_period)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Assuming this spikes object's length fits into the given period, it shifts
 all times to start from the beginning of the given period. This may be used
 to reconstruct the original spikes object from subperiods that were cut out
 previously, using the withinPeriod method.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The desired period 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: A spikes object
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17567"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17569"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/31
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501422000000000000000">
1.42.20 Method <TT>spikes/addSpikes</TT></A>
</H3><A NAME="17571"></A><A NAME="ref_spikes__addSpikes"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = addSpike(s, times)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>times</TT>:</STRONG></DT>
<DD>Times of spikes to add
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: The updated object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17573"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501422100000000000000">
1.42.21 Method <TT>spikes/plotFreqVsTime</TT></A>
</H3><A NAME="17575"></A><A NAME="ref_spikes__plotFreqVsTime"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a frequency-time graph from the spikes object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotFreqVsTime(s, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If s is a vector of spikes objects, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>timeScale</TT>:</STRONG></DT>
<DD>'s' for seconds, or 'ms' for milliseconds.
</DD>
<DT><STRONG><TT>type</TT>:</STRONG></DT>
<DD>If 'simple' plots 1/is for each spike time, 
<P>
'manhattan' uses flat lines of 1/isi height between spike times (default).
(others passed to plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
	title_str: (Optional) String to append to plot title.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17577"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17579"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501422200000000000000">
1.42.22 Method <TT>spikes/SFA</TT></A>
</H3><A NAME="17581"></A><A NAME="ref_spikes__SFA"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the spike frequency accommodation (SFA) of the 
	inter-spike-intervals (ISI).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">sfa = SFA(s, a_period)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>SFA is the ration of the last ISI to the first ISI in the period.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>s</TT>:</STRONG></DT>
<DD>A spikes object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The period where spikes were found (optional)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
sfa: Spike frequency accommodation.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17583"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17585"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050143000000000000000">
1.43 Class <TT>spikes_db</TT></A>
</H2><A NAME="17587"></A><A NAME="ref_spikes_db"></A>
<H3><A NAME="SECTION050143100000000000000">
1.43.1 Constructor <TT>spikes_db/spikes_db</TT></A>
</H3><A NAME="17589"></A><A NAME="ref_spikes_db__spikes_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of spike shape results obtained from a period in a trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_spikes_db = spikes_db(data, col_names, a_trace, a_period, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Use trace/analyzeSpikesInPeriod to 
 get an instance of this class.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>Database contents.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>The column names.
</DD>
<DT><STRONG><TT>a_trace</TT>:</STRONG></DT>
<DD>The trace where the spikes were found.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The period inside a_trace where spikes were found.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, trace, period, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17591"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17593"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17595"></A>, <TT>trace/analyzeSpikesInPeriod</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__analyzeSpikesInPeriod"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17597"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050143200000000000000">
1.43.2 Method <TT>spikes_db/plot_abstract</TT></A>
</H3><A NAME="17599"></A><A NAME="ref_spikes_db__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Visualizes the spikes_db by marking spike shapes measurements on the trace plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pm = plot_abstract(a_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A spikes_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to trace/plotData.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_pm: A trace plot.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17601"></A>, <TT>tests_db/plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17603"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17605"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050144000000000000000">
1.44 Class <TT>sql_portal</TT></A>
</H2><A NAME="17607"></A><A NAME="ref_sql_portal"></A>
<H3><A NAME="SECTION050144100000000000000">
1.44.1 Constructor <TT>sql_portal/sql_portal</TT></A>
</H3><A NAME="17609"></A><A NAME="ref_sql_portal__sql_portal"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>For import and export to external SQL engines using the Matlab Database Toolbox.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = sql_portal(db_conn, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses the Database (DB) Toolbox's connection object to read and write to
 external SQL engines.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db_conn</TT>:</STRONG></DT>
<DD>An object of the database class of the DB Toolbox.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
db_conn: The DB Toolbox connection object.
	id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17611"></A>, <TT>database</TT> (p.&nbsp;<A HREF="#ref_database"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17613"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/11/29
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050144200000000000000">
1.44.2 Method <TT>sql_portal/get</TT></A>
</H3><A NAME="17615"></A><A NAME="ref_sql_portal__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050144300000000000000">
1.44.3 Method <TT>sql_portal/sql_table</TT></A>
</H3><A NAME="17617"></A><A NAME="ref_sql_portal__sql_table"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create an SQL table from the contents of a tests_db object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">sql_table(a_sql_portal, a_tests_db, table_label, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Converter function to get a tests_db object properly annotated with
 the metadata obtained from the results of the executed SQL
 query. Currently this function is limited to importing numeric data only.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_sql_portal</TT>:</STRONG></DT>
<DD>A sql_portal object.
</DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>table_label</TT>:</STRONG></DT>
<DD>Name of the newly created table.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17619"></A>, <TT>database</TT> (p.&nbsp;<A HREF="#ref_database"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17621"></A>, <TT>sql_portal/tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_sql_portal__tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17623"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/12/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050144400000000000000">
1.44.4 Method <TT>sql_portal/subsref</TT></A>
</H3><A NAME="17625"></A><A NAME="ref_sql_portal__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050144500000000000000">
1.44.5 Method <TT>sql_portal/sql_statement</TT></A>
</H3><A NAME="17627"></A><A NAME="ref_sql_portal__sql_statement"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Run an SQL statement and discard its results.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">response_str = sql_statement(a_sql_portal, statement_string, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This function is for sending SQL statements that do not return any data,
 for such functions as inserting data into a database, creating views, and
 running administration commands. See sql_portal/tests_db to import select
 query results into Pandora.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_sql_portal</TT>:</STRONG></DT>
<DD>A sql_portal object.
</DD>
<DT><STRONG><TT>statement_string</TT>:</STRONG></DT>
<DD>An SQL statement that does not return data.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
response_str: Response string from the SQL engine.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>sql_portal/tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_sql_portal__tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17629"></A>, <TT>database</TT> (p.&nbsp;<A HREF="#ref_database"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17631"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050144600000000000000">
1.44.6 Method <TT>sql_portal/tests_db</TT></A>
</H3><A NAME="17633"></A><A NAME="ref_sql_portal__tests_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a tests_db object from the results of a SQL query.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = tests_db(a_sql_portal, query_string, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Converter function to get a tests_db object properly annotated with
 the metadata obtained from the results of the executed SQL
 query. Currently this function is limited to importing numeric data only.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_sql_portal</TT>:</STRONG></DT>
<DD>A sql_portal object.
</DD>
<DT><STRONG><TT>query_string</TT>:</STRONG></DT>
<DD>An SQL query returning numeric results.
</DD>
<DT><STRONG><TT>query_id</TT>:</STRONG></DT>
<DD>Identifier associated witht the query, to be passed
<P>
into the tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17635"></A>, <TT>database</TT> (p.&nbsp;<A HREF="#ref_database"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17637"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/11/29
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050145000000000000000">
1.45 Class <TT>stats_db</TT></A>
</H2><A NAME="17639"></A><A NAME="ref_stats_db"></A>
<H3><A NAME="SECTION050145100000000000000">
1.45.1 Constructor <TT>stats_db/stats_db</TT></A>
</H3><A NAME="17641"></A><A NAME="ref_stats_db__stats_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database of rows corresponding to statistical distribution
		properties of tests. Multiple pages can be used to
		indicate another dimension.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stats_db = stats_db(test_results, col_names, row_names, page_names, 
			id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_3D_db. Allows generating a plot, etc.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>test_results</TT>:</STRONG></DT>
<DD>The 3-d array of rows, columns, and pages.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>Test names in this db.
</DD>
<DT><STRONG><TT>row_names</TT>:</STRONG></DT>
<DD>Statistical test names for each row.
</DD>
<DT><STRONG><TT>page_names</TT>:</STRONG></DT>
<DD>Meaning of each separate page of data 
<P>
(e.g., a different invariant parameter).
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>axis_limits</TT>:</STRONG></DT>
<DD>Limits in the form of [xmin xmax ymin ymax]
<P>
for errorbar axes.
</DD>
<DT><STRONG><TT>yTicksPos</TT>:</STRONG></DT>
<DD>'left' means only put y-axis ticks to leftmost plot.
</DD>
<DT><STRONG><TT>xTicksPos</TT>:</STRONG></DT>
<DD>'bottom' means only put x-axis ticks to lowest plot.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_3D_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17643"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17645"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145200000000000000">
1.45.2 Method <TT>stats_db/get</TT></A>
</H3><A NAME="17647"></A><A NAME="ref_stats_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145300000000000000">
1.45.3 Method <TT>stats_db/compareStats</TT></A>
</H3><A NAME="17649"></A><A NAME="ref_stats_db__compareStats"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Merges multiple stats_dbs into pages of a single stats_db for comparison.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_mult_stats_db = compareStats(a_stats_db, a_2nd_stats_db, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a plot_simple object from this histogram.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A stats_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_mult_stats_db: A multi-page stats_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17651"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17653"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145400000000000000">
1.45.4 Method <TT>stats_db/plotVarMatrix</TT></A>
</H3><A NAME="17655"></A><A NAME="ref_stats_db__plotVarMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a stack of parameter-test variation plots organized in a matrix.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot_stack = plotVarMatrix(p_stats, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_stats</TT>:</STRONG></DT>
<DD>Array of invariant parameter databases obtained from
<P>
calling tests_3D_db/paramsTestsHistsStats.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_stack.
<DL>
<DD></DD>
<DT><STRONG><TT>plotMethod</TT>:</STRONG></DT>
<DD>'plotVar' uses stats_db/plotVar (default)
<P>
'plot_bars' uses stats_db/plot_bars
</DD>
<DT><STRONG><TT>rotateYLabel</TT>:</STRONG></DT>
<DD>Rotate row labels this much (default=60).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot_stack: A plot_stack with the plots organized in matrix form
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>paramsTestsHistsStats</TT> (p.&nbsp;<A HREF="#ref_paramsTestsHistsStats"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17657"></A>, <TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17659"></A>, <TT>plotVar.</TT> (p.&nbsp;<A HREF="#ref_plotVar."><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17661"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145500000000000000">
1.45.5 Method <TT>stats_db/plotColorVar</TT></A>
</H3><A NAME="17663"></A><A NAME="ref_stats_db__plotColorVar"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a color-plot of parameter-test variations in a matrix.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotColorVar(p_stats, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_stats</TT>:</STRONG></DT>
<DD>Array of invariant parameter databases obtained from
<P>
calling tests_3D_db/paramsTestsHistsStats.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_stack.
<DL>
<DD></DD>
<DT><STRONG><TT>plotMethod</TT>:</STRONG></DT>
<DD>'plotVar' uses stats_db/plotVar (default)
<P>
'plot_bars' uses stats_db/plot_bars
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract with the color plot
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>paramsTestsHistsStats</TT> (p.&nbsp;<A HREF="#ref_paramsTestsHistsStats"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17665"></A>, <TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17667"></A>, <TT>plotVar.</TT> (p.&nbsp;<A HREF="#ref_plotVar."><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17669"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145600000000000000">
1.45.6 Method <TT>stats_db/set</TT></A>
</H3><A NAME="17671"></A><A NAME="ref_stats_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145700000000000000">
1.45.7 Method <TT>stats_db/onlyRowsTests</TT></A>
</H3><A NAME="17673"></A><A NAME="ref_stats_db__onlyRowsTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a tests_db that only contains the desired 
		tests and rows (and pages).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = onlyRowsTests(obj, rows, tests, pages)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Selects the given dimensions and returns in a new tests_db object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>A logical or index vector of rows. If ':', all rows.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Cell array of test names or column indices. If ':', all tests.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>(Optional) A logical or index vector of pages. ':' for all pages.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17675"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17677"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145800000000000000">
1.45.8 Method <TT>stats_db/subsref</TT></A>
</H3><A NAME="17679"></A><A NAME="ref_stats_db__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050145900000000000000">
1.45.9 Method <TT>stats_db/plotVar</TT></A>
</H3><A NAME="17681"></A><A NAME="ref_stats_db__plotVar"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a plot of the variation between two tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotVar(a_stats_db, test1, test2, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a plot description where the mean values are used for solid lines
 and the std values of test2 is indicated with errorbars. It is assumed that 
 each page of the stats_db contains a value to be matched.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A stats_db object.
</DD>
<DT><STRONG><TT>test1</TT>:</STRONG></DT>
<DD>Test column for the x-axis, only mean values are used.
</DD>
<DT><STRONG><TT>test2</TT>:</STRONG></DT>
<DD>Test column for the y-axis, std values are indicated with errorbars.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>plotType</TT>:</STRONG></DT>
<DD>1, only errorbars (default); 2, errorbars extending from bars.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, only display given title_str.
<P>
(rest passed to plot_abstract)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plotVar</TT> (p.&nbsp;<A HREF="#ref_plotVar"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17683"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17685"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501451000000000000000">
1.45.10 Method <TT>stats_db/plot_bars_ax</TT></A>
</H3><A NAME="17687"></A><A NAME="ref_stats_db__plot_bars_ax"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Bar plot with extending errorbars for all columns in the same axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_bars_ax(a_tests_db, row, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Differs from stats_db/plot_bars because it does not open a new axis
 for each column. This is only suitable if all columns have similar extents.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A stats_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Optional title string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>putLabels</TT>:</STRONG></DT>
<DD>Put special column name labels.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17689"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17691"></A>, <TT>stats_db/plot_bars</TT> (p.&nbsp;<A HREF="node5.html#ref_stats_db__plot_bars"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17693"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501451100000000000000">
1.45.11 Method <TT>stats_db/plotYTests</TT></A>
</H3><A NAME="17695"></A><A NAME="ref_stats_db__plotYTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create an errorbar plot of database stats measures against given X-axis values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_p = plotYTests(a_stats_db, x_vals, tests, axis_labels, title_str, short_title, command, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>x_vals</TT>:</STRONG></DT>
<DD>A vector of X-axis values.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A vector or cell array of columns to correspond to each value from x_vals.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>short_title</TT>:</STRONG></DT>
<DD>(Optional) Few words that may appear in legends of multiplot.
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>(Optional) Command to do the plotting with (default: 'plot')
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>LineStyle</TT>:</STRONG></DT>
<DD>Plot line style to use. (default: 'd-')
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_p: A plot_abstract.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_p = plotYTests(a_stats_db, [0 40 100 200], ...
<BR>'IniSpontSpikeRateISI_0pA', 'PulseIni100msSpikeRateISI_D40pA', ...
<BR>'PulseIni100msSpikeRateISI_D100pA', 'PulseIni100msSpikeRateISI_D200pA', ...
<BR>'current pulse [pA]', 'firing rate [Hz]', ', f-I curves', 'neuron 1');
<BR>&#187; plotFigure(a_p);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17697"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/23
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501451200000000000000">
1.45.12 Method <TT>stats_db/plot_bars</TT></A>
</H3><A NAME="17699"></A><A NAME="ref_stats_db__plot_bars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a bar graph with errorbars for each db column. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_bars(a_stats_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Looks for 'min', 'max', and 'STD' labels in the row_idx for drawing the errorbars. 
 Each page of the DB will produce grouped bars.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A stats_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>The plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>pageVariable</TT>:</STRONG></DT>
<DD>The column used for denoting page values.
<P>
(passed to plot_bars/plot_bars)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_bars or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17701"></A>, <TT>plot_bars/plot_bars</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_bars__plot_bars"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17703"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501451300000000000000">
1.45.13 Method <TT>stats_db/plot_abstract</TT></A>
</H3><A NAME="17705"></A><A NAME="ref_stats_db__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates an error bar graph for each db columns. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(a_stats_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a plot_simple object from this histogram. Looks for 'mean',
 'min', 'max', and 'STD' labels in the row_idx for drawing the
 errorbars. Each column of a_stats_db is shown in a separate
 axis. Values from multiple pages of a_stats_db are shown as distinct
 points in the axis.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_stats_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>A title string on the plot
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_abstract or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17707"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17709"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050146000000000000000">
1.46 Class <TT>tests_3D_db</TT></A>
</H2><A NAME="17711"></A><A NAME="ref_tests_3D_db"></A>
<H3><A NAME="SECTION050146100000000000000">
1.46.1 Constructor <TT>tests_3D_db/tests_3D_db</TT></A>
</H3><A NAME="17713"></A><A NAME="ref_tests_3D_db__tests_3D_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A database multiple pages with rows of test columns. 
		Each page may represent aspects of the data that are
		different, but not defined in this object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_3D_db = tests_3D_db(data, col_names, row_names, page_names, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a subclass of tests_db. Usually it contains a RowIndex
 column that points to an original db from which this data originated. 
 The row indices can be used to reach the values associated with different
 pages of information contained in this object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>The 3-d vector of rows, columns, and pages.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>Colun names of the database.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>invarName</TT>:</STRONG></DT>
<DD>Name of the invariant parameter for this db.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
tests_db, page_idx.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17715"></A>, <TT>tests_db/invarValues</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__invarValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17717"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146200000000000000">
1.46.2 Method <TT>tests_3D_db/get</TT></A>
</H3><A NAME="17719"></A><A NAME="ref_tests_3D_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146300000000000000">
1.46.3 Method <TT>tests_3D_db/plotParamPairImage</TT></A>
</H3><A NAME="17721"></A><A NAME="ref_tests_3D_db__plotParamPairImage"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates an image plot of variation of a test with two parameters in the first page.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotParamPairImage(a_db, test, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It is assumed that the 3D DB is created by invariant combinations of two parameters,
 which are the first two columns. Each page of the db must contain a same parameter 
 values. This is the default character of tests_3D_db created by 
 params_tests_db/invarParam. Parameter values will be enumerated and then an 
 image plot is created.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>test</TT>:</STRONG></DT>
<DD>Test column to take the measure value.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties to be passed to plot_abstract.
<DL>
<DD></DD>
<DT><STRONG><TT>truncateDecDigits</TT>:</STRONG></DT>
<DD>Truncate labels to this many decimal digits.
</DD>
<DT><STRONG><TT>labelSteps</TT>:</STRONG></DT>
<DD>Skip this many labels between ticks to reduce to total number.
</DD>
<DT><STRONG><TT>maxValue</TT>:</STRONG></DT>
<DD>Maximal value to normalize colors and to annotate the colorbar.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object or one of its subclasses.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">Find relationship of two parameters against a measure:
<BR>&#187; plotFigure(plotParamPairImage(invarParam(a_db, 'NaF', 'KCNQ'), 'PulseIni100msRest2SpikeRateISI_D100pA'));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db/invarParam</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db__invarParam"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17723"></A>, <TT>plotImage</TT> (p.&nbsp;<A HREF="#ref_plotImage"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17725"></A>, <TT>plot_abstract.</TT> (p.&nbsp;<A HREF="#ref_plot_abstract."><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17727"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146400000000000000">
1.46.4 Method <TT>tests_3D_db/mergePages</TT></A>
</H3><A NAME="17729"></A><A NAME="ref_tests_3D_db__mergePages"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Merges tests from separate pages into a 2D params_tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = mergePages(db, page_tests, page_suffixes)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Keeps uniqueness by adding suffixes to test names.
 If you're using invarParams, do swapRowsPages, then join with original db to get
 the parameter values.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>page_tests</TT>:</STRONG></DT>
<DD>Cell array of list of tests to take from each page.
</DD>
<DT><STRONG><TT>page_suffixes</TT>:</STRONG></DT>
<DD>Cell array of suffixes to append to tests from each page.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17731"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17733"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146500000000000000">
1.46.5 Method <TT>tests_3D_db/diff2D</TT></A>
</H3><A NAME="17735"></A><A NAME="ref_tests_3D_db__diff2D"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a tests_db by taking the derivative of the given test.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tests_db = diff2D(a_db, test, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Applies the diff function to the chosen test, and collapses the middle
 dimension of the 3D DB to create a 2D DB and transposes it. The result is
 that the pages of the 3D DB becomes the rows of the new database, and the
 differenced rows appear as new columns, each named uniquely. The column
 index would correspons to the row index in the 3D DB. A new column
 'PageNumber' is appended to point back to the 3D DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>test</TT>:</STRONG></DT>
<DD>Test column.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tests_db: A tests_db that holds the requested differences of parameter values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>boxplot</TT> (p.&nbsp;<A HREF="#ref_boxplot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17737"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17739"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146600000000000000">
1.46.6 Method <TT>tests_3D_db/set</TT></A>
</H3><A NAME="17741"></A><A NAME="ref_tests_3D_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146700000000000000">
1.46.7 Method <TT>tests_3D_db/histograms</TT></A>
</H3><A NAME="17743"></A><A NAME="ref_tests_3D_db__histograms"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_histogram_db = histogram(db, col, num_bins)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If one wants to get histograms of test values for each single value of
 the selected invariant parameter, then swapRowsPages should be done
 first on db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>col</TT>:</STRONG></DT>
<DD>Column to find the histogram.
</DD>
<DT><STRONG><TT>num_bins</TT>:</STRONG></DT>
<DD>Number of histogram bins (Optional, default=100)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_histogram_db: A histogram_db object containing the histogram.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17745"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17747"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146800000000000000">
1.46.8 Method <TT>tests_3D_db/plotVarBox</TT></A>
</H3><A NAME="17749"></A><A NAME="ref_tests_3D_db__plotVarBox"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a boxplot of the variation between two tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotVarBox(a_db, test1, test2, notch, sym, vert, whis, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It is assumed that each page of the db contains a different parameter value.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>test1</TT>:</STRONG></DT>
<DD>Test column for the x-axis, only mean values are used.
</DD>
<DT><STRONG><TT>test2</TT>:</STRONG></DT>
<DD>Test column for the y-axis, used for boxplot.
</DD>
<DT><STRONG><TT>notch, sym, vert, whis</TT>:</STRONG></DT>
<DD>See boxplot, defaults = (1, '+', 1, 1.5).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties to be passed to plot_abstract.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>boxplot</TT> (p.&nbsp;<A HREF="#ref_boxplot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17751"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17753"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050146900000000000000">
1.46.9 Method <TT>tests_3D_db/display</TT></A>
</H3><A NAME="17755"></A><A NAME="ref_tests_3D_db__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501461000000000000000">
1.46.10 Method <TT>tests_3D_db/joinPages</TT></A>
</H3><A NAME="17757"></A><A NAME="ref_tests_3D_db__joinPages"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Joins the rows of the given db to the with_db rows matching with the PageIndex
 	column.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = joinPages(db, tests, with_db, w_tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Replicates the desired columns in the with_db with rows having a 
 page index and joins them next to desired columns from the current 3D_db. Flattens 
 the resulting 3D_db to become a 2D db. Assumes each page index only 
 appears once in with_db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_3D_db object.
</DD>
<DT><STRONG><TT>with_db</TT>:</STRONG></DT>
<DD>A tests_db object with a PageIndex column.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17759"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501461100000000000000">
1.46.11 Method <TT>tests_3D_db/corrCoefs</TT></A>
</H3><A NAME="17761"></A><A NAME="ref_tests_3D_db__corrCoefs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates correlation coefficients by comparing col1 with other cols. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_coefs_db = corrCoefs(db, col1, cols, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If db has multiple pages, then each page in db produces a row of
 coefficients and matching PageIndex. Assuming the db was created with
 invarValues, this function finds the invariant correlation coefficients
 between its columns. The invariant correlation coefficients are the
 correlation of one column value with another column value when some other
 column values are fixed.  Since there are many occurences of the invariant
 coefficients, a histogram can then be created and returned from the
 created db. The other columns that are fixed are not in this db object,
 but can be reached using the indices to the original db. The page
 number is saved in the created db, so that it can be used to find the page
 from which the coefficient came. Then row indices of the page points to
 original constant column values.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>col1</TT>:</STRONG></DT>
<DD>Column to compare.
</DD>
<DT><STRONG><TT>cols</TT>:</STRONG></DT>
<DD>Columns to be compared with col1.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>skipCoefs</TT>:</STRONG></DT>
<DD>If 1, coefficients of less confidence than <P>
will be skipped. (default=1)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_coefs_db: A corrcoefs_db of the coefficients and page indices.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17763"></A>, <TT>corrcoefs_db</TT> (p.&nbsp;<A HREF="node5.html#ref_corrcoefs_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17765"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501461200000000000000">
1.46.12 Method <TT>tests_3D_db/paramsTestsHistsStats</TT></A>
</H3><A NAME="17767"></A><A NAME="ref_tests_3D_db__paramsTestsHistsStats"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates histograms and statistics for DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[pt_hists, p_stats] = paramsTestsHistsStats(p_t3ds, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calculates histograms and statistics for all combinations of tests 
 and params and returns them in a cell array. Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>p_t3ds</TT>:</STRONG></DT>
<DD>Array of invariant parameter databases obtained by
<P>
calling the params_tests_db/invarParams method.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>statsMethod</TT>:</STRONG></DT>
<DD>method to call to get a stats_db (default='statsMeanSE')
</DD>
<DT><STRONG><TT>useDiff</TT>:</STRONG></DT>
<DD>If 1, takes the derivative with diff on the 3D DBs (default=0).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
pt_hists: An array of 3D histograms for each pair of param 
		  and test.
	p_stats: An array of stats_dbs for each param.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>invarParams</TT> (p.&nbsp;<A HREF="#ref_invarParams"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17769"></A>, <TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17771"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501461300000000000000">
1.46.13 Method <TT>tests_3D_db/swapRowsPages</TT></A>
</H3><A NAME="17773"></A><A NAME="ref_tests_3D_db__swapRowsPages"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Swaps the row dimension with the page dimension of the
		  tests_3D_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_3D_db = swapRowsPages(db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Assuming that this is a invariant parameter and tests relations db, this
 function transposes the data matrix by swapping the pages with rows. Each
 resulting page correspond to a single value of the chosen parameter, with
 each row containing a test result with different combinations of the rest
 of the parameters.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_3D_db: A tests_3D_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17775"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050147000000000000000">
1.47 Class <TT>tests_db</TT></A>
</H2><A NAME="17777"></A><A NAME="ref_tests_db"></A>
<H3><A NAME="SECTION050147100000000000000">
1.47.1 Constructor <TT>tests_db/tests_db</TT></A>
</H3><A NAME="17779"></A><A NAME="ref_tests_db__tests_db"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A generic database of test results organized in a matrix format.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = tests_db(test_results, col_names, row_names, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Defines all operations on this structure so that subclasses can use them.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>test_results</TT>:</STRONG></DT>
<DD>A matrix that contains columns associated with
<P>
tests and rows for separate observations.
</DD>
<DT><STRONG><TT>col_names</TT>:</STRONG></DT>
<DD>Cell array of column names of test_results.
</DD>
<DT><STRONG><TT>row_names</TT>:</STRONG></DT>
<DD>Cell array of row names of test_results.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>An identifying string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
data: The data matrix.
	row_idx, col_idx: Structure associating row/column names to indices.
	id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17781"></A>, <TT>params_db</TT> (p.&nbsp;<A HREF="#ref_params_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17783"></A>, <TT>test_variable_db (N/I)</TT> (p.&nbsp;<A HREF="#ref_test_variable_db__N__I_"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17785"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/01
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147200000000000000">
1.47.2 Method <TT>tests_db/setRows</TT></A>
</H3><A NAME="17787"></A><A NAME="ref_tests_db__setRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Sets the rows of observations in tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">index = setRows(obj, rows)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Sets a new set of observations to the database and returns the new DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>A matrix that contains rows for the DB.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object with the new rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17789"></A>, <TT>addRow</TT> (p.&nbsp;<A HREF="#ref_addRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17791"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17793"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147300000000000000">
1.47.3 Method <TT>tests_db/corrcoef</TT></A>
</H3><A NAME="17795"></A><A NAME="ref_tests_db__corrcoef"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates a correlation coefficient matrix by comparing cols.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_coefs_db = corrcoef(db, cols, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>cols</TT>:</STRONG></DT>
<DD>Columns to be compared.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>skipCoefs</TT>:</STRONG></DT>
<DD>If 1, coefficients of less confidence than <P>
will be skipped. (default=1)
</DD>
<DT><STRONG><TT>alpha</TT>:</STRONG></DT>
<DD>Skip coefs with p values lower than this (default=0.05).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_coefs_db: A tests_3D_db of the coefficients.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17797"></A>, <TT>corrcoefs_db</TT> (p.&nbsp;<A HREF="node5.html#ref_corrcoefs_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17799"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147400000000000000">
1.47.4 Method <TT>tests_db/allocateRows</TT></A>
</H3><A NAME="17801"></A><A NAME="ref_tests_db__allocateRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Preallocates a NaN-filled num_rows rows in tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = allocateRows(obj, num_rows)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Allocates the desired number of rows to speed up filling up the data matrix 
   using assignRowsTests. Using addRow after this operation is still expensive.
   The method of allocating a matrix, filling it up, and then providing it to 
   the tests_db constructor is the preferred method of creating tests_db objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>num_rows</TT>:</STRONG></DT>
<DD>The predicted number of observations for this tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>assignRowsTests</TT> (p.&nbsp;<A HREF="#ref_assignRowsTests"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17803"></A>, <TT>addRow</TT> (p.&nbsp;<A HREF="#ref_addRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17805"></A>, <TT>setRows</TT> (p.&nbsp;<A HREF="#ref_setRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17807"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17809"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147500000000000000">
1.47.5 Method <TT>tests_db/matchingRow</TT></A>
</H3><A NAME="17811"></A><A NAME="ref_tests_db__matchingRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a criterion database for matching the tests of a row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">crit_db = matchingRow(db, row, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Copies selected test values from row as the first row into the criterion
 db. Adds a second row for the STD of each column in the db.  Calculates the
 covariance for using the Mahalonobis distance in the ranking.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row index to match.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>distDB</TT>:</STRONG></DT>
<DD>Take the standard deviation and covariance of this db instead.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
crit_db: A tests_db with two rows for values and STDs.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; crit_db = matchingRow(phys_control_compare_db, 
<BR>
find(phys_control_compare_db(:, 'TracesetIndex') == 61))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rankMatching</TT> (p.&nbsp;<A HREF="#ref_rankMatching"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17813"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17815"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17817"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147600000000000000">
1.47.6 Method <TT>tests_db/get</TT></A>
</H3><A NAME="17819"></A><A NAME="ref_tests_db__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147700000000000000">
1.47.7 Method <TT>tests_db/tests2idx</TT></A>
</H3><A NAME="17821"></A><A NAME="ref_tests_db__tests2idx"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Find dimension indices from a test names/numbers specification.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">idx = tests2idx(db, dim_name, tests)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim_name</TT>:</STRONG></DT>
<DD>String indicating 'row', 'col', or 'page'
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Either a single or array of column numbers, or a single
<P>
test name or a cell array of test names. If ':', all
tests. For name strings, regular expressions are
supported if quoted with slashes (e.g., '/a.*/')
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
idx: Array of column indices.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; cols = tests2idx(a_db, 'col', 'col1', '/col2+/');
<BR>
will return indices of col1 and columns like col2, col22, col22, etc.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17823"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17825"></A>, <TT>regexp</TT> (p.&nbsp;<A HREF="#ref_regexp"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17827"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147800000000000000">
1.47.8 Method <TT>tests_db/invarValues</TT></A>
</H3><A NAME="17829"></A><A NAME="ref_tests_db__invarValues"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds all sets in which given columns vary while the rest are invariant.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tests_3D_db = invarValues(db, cols, in_page_unique_cols)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Useful when trying to find relationships between some columns
 independent of other columns. In a database that contains results of a
 multivariate function, this function can find the effect of one or more
 parameters when other parameters are kept constant (i.e., invariant). Rows
 with the values of the desired columns are separated into the pages of a
 tests_3D_db for each unique set of the other column values. These
 invariant values of the other columns are missing from the resulting
 tests_3D_db, instead a RowIndex is kept pointing to the db in which they
 can be found. See joinRows for joining the results back with the invariant
 columns.
   In databases that contain all unique combinations of certain parameters,
 the resulting 3D database becomes symmetric. This function row-sorts the
 database to ensure that each page has the same parameter values in the
 same rows. This is important because when the rows and pages of database
 is swapped (see tests_3D_db/swapRowsPages) each page has the same value of
 the in_page_unique_cols variables. Other functions such as
 tests_3D_db/mergePages also depend on this property.
   However, for databases with missing combinations, in_page_unique_cols
 specifies which columns is used to guide which rows of the page to place
 values found. This function will fail if you do not have such a column.
 Note: the trial column will be ignored before finding invariant values.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>cols</TT>:</STRONG></DT>
<DD>Vector of column numbers to find values when others are
<P>
invariant. Include result columns here.
</DD>
<DT><STRONG><TT>in_page_unique_cols</TT>:</STRONG></DT>
<DD>Vector of columns that have the same unique values in each page 

<P>
(Optional; used only if database is not symmetric, to ignore 
missing values of in_page_unique_cols)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tests_3D_db: A tests_3D_db object of organized values.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_db = tests_db([ ... ], 'par1', 'par2', 'measure1', 'measure2')
<BR>&#187; a_3d_db = invarValues(a_db, [2:4], 'par2')
<BR>&#187; &#187; joined_3d_db = joinRows(a_db, a_3d_db)
<BR>&#187; displayRows(joined_3d_db(:, :, 1))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17831"></A>, <TT>tests_3D_db/corrCoefs</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__corrCoefs"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17833"></A>, <TT>tests_3D_db/plotPair</TT> (p.&nbsp;<A HREF="#ref_tests_3D_db__plotPair"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17835"></A>, <TT>joinRows</TT> (p.&nbsp;<A HREF="#ref_joinRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17837"></A>, <TT>tests_3D_db/swapRowsPages</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__swapRowsPages"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17839"></A>, <TT>tests_3D_db/mergePages</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__mergePages"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17841"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050147900000000000000">
1.47.9 Method <TT>tests_db/shufflerows</TT></A>
</H3><A NAME="17843"></A><A NAME="ref_tests_db__shufflerows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a db with rows of given test columns are shuffled. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = shufflerows(db, tests, grouped)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Can be used for shuffle prediction. Basically, shuffle rows of tests and rerun
 high order analyses. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Tests to shuffle.
</DD>
<DT><STRONG><TT>grouped</TT>:</STRONG></DT>
<DD>If 1 then shuffle tests all together, 
<P>
if 0 shuffle each test separately.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The shuffled db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17845"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471000000000000000">
1.47.10 Method <TT>tests_db/plot</TT></A>
</H3><A NAME="17847"></A><A NAME="ref_tests_db__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method to plot a tests_db or a subclass. Requires a 
	plot_abstract method to be defined for this object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plot(a_tests_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A histogram_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_abstract.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
h: The figure handle created.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17849"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17851"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471100000000000000">
1.47.11 Method <TT>tests_db/setProp</TT></A>
</H3><A NAME="17853"></A><A NAME="ref_tests_db__setProp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting optional object properties.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = setProp(obj, prop1, val1, prop2, val2, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Modifies or adds property values. As many property name-value 
 pairs can be specified.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>Any object that has a props field.
</DD>
<DT><STRONG><TT>attr</TT>:</STRONG></DT>
<DD>Property name
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>Property value.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with the updated properties.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471200000000000000">
1.47.12 Method <TT>tests_db/physiol_bundle</TT></A>
</H3><A NAME="17855"></A><A NAME="ref_tests_db__physiol_bundle"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a physiol_bundle from a raw physiology database.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pbundle = physiol_bundle(phys_dball, phys_dataset, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Removes small bias currents, calculates input resistance by averaging
 negative CIP traces, averages multiple traces with similar treatments,
 selects certain CIP levels collapses its rows to create a
 one-neuron-per-pow database. It includes post-DB calculated columns
 such as rate ratios between spont and recovery periods.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>phys_dball</TT>:</STRONG></DT>
<DD>A raw database obtained by loading traces from the tracesets.
</DD>
<DT><STRONG><TT>phys_dataset</TT>:</STRONG></DT>
<DD>Dataset object passed to physiol_bundle.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional parameters.
<DL>
<DD></DD>
<DT><STRONG><TT>weedCols</TT>:</STRONG></DT>
<DD>Cell array of parameter columns to be weed-out before averaging rows
<P>
that are same w.r.t other parameters.
(default='pulseOn', 'pulseOff', 'traceEnd', 'pAbias', 'ItemIndex').
</DD>
<DT><STRONG><TT>drugCols</TT>:</STRONG></DT>
<DD>Cell array of drug names that need to be zero for the
<P>
control db (default='TTX', 'Apamin', 'EBIO', 'XE991', 'Cadmium', 'drug_4AP').
</DD>
</DL></DD>
<DT><STRONG><TT>CIPList</TT>:</STRONG></DT>
<DD>row array specifying the CIP levels to choose (eliminate the

<P>
others), default is an empty array, which means to choose all.<DL>
<DD></DD>
<DT><STRONG><TT>biasLimit</TT>:</STRONG></DT>
<DD>Limit in pA, biases larger +/- than which will be
<P>
eliminated. (default=30)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
phys_joined_db: Final one row per cip and neuron db.
	phys_joined_control_db: Rows where all drug treatments are zero.
	phys_db: Original db only with parameter and including the weedCols.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_bundle</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_bundle"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17857"></A>, <TT>params_tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17859"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471300000000000000">
1.47.13 Method <TT>tests_db/mtimes</TT></A>
</H3><A NAME="17861"></A><A NAME="ref_tests_db__mtimes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Multiplies the DB with a scalar.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = mtimes(left_obj, right_obj)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the multiplication. One or more must be of type tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/times</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__times"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17863"></A>, <TT>mtimes</TT> (p.&nbsp;<A HREF="#ref_mtimes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17865"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471400000000000000">
1.47.14 Method <TT>tests_db/addColumns</TT></A>
</H3><A NAME="17867"></A><A NAME="ref_tests_db__addColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Inserts new columns to tests_db.
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Adds new test columns to the database and returns the new DB.
 Usage 2 concatanates two DBs columnwise. This operation is 
 expensive in the sense that the whole database matrix needs to be 
 enlarged just to add a single new column. The method of allocating
 a matrix, filling it up, and then providing it to the tests_db 
 constructor is the preferred method of creating tests_db objects. 
 This method may be used for measures obtained by operating on raw measures.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj, b_obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>test_names</TT>:</STRONG></DT>
<DD>A cell array of test names to be added.
</DD>
<DT><STRONG><TT>test_columns</TT>:</STRONG></DT>
<DD>Data matrix of columns to be added.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that includes the new columns.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17869"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17871"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471500000000000000">
1.47.15 Method <TT>tests_db/rdivide</TT></A>
</H3><A NAME="17873"></A><A NAME="ref_tests_db__rdivide"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Adds a DB to another or to a scalar.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = rdivide(left_obj, right_obj)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If DBs have mismatching columns only the common columns will be kept.
 In any case, the resulting DB columns will be sorted in the order of the
 left-hand-side DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the addition. One must be of type tests_db
<P>
and the other can be a scalar or tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>rdivide</TT> (p.&nbsp;<A HREF="#ref_rdivide"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17875"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471600000000000000">
1.47.16 Method <TT>tests_db/tests2log</TT></A>
</H3><A NAME="17877"></A><A NAME="ref_tests_db__tests2log"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Return logical array of indices from a test names/numbers specification.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_log = tests2log(db, dim_name, tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>See tests2idx.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim_name</TT>:</STRONG></DT>
<DD>String indicating 'row', 'col', or 'page'
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Either a single or array of column numbers, or a single
<P>
test name or a cell array of test names. If ':', all
tests. For name strings, regular expressions are
supported if quoted with slashes (e.g., '/a.*/')
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_log: Array of column indices.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; cols = tests2log(a_db, 'col', 'col1', '/col2+/');
<BR>&#187; stripped_db = a_db(:,&nbsp;cols)
<BR>
will remove columns col1 and col2, col22, col22, etc. from stripped_db.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17879"></A>, <TT>tests2cols</TT> (p.&nbsp;<A HREF="#ref_tests2cols"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17881"></A>, <TT>regexp</TT> (p.&nbsp;<A HREF="#ref_regexp"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17883"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/05/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471700000000000000">
1.47.17 Method <TT>tests_db/transpose</TT></A>
</H3><A NAME="17885"></A><A NAME="ref_tests_db__transpose"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Transposes data matrix and swaps row and columns metadata as well.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = transpose(a_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The transposed tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>transpose</TT> (p.&nbsp;<A HREF="#ref_transpose"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17887"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/02/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471800000000000000">
1.47.18 Method <TT>tests_db/joinRows</TT></A>
</H3><A NAME="17889"></A><A NAME="ref_tests_db__joinRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Joins a_db rows with w_db rows having matching RowIndex values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = joinRows(a_db, w_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Concatenates columns of rows matching the join condition from the two
 databases. Each row index must appear only once in w_db. The created db
 preserves the ordering of w_db. See the multipleIndices option if there
 are several redundant index columns. Multiple pages in w_db are accepted
 (see keepNaNs option).
   This function is the equivalent of a "right outer join" command in SQL, w_db
 being the database table on the right.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>w_db</TT>:</STRONG></DT>
<DD>A tests_db object with a row index column.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>indexColName</TT>:</STRONG></DT>
<DD>(Optional) Name of row index column
<P>
(default='RowIndex').
</DD>
<DT><STRONG><TT>keepNaNs</TT>:</STRONG></DT>
<DD>If 1, substitute NaN values for NaN indices. 
<P>
(default=1, for multi-page DBs; 0, otherwise).
</DD>
<DT><STRONG><TT>multipleIndices</TT>:</STRONG></DT>
<DD>If 1, search for substitute RowIndex* columns for

<P>
indices with NaN values. It will fail if all indices are NaNs. (default=0)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17891"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501471900000000000000">
1.47.19 Method <TT>tests_db/histogram</TT></A>
</H3><A NAME="17893"></A><A NAME="ref_tests_db__histogram"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns histogram of chosen database column.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_histogram_db = histogram(db, col, num_bins, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a histogram_db object with rows corresponding to histogram
 entries. If an array of DBs is given, finds and uses common histogram bin centers.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>col</TT>:</STRONG></DT>
<DD>Column to find the histogram.
</DD>
<DT><STRONG><TT>num_bins</TT>:</STRONG></DT>
<DD>Number of histogram bins (Optional, default=100), or
<P>
vector of histogram bin centers.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>normalized</TT>:</STRONG></DT>
<DD>If 1, normalize histogram counts.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_histogram_db: A histogram_db object containing the histogram.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_hist_db = histogram(my_db, 'spike_width');
<BR>&#187; plot(a_hist_db);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17895"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17897"></A>, <TT>hist</TT> (p.&nbsp;<A HREF="#ref_hist"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17899"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472000000000000000">
1.47.20 Method <TT>tests_db/enumerateColumns</TT></A>
</H3><A NAME="17901"></A><A NAME="ref_tests_db__enumerateColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Replaces each value with an integer pointing to the index of enumerated unique values in a column.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = enumerateColumns(a_db, tests, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Finds unique values of each column, and replaces the original values
 with the enumerated indices of these unique values. Useful for normalizing all 
 parameter values in a hypercube.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Array of tests to be enumerated.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>truncateDecDigits</TT>:</STRONG></DT>
<DD>Use only up to this many decimal digits after the point 
<P>
when checking for uniqueness.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The modified DB.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; enumerated_db = enumerateColumns(a_db(:, 1:9));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>uniqueValues</TT> (p.&nbsp;<A HREF="#ref_uniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17903"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472100000000000000">
1.47.21 Method <TT>tests_db/uop</TT></A>
</H3><A NAME="17905"></A><A NAME="ref_tests_db__uop"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Unary operation.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = uop(left_obj, op_func, op_id)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Applies the operation to the database contents and updates its id
 field. Unary minus (uminus) uses this function.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj</TT>:</STRONG></DT>
<DD>Operands of the operation.
</DD>
<DT><STRONG><TT>op_func</TT>:</STRONG></DT>
<DD>Operation function (e.g., @plus).
</DD>
<DT><STRONG><TT>op_id</TT>:</STRONG></DT>
<DD>A string to represent the operation that will show up in the
<P>
returned id.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/uminus</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__uminus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17907"></A>, <TT>uminus</TT> (p.&nbsp;<A HREF="#ref_uminus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17909"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472200000000000000">
1.47.22 Method <TT>tests_db/approxMappingNNet</TT></A>
</H3><A NAME="17911"></A><A NAME="ref_tests_db__approxMappingNNet"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Approximates the desired input-output mapping using a Matlab neural network.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[an_approx_db, a_nnet] = approxMappingNNet(a_db, input_cols, output_cols, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Approximates the mapping between the given inputs to outputs
 using the Matlab Neural Network Toolbox. By default it creates a
 feed-forward network to be trained with a Levenberg-Marquardt training
 algorithm (see newff). Returns and the trained network object and a
 database with output columns obtained from the approximator. The outputs
 can then be compared to the original database to test the success of the
 approximation. If 'warning on verbose' is issued prior to running, it
 provides additional debug info.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>input_cols, output_cols</TT>:</STRONG></DT>
<DD>Input and output columns to be mapped
<P>
(see tests2cols for accept column specifications).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>nnetFcn</TT>:</STRONG></DT>
<DD>Neural network classifier function (default='newff')
</DD>
<DT><STRONG><TT>nnetParams</TT>:</STRONG></DT>
<DD>Cell array of parameters passed to nnetFcn after
<P>
inputs and outputs.
</DD>
<DT><STRONG><TT>trainMode</TT>:</STRONG></DT>
<DD>'batch' or 'incr'.
</DD>
<DT><STRONG><TT>classProbs</TT>:</STRONG></DT>
<DD>'prob': use probabilistic sampling to normalize

<P>
prior class probabilities.
</DD>
<DT><STRONG><TT>maxEpochs</TT>:</STRONG></DT>
<DD>maximum number of epochs to train for.

<P>
(Rest passed to balanceInputProbs and tests_db)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
an_approx_db: A tests_db object containing the original inputs and
			the approximated outputs.
	a_nnet: The Matlab neural network approximator object.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; [a_class_db, a_nnet = approxMappingNNet(my_db, 'NaF', 'Kv3', 'spike_width');
<BR>&#187; plotFigure(plot_superpose(plotScatter(my_db, 'NaF', 'spike_width'),
<BR>
plotScatter(a_class_db, 'NaF', 'spike_width')))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17913"></A>, <TT>newff</TT> (p.&nbsp;<A HREF="#ref_newff"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17915"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/12
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472300000000000000">
1.47.23 Method <TT>tests_db/isnan</TT></A>
</H3><A NAME="17917"></A><A NAME="ref_tests_db__isnan"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns logical row indices of NaN-valued columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = isnan(db, col)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>col</TT>:</STRONG></DT>
<DD>Column to check (Optional, default = 1)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical column vector of rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>isnan</TT> (p.&nbsp;<A HREF="#ref_isnan"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17919"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17921"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472400000000000000">
1.47.24 Method <TT>tests_db/minus</TT></A>
</H3><A NAME="17923"></A><A NAME="ref_tests_db__minus"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Subtracts a DB from another or from a scalar.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = minus(left_obj, right_obj)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If DBs have mismatching columns only the common columns will be kept.
 In any case, the resulting DB columns will be sorted in the order of the
 left-hand-side DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the subtraction. One must be of type tests_db
<P>
and the other can be a scalar.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>minus</TT> (p.&nbsp;<A HREF="#ref_minus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17925"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472500000000000000">
1.47.25 Method <TT>tests_db/displayRowsTeX</TT></A>
</H3><A NAME="17927"></A><A NAME="ref_tests_db__displayRowsTeX"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a LaTeX table that lists rows of this DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = displayRowsTeX(a_db, caption, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>By default table is rotated 90 degrees and scaled to 90
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Table caption.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to TeXtable.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX string for table float.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; string2File(displayRowsTeX(a_db(1:10, 4:7), 'some values',
<BR>
struct('rotate', 0)), 'table.tex')
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>displayRows</TT> (p.&nbsp;<A HREF="#ref_displayRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17929"></A>, <TT>TeXtable</TT> (p.&nbsp;<A HREF="#ref_TeXtable"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17931"></A>, <TT>cell2TeX</TT> (p.&nbsp;<A HREF="#ref_cell2TeX"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17933"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472600000000000000">
1.47.26 Method <TT>tests_db/plotXRows</TT></A>
</H3><A NAME="17935"></A><A NAME="ref_tests_db__plotXRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a scatter plot with a test versus the row numbers on the X-axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_p = plotXRows(a_db, test_y, title_str, short_title, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>test_y</TT>:</STRONG></DT>
<DD>Y variable.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>short_title</TT>:</STRONG></DT>
<DD>(Optional) Few words that may appear in legends of multiplot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to
<P>
plotScatter.<DL>
<DD></DD>
<DT><STRONG><TT>RowName</TT>:</STRONG></DT>
<DD>Label to show on X-axis, becomes a db column (default='RowNumber')
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_p: A plot_abstract.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472700000000000000">
1.47.27 Method <TT>tests_db/plotCovar</TT></A>
</H3><A NAME="17937"></A><A NAME="ref_tests_db__plotCovar"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates an image plot of the covariance-type values in a_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotCovar(a_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object that resulted from a function like cov.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>inverse</TT>:</STRONG></DT>
<DD>If 1, take inverse of the data matrix.
</DD>
<DT><STRONG><TT>corrcoef</TT>:</STRONG></DT>
<DD>If 1, normalize matrix elements to get corrcoef values.
</DD>
<DT><STRONG><TT>logScale</TT>:</STRONG></DT>
<DD>If 1, take logarithm of values before plotting.
</DD>
<DT><STRONG><TT>localityIters</TT>:</STRONG></DT>
<DD>Apply a locality optimization algorithm with
<P>
this many iterations.
(rest passed to plot_image.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object or one of its subclasses.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plotCovar(cov(get(constrainedMeasuresPreset(pbundle2, 6), 'joined_control_db'))));
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/cov</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__cov"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17939"></A>, <TT>plotImage</TT> (p.&nbsp;<A HREF="#ref_plotImage"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17941"></A>, <TT>tests_db/matchingRow</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17943"></A>, <TT>corrcoefs.</TT> (p.&nbsp;<A HREF="#ref_corrcoefs."><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17945"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/05/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472800000000000000">
1.47.28 Method <TT>tests_db/noNaNRows</TT></A>
</H3><A NAME="17947"></A><A NAME="ref_tests_db__noNaNRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a DB by removing rows containing any NaN or Inf.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = noNaNRows(a_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: DB with missing rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/isnanrows</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__isnanrows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17949"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501472900000000000000">
1.47.29 Method <TT>tests_db/princomp</TT></A>
</H3><A NAME="17951"></A><A NAME="ref_tests_db__princomp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a database of the principal components of given DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pca_db = princomp(db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>normalized</TT>:</STRONG></DT>
<DD>If specified zscore is used before princomp.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_pca_db: A tests_db where each row is a principal component.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>princomp</TT> (p.&nbsp;<A HREF="#ref_princomp"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17953"></A>, <TT>zscore</TT> (p.&nbsp;<A HREF="#ref_zscore"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17955"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/21
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473000000000000000">
1.47.30 Method <TT>tests_db/cov</TT></A>
</H3><A NAME="17957"></A><A NAME="ref_tests_db__cov"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a database of the covariance of given DB.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cov_db = cov(db, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>keepOrigDB</TT>:</STRONG></DT>
<DD>Keep db as origDB in the props.
<P>
(others passed to tests_db)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cov_db: A tests_db which contains the covariance matrix.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cov</TT> (p.&nbsp;<A HREF="#ref_cov"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17959"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/05/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473100000000000000">
1.47.31 Method <TT>tests_db/set</TT></A>
</H3><A NAME="17961"></A><A NAME="ref_tests_db__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473200000000000000">
1.47.32 Method <TT>tests_db/plotrow</TT></A>
</H3><A NAME="17963"></A><A NAME="ref_tests_db__plotrow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a plot_abstract describing the desired db row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotrow(a_tests_db, row, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row number to visualize.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Optional title string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>putLabels</TT>:</STRONG></DT>
<DD>Put special column name labels.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17965"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17967"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473300000000000000">
1.47.33 Method <TT>tests_db/checkConsistentCols</TT></A>
</H3><A NAME="17969"></A><A NAME="ref_tests_db__checkConsistentCols"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Check if two DBs have exactly the same columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[col_names, with_col_names] = checkConsistentCols(db, with_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>with_db</TT>:</STRONG></DT>
<DD>A tests_db object whose column names are checked for consistency.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>useCommon</TT>:</STRONG></DT>
<DD>Tolerate mismatching column names and only return
<P>
the common columns.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
col_names, with_col_names: list of column names of each DB.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>vertcat</TT> (p.&nbsp;<A HREF="#ref_vertcat"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17971"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17973"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/01/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473400000000000000">
1.47.34 Method <TT>tests_db/ne</TT></A>
</H3><A NAME="17975"></A><A NAME="ref_tests_db__ne"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns logical indices of db rows that doesn't match with given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = ne(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>ne</TT> (p.&nbsp;<A HREF="#ref_ne"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17977"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17979"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473500000000000000">
1.47.35 Method <TT>tests_db/std</TT></A>
</H3><A NAME="17981"></A><A NAME="ref_tests_db__std"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the std of the data matrix of a_db. Ignores NaN values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_db, n] = std(a_db, sflag, dim)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Does a recursive operation over dimensions in order to remove NaN values.
 This takes considerable amount of time compared with a straightforward std
 operation. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim</TT>:</STRONG></DT>
<DD>Work down dimension.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The DB with std values.
	n: (Optional) Numbers of non-NaN rows included in calculating each column.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>std</TT> (p.&nbsp;<A HREF="#ref_std"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17983"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17985"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473600000000000000">
1.47.36 Method <TT>tests_db/sortrows</TT></A>
</H3><A NAME="17987"></A><A NAME="ref_tests_db__sortrows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a sorted_db according to given columns. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[sorted_db, idx] = sortrows(db, cols)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>WARNING: For multi-page dbs, sorts only the first page and applies the ordering 
 to all other pages which may produce wrong results for some applications.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>cols</TT>:</STRONG></DT>
<DD>Columns to use for sorting.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
sorted_db: The sorted tests_db.
	idx: The row index permutation vector, such that sorted_db = db(idx, :). 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>sortrows</TT> (p.&nbsp;<A HREF="#ref_sortrows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17989"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17991"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473700000000000000">
1.47.37 Method <TT>tests_db/processDimNonNaNInf</TT></A>
</H3><A NAME="17993"></A><A NAME="ref_tests_db__processDimNonNaNInf"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Recursively process the specified dimension with the desired function after removing NaNs and Infs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_db, n] = processDimNonNaNInf(a_db, dim, a_func, a_func_name)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Does a recursive operation over other dimensions in order to remove
 NaN and Inf values. This takes more time than applying the function directly. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim</TT>:</STRONG></DT>
<DD>Work down dimension (see mean).
</DD>
<DT><STRONG><TT>a_func</TT>:</STRONG></DT>
<DD>A function name or handle to be passed to feval that
<P>
takes the data as the first argument and dimension to
work as second.
</DD>
<DT><STRONG><TT>a_func_name</TT>:</STRONG></DT>
<DD>(Optional) A name to add to the id of a_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The DB with one row of max values, with selected dimension
	replaced by the output of the given function.
   n: (Optional) Numbers of used values in each call of a_func.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">a_db = tests_3D_db(rand(5, 5, 5));
<BR>&#187; b_db = processDimNonNaNInf(a_db, 1, 'mean')
<BR>
will find the mean of rows in each page of the random 3D matrix.
<BR>&#187; b_db = processDimNonNaNInf(a_db, 1, @(x,y)(max(x, [], y)), 'max')
<BR>
more complex function form with 'max'.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>max</TT> (p.&nbsp;<A HREF="#ref_max"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17995"></A>, <TT>mean</TT> (p.&nbsp;<A HREF="#ref_mean"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17997"></A>, <TT>feval</TT> (p.&nbsp;<A HREF="#ref_feval"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="17999"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18001"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/05/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473800000000000000">
1.47.38 Method <TT>tests_db/factoran</TT></A>
</H3><A NAME="18003"></A><A NAME="ref_tests_db__factoran"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a database of factor loadings obtained from the 
		factor analysis of db with factoran. Each row corresponds
		to a rotated factor and columns represent observed variables.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_factors_db = factoran(db, num_factors, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses the promax method to rotate common factors.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>num_factors</TT>:</STRONG></DT>
<DD>Number of common factors to look for.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_factors_db: A corrcoefs_db of the coefficients and page indices.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18005"></A>, <TT>corrcoefs_db</TT> (p.&nbsp;<A HREF="node5.html#ref_corrcoefs_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18007"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501473900000000000000">
1.47.39 Method <TT>tests_db/plotrows</TT></A>
</H3><A NAME="18009"></A><A NAME="ref_tests_db__plotrows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a plot_stack describing the db rows.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotrows(a_tests_db, axis_limits, orientation, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>axis_limits</TT>:</STRONG></DT>
<DD>If given, all plots contained will have these axis limits.
</DD>
<DT><STRONG><TT>orientation</TT>:</STRONG></DT>
<DD>Stack orientation 'x' for horizontal, 'y' for vertical, etc.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Optional title string.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties passed to plot_stack.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_stack object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18011"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18013"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474000000000000000">
1.47.40 Method <TT>tests_db/mean</TT></A>
</H3><A NAME="18015"></A><A NAME="ref_tests_db__mean"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the mean of the data matrix of a_db. Ignores NaN values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_db, n] = mean(a_db, dim)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Does a recursive operation over dimensions in order to remove NaN values.
 This takes more time than a straightforward mean operation. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim</TT>:</STRONG></DT>
<DD>Work down dimension.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The DB with one row of mean values.
	n: (Optional) Numbers of non-NaN rows included in calculating each column.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>mean</TT> (p.&nbsp;<A HREF="#ref_mean"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18017"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18019"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474100000000000000">
1.47.41 Method <TT>tests_db/rankMatching</TT></A>
</H3><A NAME="18021"></A><A NAME="ref_tests_db__rankMatching"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a ranking db of row distances of db to given criterion db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_ranked_db = rankMatching(db, crit_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>crit_db can be created with the matchingRow method. TestWeights modify the importance 
 of each measure.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db to rank.
</DD>
<DT><STRONG><TT>crit_db</TT>:</STRONG></DT>
<DD>A tests_db object holding the match criterion tests and stds.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>limitSTD</TT>:</STRONG></DT>
<DD>limit any measure to this many STDs max.
</DD>
<DT><STRONG><TT>tolerateNaNs</TT>:</STRONG></DT>
<DD>If 0, rows with any NaN values are skipped
<P>
, if 1, NaN values are given a fixed 3xSTD penalty (default=1).
</DD>
<DT><STRONG><TT>testWeights</TT>:</STRONG></DT>
<DD>Structure array associating tests and multiplicative weights.
</DD>
<DT><STRONG><TT>restoreWeights</TT>:</STRONG></DT>
<DD>Reverse the testWeights application after
<P>
calculating distances.
</DD>
<DT><STRONG><TT>topRows</TT>:</STRONG></DT>
<DD>If given, only return this many of the top rows.
</DD>
<DT><STRONG><TT>useMahal</TT>:</STRONG></DT>
<DD>Use the Mahalonobis distance from the covariance

<P>
matrix in crit_db.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_ranked_db: A ranked_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>matchingRow</TT> (p.&nbsp;<A HREF="#ref_matchingRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18023"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18025"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474200000000000000">
1.47.42 Method <TT>tests_db/onlyRowsTests</TT></A>
</H3><A NAME="18027"></A><A NAME="ref_tests_db__onlyRowsTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a tests_db that only contains the desired 
		tests and rows (and pages).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = onlyRowsTests(obj, rows, tests, pages)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Selects the given dimensions and returns in a new tests_db object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows, tests</TT>:</STRONG></DT>
<DD>A logical or index vector of rows, or cell array of
<P>
names of rows. If ':', all rows. For names, regular expressions are
supported if quoted with slashes (e.g., '/a.*/'). See tests2idx.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>(Optional) A logical or index vector of pages. ':' for all pages.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18029"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18031"></A>, <TT>tests2idx</TT> (p.&nbsp;<A HREF="#ref_tests2idx"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18033"></A>, <TT>regexp</TT> (p.&nbsp;<A HREF="#ref_regexp"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18035"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474300000000000000">
1.47.43 Method <TT>tests_db/times</TT></A>
</H3><A NAME="18037"></A><A NAME="ref_tests_db__times"></A><DL>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = mtimes(left_obj, right_obj)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the multiplication. One or more must be of type tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/times</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__times"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18039"></A>, <TT>mtimes</TT> (p.&nbsp;<A HREF="#ref_mtimes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18041"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474400000000000000">
1.47.44 Method <TT>tests_db/renameColumns</TT></A>
</H3><A NAME="18043"></A><A NAME="ref_tests_db__renameColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Rename an existing column or columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = renameColumns(a_db, test_names, new_names)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is a cheap operation than modifies meta-data kept in object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>test_names</TT>:</STRONG></DT>
<DD>A cell array of existing test names.
</DD>
<DT><STRONG><TT>new_names</TT>:</STRONG></DT>
<DD>New names to replace existing ones.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The tests_db object that includes the new columns.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; new_db = renameColumns(a_db, 'PulseIni100msSpikeRateISI_D40pA', 'Firing_rate');
<BR>&#187; new_db = renameColumns(a_db, 'a', 'b', 'c', 'd');
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18045"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18047"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474500000000000000">
1.47.45 Method <TT>tests_db/statsAll</TT></A>
</H3><A NAME="18049"></A><A NAME="ref_tests_db__statsAll"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Makes a stats_db with rows of mean, STD, SE, and CV of the tests' distributions in db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stats_db = statsAll(db, tests, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A selection of tests (see onlyRowsTests).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties for stats_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_stats_db: A stats_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18051"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474600000000000000">
1.47.46 Method <TT>tests_db/getColNames</TT></A>
</H3><A NAME="18053"></A><A NAME="ref_tests_db__getColNames"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Gets column names.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">col_names = getColNames(db, tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Performs a light operation without touching the data matrix.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Columns for which to get names (Optional, default = ':')
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
col_names: A cell array of strings.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>getColNames</TT> (p.&nbsp;<A HREF="#ref_getColNames"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18055"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18057"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474700000000000000">
1.47.47 Method <TT>tests_db/addLastRow</TT></A>
</H3><A NAME="18059"></A><A NAME="ref_tests_db__addLastRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Inserts a row of observations at the end of tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">index = addLastRow(obj, row)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Adds a new set of observations to the database and returns its row index.
   This operation is expensive because the whole 
   database matrix needs to be duplicated and resized in order to add a 
   single new row. The method of allocating a matrix, filling it up, and
   then providing it to the tests_db constructor is the preferred method 
   of creating tests_db objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row vector that contains values for each DB column.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that includes the new row.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18061"></A>, <TT>addRow</TT> (p.&nbsp;<A HREF="#ref_addRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18063"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18065"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474800000000000000">
1.47.48 Method <TT>tests_db/isinf</TT></A>
</H3><A NAME="18067"></A><A NAME="ref_tests_db__isinf"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns logical row indices of Inf-valued columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = isinf(db, col)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>col</TT>:</STRONG></DT>
<DD>Column to check (Optional, default = 1)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical column vector of rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>isinf</TT> (p.&nbsp;<A HREF="#ref_isinf"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18069"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18071"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501474900000000000000">
1.47.49 Method <TT>tests_db/subsref</TT></A>
</H3><A NAME="18073"></A><A NAME="ref_tests_db__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines indexing for tests_db objects for () and . operations. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = obj(rows, tests)
 obj = obj.attribute</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Returns attributes or selects the given test columns and rows
 and returns in a new tests_db object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>A logical or index vector of rows. If ':', all rows.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Cell array of test names or column indices. If ':', all tests.
</DD>
<DT><STRONG><TT>attribute</TT>:</STRONG></DT>
<DD>A tests_db class attribute.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18075"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18077"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475000000000000000">
1.47.50 Method <TT>tests_db/eq</TT></A>
</H3><A NAME="18079"></A><A NAME="ref_tests_db__eq"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Equality (==) operator. Returns logical indices of db rows 
	that match with given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = eq(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>eq</TT> (p.&nbsp;<A HREF="#ref_eq"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18081"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18083"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475100000000000000">
1.47.51 Method <TT>tests_db/plotBox</TT></A>
</H3><A NAME="18085"></A><A NAME="ref_tests_db__plotBox"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a boxplot from each column in tests_db in separate axes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotBox(a_tests_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>Optional title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>putLabels</TT>:</STRONG></DT>
<DD>Put special column name labels.
</DD>
<DT><STRONG><TT>notch</TT>:</STRONG></DT>
<DD>If 1, put notches on boxplots (default=1).
</DD>
<DT><STRONG><TT>whis</TT>:</STRONG></DT>
<DD>Whisker size passed to boxplotp (default=1.5);
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18087"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18089"></A>, <TT>boxplotp</TT> (p.&nbsp;<A HREF="#ref_boxplotp"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18091"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475200000000000000">
1.47.52 Method <TT>tests_db/uminus</TT></A>
</H3><A NAME="18093"></A><A NAME="ref_tests_db__uminus"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Unary minus or negation.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = uminus(left_obj)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>uminus</TT> (p.&nbsp;<A HREF="#ref_uminus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18095"></A>, <TT>uop</TT> (p.&nbsp;<A HREF="#ref_uop"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18097"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/16
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475300000000000000">
1.47.53 Method <TT>tests_db/subsasgn</TT></A>
</H3><A NAME="18099"></A><A NAME="ref_tests_db__subsasgn"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic index-based assignment for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475400000000000000">
1.47.54 Method <TT>tests_db/dbsize</TT></A>
</H3><A NAME="18101"></A><A NAME="ref_tests_db__dbsize"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the size of the data matrix of db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = dbsize(db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: The size values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>size</TT> (p.&nbsp;<A HREF="#ref_size"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18103"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18105"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475500000000000000">
1.47.55 Method <TT>tests_db/sqrt</TT></A>
</H3><A NAME="18107"></A><A NAME="ref_tests_db__sqrt"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Takes the square root of a_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = sqrt(a_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Overloaded sqrt function.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>sqrt</TT> (p.&nbsp;<A HREF="#ref_sqrt"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18109"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475600000000000000">
1.47.56 Method <TT>tests_db/plotTestsHistsMatrix</TT></A>
</H3><A NAME="18111"></A><A NAME="ref_tests_db__plotTestsHistsMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a matrix plot of test histograms.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pm = plotTestsHistsMatrix(a_db, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Skips the 'ItemIndex' test.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_abstract.
<DL>
<DD></DD>
<DT><STRONG><TT>orient</TT>:</STRONG></DT>
<DD>Orientation of the plot_stack. 'x', 'y', or 'matrix' (default).
</DD>
<DT><STRONG><TT>histBins</TT>:</STRONG></DT>
<DD>Number of histogram bins.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>Don't put the DB id on the title.
</DD>
<DT><STRONG><TT>axisLimits</TT>:</STRONG></DT>
<DD>Only x-ranges are used from this expression.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_pm: A plot_stack with the plots organized in matrix form
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18113"></A>, <TT>plotVar</TT> (p.&nbsp;<A HREF="#ref_plotVar"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18115"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475700000000000000">
1.47.57 Method <TT>tests_db/anyRows</TT></A>
</H3><A NAME="18117"></A><A NAME="ref_tests_db__anyRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns db rows matching any of the given rows.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">idx = anyRows(db, rows)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The db rows are compared to each row and row indices succeeding any of
 these comparisons are returned.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Row array, matrix or database to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
idx: A logical column vector of matching db row indices. 
	rows_idx: Indices of rows entries corresponding to each db
		row. Non-matching entries were left as NaN.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; db(anyRows(db(:, 'trial') == [12; 46; 37]), :)
<BR>
returns a db with rows having trial equal to any of the given values.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>compareRows</TT> (p.&nbsp;<A HREF="#ref_compareRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18119"></A>, <TT>eq</TT> (p.&nbsp;<A HREF="#ref_eq"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18121"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18123"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475800000000000000">
1.47.58 Method <TT>tests_db/gt</TT></A>
</H3><A NAME="18125"></A><A NAME="ref_tests_db__gt"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Greater than (&gt;) operator. Returns logical indices of db rows 
	that are greater than given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = gt(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>gt</TT> (p.&nbsp;<A HREF="#ref_gt"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18127"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18129"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501475900000000000000">
1.47.59 Method <TT>tests_db/hist3</TT></A>
</H3><A NAME="18131"></A><A NAME="ref_tests_db__hist3"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a 2D histogram of chosen two database columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_histogram_db = hist3(db, col1, col2, num_bins, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Generates a histogram_db object with rows corresponding to histogram
 entries. If an array of DBs is given, finds and uses common histogram bin centers.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>col1, col2</TT>:</STRONG></DT>
<DD>Columns to find the histogram.
</DD>
<DT><STRONG><TT>num_bins</TT>:</STRONG></DT>
<DD>Two numbers for histogram bins (Optional, default=100), or
<P>
Two-element cell of histogram bin centers vectors.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>normalized</TT>:</STRONG></DT>
<DD>If 1, normalize histogram counts.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_histogram_db: A histogram_db object containing the 2D histogram.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_hist_db = hist3(my_db, 'spike_width', 'spike_rate');
<BR>&#187; plot(a_hist_db);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18133"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18135"></A>, <TT>hist3</TT> (p.&nbsp;<A HREF="#ref_hist3"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18137"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476000000000000000">
1.47.60 Method <TT>tests_db/kmeansCluster</TT></A>
</H3><A NAME="18139"></A><A NAME="ref_tests_db__kmeansCluster"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a database of cluster centers obtained from a k-means cluster analysis with the command kmeans.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_cluster_db = kmeansCluster(db, num_clusters, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>num_clusters</TT>:</STRONG></DT>
<DD>Number of clusters to form.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>DistanceMeasure</TT>:</STRONG></DT>
<DD>Choose one appropriate for kmeans.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_cluster_db: A tests_db where each row is a cluster center.
	a_hist_db: histogram_db showing cluster membership from original db.
	idx: Cluster indices of each row or original db.
	sum_distances: Quality of clustering indicated by total distance from
			centroid for each cluster.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18141"></A>, <TT>histogram_db</TT> (p.&nbsp;<A HREF="node5.html#ref_histogram_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18143"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476100000000000000">
1.47.61 Method <TT>tests_db/compareRows</TT></A>
</H3><A NAME="18145"></A><A NAME="ref_tests_db__compareRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns differing rows of db and the given row(s).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[idx, compared] = compareRows(db, rows)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It can compare all db rows to corresponding row entries or to a single
 row. For the case with only one entry, returns all db rows that do not
 match the given row in idx, and the result of the differences in
 compared. For the case of multiple rows, rows must have exactly same number
 of rows with db. In both cases, idx must be negated to test for equality.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Row array, matrix or database to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
idx: A inverted logical column vector of comparison results. 
		(false if db == rows, true otherwise)
	compared: A column vector of differences of each DB row to the
 		given row (i.e., compared = db - rows).
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; db(db(:, 'trial') &gt; [12]), :)
<BR>
calls gt which calls compareRows to check for equality. Returns a db
<BR>
only containing rows with trial numbers greater than 12.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>eq</TT> (p.&nbsp;<A HREF="#ref_eq"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18147"></A>, <TT>anyRows</TT> (p.&nbsp;<A HREF="#ref_anyRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18149"></A>, <TT>ge</TT> (p.&nbsp;<A HREF="#ref_ge"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18151"></A>, <TT>gt</TT> (p.&nbsp;<A HREF="#ref_gt"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18153"></A>, <TT>le</TT> (p.&nbsp;<A HREF="#ref_le"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18155"></A>, <TT>lt</TT> (p.&nbsp;<A HREF="#ref_lt"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18157"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18159"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476200000000000000">
1.47.62 Method <TT>tests_db/statsMeanStd</TT></A>
</H3><A NAME="18161"></A><A NAME="ref_tests_db__statsMeanStd"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a stats_db object with mean, STD, and number of observations of the tests' distributions.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stats_db = statsMeanStd(db, tests, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A selection of tests (see onlyRowsTests).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties for stats_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_stats_db: A stats_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18163"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476300000000000000">
1.47.63 Method <TT>tests_db/display</TT></A>
</H3><A NAME="18165"></A><A NAME="ref_tests_db__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476400000000000000">
1.47.64 Method <TT>tests_db/groupBy</TT></A>
</H3><A NAME="18167"></A><A NAME="ref_tests_db__groupBy"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Groups same values of column(s) into separate pages of a 3D db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tests_3D_db = groupBy(db, cols)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Functionality similar to SQL's GROUP BY keyword. This function uses invarValues.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>cols</TT>:</STRONG></DT>
<DD>Columns whose same values will be in one page (see tests2cols
<P>
for column representation).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tests_3D_db: A tests_3D_db object of organized values.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_db = tests_db([ ... ], 'par1', 'par2', 'measure1', 'measure2')
<BR>&#187; a_3d_db = groupBy(a_db, 'par1')
<BR>&#187; &#187; joined_3d_db = joinRows(a_db, a_3d_db)
<BR>&#187; displayRows(joined_3d_db(:, :, 1))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>invarValues</TT> (p.&nbsp;<A HREF="#ref_invarValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18169"></A>, <TT>tests_3D_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18171"></A>, <TT>tests_3D_db/corrCoefs</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__corrCoefs"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18173"></A>, <TT>tests_3D_db/plotPair</TT> (p.&nbsp;<A HREF="#ref_tests_3D_db__plotPair"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18175"></A>, <TT>joinRows</TT> (p.&nbsp;<A HREF="#ref_joinRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18177"></A>, <TT>tests_3D_db/swapRowsPages</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__swapRowsPages"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18179"></A>, <TT>tests_3D_db/mergePages</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_3D_db__mergePages"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18181"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/05/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476500000000000000">
1.47.65 Method <TT>tests_db/statsBounds</TT></A>
</H3><A NAME="18183"></A><A NAME="ref_tests_db__statsBounds"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a stats_db object with three rows corresponding to the mean, min, max and number of observations of the tests' distributions. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stats_db = statsBounds(a_db, tests, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>A page is generated for each page of data in db.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A selection of tests (see onlyRowsTests).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties for stats_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_stats_db: A stats_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18185"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476600000000000000">
1.47.66 Method <TT>tests_db/end</TT></A>
</H3><A NAME="18187"></A><A NAME="ref_tests_db__end"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Overloaded primitive matlab function, returns maximal dimension size.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = end(db, index, total)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: The size.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>size</TT> (p.&nbsp;<A HREF="#ref_size"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18189"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18191"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476700000000000000">
1.47.67 Method <TT>tests_db/isnanrows</TT></A>
</H3><A NAME="18193"></A><A NAME="ref_tests_db__isnanrows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds rows with any NaN values. Returns logical indices of db rows.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = isnanrows(db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Some operations need that no NaN values exist in the matrix. This method
 can be used to find and then remove NaN-contaminated rows from DB. Note
 that sometimes no rows can  be found, and some columns should be discarded
 before this operation.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical vector to be used in indexing db objects or passed
		through other logical operators. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>isnan</TT> (p.&nbsp;<A HREF="#ref_isnan"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18195"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18197"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476800000000000000">
1.47.68 Method <TT>tests_db/le</TT></A>
</H3><A NAME="18199"></A><A NAME="ref_tests_db__le"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Less or equal (&lt;=) operator. Returns logical indices of db rows 
	that are less than or equal to given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = le(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>le</TT> (p.&nbsp;<A HREF="#ref_le"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18201"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18203"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501476900000000000000">
1.47.69 Method <TT>tests_db/plotUniquesStatsStacked3D</TT></A>
</H3><A NAME="18205"></A><A NAME="ref_tests_db__plotUniquesStatsStacked3D"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Stack of 2D image plots of a column mean at unique values of three other columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stacked_plot = plotUniquesStatsStacked3D(a_db, unique_test1, unique_test2, 
 					unique_test3, stat_test, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db.
</DD>
<DT><STRONG><TT>unique_test1, unique_test2</TT>:</STRONG></DT>
<DD>Columns whose unique values make up the X
<P>
&amp; Y of the 2D image plot.
</DD>
<DT><STRONG><TT>unique_test3</TT>:</STRONG></DT>
<DD>Column whose unique values make up stacked dimension.
</DD>
<DT><STRONG><TT>stat_test</TT>:</STRONG></DT>
<DD>Column for which statsMeanSTD will be calculated for each

<P>
unique value.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.

<P>
(rest passed to plotUniquesStats2D and plot_stack).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_stacked_plot: A plot_abstract object to be plotted.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18207"></A>, <TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18209"></A>, <TT>statsMeanStd</TT> (p.&nbsp;<A HREF="#ref_statsMeanStd"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18211"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18213"></A>, <TT>plotImage</TT> (p.&nbsp;<A HREF="#ref_plotImage"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18215"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477000000000000000">
1.47.70 Method <TT>tests_db/plotScatter3D</TT></A>
</H3><A NAME="18217"></A><A NAME="ref_tests_db__plotScatter3D"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a 3D scatter plot of the given three tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_p = plotScatter3D(a_db, test1, test2, test3, title_str, short_title, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>test1, test2, test3</TT>:</STRONG></DT>
<DD>X, Y, &amp; Z variables.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>short_title</TT>:</STRONG></DT>
<DD>(Optional) Few words that may appear in legends of multiplot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>LineStyle</TT>:</STRONG></DT>
<DD>Plot line style to use. (default: 'x')
</DD>
<DT><STRONG><TT>Regress</TT>:</STRONG></DT>
<DD>Calculate and plot a linear regression.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_p: A plot_abstract.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/11/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477100000000000000">
1.47.71 Method <TT>tests_db/rop</TT></A>
</H3><A NAME="18219"></A><A NAME="ref_tests_db__rop"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Prepares aligned columns in two DBs or one DB and a scalar for an array arithmetic operation.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = rop(left_obj, right_obj, op_func, op_id)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If DBs have mismatching columns only the common columns will be kept.
 In any case, the resulting DB columns will be sorted in the order of the
 left-hand-side DB. Array addition (plus), subtraction (minus),
 multiplication (mtimes) and division (rdivide) use this function to
 align columns.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the operation. One must be of type tests_db
<P>
and the other can be a scalar or tests_db.
</DD>
<DT><STRONG><TT>op_func</TT>:</STRONG></DT>
<DD>Operation function (e.g., @plus).
</DD>
<DT><STRONG><TT>op_id</TT>:</STRONG></DT>
<DD>A string to represent the operation that will show up in the

<P>
returned id.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/plus</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__plus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18221"></A>, <TT>tests_db/minus</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__minus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18223"></A>, <TT>tests_db/mtimes</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__mtimes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18225"></A>, <TT>tests_db/rdivide</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__rdivide"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18227"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477200000000000000">
1.47.72 Method <TT>tests_db/vertcat</TT></A>
</H3><A NAME="18229"></A><A NAME="ref_tests_db__vertcat"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Vertical concatanation [db;with_db;...] operator.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = vertcat(db, with_db, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Concatanates rows of with_db to rows of db. Overrides the built-in
 vertcat function that is called when [db;with_db] is executed. If the 
 first argument is a array of DBs, then this functionality is not needed;
 built-in vertcat is called.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>with_db</TT>:</STRONG></DT>
<DD>A tests_db object whose rows are concatanated to db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db that contains rows of db and with_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>vertcat</TT> (p.&nbsp;<A HREF="#ref_vertcat"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18231"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18233"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/01/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477300000000000000">
1.47.73 Method <TT>tests_db/lt</TT></A>
</H3><A NAME="18235"></A><A NAME="ref_tests_db__lt"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Less than (&lt;) operator. Returns logical indices of db rows 
	that are less than given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = lt(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>lt</TT> (p.&nbsp;<A HREF="#ref_lt"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18237"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18239"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477400000000000000">
1.47.74 Method <TT>tests_db/unique</TT></A>
</H3><A NAME="18241"></A><A NAME="ref_tests_db__unique"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns DB with unique rows.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = unique(a_db)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>tests_db from which to find uniques.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>unique</TT> (p.&nbsp;<A HREF="#ref_unique"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18243"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/11/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477500000000000000">
1.47.75 Method <TT>tests_db/addRow</TT></A>
</H3><A NAME="18245"></A><A NAME="ref_tests_db__addRow"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Inserts a row of observations to tests_db at the given row index.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">index = addRow(obj, row, index)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Adds a new set of observations to the database and returns the new DB.
   This operation is expensive in the sense that the whole database matrix
   needs to be copied to be passed to this function just to add a 
   single new row. The method of allocating a matrix, filling it up, and
   then providing it to the tests_db constructor is the preferred method 
   of creating tests_db objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row vector that contains values for each DB column.
</DD>
<DT><STRONG><TT>index</TT>:</STRONG></DT>
<DD>The row index.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that includes the new row.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>addLastRow</TT> (p.&nbsp;<A HREF="#ref_addLastRow"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18247"></A>, <TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18249"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18251"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477600000000000000">
1.47.76 Method <TT>tests_db/plotYTests</TT></A>
</H3><A NAME="18253"></A><A NAME="ref_tests_db__plotYTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a plot given database measures against given X-axis values, for each row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_p = plotYTests(a_db, x_vals, tests, axis_labels, title_str, short_title, command, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>x_vals</TT>:</STRONG></DT>
<DD>A vector of X-axis values.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A vector or cell array of columns to correspond to each value from x_vals.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>short_title</TT>:</STRONG></DT>
<DD>(Optional) Few words that may appear in legends of multiplot.
</DD>
<DT><STRONG><TT>command</TT>:</STRONG></DT>
<DD>(Optional) Command to do the plotting with (default: 'plot')
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>LineStyle</TT>:</STRONG></DT>
<DD>Plot line style to use. (default: 'd-')
</DD>
<DT><STRONG><TT>ShowErrorbars</TT>:</STRONG></DT>
<DD>If 1, errorbars are added to each point.
</DD>
<DT><STRONG><TT>StatsDB</TT>:</STRONG></DT>
<DD>If given, use this stats_db for the errorbar (default=statsMeanStd(a_db)).
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_p: A plot_abstract.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_p = plotYTests(a_db_row, [0 40 100 200], ...
<BR>'IniSpontSpikeRateISI_0pA', 'PulseIni100msSpikeRateISI_D40pA', ...
<BR>'PulseIni100msSpikeRateISI_D100pA', 'PulseIni100msSpikeRateISI_D200pA', ...
<BR>'current pulse [pA]', 'firing rate [Hz]', ', f-I curves', 'neuron 1');
<BR>&#187; plotFigure(a_p);
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18255"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/23
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477700000000000000">
1.47.77 Method <TT>tests_db/plotUniquesStatsBars</TT></A>
</H3><A NAME="18257"></A><A NAME="ref_tests_db__plotUniquesStatsBars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a mean-STD bar plot of a column for unique values of another column.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_bar_plot = plotUniquesStatsBars(a_db, unique_test, stat_test, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db.
</DD>
<DT><STRONG><TT>unique_test</TT>:</STRONG></DT>
<DD>Column for which to generate bars for each of its unique values.
</DD>
<DT><STRONG><TT>stat_test</TT>:</STRONG></DT>
<DD>Column for which statsMeanSTD will be calculated for each bar.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>popMean</TT>:</STRONG></DT>
<DD>If specified, plot a dotted line specifying the
<P>
population mean. If NaN, calculated from a_db.
</DD>
<DT><STRONG><TT>yLims</TT>:</STRONG></DT>
<DD>two-element vector for specifying y axis limits showing
<P>
interesting part of the bar plot.
</DD>
<DT><STRONG><TT>uniqueVals</TT>:</STRONG></DT>
<DD>Use these unique values for unique_test.

<P>
(rest passed to plot_bars [and plot_superpose if popMean]).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_bar_plot: A plot_abstract object to be plotted.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plotUniquesStatsBars(triplet_param_success_db, 'F_tau_m', ...
<BR>'successDefault', 'across triplets', ...
<BR>
struct('fixedSize', [4 2], 'yLims', [.7 .9], ...
<BR>'popMean', 0.82, 'quiet', 1)))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18259"></A>, <TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18261"></A>, <TT>statsMeanStd</TT> (p.&nbsp;<A HREF="#ref_statsMeanStd"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18263"></A>, <TT>plot_bars</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_bars"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18265"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477800000000000000">
1.47.78 Method <TT>tests_db/plotScatter</TT></A>
</H3><A NAME="18267"></A><A NAME="ref_tests_db__plotScatter"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a scatter plot of the given two tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_p = plotScatter(a_db, test1, test2, title_str, short_title, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>test1, test2</TT>:</STRONG></DT>
<DD>X &amp; Y variables.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>short_title</TT>:</STRONG></DT>
<DD>(Optional) Few words that may appear in legends of multiplot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>LineStyle</TT>:</STRONG></DT>
<DD>Plot line style to use. (default: 'x')
</DD>
<DT><STRONG><TT>Regress</TT>:</STRONG></DT>
<DD>Calculate and plot a linear regression.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, don't include database name on title.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_p: A plot_abstract.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/29
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501477900000000000000">
1.47.79 Method <TT>tests_db/assignRowsTests</TT></A>
</H3><A NAME="18269"></A><A NAME="ref_tests_db__assignRowsTests"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Assign the values to the tests and rows (and pages) of the tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = assignRowsTests(obj, val, rows, tests, pages)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Selects the given dimensions and returns in a new tests_db object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>DB object or data matrix to be assigned to the addressed indices.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>A logical or index vector of rows. If ':', all rows.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Cell array of test names or column indices. If ':', all tests.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>(Optional) A logical or index vector of pages. ':' for all pages.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>subsref</TT> (p.&nbsp;<A HREF="#ref_subsref"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18271"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18273"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/02/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478000000000000000">
1.47.80 Method <TT>tests_db/plus</TT></A>
</H3><A NAME="18275"></A><A NAME="ref_tests_db__plus"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Adds a DB to another or to a scalar.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = plus(left_obj, right_obj)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If DBs have mismatching columns only the common columns will be kept.
 In any case, the resulting DB columns will be sorted in the order of the
 left-hand-side DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>left_obj, right_obj</TT>:</STRONG></DT>
<DD>Operands of the addition. One must be of type tests_db
<P>
and the other can be a scalar or tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plus</TT> (p.&nbsp;<A HREF="#ref_plus"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18277"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478100000000000000">
1.47.81 Method <TT>tests_db/displayRows</TT></A>
</H3><A NAME="18279"></A><A NAME="ref_tests_db__displayRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Displays rows of data with associated column labels.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = displayRows(db, rows, pages)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>Pages of db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: A cell array of trasposed database contents, prefixed with 
	   column names on each row. Meant to be displayed on the screen.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18281"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478200000000000000">
1.47.82 Method <TT>tests_db/testsHists</TT></A>
</H3><A NAME="18283"></A><A NAME="ref_tests_db__testsHists"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates histograms for all tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">t_hists = testsHists(a_db, num_bins)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>num_bins</TT>:</STRONG></DT>
<DD>Number of histogram bins (Optional, default=100), or
<P>
vector of histogram bin centers.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
t_hists: An array of histograms for each test in a_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>params_tests_profile</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18285"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478300000000000000">
1.47.83 Method <TT>tests_db/ge</TT></A>
</H3><A NAME="18287"></A><A NAME="ref_tests_db__ge"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Greater or equal to (&gt;=) operator. Returns logical indices of db rows 
	that are greater than or equal to given row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">rows = ge(db, row)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>Row array to be compared with db rows.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A logical or index vector to be used in indexing db objects. 
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>ge</TT> (p.&nbsp;<A HREF="#ref_ge"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18289"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18291"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478400000000000000">
1.47.84 Method <TT>tests_db/swapRowsPages</TT></A>
</H3><A NAME="18293"></A><A NAME="ref_tests_db__swapRowsPages"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Swaps the row dimension with the page dimension of the tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = swapRowsPages(db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Watered-down version of the tests_3D_db/swapRowsPages function that
 does not touch row indices. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18295"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478500000000000000">
1.47.85 Method <TT>tests_db/plotUniquesStats2D</TT></A>
</H3><A NAME="18297"></A><A NAME="ref_tests_db__plotUniquesStats2D"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>2D image plot of the change in column mean for unique values of two other columns.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">an_image_plot = plotUniquesStats2D(a_db, unique_test1, unique_test2, 
 					stat_test, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db.
</DD>
<DT><STRONG><TT>unique_test1, unique_test2</TT>:</STRONG></DT>
<DD>Columns whose unique values make up the X
<P>
&amp; Y of the 2D image plot.
</DD>
<DT><STRONG><TT>stat_test</TT>:</STRONG></DT>
<DD>Column for which statsMeanSTD will be calculated for each

<P>
unique value.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>popMean</TT>:</STRONG></DT>
<DD>If specified, plot a dotted line specifying the
<P>
population mean. If NaN, calculate from given a_db.
</DD>
<DT><STRONG><TT>popDev</TT>:</STRONG></DT>
<DD>Use this value +/- to choose colorbar extents

<P>
(default=.3 or 2*STD if popMean=NaN).
</DD>
<DT><STRONG><TT>colorbar</TT>:</STRONG></DT>
<DD>Show vertical colorbar axis (see plotImage).
</DD>
<DT><STRONG><TT>uniqueVals1,uniqueVals2</TT>:</STRONG></DT>
<DD>Use these unique values for

<P>
unique_test1,unique_test2.
</DD>
<DT><STRONG><TT>statsFunc</TT>:</STRONG></DT>
<DD>tests_db/stats* method to use (default: statsMeanStd).
</DD>
<DT><STRONG><TT>statsRow</TT>:</STRONG></DT>
<DD>The row to pick from statsFunc results (default: mean).

<P>
(rest passed to plotImage and plot_abstract).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
an_image_plot: A plot_abstract object to be plotted.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plotFigure(plotUniquesStats2D(triplet_param_success_db, ...
<BR>'F_tau_m', 'S_tau_m', 'successDefault', ...
<BR>'accross triplets', 
<BR>
struct('fixedSize', [4 3], 'popMean', NaN, ...
<BR>'colorbar', 1, 'quiet', 1, 'border', [0.03 0 0.03 0])))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18299"></A>, <TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18301"></A>, <TT>statsMeanStd</TT> (p.&nbsp;<A HREF="#ref_statsMeanStd"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18303"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18305"></A>, <TT>plotImage</TT> (p.&nbsp;<A HREF="#ref_plotImage"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18307"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/04/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478600000000000000">
1.47.86 Method <TT>tests_db/plot_bars</TT></A>
</H3><A NAME="18309"></A><A NAME="ref_tests_db__plot_bars"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a bar graph comparing all DB rows in groups, with a separate axis for each column.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_bars(a_tests_db, title_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_tests_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) The plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A object of plot_bars or one of its subclasses.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18311"></A>, <TT>plot_simple</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_simple"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18313"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/03/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478700000000000000">
1.47.87 Method <TT>tests_db/diff</TT></A>
</H3><A NAME="18315"></A><A NAME="ref_tests_db__diff"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a tests_db by taking the derivative of all tests.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = diff(a_db, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Applies the diff function to whole DB. The resulting DB will have one less row.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>diff</TT> (p.&nbsp;<A HREF="#ref_diff"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18317"></A>, <TT>tests_3D_db/getDiff2DDB</TT> (p.&nbsp;<A HREF="#ref_tests_3D_db__getDiff2DDB"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18319"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478800000000000000">
1.47.88 Method <TT>tests_db/delColumns</TT></A>
</H3><A NAME="18321"></A><A NAME="ref_tests_db__delColumns"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Deletes columns from tests_db.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = delColumns(obj, tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Deletes test columns from the database and returns the new DB.
   This operation is expensive in the sense that the whole database matrix
   needs to be copied just to delete a 
   single column. The method of allocating a matrix, filling it up, and
   then providing it to the tests_db constructor is the preferred method 
   of creating tests_db objects. This method may be used for 
   measures obtained by operating on raw measures.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Numbers or names of tests (see tests2cols)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The tests_db object that is missing the columns.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18323"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18325"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501478900000000000000">
1.47.89 Method <TT>tests_db/meanDuplicateRows</TT></A>
</H3><A NAME="18327"></A><A NAME="ref_tests_db__meanDuplicateRows"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Row-reduces a db by finding sets of rows with same main_cols values, and replacing each set with a single row containing main_cols and the mean of rest_cols.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tests_db = meanDuplicateRows(db, main_cols, rest_cols)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The database is sorted for the values of the columns of 
 interest (main_cols) and all rows with duplicate values of 
 these columns are identified. The rest of the columns (rest_cols) 
 are averaged and reduced to a single row, and attached to the
 nominal values of main_cols. Two additional parameter columns will be added to the
 database created. The NumDuplicates column is the the number of duplicates 
 used in the mean operation, and RowIndex is the row number points 
 to the first of a set of duplicate values.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>main_cols</TT>:</STRONG></DT>
<DD>Vector of columns in which to find duplicates.
</DD>
<DT><STRONG><TT>rest_cols</TT>:</STRONG></DT>
<DD>Vector of columns to be averaged for duplicate main_cols.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tests_db: The db object of with the means on page 1 
		    and standard deviations on page 2.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db/mean</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__mean"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18329"></A>, <TT>tests_db/std</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db__std"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18331"></A>, <TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18333"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479000000000000000">
1.47.90 Method <TT>tests_db/statsMeanSE</TT></A>
</H3><A NAME="18335"></A><A NAME="ref_tests_db__statsMeanSE"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generates a stats_db object with two rows corresponding to 
		the mean and standard error (SE) of the tests' distributions.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_stats_db = statsMeanSE(db, tests, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>A selection of tests (see onlyRowsTests).
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties for stats_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_stats_db: A stats_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18337"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479100000000000000">
1.47.91 Method <TT>tests_db/sum</TT></A>
</H3><A NAME="18339"></A><A NAME="ref_tests_db__sum"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a tests_db by summing all rows.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_db = sum(a_db, dim, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Applies the sum function to whole DB. The resulting DB will have one row.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>Optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The resulting tests_db.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>sum</TT> (p.&nbsp;<A HREF="#ref_sum"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18341"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/05/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479200000000000000">
1.47.92 Method <TT>tests_db/plot_abstract</TT></A>
</H3><A NAME="18343"></A><A NAME="ref_tests_db__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Default visualization for a database.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_pm = plot_abstract(a_db, title_str)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Calls plotTestsHistsMatrix. Subclasses should override this method
 to provide their own visualization.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A params_tests_db object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) A string to be concatanated to the title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_pm: A plot_stack with the plots organized in matrix form
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; plot(my_db, ': first impression')
<BR>
will call this function and send the generated plot to the plotFigure function.
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract/plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract__plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18345"></A>, <TT>plotTestsHistsMatrix</TT> (p.&nbsp;<A HREF="#ref_plotTestsHistsMatrix"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18347"></A>, <TT>plotFigure</TT> (p.&nbsp;<A HREF="#ref_plotFigure"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18349"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479300000000000000">
1.47.93 Method <TT>tests_db/crossProd</TT></A>
</H3><A NAME="18351"></A><A NAME="ref_tests_db__crossProd"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Create a DB by taking the cross product of two database row sets.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">cross_db = crossProd(a_db, b_db)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This is not a vector cross product operation. Each row of the two DBs are matched 
 and added as a new row to a DB. The end is a DB with all combinations 
 of rows from both DBs. The final DB contains columns of both DBs.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db, b_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
cross_db: The tests_db object with all combinations of rows.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>allocateRows</TT> (p.&nbsp;<A HREF="#ref_allocateRows"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18353"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18355"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/10/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479400000000000000">
1.47.94 Method <TT>tests_db/tests2cols</TT></A>
</H3><A NAME="18357"></A><A NAME="ref_tests_db__tests2cols"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Find column numbers from a test names/numbers specification.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">cols = tests2cols(db, tests)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses tests2idx.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>tests</TT>:</STRONG></DT>
<DD>Either a single or array of column numbers, or a single
<P>
test name or a cell array of test names. If ':', all
tests. For name strings, regular expressions are
supported if quoted with slashes (e.g., '/a.*/'). 
See tests2idx for more.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
cols: Array of column indices.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18359"></A>, <TT>tests2idx</TT> (p.&nbsp;<A HREF="#ref_tests2idx"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18361"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479500000000000000">
1.47.95 Method <TT>tests_db/rows2Struct</TT></A>
</H3><A NAME="18363"></A><A NAME="ref_tests_db__rows2Struct"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert given rows of database to a structure array.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">s = rows2Struct(db, rows, pages)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>rows</TT>:</STRONG></DT>
<DD>Indices of rows in db.
</DD>
<DT><STRONG><TT>pages</TT>:</STRONG></DT>
<DD>Pages of db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
s: A structure of column name and value pairs.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18365"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/08/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501479600000000000000">
1.47.96 Method <TT>tests_db/max</TT></A>
</H3><A NAME="18367"></A><A NAME="ref_tests_db__max"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the max of the data matrix of a_db. Ignores NaN and Inf values.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[a_db, n] = max(a_db, dim)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Does a recursive operation over dimensions in order to remove NaN and
 Inf values. This takes more time than a straightforward max operation. 
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_db</TT>:</STRONG></DT>
<DD>A tests_db object.
</DD>
<DT><STRONG><TT>dim</TT>:</STRONG></DT>
<DD>Work down dimension.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_db: The DB with one row of max values.
	n: (Optional) Numbers of non-NaN rows included in calculating each column.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>max</TT> (p.&nbsp;<A HREF="#ref_max"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18369"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18371"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/06
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050148000000000000000">
1.48 Class <TT>trace</TT></A>
</H2><A NAME="18373"></A><A NAME="ref_trace"></A>
<H3><A NAME="SECTION050148100000000000000">
1.48.1 Constructor <TT>trace/trace</TT></A>
</H3><A NAME="18375"></A><A NAME="ref_trace__trace"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>A generic trace from a cell. It can be voltage, current, etc.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = trace(data_src, dt, dy, id, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Traces for specific experimental or simulation protocols can extend 
 this class for adding new parameters. This object is designed to recognize
 most data file formats. See the data_src parameter below.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data_src</TT>:</STRONG></DT>
<DD>Trace data as a column vector OR name of a data file generated by either 
<P>
Genesis (.bin, .gbin, .genflac), PCDX (.all), or Matlab (.mat).
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>Identification string
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>scale_y</TT>:</STRONG></DT>
<DD>Y-axis scale to be applied to loaded data.
</DD>
<DT><STRONG><TT>offset_y</TT>:</STRONG></DT>
<DD>Y-axis offset to be added to loaded and scaled data.
</DD>
<DT><STRONG><TT>trace_time_start</TT>:</STRONG></DT>
<DD>Samples in the beginning to discard [dt]
</DD>
<DT><STRONG><TT>baseline</TT>:</STRONG></DT>
<DD>Resting potential.
</DD>
<DT><STRONG><TT>channel</TT>:</STRONG></DT>
<DD>Channel to read from file Genesis, PCDX, or Neuron file.
</DD>
<DT><STRONG><TT>file_type</TT>:</STRONG></DT>
<DD>Specify file type instead of guessing from extension:
<P>
'genesis': Raw binary files created with Genesis disk_out method.
'genesis_flac': Compressed Genesis binary files.
'neuron': Binary files created with Neuron's Vector.vwrite method.
'pcdx': .ALL data acquisition files from PCDX program.
'matlab': Matlab .MAT binary files with matrix data.
</DD>
<DT><STRONG><TT>traces</TT>:</STRONG></DT>
<DD>Traces to read from PCDX file.
</DD>
<DT><STRONG><TT>spike_finder</TT>:</STRONG></DT>
<DD>Method of finding spikes 

<P>
(1 for findFilteredSpikes, 2 for findspikes).
</DD>
<DT><STRONG><TT>threshold</TT>:</STRONG></DT>
<DD>Spike threshold used in spike finders.
</DD>
<DT><STRONG><TT>init_Vm_method</TT>:</STRONG></DT>
<DD>Method of finding spike thresholds during spike

<P>
shape calculation (see spike_shape/spike_shape).
</DD>
<DT><STRONG><TT>init_threshold</TT>:</STRONG></DT>
<DD>Spike initiation threshold (deriv or accel).

<P>
(see above methods and implementation in calcInitVm)
</DD>
<DT><STRONG><TT>init_lo_thr, init_hi_thr</TT>:</STRONG></DT>
<DD>Low and high thresholds for slope.
</DD>
</DL></DD>
<DT><STRONG><TT>custom_filter</TT>:</STRONG></DT>
<DD>Recommended if sampling rate differs appreciably from 10 kHz.

<P>
If custom_filter == 1, a filter with custom lowpass and highpass
cutoffs can be specified. This allows for fast and accurate spike
discrimination. The filter type used is a 2-pole
butterworth, different than the default high-order
Cheby2. Creates new prop called 'butterWorth' to
hold the filter.
</DD>
<DT><STRONG><TT>lowPassFreq</TT>:</STRONG></DT>
<DD>if set, it sets a new low pass cutoff. default is 3000Hz
</DD>
<DT><STRONG><TT>highPassFreq</TT>:</STRONG></DT>
<DD>if set it sets a new high pass cutoff. default is 50 Hz
<DL>
<DD></DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, reduces the amount of textual description in plots, etc.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
data: The trace column matrix.
	dt, dy, id, props (see above)
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18377"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18379"></A>, <TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18381"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18383"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148200000000000000">
1.48.2 Method <TT>trace/plotData</TT></A>
</H3><A NAME="18385"></A><A NAME="ref_trace__plotData"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plotData(t, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If t is a vector of traces, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>timeScale</TT>:</STRONG></DT>
<DD>'s' for seconds, or 'ms' for milliseconds.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>If 1, only display given title_str.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18387"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18389"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18391"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148300000000000000">
1.48.3 Method <TT>trace/get</TT></A>
</H3><A NAME="18393"></A><A NAME="ref_trace__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148400000000000000">
1.48.4 Method <TT>trace/plot</TT></A>
</H3><A NAME="18395"></A><A NAME="ref_trace__plot"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plot(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to plot_abstract.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
h: Handle to figure object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18397"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18399"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148500000000000000">
1.48.5 Method <TT>trace/setProp</TT></A>
</H3><A NAME="18401"></A><A NAME="ref_trace__setProp"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting optional object properties.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = setProp(obj, prop1, val1, prop2, val2, ...)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Modifies or adds property values. As many property name-value 
 pairs can be specified.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>obj</TT>:</STRONG></DT>
<DD>Any object that has a props field.
</DD>
<DT><STRONG><TT>attr</TT>:</STRONG></DT>
<DD>Property name
</DD>
<DT><STRONG><TT>val</TT>:</STRONG></DT>
<DD>Property value.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: The new object with the updated properties.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148600000000000000">
1.48.6 Method <TT>trace/calcMax</TT></A>
</H3><A NAME="18403"></A><A NAME="ref_trace__calcMax"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the maximal value of the given period 
 		of the trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[max_val, max_idx] = calcMax(t, period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>A period object (optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
max_val: The max value.
	max_idx: Its index in the trace.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18405"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18407"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148700000000000000">
1.48.7 Method <TT>trace/getResults</TT></A>
</H3><A NAME="18409"></A><A NAME="ref_trace__getResults"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Runs all tests defined by this class and return them in a 
		structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = getResults(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: A structure associating test names to values 
		in ms and mV (or mA).
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18411"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148800000000000000">
1.48.8 Method <TT>trace/calcAvg</TT></A>
</H3><A NAME="18413"></A><A NAME="ref_trace__calcAvg"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the average value of the given period 
 		of the trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">avg_val = calcAvg(t, period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>A period object (optional).
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18415"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18417"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050148900000000000000">
1.48.9 Method <TT>trace/analyzeSpikesInPeriod</TT></A>
</H3><A NAME="18419"></A><A NAME="ref_trace__analyzeSpikesInPeriod"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns results and a db of spikes by collecting test results of a cip_trace, analyzing each individual spike.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[results period_spikes a_spikes_db spikes_stats_db spikes_hists_dbs] =
      analyzeSpikesInPeriod(a_cip_trace, a_spikes, period, prefix_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cip_trace</TT>:</STRONG></DT>
<DD>A cip_trace object.
</DD>
<DT><STRONG><TT>a_spikes</TT>:</STRONG></DT>
<DD>A spikes object from the a_cip_trace object.
</DD>
<DT><STRONG><TT>period</TT>:</STRONG></DT>
<DD>A period of object of a_cip_trace object of interest.
</DD>
<DT><STRONG><TT>prefix_str</TT>:</STRONG></DT>
<DD>Prefix string to be added to spike shape results.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: Results structure names prefixed with prefix_str.
	period_spikes: Corrected spikes object for this period.
	a_spikes_db: A mini spikes database of results from each spike in period. 
	spikes_stats_db: Statistics from the mini spikes database.
	spikes_hists_dbs: Cell array of histograms from the mini spikes database.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_trace</TT> (p.&nbsp;<A HREF="node5.html#ref_cip_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18421"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18423"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18425"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18427"></A>, <TT>getProfileAllSpikes</TT> (p.&nbsp;<A HREF="#ref_getProfileAllSpikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18429"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/05/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481000000000000000">
1.48.10 Method <TT>trace/set</TT></A>
</H3><A NAME="18431"></A><A NAME="ref_trace__set"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Generic method for setting object attributes.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481100000000000000">
1.48.11 Method <TT>trace/periodWhole</TT></A>
</H3><A NAME="18433"></A><A NAME="ref_trace__periodWhole"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the boundaries of the whole period of trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">whole_period = periodWhole(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18435"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18437"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481200000000000000">
1.48.12 Method <TT>trace/subsref</TT></A>
</H3><A NAME="18439"></A><A NAME="ref_trace__subsref"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic indexing for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481300000000000000">
1.48.13 Method <TT>trace/calcMin</TT></A>
</H3><A NAME="18441"></A><A NAME="ref_trace__calcMin"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Calculates the minimal value of the given period 
 		of the trace, t. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[min_val, min_idx] = calcMin(t, a_period)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>A period object (optional).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
min_val: The min value.
	min_idx: Its index in the trace.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18443"></A>, <TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18445"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481400000000000000">
1.48.14 Method <TT>trace/runAvg</TT></A>
</H3><A NAME="18447"></A><A NAME="ref_trace__runAvg"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a trace which is the running average of this trace.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">avg_t = runAvg(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
avg_t: A trace object that contains the running average of this trace.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18449"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/05/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481500000000000000">
1.48.15 Method <TT>trace/display</TT></A>
</H3><A NAME="18451"></A><A NAME="ref_trace__display"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481600000000000000">
1.48.16 Method <TT>trace/getDy</TT></A>
</H3><A NAME="18453"></A><A NAME="ref_trace__getDy"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns dy.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">dy = getDy(t)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
dy: The dy value.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18455"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/31
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481700000000000000">
1.48.17 Method <TT>trace/findFilteredSpikes</TT></A>
</H3><A NAME="18457"></A><A NAME="ref_trace__findFilteredSpikes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Runs a frequency filter over the data and then 
			finds all peaks using findspikes.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[times, peaks, n] = 
	findFilteredSpikes(t, a_period, plotit, minamp)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Runs a 50-300 Hz band-pass filter over the data and then calls findspikes.
   The filter both removes low-frequency offset changes, such as 
   cip period effects, and high-frequency noise that is detected 
   as local peaks by findspikes. The spikes found are 
   post-processed to make sure the rise and fall times are consistent.
   Note: The filter employed only works with data sampled at 10kHz.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>Trace object
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>Period of interest.
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>Plots the spikes found if 1.
<P>
minamp (optional): minimum amplitude above baseline that must be reached.
-&gt; adjust as necessary to discriminate spikes from EPSPs.
</DD>
<DT><STRONG><TT>Returns</TT>:</STRONG></DT>
<DD><P>
</DD>
<DT><STRONG><TT>times</TT>:</STRONG></DT>
<DD>The times of spikes [dt].
</DD>
<DT><STRONG><TT>peaks</TT>:</STRONG></DT>
<DD>The peaks corresponding to the times of spikes.
</DD>
<DT><STRONG><TT>n</TT>:</STRONG></DT>
<DD>The number of spikes.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>findspikes</TT> (p.&nbsp;<A HREF="#ref_findspikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18459"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18461"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481800000000000000">
1.48.18 Method <TT>trace/withinPeriod</TT></A>
</H3><A NAME="18463"></A><A NAME="ref_trace__withinPeriod"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a trace object valid only within the given period.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = withinPeriod(t, a_period, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>The desired period
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>useAvailable</TT>:</STRONG></DT>
<DD>If 1, don't stop if period not available, use maximum available.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
obj: A trace object
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18465"></A>, <TT>period</TT> (p.&nbsp;<A HREF="node5.html#ref_period"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18467"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/25
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501481900000000000000">
1.48.19 Method <TT>trace/spikes</TT></A>
</H3><A NAME="18469"></A><A NAME="ref_trace__spikes"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert trace to spikes object for spike timing calculations.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = spikes(trace, a_period, plotit, minamp)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spikes object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>a_period</TT>:</STRONG></DT>
<DD>A period object denoting the part of trace of interest 
<P>
(optional, if empty vector, taken as wholePeriod).
</DD>
<DT><STRONG><TT>plotit</TT>:</STRONG></DT>
<DD>If non-zero, a plot is generated for showing spikes found

<P>
(optional).
</DD>
<DT><STRONG><TT>minamp</TT>:</STRONG></DT>
<DD>minimum amplitude that must be reached if using findFilteredSpikes.

<P>
-&gt; adjust as needed to discriminate spikes from EPSPs.
(optional)
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18471"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/07/30
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501482000000000000000">
1.48.20 Method <TT>trace/getSpike</TT></A>
</H3><A NAME="18473"></A><A NAME="ref_trace__getSpike"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert a spike in the trace to a spike_shape object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = getSpike(trace, spikes, spike_num, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spike_shape object from desired spike. It is more efficient if
 you already have the spikes object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>spikes</TT>:</STRONG></DT>
<DD>(Optional) A spikes object obtained from trace, 
<P>
calculated automatically if given as [].
</DD>
<DT><STRONG><TT>spike_num</TT>:</STRONG></DT>
<DD>The index of spike to extract.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>spike_id</TT>:</STRONG></DT>
<DD>A prefix string added to the spike_shape object's id.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">This will create an annotated plot of the third spike in my_trace:
<BR>&#187; plotFigure(plotResults(getSpike(my_trace, [], 3)))
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18475"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/19
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501482100000000000000">
1.48.21 Method <TT>trace/plot_abstract</TT></A>
</H3><A NAME="18477"></A><A NAME="ref_trace__plot_abstract"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots a trace by calling plotData.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_plot = plot_abstract(t, title_str, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If t is a vector of traces, returns a vector of plot objects.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>t</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>title_str</TT>:</STRONG></DT>
<DD>(Optional) String to append to plot title.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>timeScale</TT>:</STRONG></DT>
<DD>'s' for seconds, or 'ms' for milliseconds.
<P>
(rest passed to plot_abstract.)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_plot: A plot_abstract object that can be visualized.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18479"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18481"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18483"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501482200000000000000">
1.48.22 Method <TT>trace/spike_shape</TT></A>
</H3><A NAME="18485"></A><A NAME="ref_trace__spike_shape"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Convert averaged spikes in the trace to a spike_shape object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">obj = spike_shape(trace, spikes, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Creates a spike_shape object.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>trace</TT>:</STRONG></DT>
<DD>A trace object.
</DD>
<DT><STRONG><TT>spikes</TT>:</STRONG></DT>
<DD>A spikes object on trace.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18487"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/04
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050149000000000000000">
1.49 Class <TT>trace_profile</TT></A>
</H2><A NAME="18489"></A><A NAME="ref_trace_profile"></A>
<H3><A NAME="SECTION050149100000000000000">
1.49.1 Constructor <TT>trace_profile/trace_profile</TT></A>
</H3><A NAME="18491"></A><A NAME="ref_trace_profile__trace_profile"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates and collects test results of a trace.
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The first usage is fully customizable to be used from subclass constructors.
 The second usage generates the spikes and spike_shape objects, and
 collects some generic test results from them. This usage is only provided
 as an example and is not used practically.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data_src</TT>:</STRONG></DT>
<DD>The trace column OR the .MAT filename.
</DD>
<DT><STRONG><TT>dt</TT>:</STRONG></DT>
<DD>Time resolution [s]
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>y-axis resolution [ISI (V, A, etc.)]
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>See trace object.
</DD>
</DL></DD>
<DT><STRONG>Returns a structure object with the following fields:</STRONG></DT>
<DD>&nbsp;

<P>
trace, spikes, spike_shape, results, id, props.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18493"></A>, <TT>spikes</TT> (p.&nbsp;<A HREF="node5.html#ref_spikes"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18495"></A>, <TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18497"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050149200000000000000">
1.49.2 Method <TT>trace_profile/get</TT></A>
</H3><A NAME="18499"></A><A NAME="ref_trace_profile__get"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Defines generic attribute retrieval for objects.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/14
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H2><A NAME="SECTION050150000000000000000"></A><A NAME="ref_utils"></A>
<BR>
1.50 Utility functions
</H2>
<H3><A NAME="SECTION050150100000000000000">
1.50.1 Function <TT>functions/TeXtable</TT></A>
</H3><A NAME="18501"></A><A NAME="ref_functions__TeXtable"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates the LaTeX string for a floating table containing given contents.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = TeXtable(contents, caption, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>contents</TT>:</STRONG></DT>
<DD>Table contents in LaTeX.
</DD>
<DT><STRONG><TT>caption</TT>:</STRONG></DT>
<DD>Table caption.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>rotate</TT>:</STRONG></DT>
<DD>Degrees to rotate.
</DD>
<DT><STRONG><TT>width</TT>:</STRONG></DT>
<DD>Resize to this width.
</DD>
<DT><STRONG><TT>height</TT>:</STRONG></DT>
<DD>Resize to this height
</DD>
<DT><STRONG><TT>center</TT>:</STRONG></DT>
<DD>Align to center.
</DD>
<DT><STRONG><TT>shortCaption</TT>:</STRONG></DT>
<DD>Short version of caption to appear at list of tables.
</DD>
<DT><STRONG><TT>floatType</TT>:</STRONG></DT>
<DD>LaTeX float to use (default='table').
</DD>
<DT><STRONG><TT>label</TT>:</STRONG></DT>
<DD>Used for internal LaTeX references.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX string for table float.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150200000000000000">
1.50.2 Function <TT>functions/logLevels</TT></A>
</H3><A NAME="18503"></A><A NAME="ref_functions__logLevels"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns a logarithmic-scaled series between min_val and max_val with num_levels elements.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">levels = logLevels(min_val, max_val, num_levels)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>min_val, max_val</TT>:</STRONG></DT>
<DD>The low and high boundaries for the output value.
</DD>
<DT><STRONG><TT>num_levels</TT>:</STRONG></DT>
<DD>Number of elements to produce, including the boundaries.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
levels: A column vector of logarithmic series between min_val and max_val.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150300000000000000">
1.50.3 Function <TT>functions/parseGenesisFilename</TT></A>
</H3><A NAME="18505"></A><A NAME="ref_functions__parseGenesisFilename"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Parses the GENESIS filename to get names and values of simulation parameters.
 Usage:
 names_vals = parseGenesisFilename(filename)
</DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Parameters:
		filename: GENESIS filename (no need to exist)
</DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
names_vals: A two-column cell array with names and values.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cip_profile</TT> (p.&nbsp;<A HREF="#ref_cip_profile"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18507"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150400000000000000">
1.50.4 Function <TT>functions/gettracelist2</TT></A>
</H3><A NAME="18509"></A><A NAME="ref_functions__gettracelist2"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Gets a list of the form: '1 3 7-10' and returns a column vector with the traces numbers, and the number of traces.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[traces, ntraces] = gettracelist2('list');</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Please note the space between single traces and the dash for ranges of
traces.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>&lt;adelgado@biology.emory.edu&gt;</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150500000000000000">
1.50.5 Function <TT>functions/loadtraces</TT></A>
</H3><A NAME="18511"></A><A NAME="ref_functions__loadtraces"></A><DL>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>file</TT>:</STRONG></DT>
<DD>PCDX file.
</DD>
<DT><STRONG><TT>tracelist</TT>:</STRONG></DT>
<DD>A string of trace description, such as '1-10'.
</DD>
<DT><STRONG><TT>channel</TT>:</STRONG></DT>
<DD>Channel to read from.
</DD>
<DT><STRONG><TT>quiet</TT>:</STRONG></DT>
<DD>(Optional) If 1, produce on print outs.
</DD>
</DL></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>&lt;adelgado@biology.emory.edu&gt;</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150600000000000000">
1.50.6 Function <TT>functions/properTeXLabel</TT></A>
</H3><A NAME="18513"></A><A NAME="ref_functions__properTeXLabel"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Replaces characters in string or cell array of strings to make it valid in TeX documents.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_label = properTeXLabel( a_label )</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It will replace characters like space, '/', '.', etc.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_label</TT>:</STRONG></DT>
<DD>A label string.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_label: The corrected proper a_label.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a_label = properTeXLabel('this_day')
<BR>
ans = 'this<!-- MATH
 $\backslash$
 -->
<IMG
 WIDTH="12" HEIGHT="32" ALIGN="MIDDLE" BORDER="0"
 SRC="img29.png"
 ALT="$ \backslash$">_day' 
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/01/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150700000000000000">
1.50.7 Function <TT>functions/defaultValue</TT></A>
</H3><A NAME="18515"></A><A NAME="ref_functions__defaultValue"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>If variable unset (either nonexistent or empty), assign it
 a default value. Otherwise the variable remains unchanged.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">var = defaultValue(varname, a_defaultvalue)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>If the variable has already been defined, it keeps unchanged. If the
 variable doesn't exist or is an empty matrix, it will be assigned a
 default value to it.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>varname</TT>:</STRONG></DT>
<DD>a string. the name of the variable.
</DD>
<DT><STRONG><TT>a_defaultvalue</TT>:</STRONG></DT>
<DD>value for the variable.
</DD>
</DL></DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">SamplingRate = defaultValue('SamplingRate', 10);
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li, Su
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150800000000000000">
1.50.8 Function <TT>functions/ns_CIPlist</TT></A>
</H3><A NAME="18517"></A><A NAME="ref_functions__ns_CIPlist"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Dawid Kurzyniec
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION050150900000000000000">
1.50.9 Function <TT>functions/diff3T</TT></A>
</H3><A NAME="18519"></A><A NAME="ref_functions__diff3T"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimate of third derivative using Taylor expansion.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">deriv3 = diff3T(x, dy)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>d&#94;3 x     x(k-3) - 8 * x(k-2) + 13 * x(k-1) - 13 * x(k+1) + 8 * x(k+2) - x(k+3)
  ---- = ------------------------------------
   dy&#94;3			        8 * dy&#94;3
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector of x = f(y).
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>The resolution of the discrete points in the vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
deriv3: Estimate of the derivative.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501000000000000000">
1.50.10 Function <TT>functions/findspikes_old</TT></A>
</H3><A NAME="18521"></A><A NAME="ref_functions__findspikes_old"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>&lt;adelgado@biology.emory.edu&gt;, 2003-03-31
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501100000000000000">
1.50.11 Function <TT>functions/properTeXFilename</TT></A>
</H3><A NAME="18523"></A><A NAME="ref_functions__properTeXFilename"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Replaces characters in string to make it a valid filename for inclusion in TeX documents.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">filename = properTeXFilename( filename )</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>It will replace characters like space, '/', '.', etc.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>filename</TT>:</STRONG></DT>
<DD>An input filename string (without extension!).
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
filename: The corrected proper filename.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; fname = properTeXFilename('hello world/1')
<BR>
ans = 'hello_world+1' 
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/12/20
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501200000000000000">
1.50.12 Function <TT>functions/mergeStructs</TT></A>
</H3><A NAME="18525"></A><A NAME="ref_functions__mergeStructs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Merges all the structures given as arguments and makes a single structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">results = mergeStructs( struct1 [, struct2, ...] )</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>The fields will in earlier arguments will have priority. So, while merging two
 structs, if there are duplicate fields, the fields in the first will be preserved.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;

<P>
struct(n): A structure.</DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
results: The merged structure.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">mergeStructs( struct('hello', 1), struct('bye', 2) );
<BR>=&gt; struct('hello', 1, 'bye', 2)
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501300000000000000">
1.50.13 Function <TT>functions/fillederrorbar</TT></A>
</H3><A NAME="18527"></A><A NAME="ref_functions__fillederrorbar"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Plots an errorbar with the middle points filled with the pen color.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handles = fillederrorbar(...)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;

<P>
(see errorbar)</DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handles: Handles to graphics objects.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501400000000000000">
1.50.14 Function <TT>functions/struct2DB</TT></A>
</H3><A NAME="18529"></A><A NAME="ref_functions__struct2DB"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Converts a structure array to a tests_db object.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tests_db = struct2DB(a_struct, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Field names become column names in the DB.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_struct</TT>:</STRONG></DT>
<DD>A structure to convert.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties, passed to tests_db.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tests_db: A tests_db object.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501500000000000000">
1.50.15 Function <TT>functions/getfuzzyfield</TT></A>
</H3><A NAME="18531"></A><A NAME="ref_functions__getfuzzyfield"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li, Su - 2007
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501600000000000000">
1.50.16 Function <TT>functions/maxima</TT></A>
</H3><A NAME="18533"></A><A NAME="ref_functions__maxima"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Find all local maxima.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">x_idx = maxima(x)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Finds derivative sign-flipping points where the second derivative is 
 less than zero.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
x_idx: Indices of maxima.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501700000000000000">
1.50.17 Function <TT>functions/chanTables2DB</TT></A>
</H3><A NAME="18535"></A><A NAME="ref_functions__chanTables2DB"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates a DB with channel tables exported from Genesis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_chans_db = chanTables2DB(tables, id, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>tables</TT>:</STRONG></DT>
<DD>Structures returned from the dump files generated by dump_chans.g.
</DD>
<DT><STRONG><TT>id</TT>:</STRONG></DT>
<DD>String that identify the source of the tables structure.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<P>
(rest passed to tests_db.)
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_chans_db: A tests_db object containing channel tables.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>trace</TT> (p.&nbsp;<A HREF="node5.html#ref_trace"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18537"></A>, <TT>trace/plot</TT> (p.&nbsp;<A HREF="node5.html#ref_trace__plot"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18539"></A>, <TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18541"></A>, <TT>GP/common/dump_chans.g (Genesis)</TT> (p.&nbsp;<A HREF="#ref_GP__common"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18543"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/03/07
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501800000000000000">
1.50.18 Function <TT>functions/prefixStruct</TT></A>
</H3><A NAME="18545"></A><A NAME="ref_functions__prefixStruct"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Adds the given prefix to each of the field names in the structure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">new_struct = prefixStruct(a_struct, prefix_str)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_struct</TT>:</STRONG></DT>
<DD>A structure.
</DD>
<DT><STRONG><TT>prefix_str</TT>:</STRONG></DT>
<DD>A string to be prefixed to each field name.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
new_struct: The new structure.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">prefixStruct( struct('bye', 1), 'hello');
<BR>=&gt; struct('hellobye', 1)
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/22
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501501900000000000000">
1.50.19 Function <TT>functions/array2str</TT></A>
</H3><A NAME="18547"></A><A NAME="ref_functions__array2str"></A><DL>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">string = array2str([1 2 3 4 6 8 10 12 13])
<BR>
the output is '[1:4 6:2:12 13]'
<BR></div>
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502000000000000000">
1.50.20 Function <TT>functions/cell2TeX</TT></A>
</H3><A NAME="18549"></A><A NAME="ref_functions__cell2TeX"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Creates LaTeX string of a formatted table with the cell array's contents.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">tex_string = cell2TeX(a_cell, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_cell</TT>:</STRONG></DT>
<DD>A cell matrix to be tabularized.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>hasTitleRow</TT>:</STRONG></DT>
<DD>The first row contains titles.
</DD>
<DT><STRONG><TT>hasTitleCol</TT>:</STRONG></DT>
<DD>The first column contains titles.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
tex_string: LaTeX formatted table string.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502100000000000000">
1.50.21 Function <TT>functions/setAxisNonNaN</TT></A>
</H3><A NAME="18551"></A><A NAME="ref_functions__setAxisNonNaN"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the limits of the current axis replaced with the non-NaN elements of the given vector.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">new_axis = setAxisNonNaN(layout_axis)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>layout_axis</TT>:</STRONG></DT>
<DD>The axis position to layout this plot. 
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
new_axis: Modified axis.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; axis(setAxisNonNaN([0 100 NaN NaN])) </div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>plot_abstract</TT> (p.&nbsp;<A HREF="node5.html#ref_plot_abstract"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18553"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2007/10/29
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502200000000000000">
1.50.22 Function <TT>functions/diff2T_h4</TT></A>
</H3><A NAME="18555"></A><A NAME="ref_functions__diff2T_h4"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimate of second derivative using Taylor expansion (derived with same method as diffT).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">deriv2 = diff2T_h4(x, dy)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>d&#94;2 x     x(k-2) - x(k-1) - x(k+1) + x(k+2)
  ---- = ------------------
   dy&#94;2		6 * dy&#94;2
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector of x = f(y).
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>The resolution of the discrete points in the vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
deriv2: Estimate of the derivative.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502300000000000000">
1.50.23 Function <TT>functions/plotImage</TT></A>
</H3><A NAME="18557"></A><A NAME="ref_functions__plotImage"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Function that plots a color matrix on current figure.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">h = plotImage(image_data, a_colormap, num_colors, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>image_data</TT>:</STRONG></DT>
<DD>2D matrix with image data.
</DD>
<DT><STRONG><TT>a_colormap</TT>:</STRONG></DT>
<DD>Colormap vector, function name or handle to colormap (e.g., 'jet').
</DD>
<DT><STRONG><TT>num_colors</TT>:</STRONG></DT>
<DD>Parameter to be passed to the a_colormap.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>colorbar</TT>:</STRONG></DT>
<DD>If defined, show colorbar on plot.
</DD>
<DT><STRONG><TT>colorbarProps</TT>:</STRONG></DT>
<DD>Set colorbar axis properties.
</DD>
<DT><STRONG><TT>colorbarLabel</TT>:</STRONG></DT>
<DD>Set colorbar y-axis label.
</DD>
<DT><STRONG><TT>truncateDecDigits</TT>:</STRONG></DT>
<DD>Truncate labels to this many decimal digits.
</DD>
<DT><STRONG><TT>maxValue</TT>:</STRONG></DT>
<DD>Maximal value at num_colors to annotate the colorbar.
</DD>
<DT><STRONG><TT>reverseYaxis</TT>:</STRONG></DT>
<DD>If 1, display y-axis values in reverse (default=0).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
colors: RGB color matrix to be passed to colormap function.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>colormap</TT> (p.&nbsp;<A HREF="#ref_colormap"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18559"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502400000000000000">
1.50.24 Function <TT>functions/diff2T</TT></A>
</H3><A NAME="18561"></A><A NAME="ref_functions__diff2T"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimate of second derivative using Taylor expansion.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">deriv2 = diff2T(x, dy)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>d&#94;2 x     - x(k-2) + 16 * x(k-1) - 30 * x(k) + 16 * x(k+1) - x(k+2)
  ---- = ------------------------------
   dy&#94;2			        12 * dy&#94;2
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector of x = f(y).
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>The resolution of the discrete points in the vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
deriv2: Estimate of the derivative.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502500000000000000">
1.50.25 Function <TT>functions/boxutilp</TT></A>
</H3><A NAME="18563"></A><A NAME="ref_functions__boxutilp"></A><DL>
<DT><STRONG>Description:</STRONG></DT>
<DD>BOXUTILP(X) is a utility function for BOXPLOT, which calls
   BOXUTILP once for each column of its first argument. Use
   BOXPLOT rather than BOXUTILP. 
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502600000000000000">
1.50.26 Function <TT>functions/ns_load_tracesets</TT></A>
</H3><A NAME="18565"></A><A NAME="ref_functions__ns_load_tracesets"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Return a set of physiol_cip_traceset objects loaded from a single NeuroSAGE HDF5 file.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">a_tss = ns_load_tracesets(data_src, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>This allows customized loading each NeuroSAGE file separately. Only
 loads traces that has the word 'cip' or 'spont' in the NeuroSAGE sequence name.
 Sample rate, channel gain and dy values are read from the acquisition data.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data_src</TT>:</STRONG></DT>
<DD>A pattern for one or more NeuroSAGE filename or structure output of ns_open_file.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>VmChan</TT>:</STRONG></DT>
<DD>(Optional) If a string, read voltage trace from channel having
<P>
this string (e.g., 'Amp1 Vm'). If numeric, use as channel
number. Added to the neuron_id to distinguish
multiple neurons recorded in same file. If not
specified, the first voltage channel is used.
</DD>
<DT><STRONG><TT>ImChan</TT>:</STRONG></DT>
<DD>(Optional) Similar to VmChan for reading current
<P>
trace. Does not affect neuron_id.
</DD>
<DT><STRONG><TT>VGain, IGain</TT>:</STRONG></DT>
<DD>for HDF5 files, these two fields only works as a default value

<P>
when the gains are not specified in the file.
</DD>
<DT><STRONG><TT>IncludeSeq</TT>:</STRONG></DT>
<DD>A string or cell array of strings specifying keywords in

<P>
sequence name to look for.
</DD>
<DT><STRONG><TT>ExcludeSeq</TT>:</STRONG></DT>
<DD>A string or cell array of strings specifying keywords in

<P>
sequence name to avoid searching.
</DD>
<DT><STRONG><TT>addTreats</TT>:</STRONG></DT>
<DD>Structure of default treatment names and their

<P>
values for this traceset to keep consistent accross 
tracesets. Use only lowercase in treatment names.
</DD>
<DT><STRONG><TT>fixTreats</TT>:</STRONG></DT>
<DD>Override wrong treatment information with

<P>
these. Same format as addTreats.
</DD>
<DT><STRONG><TT>renameTreats</TT>:</STRONG></DT>
<DD>Structure with from-&gt;to rename pairs.
</DD>
<DT><STRONG><TT>trials</TT>:</STRONG></DT>
<DD>A vector of trials to load from the file. All others

<P>
are skipped.
(All other props are passed to physiol_cip_traceset)
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
a_tss: Cell array of physiol_cip_traceset objects.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>physiol_cip_traceset_fileset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset_fileset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18567"></A>, <TT>physiol_cip_traceset</TT> (p.&nbsp;<A HREF="node5.html#ref_physiol_cip_traceset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18569"></A>, <TT>params_tests_dataset</TT> (p.&nbsp;<A HREF="node5.html#ref_params_tests_dataset"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18571"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li, Su; Cengiz Gunay &lt;cgunay@emory.edu&gt;; and Jeremy Edgerton, 2007/12/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502700000000000000">
1.50.27 Function <TT>functions/string2File</TT></A>
</H3><A NAME="18573"></A><A NAME="ref_functions__string2File"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Writes string verbatim into a file.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">string2File(string, filename, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>string</TT>:</STRONG></DT>
<DD>To be written into file.
</DD>
<DT><STRONG><TT>filename</TT>:</STRONG></DT>
<DD>The file to be created.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>append</TT>:</STRONG></DT>
<DD>If 1, append to existing file.
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>cell2TeX</TT> (p.&nbsp;<A HREF="#ref_cell2TeX"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18575"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/12/10
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502800000000000000">
1.50.28 Function <TT>functions/sortedUniqueValues</TT></A>
</H3><A NAME="18577"></A><A NAME="ref_functions__sortedUniqueValues"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Find unique rows in an already sorted matrix (or column vector).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[rows, idx] = sortedUniqueValues(data)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses the derivation by Matlab diff function method.  Redundant with the
 Matlab function UNIQUE doing the same job:
 [rows, idx]= unique(data, 'rows', 'first'). 
 However, sortedUniqueValues is more efficient if the input data is already
 sorted for some other reason (see usage in tests_db/invarValues).
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>A ascending row-sorted matrix or column vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A matrix or column vector of unique rows.
   idx: Indices of the unique rows in the original data matrix.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>uniqueValues</TT> (p.&nbsp;<A HREF="#ref_uniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18579"></A>, <TT>unique</TT> (p.&nbsp;<A HREF="#ref_unique"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18581"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/27
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501502900000000000000">
1.50.29 Function <TT>functions/meanSpikeFreq</TT></A>
</H3><A NAME="18583"></A><A NAME="ref_functions__meanSpikeFreq"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the mean firing frequency in Hz according to mean <!-- MATH
 $\backslash$
 -->
<IMG
 WIDTH="12" HEIGHT="32" ALIGN="MIDDLE" BORDER="0"
 SRC="img30.png"
 ALT="$ \backslash$">
  	    inter-spike-interval of the given spike train and the 
	    time resolution dt.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">meanFreq = meanSpikeFreq( spike_train, dt, period )</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Parameters:
		spike_train: Spike times returned by findspikes
		dt: Time step size [s]
		period: Duration of the total time period [dt]
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/03/08
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503000000000000000">
1.50.30 Function <TT>functions/diff3T_h4</TT></A>
</H3><A NAME="18585"></A><A NAME="ref_functions__diff3T_h4"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimate of third derivative using Taylor expansion (derived with same method as diffT and diff2T_h4).
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">deriv2 = diff3T_h4(x, dy)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>d&#94;3 x     - x(k-2) + 2 * x(k-1) - 2 * x(k+1) + x(k+2)
  ---- = -----------------------
   dy&#94;3			12 * dy&#94;3
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector of x = f(y).
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>The resolution of the discrete points in the vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
deriv2: Estimate of the derivative.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/18
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503100000000000000">
1.50.31 Function <TT>functions/growRange</TT></A>
</H3><A NAME="18587"></A><A NAME="ref_functions__growRange"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Returns the maximal range from rows of axis limits. 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">range = growRange(ranges)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>ranges</TT>:</STRONG></DT>
<DD>A matrix where each row is return val of axis func.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
range: The maximal range obtained that includes all given axes.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/10/13
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503200000000000000">
1.50.32 Function <TT>functions/readNeuronVecBin</TT></A>
</H3><A NAME="18589"></A><A NAME="ref_functions__readNeuronVecBin"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Konstantin Miller &lt;miller@cs.tu-berlin.de&gt;, Aug 09, 2005.</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503300000000000000">
1.50.33 Function <TT>functions/colormapBlueCrossRed</TT></A>
</H3><A NAME="18591"></A><A NAME="ref_functions__colormapBlueCrossRed"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Blue to red crossing colormap, with a black-colored zero-crossing.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">colors = colormapBlueCrossRed(num_half_colors)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Colormap contains (2 * num_half_colors + 1) colors, where (num_half_colors + 1) is the 
 zero crossing.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>num_half_colors</TT>:</STRONG></DT>
<DD>Number of colors to generate on one of the red or blue scales.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
colors: RGB color matrix to be passed to colormap function.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>colormap</TT> (p.&nbsp;<A HREF="#ref_colormap"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18593"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/06/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503400000000000000">
1.50.34 Function <TT>functions/diffT</TT></A>
</H3><A NAME="18595"></A><A NAME="ref_functions__diffT"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Estimate of first derivative using Taylor expansion.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">deriv = diffT(x, dy)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>dx     x(k-2) - 8 * x(k-1) + 8 * x(k+1) - x(k+2)
  -- = ---------------------
   dy			12 * dy
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x</TT>:</STRONG></DT>
<DD>A vector.
</DD>
<DT><STRONG><TT>dy</TT>:</STRONG></DT>
<DD>The resolution of the discrete points in the vector.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
deriv: Estimate of the first derivative.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/11/15
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503500000000000000">
1.50.35 Function <TT>functions/interpValByIndex</TT></A>
</H3><A NAME="18597"></A><A NAME="ref_functions__interpValByIndex"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds the interpolated value by using the real valued index from the data vector.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">val = interpValByIndex(idx, data)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Parameters:
	idx: A real-valued index.
	data: A data vector.
</DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
val: the value taken from the nearest integer indices of data and interpolated.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">&#187; a= [1 2 3];
<BR>&#187; interpValByIndex(1.5, a)
<BR>
ans =
<BR>
1.5000
<BR></div>
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>spike_shape</TT> (p.&nbsp;<A HREF="node5.html#ref_spike_shape"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18599"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/08/02
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503600000000000000">
1.50.36 Function <TT>functions/subTextLabel</TT></A>
</H3><A NAME="18601"></A><A NAME="ref_functions__subTextLabel"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Draws a text label on a plot.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">handle = subTextLabel(x, y, text_str, props)</div></DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>x, y</TT>:</STRONG></DT>
<DD>2D coordinates.
</DD>
<DT><STRONG><TT>text_str</TT>:</STRONG></DT>
<DD>String to be drawn on plot.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>Units</TT>:</STRONG></DT>
<DD>position units for the coordinates (see Units in axes properties).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
handle: Text object handle.
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/04/11
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503700000000000000">
1.50.37 Function <TT>functions/calcGraphNormPtsRatio</TT></A>
</H3><A NAME="18603"></A><A NAME="ref_functions__calcGraphNormPtsRatio"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Return the ratios of normalized to point units for dimensions of axis.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[ratio_x, ratio_y] = calcGraphNormPtsRatio(grfx_handle)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Used for findind character sizes given the size of an axis. Normally if the plot
 is resized, the characters may take up too much space or may not fit anymore unless
 the spacing is corrected.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>grfx_handle</TT>:</STRONG></DT>
<DD>A graphics handle to an existing axis or figure.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
ratio_x, ratio_y: Normalized to points ratio for axis width and height, respectively.
</DD>
<DT><STRONG>Example:</STRONG></DT>
<DD>&nbsp;
<div style="font-family: monospace">To find the normalized distance for a 10pt character:
<BR>&#187; dx = 10 * calcGraphNormPtsRatio(my_figure_handle);
<BR></div>
</DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2006/03/05
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503800000000000000">
1.50.38 Function <TT>functions/balanceInputProbs</TT></A>
</H3><A NAME="18605"></A><A NAME="ref_functions__balanceInputProbs"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Balances samples according to prior class probabilities of the outputs.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[new_inputs, new_outputs] = balanceInputProbs(a_class_inputs, a_class_outputs, balance_ratio, props)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Uses the method in Lawrence, burns, Back, Tsoi and Lee Giles "Neural
 network classification and prior class probabilities" for
 probabilitic balancing of input and output samples when the number of
 samples in each class is vastly different and causes problems with
 classification without balancing.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>a_class_inputs, a_class_outputs</TT>:</STRONG></DT>
<DD>Input and output vectors.
</DD>
<DT><STRONG><TT>balance_ratio</TT>:</STRONG></DT>
<DD>c_s, between 0 and 1. If 1, equal samples from
<P>
each class if used. If 0, prior class probabilities are followed.
</DD>
<DT><STRONG><TT>props</TT>:</STRONG></DT>
<DD>A structure with any optional properties.
<DL>
<DD></DD>
<DT><STRONG><TT>repeatSamples</TT>:</STRONG></DT>
<DD>If 1, repeats the smaller class samples to match
<P>
with larger class. Otherwise, takes the minimal number of
samples to avoid repetitions (default=1).
</DD>
</DL></DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
new_inputs, new_outputs: New input and output vectors.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>approxMappingNNet</TT> (p.&nbsp;<A HREF="#ref_approxMappingNNet"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18607"></A>, <TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18609"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2008/01/09
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501503900000000000000">
1.50.39 Function <TT>functions/findspikes</TT></A>
</H3><A NAME="18611"></A><A NAME="ref_functions__findspikes"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>Li, Su based on the original of Alfonso Delagado-Reyes
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501504000000000000000">
1.50.40 Function <TT>functions/boxplotp</TT></A>
</H3><A NAME="18613"></A><A NAME="ref_functions__boxplotp"></A><DL>
<DT><STRONG>Description:</STRONG></DT>
<DD>BOXPLOTP(X,NOTCH,SYM,VERT,WHIS,PROPS) produces a box and whisker plot for
   each column of X. The box has lines at the lower quartile, median, 
   and upper quartile values. The whiskers are lines extending from 
   each end of the box to show the extent of the rest of the data. 
   Outliers are data with values beyond the ends of the whiskers.
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501504100000000000000">
1.50.41 Function <TT>functions/collectspikes</TT></A>
</H3><A NAME="18615"></A><A NAME="ref_functions__collectspikes"></A><DL>
<DT><STRONG>Author:</STRONG></DT>
<DD>&lt;adelgado@biology.emory.edu&gt;</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501504200000000000000">
1.50.42 Function <TT>functions/findVectorInMatrix</TT></A>
</H3><A NAME="18617"></A><A NAME="ref_functions__findVectorInMatrix"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Finds rows of data that match row.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">idx = findVectorInMatrix(data, row)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Matlab's eq (==) command unfortunately doesn't allow this directly.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>A matrix or column vector.
</DD>
<DT><STRONG><TT>row</TT>:</STRONG></DT>
<DD>A row vector.
</DD>
<DT><STRONG><TT>Returns</TT>:</STRONG></DT>
<DD><P>
</DD>
<DT><STRONG><TT>idx</TT>:</STRONG></DT>
<DD>Indices of matching rows in the original data matrix.
</DD>
</DL></DD>
<DT><STRONG>See also:</STRONG></DT>
<DD></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2005/09/1
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501504300000000000000">
1.50.43 Function <TT>functions/uniqueValues</TT></A>
</H3><A NAME="18619"></A><A NAME="ref_functions__uniqueValues"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Find unique rows in a matrix (or column vector). 
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">[rows, idx] = uniqueValues(data)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Version which makes use of sort and diff. Maintains order of the
 original input.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>data</TT>:</STRONG></DT>
<DD>A matrix or column vector
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
rows: A matrix or column vector of unique rows.
   idx: Indices of the unique rows in the original data matrix.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>sortedUniqueValues</TT> (p.&nbsp;<A HREF="#ref_sortedUniqueValues"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18621"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/24
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<H3><A NAME="SECTION0501504400000000000000">
1.50.44 Function <TT>functions/makeIdx</TT></A>
</H3><A NAME="18623"></A><A NAME="ref_functions__makeIdx"></A><DL>
<DT><STRONG>Summary:</STRONG></DT>
<DD>Prepare the idx structure from names.
</DD>
<DT><STRONG>Usage:</STRONG></DT>
<DD>&nbsp;<div style="font-family: monospace">idx = makeIdx(names)</div></DD>
<DT><STRONG>Description:</STRONG></DT>
<DD>Helper function.
</DD>
<DT><STRONG>Parameters:</STRONG></DT>
<DD>&nbsp;
<DL>
<DD></DD>
<DT><STRONG><TT>names</TT>:</STRONG></DT>
<DD>Cell array of names for a db dimension.
</DD>
</DL></DD>
<DT><STRONG>Returns:</STRONG></DT>
<DD>&nbsp;

<P>
idx: Structure associating names to array indices.
</DD>
<DT><STRONG>See also:</STRONG></DT>
<DD><TT>tests_db</TT> (p.&nbsp;<A HREF="node5.html#ref_tests_db"><IMG  ALIGN="BOTTOM" BORDER="1" ALT="[*]" SRC="crossref.png"></A>)<A NAME="18625"></A></DD>
<DT><STRONG>Author:</STRONG></DT>
<DD>Cengiz Gunay &lt;cgunay@emory.edu&gt;, 2004/09/17
</DD>
</DL>
<BR>
<BR> <HR>width <BR>
<BR>
<P>


<P>

<HR>
<!--Navigation Panel-->
<A NAME="tex2html1876"
  HREF="node6.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next" SRC="next.png"></A> 
<A NAME="tex2html1870"
  HREF="prog-manual.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up" SRC="up.png"></A> 
<A NAME="tex2html1864"
  HREF="node4.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous" SRC="prev.png"></A> 
<A NAME="tex2html1872"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents" SRC="contents.png"></A> 
<A NAME="tex2html1874"
  HREF="node6.html">
<IMG WIDTH="43" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A NAME="tex2html1877"
  HREF="node6.html">Index</A>
<B> Up:</B> <A NAME="tex2html1871"
  HREF="prog-manual.html">Plotting and Analysis for</A>
<B> Previous:</B> <A NAME="tex2html1865"
  HREF="node4.html">Bibliography</A>
 &nbsp; <B>  <A NAME="tex2html1873"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A NAME="tex2html1875"
  HREF="node6.html">Index</A></B> 
<!--End of Navigation Panel-->
<ADDRESS>
Cengiz Gunay
2008-10-13
</ADDRESS>
</BODY>
</HTML>
